---
title: 'The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata'
keywords:
- to do
lang: en-US
date-meta: '2021-05-03'
author-meta:
- Adriano Rutz
- Maria Sorokina
- Jakub Galgonek
- Daniel Mietchen
- Egon Willighagen
- Arnaud Gaudry
- James G. Graham
- Ralf Stephan
- Roderic Page
- Jiří Vondrášek
- Christoph Steinbeck
- Guido F. Pauli
- Jean-Luc Wolfender
- Jonathan Bisson
- Pierre-Marie Allard
header-includes: |-
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta name="citation_title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta property="og:title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta property="twitter:title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta name="dc.date" content="2021-05-03" />
  <meta name="citation_publication_date" content="2021-05-03" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Adriano Rutz" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-0443-9902" />
  <meta name="twitter:creator" content="@adafede" />
  <meta name="citation_author" content="Maria Sorokina" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-9359-7149" />
  <meta name="twitter:creator" content="@ms_sorok" />
  <meta name="citation_author" content="Jakub Galgonek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-7038-544X" />
  <meta name="twitter:creator" content="@JGalgonek" />
  <meta name="citation_author" content="Daniel Mietchen" />
  <meta name="citation_author_institution" content="School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, Virginia 22904, United States" />
  <meta name="citation_author_orcid" content="0000-0001-9488-1870" />
  <meta name="twitter:creator" content="@EvoMRI" />
  <meta name="citation_author" content="Egon Willighagen" />
  <meta name="citation_author_institution" content="Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands" />
  <meta name="citation_author_orcid" content="0000-0001-7542-0286" />
  <meta name="twitter:creator" content="@egonwillighagen" />
  <meta name="citation_author" content="Arnaud Gaudry" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-3648-7362" />
  <meta name="citation_author" content="James G. Graham" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0002-7114-8921" />
  <meta name="citation_author" content="Ralf Stephan" />
  <meta name="citation_author_institution" content="Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada" />
  <meta name="citation_author_orcid" content="0000-0002-4650-631X" />
  <meta name="citation_author" content="Roderic Page" />
  <meta name="citation_author_institution" content="IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom" />
  <meta name="citation_author_orcid" content="0000-0002-7101-9767" />
  <meta name="twitter:creator" content="@rdmpage" />
  <meta name="citation_author" content="Jiří Vondrášek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-6066-973X" />
  <meta name="citation_author" content="Christoph Steinbeck" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-6966-0814" />
  <meta name="twitter:creator" content="@csteinbeck" />
  <meta name="citation_author" content="Guido F. Pauli" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1022-4326" />
  <meta name="citation_author" content="Jean-Luc Wolfender" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-0125-952X" />
  <meta name="citation_author" content="Jonathan Bisson" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1640-9989" />
  <meta name="twitter:creator" content="@Bjonnh" />
  <meta name="citation_author" content="Pierre-Marie Allard" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-3389-2191" />
  <meta name="twitter:creator" content="@NatprodCbn" />
  <link rel="canonical" href="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="og:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="twitter:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_pdf_url" content="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://lotusnprod.github.io/lotus-manuscript/v/21649de2f311fbd06652228489145cac25babfc6/" />
  <meta name="manubot_html_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/21649de2f311fbd06652228489145cac25babfc6/" />
  <meta name="manubot_pdf_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/21649de2f311fbd06652228489145cac25babfc6/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <meta property="og:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/21649de2f311fbd06652228489145cac25babfc6/content/images/thumbnail.png" />
  <meta property="twitter:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/21649de2f311fbd06652228489145cac25babfc6/content/images/thumbnail.png" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
bibliography:
- content/manual-references.json
manubot-output-bibliography: output/references.json
manubot-output-citekeys: output/citations.tsv
manubot-requests-cache-path: ci/cache/requests-cache
manubot-clear-requests-cache: false
...






<small><em>
This manuscript
([permalink](https://lotusnprod.github.io/lotus-manuscript/v/21649de2f311fbd06652228489145cac25babfc6/))
was automatically generated
from [lotusnprod/lotus-manuscript@21649de](https://github.com/lotusnprod/lotus-manuscript/tree/21649de2f311fbd06652228489145cac25babfc6)
on May 3, 2021.
</em></small>

## Authors



+ **Adriano Rutz** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-0443-9902](https://orcid.org/0000-0003-0443-9902)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [adafede](https://github.com/adafede)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [adafede](https://twitter.com/adafede)<br>
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
  </small>

+ **Maria Sorokina** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0001-9359-7149](https://orcid.org/0000-0001-9359-7149)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [mSorok](https://github.com/mSorok)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [ms_sorok](https://twitter.com/ms_sorok)<br>
  <small>
     Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
  </small>

+ **Jakub Galgonek** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-7038-544X](https://orcid.org/0000-0002-7038-544X)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [galgonek](https://github.com/galgonek)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [JGalgonek](https://twitter.com/JGalgonek)<br>
  <small>
     Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
  </small>

+ **Daniel Mietchen** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0001-9488-1870](https://orcid.org/0000-0001-9488-1870)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [Daniel-Mietchen](https://github.com/Daniel-Mietchen)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [EvoMRI](https://twitter.com/EvoMRI)<br>
  <small>
     School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, Virginia 22904, United States
  </small>

+ **Egon Willighagen** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0001-7542-0286](https://orcid.org/0000-0001-7542-0286)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [egonw](https://github.com/egonw)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [egonwillighagen](https://twitter.com/egonwillighagen)<br>
  <small>
     Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands
  </small>

+ **Arnaud Gaudry** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-3648-7362](https://orcid.org/0000-0002-3648-7362)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [ArnaudGaudry](https://github.com/ArnaudGaudry)<br>
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
  </small>

+ **James G. Graham** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-7114-8921](https://orcid.org/0000-0002-7114-8921)<br>
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
  </small>

+ **Ralf Stephan** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-4650-631X](https://orcid.org/0000-0002-4650-631X)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [rwst](https://github.com/rwst)<br>
  <small>
     Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada
  </small>

+ **Roderic Page** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-7101-9767](https://orcid.org/0000-0002-7101-9767)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [rdmpage](https://github.com/rdmpage)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [rdmpage](https://twitter.com/rdmpage)<br>
  <small>
     IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom
  </small>

+ **Jiří Vondrášek** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-6066-973X](https://orcid.org/0000-0002-6066-973X)<br>
  <small>
     Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
  </small>

+ **Christoph Steinbeck** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0001-6966-0814](https://orcid.org/0000-0001-6966-0814)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [steinbeck](https://github.com/steinbeck)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [csteinbeck](https://twitter.com/csteinbeck)<br>
  <small>
     Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
  </small>

+ **Guido F. Pauli** <br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-1022-4326](https://orcid.org/0000-0003-1022-4326)<br>
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
  </small>

+ **Jean-Luc Wolfender** ^[✉](#correspondence)^<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0002-0125-952X](https://orcid.org/0000-0002-0125-952X)<br>
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
  </small>

+ **Jonathan Bisson** ^[✉](#correspondence)^<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-1640-9989](https://orcid.org/0000-0003-1640-9989)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [bjonnh](https://github.com/bjonnh)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [Bjonnh](https://twitter.com/Bjonnh)<br>
  <small>
     Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
  </small>

+ **Pierre-Marie Allard** ^[✉](#correspondence)^<br>
    ![ORCID icon](images/orcid.svg){.inline_icon}
    [0000-0003-3389-2191](https://orcid.org/0000-0003-3389-2191)
    · ![GitHub icon](images/github.svg){.inline_icon}
    [oolonek](https://github.com/oolonek)
    · ![Twitter icon](images/twitter.svg){.inline_icon}
    [NatprodCbn](https://twitter.com/NatprodCbn)<br>
  <small>
     School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
  </small>


::: {#correspondence}
✉ — correspondence preferred via [GitLab Issues](https://gitlab.com/groups/lotus7/-/issues).
Otherwise, address correspondence to <jean-luc.wolfender@unige.ch>, <bjo@uic.edu>, and <pierre-marie.allard@unige.ch>.
:::


## Abstract {.page_break_before}

As bio- and chemoinformatics are reshaping natural products (NP) research, existing databases impose intrinsic and practical limits regarding access, chemical or taxonomic scope, field standardization, and interoperability. Further limitations result from essential but missing links to the primary literature, which contain the experimental information that documents structures, organisms and their relationships. Sharing such consolidated knowledge via an open platform has strong transformative potential for NP research. The LOTUS initiative has now completed the first steps towards the harmonization, curation, validation, and open dissemination of 500,000+ referenced structure-organism pairs. LOTUS data is hosted on Wikidata and regularly mirrored on [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net). Data sharing within the Wikidata framework broadens data access, chemical and taxonomic scope, and interoperability, thereby overcoming many of the limitations of existing electronic resources. Furthermore, embedding LOTUS data into the vast Wikidata knowledge graph facilitates new biological and chemical insights. This opens new possibilities for community curation of data and evolving publication models. All code developed within the LOTUS initiative for data gathering, curation, and dissemination are publicly available on [https://gitlab.com/lotus7](https://gitlab.com/lotus7) and [https://github.com/mSorok/LOTUSweb](https://github.com/mSorok/LOTUSweb). Collectively, LOTUS evolves NP knowledge management to function in a world of openly shared electronic resources.

![](images/graphicalAbstract.svg){graphical-abstract}


## Introduction {.page_break_before}

### Evolution of Electronic Natural Product Resources

Natural products (NP) research is a transdisciplinary field with scopes ranging from fundamental structural aspects of naturally-occuring molecular entities to their effects on living organisms, and even the study of chemically-mediated interactions within entire ecosystems. Technological and methodological developments are constantly reshaping NP research, a field steeped in history, with a rich legacy of experimental practices (Allard et al., 2018). In particular, contemporary bioinformatic approaches enable the (re-)interpretation and (re-)annotation of datasets documenting molecular aspects of biodiversity (Olivon et al., 2017). To efficiently annotate previously reported NP, or to identify and report new entities, these tools rely on properly maintained NP database (DB) (Tsugawa, 2018). Despite the ambiguous definition of “natural” (“All natural,” 2007), assuming that it is reasonable to consider a NP is a chemical entity found in a living organism, a future-oriented electronic NP resource should contain, at minimum, a list of chemical entities, organisms, and references to the work(s) establishing the links between them. If large, well-structured, and freely accessible DB, composed only of chemical structures (e.g. PubChem (Kim et al., 2019), with over 100M entries) or biological organisms (e.g. GBIF (“GBIF.org,” 2020), with over 1,400M entries) exists, their scarce interlinkage limits their applications. Currently, no open, cross-kingdom and comprehensive electronic NP resource links NP and their producing organisms, along with information about the experimental works describing those links. If referenced structure-organism pairs are critical evidence for pharmacognosy and related NP research, they remain poorly accessible (Cordell, 2017a).


Pioneering efforts to address such issues led to the establishment of KNApSAck (Shinbo et al., 2006), which is likely the first public curated electronic NP resource of referenced structure-organism pairs. KNApSAck currently contains over 50,000 structures and over 100,000 structure-organism pairs. However, the organism field is not standardized and downloads are complicated. The NAPRALERT dataset (Graham and Farnsworth, 2010), compiled over five decades, gathers annotated data derived from over 200,000 primary NP literature sources and contains 200,000 distinct compound names and structural elements, along with over 500,000 records of distinct structure-organism pairs. In total, it represents over 900,000 total records of pairs due to equivalent structure-organism pairs reported in different citations. NAPRALERT however uses an access model with limited free searches. Finally, the NPAtlas (Santen et al., 2019) is a recent project aimed at complying with the FAIR (Findability, Accessibility, Interoperability, and Reuse) guidelines for digital assets (Wilkinson et al., 2016) and offering web access. While the NPAtlas encourages submission of new compounds with their biological source, it focuses on microbial NP and ignores a wide range of biosynthetically active organisms, such as the plant kingdom.


The majority of available electronic NP resources provide entries without referencing their origin, thereby breaking the precious link required for tracing information back to the original data and assessing its quality. Even valuable efforts for compiling NP data made by commercial products, such as the Dictionary of Natural Products (DNP), are missing proper documentation of this critical information, thereby precluding further computational use or exhaustive review. 
Building on experience with the recently published COlleCtion of Open NatUral producTs (COCONUT) (Sorokina et al., 2021), the LOTUS project addresses these shortcomings. At its current stage of development, LOTUS features >500,000 distributed, referenced structure-organism pairs. After extensive data curation and harmonization, each documented structure-organism pair became standardized at the chemical, biological, and reference level. Collectively, LOTUS offers a high quality, computer-interpretable knowledge base. 

### Accommodating Principles of FAIRness and TRUSTworthiness

In awareness of the multi-facetted pitfalls associated with implementing, using, and maintaining classical scientific DBs (Helmy et al., 2016), and to enhance current and future sharing, the LOTUS project selected the Wikidata platform for disseminating its resources. Since its creation, Wikidata has focused on cross-disciplinary and multilingual support. It is curated and governed collaboratively by a global community of volunteers, about 20,000 of which are contributing monthly. Wikidata currently contains more than 1 billion statements in the form of subject-predicate-object triples. They are machine-interpretable and can be enriched with qualifiers and references. They correspond to ~90 million entries, which can be grouped into classes as diverse as countries, songs, disasters, or chemical compounds. They are closely integrated with Wikipedia and serve as the source for its infoboxes. Various workflows have been established for reporting such classes, particularly those of interest to life sciences, such as genes, proteins, diseases, drugs, or biological taxa (Waagmeester et al., 2020).
Building on the principles and experiences described above, this article introduces the development and implementation of a workflow for NP occurrences curation and dissemination using both FAIR and TRUST (Transparency, Responsibility, User focus, Sustainability, and Technology) principles (Lin et al., 2020). The presented data upload and retrieval procedures ensure optimal accessibility by the research community, allowing any researcher to contribute and reuse the data with a clear and open license (Creative Commons 0).


Despite many advantages, Wikidata hosting of the LOTUS project has some drawbacks. While the SPARQL query language offers a powerful way to query available data, it can also appear intimidating at first for an inexperienced user. Furthermore, some typical queries of molecular electronic NP resources such as structural or spectral search are not yet available in Wikidata. To bridge this gap, LOTUS is hosted in parallel at https://lotus.naturalproducts.net (LNPN) within the naturalproducts.net ecosystem. LNPN is periodically updated with the latest LOTUS data. The advantage of this dual hosting is that it yields an integrated,  community-curated, and vast knowledge base (via Wikidata), as well as a NP community-oriented product with tailored search modes (via LNPN). The LOTUS project and its multiple data interaction options provide the basis for transparent and sustainable ways to access, share, and create knowledge on NP occurrence. More broadly, the LOTUS project will foster cross-fertilization of the fields of chemistry and biology and associated disciplines.


### Structure of this Article 

The first section presents an overview of the LOTUS project at its current stage of development, keeping in mind that it is a constant work-in-progress. As part of this overview, the central curation and dissemination elements of the LOTUS project are explained in detail. The second section addresses the interaction between LOTUS and its end-users, including data retrieval, addition, and editing. The third section is dedicated to the interpretation of LOTUS data and illustrates the dimensions and qualities of the current LOTUS dataset from the chemical and biological perspectives.


## Results & Discussion {.page_break_before}

### Outline of the LOTUS Blueprint

Building on the standards established by each related Wikidata projects (chemistry (Wikidata:WikiProject Chemistry), taxonomy (Wikidata:WikiProject Taxonomy), and source metadata (Wikidata:WikiProject Source MetaData)), a NP chemistry oriented subproject was created (Wikidata:WikiProject Chemistry/Natural products). Its central data is constituted of 3 minimal sufficient objects: 

- A chemical compound object, with associated Simplified Molecular Input Line Entry System (SMILES) (Weininger, 1988), International Chemical Identifier (InChI) (Heller et al., 2013), and InChIKey (a hashed version of the InChI).
- A biological organism object, with associated taxon name, the taxonomic DB where it was described, and the taxon ID in this DB.
- A reference object describing the structure-organism pair, with the associated article title and a Digital Object Identifier (DOI), a PubMed ID (PMID), or a PubMed Central ID (PMCID).

As data formats are largely inhomogeneous among existing electronic NP resources, fields related to chemical structure, biological organism, and literature reference are variable and essentially not standardized. Therefore, LOTUS implements multiple stages of harmonization, cleaning, validation, and dissemination (Figure 1, stages 1 to 4). LOTUS employs the Single Source of Truth approach (SSOT, Single_source_of_truth) to ensure data reliability and continuous availability of the latest curated version of LOTUS data in both Wikidata and LNPN. The SSOT approach consists of a PostgreSQL DB that structures links and data schemes such that every data element is in a single place. To accommodate the addition of data directly from new data sources or at the Wikidata level), LOTUS’ processing pipeline is tailored to efficiently include and diffuse novel or curated data (Figure 1, stage 4). This iterative workflow relies both on data addition and retrieval actions described in the Data Interaction section. The overall process leading to referenced and curated structure-organisms pairs is illustrated in Figure @fig:workflow and detailed below.


![
**Overview of the workflow in the LOTUS project** The process consists of four stages: (1) Harmonization, (2) Cleaning, (3) Validation, and (4) Dissemination. The process was designed to incorporate (5) future contributions either in the form of new data addition (a and b) or via curation of existing data (c). This an iterative process that allows the community to participate in the global effort to document NP occurrences (the “virtuous circle” of NP).
](images/workflow.svg "Workflow"){#fig:workflow}

All stages of the process are described in the https://gitlab.com/lotus7 project and at https://github.com/mSorok/LOTUSweb. At the time of submission, over 742,000 LOTUS entries contained a curated chemical structure, biological organism, and reference and are available on both Wikidata and LNPN. As the LOTUS data volume is expected to increase over time, a frozen (as of 2021-02-23) tabular version of this dataset with its associated metadata is made available at https://osf.io/hgjdb/.

### Data Harmonization

Multiple data sources were processed and are described hereafter. All publicly accessible electronic NP resources included in COCONUT that contain referenced structure-organism pairs were processed. They were complemented with COCONUT’s own structure-organism documented pairs (Sorokina and Steinbeck, 2020a) and the following additional electronic NP resources: Dr. Duke (“U.S. Department of Agriculture, Agricultural Research Service. Dr. Duke’s Phytochemical and Ethnobotanical Databases” 1992), Cyanometdb (Jones et al., 2020), Datawarrior (López-López et al., 2019), a subset of NAPRALERT (Graham and Farnsworth, 2010), Wakankensaku (“Main Page - WAKANKENSAKU,” n.d.), and DiaNat-DB (Madariaga-Mazón, 2021). The owners of the electronic NP resources not explicitly licensed as open were individually contacted for permission to access and reuse data. The complete list of data sources and related information is available as SI-1. All necessary scripts for gathering and harmonization can be found in the lotusProcessor repository in the src/1_gathering directory. All subsequent iterations with additional data sources (either the updated versions of the same data sources or new ones), will start by comparing the new data sources with previously gathered ones at the SSOT level to curate data only once.

### Data Cleaning & Vaidation

As described in Figure 1, the data curation process was divided into four stages: harmonization, cleaning, and validation. Cleaning of each of the three central objects (chemical, biological, and reference) of the referenced pairs was performed. Given the data size with over 2.5M initial entries, manual validation was unfeasible. Curating the references turned out to be a particularly problematic point of the process. Whereas organisms are typically reported by at least their vernacular or scientific denomination and structures by their SMILES, InChI, InChIKey, or image (case not covered in this work), references suffer from largely insufficient reporting standards. Better reporting, together with new tools, such as Scholia (Nielsen et al., 2017; Rasberry et al., 2019), relying on Wikidata, Fatcat, or Semantic Scholar should lead towards improved references-related information retrieval in the future. Despite the poor standardization of the initial reference field, this latter is the only way to establish the validity of the structure-organism pair. Therefore, in addition to the entries curated during data processing, 420 referenced structure-organism pairs were analyzed manually to establish rules for automatic filtering of the entries. This filter was then applied to all entries. To confirm the efficacy of the filtering process, a subset of 100 diverse automatically curated and validated entries was manually checked, and showed a rate of 97% of true positives. The detailed results of the two manual validation steps are reported in Supporting Information SI-2. The resulting data is also available in the dataset shared at https://osf.io/hgjdb/.

| | Structure | Organism | Reference | 
|:--------:|:-------------:|:-------------:|:-------------:|
| **Before curation** | Cyathocaline | Stem bark of Cyathocalyx zeylanica CHAMP. ex HOOK. f. & THOMS. (Annonaceae) | Wijeratne E. M. K., de Silva L. B., Kikuchi T., Tezuka Y., Gunatilaka A. A. L., Kingston D. G. I., J. Nat. Prod., 58, 459-462 (1995). | 
| **After curation** |  VFIIVOHWCNHINZ-UHFFFAOYSA-N | Cyathocalyx zeylanicus | 10.1021/NP50117A020 | 

Table: Example of a given referenced structure-organism pair before and after curation {#tbl:examples}

Table 1 shows an example of a referenced structure-organism pair before and after curation, which resolved the structure to an InChIKey, the organism to a valid taxonomic name, and the reference to a DOI. Challenging examples encountered during the development of the curation process were compiled in an edge case table (tests/tests.tsv) to allow for automatic unit testing. These tests allow a continuous revalidation of any change made to the code, ensuring that no corrected error can reappear.

![
**Alluvial plot of the data transformation flow within the LOTUS project during the automated curation and validation processes**. The graph demonstrates how curation and validation of each informational object eventually yields curated information. The figure also reflects the relative proportions of the stream of data by showing the contributions from the various sources (“source” block), the composition of the harmonized subcategories (“original subcategory” block), and the validated data after curation (“cleaned category” block). Automatically validated entries are represented in green and rejected ones in blue.

](images/alluvial.svg "Alluvial"){#fig:alluvial}

The alluvial plot in Figure 2 illustrates the individual contribution of each source and original subcategory that led to the cleaned categories of structure, organism, and reference. For example, the graph highlights the important contribution of the DOI category of references contained in NAPRALERT towards the current set of validated references. The combination of the results of the automated curation pipeline and the manually curated entries led to the establishment of four categories (manually validated, manually rejected, automatically validated, and automatically rejected) of the documented structure-organism pairs that formed the processed part of the SSOT. Out of a total of more than 2M pairs, the manual and automatic validation retained 740,000+ pairs (~30 %), which were then selected for dissemination on Wikidata. The disseminated data contains 250,000+ unique chemical structures, 30,000+ distinct organisms, and 75,000+ references. 

### Data Dissemination

Researchers should benefit from all results of published scientific studies immediately upon publication. This is considered as the foundation of scientific investigation and a prerequisite for effectively directing new research efforts based on prior available information. To achieve this, research results have to be made publicly available and reusable. As computers are now the main instrument of any scientist, all research data including those in publications should be disseminated in computer-readable format and following the FAIR principles. LOTUS employs Wikidata as the repository for the referenced structure-organism pairs as this enables the documented research data to be integrated with a pre-existing vast body of chemical and biological knowledge. 


The dynamic nature of the Wikidata platform fosters the continuous curation of deposited data through its user communities. Independence from institutional funding represents another major advantage of Wikidata. Wikidata knowledge base and the option to use elaborated SPARQL queries allow the exploration of the dataset from a sheer unlimited number of angles. The openness of Wikidata also offers unique opportunities for community curation, which will support, if not guarantee, a dynamic and evolving data repository. At the same time, certain limitations of this approach can be anticipated. Despite their power, SPARQL queries are complex and often require a relatively in-depth understanding of the models and data structure. This involves a steep learning curve and can discourage end-users. Furthermore, traditional ways to query NP electronic NP resources such as structural or spectral searches are currently not within the scope of Wikidata. 


Using the pre-existing COCONUT template, LNPN hosting allows the user to perform structural searches in a more classical way, e.g., by drawing a molecule. It addresses the shortcomings of the current lack of structural searches in Wikidata for the community. Future versions of LOTUS and COCONUT are envisioned to be augmented by predicted MS spectra and are expected to be hosted at the naturalproducts.net portal to allow mass, fragment, and spectral-based queries.
To facilitate queries focused on specific taxa (e.g., “return all molecules found in the Asteraceae family”), a unified taxonomy is paramount. As the taxonomy of living organisms is a complex and always evolving field, all the taxon identifiers from all accepted taxonomic DB for a given taxon name were kept. This implies that for a given name, multiple taxonomies from different taxonomic DB are allowed. Initiatives such as the Open Tree of Life (OTL) (Rees and Cranston, 2017) will help to gradually reduce these discrepancies, the Wikidata platform should support this development. OTL also benefits from regular expert curation and new data. As the taxonomic identifier property for this DB did not exist in Wikidata, its creation was requested and obtained (P9157). 


Following the previously described curation process, all validated entries have been made available through Wikidata and LNPN. LNPN will be regularly mirroring Wikidata LOTUS through the SSOT as described in Figure 1. 


### Data Interaction from the User Point-of-view

With the data being available in multiple formats, the possibilities to interact with the LOTUS data are numerous. Some basic and more advanced examples on how to retrieve, add and edit LOTUS data are provided hereafter.

#### Data Retrieval

LOTUS data can be queried and retrieved either on Wikidata directly or on LNPN, both of these options presenting unique advantages. Wikidata offers modularity at the cost of potentially complex access to the data. LNPN offers a Graphical User Interface (GUI) with chemical structure drawing possibility, easy structural or biological filtering, and advanced chemical descriptors, but with a more rigid structure. A frozen (2021-02-23) version of LOTUS data is available at https://osf.io/zupqj/. Hereafter, finer approaches to directly interrogate the up-to-date LOTUS data both in Wikidata and LNPN are detailed.

##### Wikidata

The simplest way to search for NP occurrence information in Wikidata is by typing directly the name of a chemical structure in the “Search Wikidata'' field. For example by typing “erysodine” the user lands on the Wikidata page of this compound (Q27265641). Scrolling down to the “found in taxon” statement gives a view of the biological organisms reported to contain this chemical compound. Under each taxon name, clicking on the reference link will then display the scientific publication documenting the occurrence.

![
**Illustration of the “found in taxon” statement section on the Wikidata
page** of erysodine [Q27265641](https://www.wikidata.org/wiki/Q27265641) showing a selection of containing taxa and the references documenting these occurrences.
](images/erysodine.png "wdpage"){#fig:wdpage}

For more elaborated queries, the most efficient way is to write SPARQL queries using the Wikidata Query Service or a direct connection to the SPARQL endpoint. Below are some examples, from simple to more elaborated queries demonstrating what can be done using this approach. The full text queries are also available with some explanations in SI-3.


| Question | Wikidata SPARQL query | 
|:------------------------:|:-------------:|
| What are the compounds present in Mouse-ear cress (*Arabidopsis thaliana*)? | [https://w.wiki/32y8](https://w.wiki/32y8) | 
| Which organisms are known to contain compounds sharing the planar structure of β-sitosterol? | [https://w.wiki/334q](https://w.wiki/334q)  | 
| Which organisms are known to contain stereoisomers of β-sitosterol? | [https://w.wiki/334s](https://w.wiki/334s) | 
| Which pigments are found in which taxa, according to which reference? | [https://w.wiki/38Rt](https://w.wiki/38Rt) |
| What are examples of organisms where compounds were reported to be produced by an organism sharing the same parent taxon, but not the organism itself? | [https://w.wiki/3359](https://w.wiki/3359) |
| Which *Zephyranthes* species lack compounds known from at least two sister species? | [https://w.wiki/335x](https://w.wiki/335x) |
| How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in. | [https://w.wiki/32Qb](https://w.wiki/32Qb) |
| Which compounds are found in a biological organism, according to which references? | [https://w.wiki/335C](https://w.wiki/335C) |
| Which organisms contain indolic scaffold? Results are grouped by the parent taxon and ordered. | [https://w.wiki/32KZ](https://w.wiki/32KZ) |
| How many structure-organism pairs have been referenced by these authors? (Here, two senior natural products chemists and co-authors of this paper are compared to the late Ferdinand Bohlmann). | [https://w.wiki/32$m](https://w.wiki/32$m) |

Table: Example of a given referenced structure-organism pair before and after curation {#tbl:examples}

The queries presented in Table 2 are only selected examples and many other ways to interrogate LOTUS data can be formulated. A combination of these queries can be used, for example, for hypothesis generation when starting a research project.
For instance, a generic SPARQL query - listed in Table 2 as "Which compounds are found in a biological organism, according to which references?" - retrieves all chemical compounds (Q11173) or groups of stereoisomers (Q59199015) with found in taxon (P703) statements supported by a bibliographic reference (Q10358455): https://w.wiki/335C. Data can then be exported in various formats, such as classical tabular formats, json or html tables. At the time of publication, it returned 798,853 entries. A frozen result of the query is available at https://osf.io/xgyhm/. 
Targeted queries allowing to interrogate LOTUS data from the perspective of one of the three objects forming the referenced structure-organism pairs can be also built. Users can, for example, retrieve a list of all reported structures in a given organism (e.g., structures found in Citrus aurantium (Q61127949) https://w.wiki/sFp). Alternatively, all organisms containing a given chemical structure can be queried (e.g., here all organisms in which β-sitosterol (Q121802) was reported https://w.wiki/dFz). For programmatic access, the WikidataLotusExporter repository also allows retrieval in RDF format and as tsv tables. As previously mentioned, some typical queries of molecular electronic resources such as structure or similarity search are not directly available in Wikidata. The SPARQL language does not natively support a simple integration of such queries. To address this issue, Galgonek et al. developed an in-house SPARQL engine that allows utilization of Sachem, their high-performance chemical DB cartridge for PostgreSQL for fingerprint-guided substructure and similarity search (Kratochvíl, 2018). The engine is used by the Integrated Database of Small Molecules (IDSM) that operates, among other things, several dedicated endpoints allowing structural search in selected small-molecule datasets via SPARQL (Kratochvíl et al., 2019). To allow substructure and similarity searches via SPARQL also on compounds from Wikidata, a dedicated IDSM/Sachem endpoint was created for this project. An example of a query containing substructure search is given in Table 2 (https://w.wiki/32KZ). The endpoint indexes isomeric (P2017) and canonical (P233) SMILES code available in Wikidata. To ensure that data are kept up-to-date, SMILES codes are downloaded from Wikidata automatically daily. The endpoint allows users to run federated queries and thus proceed to structure-oriented searches on the LOTUS data hosted at Wikidata. For example, the following SPARQL query, https://w.wiki/32KZ, will return a list of all organisms producing compounds with an indolic scaffold. The list is aggregated at the parent taxa level of the containing organisms and ranked by the number of scaffold occurrences.

##### Lotus.NaturalProducts.Net (LNPN)

In the search field of the LNPN interface, simple queries can be achieved by typing in the molecule name (e.g. protopine) or pasting a SMILES, InChI, InChIKey string, or a Wikidata identifier. All compounds found in a given organism can be found by typing the organism name at the species or any higher taxa level (e.g. Tabernanthe iboga). The compound search by chemical class is also enabled in the same way.
Alternatively, a structure can be directly drawn in the structure search interface (https://lotus.naturalproducts.net/search/structure), where the user can also decide on the nature of the structure search (exact, similarity, substructure search). Refined search mode combining multiple search criteria, in particular physicochemical properties, is available in the advanced search interface (https://lotus.naturalproducts.net/search/advanced).
From LNPN the bulk data can be retrieved as an SDF or SMILES file, or as a complete MongoDB dump via https://lotus.naturalproducts.net/download. Extensive documentation, describing the search possibilities and data entries is available at https://lotus.naturalproducts.net/documentation. LNPN can also be queried programmatically via the API, which is also described in the documentation.

#### Data Addition

One strong advantage of LOTUS consists in the possibility given to users to contribute to the NP occurrences documentation effort by adding new data or editing existing data. As all of the LOTUS data is stored in the SSOT, it is also used to avoid reprocessing of previously treated elements. However, at the moment, the SSOT is not open for direct write access to the public to maintain its coherence and allow us to evolve the scheme. To add or modify data in LOTUS, the users can employ the following approaches.

##### Sources

The LOTUS process will regularly re-import both the current sources and new ones. New and edited information from these electronic NP resources will be checked against the SSOT and if not present or different, it will follow the curation pipeline and will be further stored into the SSOT. Any researcher can, thus, contribute to these external electronic NP resources as means of providing new data for LOTUS, keeping in mind the delay between data addition and subsequent inclusion into LOTUS.

##### Wikidata

The currently favored approach to add new data to LOTUS is to edit Wikidata directly. This data will then be automatically imported into the SSOT DB. There are several ways to interact with Wikidata which depend on the technical skills of the user and the volume of data to be imported/modified.

###### Manual Upload

Any researcher interested in reporting NP occurrences can manually add the data directly in Wikidata, without any particular technical knowledge requirement. The only prerequisite is to have a Wikidata account and to follow the general object editing guidelines. Regarding the addition of NP-centered objects (i.e., documented structure-organisms pairs) refer to the WikiProject Chemistry/Natural products group page. 
A tutorial for the manual creation and upload to Wikidata of a documented structure-organism pair is available in SI-3. While direct Wikidata upload is possible, contributors are encouraged to use the LOTUS curation pipeline as a preliminary step to strengthen the initial data quality. The added data will therefore benefit from the curation and validation stages implemented in the LOTUS processing pipeline.

###### Batch and Automated Upload

At the end of the previously described initial curation process, more than 500,000 referenced structure-organism pairs were validated for uploading on Wikidata. To automate the Wikidata upload process, a set of programs were written to automatically process the curated outputs, group references, organisms, and compounds, check if they are already present in Wikidata (using SPARQL and direct Wikidata querying), and insert or update the entities as needed (i.e., upserting). These scripts can be used for batch upload of properly curated and referenced structure-organism pairs to Wikidata. ScriptsPrograms for data addition to Wikidata can be found in the repository WikidataLotusImporter. The following Xtools page offers an overview of the latest activity performed by our NPimporterBot, using those programs.

#### Data Edition

Even if correct at a given time point, scientific advances can invalidate or correct previously discovered data. Thus, the possibility to continuously edit the data is desirable and guarantees data quality and sustainability. Community-maintained knowledge bases such as Wikidata encourage such a process. Wikidata presents the advantage of allowing both manual and automated correction. Field-specific robots such as SuccuBot, KrBot, Pi_bot, and ProteinBoxBot or our NPImporterBot went through an approval process. They often do thousands of editions without any need for human input. This helps to reduce automatically the amount of incorrect data that would otherwise require manual editing. However, manual curation by human experts remains the highest standard. Valuing this, interested users are invited to follow the manual curation tutorial in SI-4. The Scholia platform currently does not offer a user-friendly edition interface for scientific references. The adaptation of such a framework to edit the documented structure-pairs in the LOTUS project could facilitate future expert curation, including manual efforts that cannot be replaced by automated scripts or AI efforts.

### Data Interpretation

To illustrate the nature and dimensions of the LOTUS dataset some selected examples of data interpretation are shown. First, the distribution of biological organisms according to the number of related chemical structures and likewise the distribution of chemical structures across biological organisms are illustrated (Figure 4). Then individual electronic NP resources participation to LOTUS data is resumed using the upset plot depiction, which allows the visualization of dataset intersection (Figure 5). In these two previous examples, the cases of β-sitosterol for the chemical structure and of Arabidopsis thaliana for the biological organism, are explored to provide well-documented entries to the reader. Finally, a biologically-interpreted chemical tree and a chemically-interpreted biological tree are presented (Figure 6 and Figure 7). They illustrate the overall chemical and biological coverage of LOTUS by linking family-specific classes of chemical structures to their taxonomic position. Figure 4, Figure 6, and Figure 7 were generated using the frozen data (as of 2021-02-23) available for download at the following link: https://osf.io/hgjdb/. Figure 5 required a dataset containing information from the commercial DNP and the complete data used for its generation is therefore unfortunately not available for public distribution. All scripts used for the generation of the figures (including SI-5) are available in the lotusProcessor repository in the src/4_visualizing directory for reproducibility purposes.

#### Organisms per Structure and Structure per Organisms Distribution

As depicted in Figure 4, three organisms on average are reported per chemical structure and eleven structures per organism. Half of the structures are reported in only one organism and five structures or fewer are reported in half of the organisms. Metabolomics studies suggest that these numbers are clearly underrated (Noteborn et al., 2000; Wang et al., 2019) and these numbers suggest that a better reporting of the metabolites during a phytochemical investigation could greatly improve coverage. This incomplete coverage may partially be explained by the fact that, usually, only newly described or bioactive structures are accepted for publication in classical NP research journals.

![
**Distribution of "structures found in organisms'' and "organisms containing
structures"**. Number of organisms linked to the 2D structure of β-sitosterol (KZJWDPNRJALLNS) and the chemical diversity of Arabidopsis thaliana are highlighted as two notable examples.
](images/distribution.svg "distribution"){#fig:distribution}

#### Individual Electronic NP Resources Contribution to LOTUS

The added value of assembling multiple NP electronic NP resources in LOTUS is illustrated in Figure 5, showing the individual electronic NP resources contribution to the ensemble of chemical structures found in Arabidopsis thaliana (“Mouse-ear cress”; Q147096) (A) and to the ensemble of taxa containing the two-dimensional structure corresponding to β-sitosterol (Q121802) and (Q63409374) (B), a compound of ubiquitous occurrence in higher plants.

![
**Upset plots of the individual electronic NP resource contribution** to 2D structures found in Arabidopsis thaliana (A) and organisms containing the 2D structure of β-sitosterol (KZJWDPNRJALNS) (B). Upset plots are evolved Venn diagrams, allowing to represent intersections between multiple sets. The horizontal bars on the lower left represent the number of corresponding entries per electronic NP resource. The dots and their connecting line represent the intersection between two sets. The vertical bars indicate the number of entries in the intersection. For example, 479 organisms containing the structure of β-sitosterol are present in both UNPD and NAPRALERT, which in turn, respectively report 1164 and 2085 organisms containing the structure of β-sitosterol.
](images/upset_combined.svg "upset"){#fig:upset}

Figure 5 A. shows that the chemical pathways distribution (according to NPClassifier (Kim et al., n.d.) across electronic NP resources is not conserved. Note that being specially tailored for NP, NPClassifier was preferred over ClassyFire (Feunang et al., 2016) but both chemical taxonomies are available as metadata in the frozen LOTUS export (https://osf.io/hgjdb/) and LNPN. Both classification tools return a chemical taxonomy for individual structures, thus allowing their grouping at higher hierarchical levels, in the same way as it is done for biological taxonomies. This upset plot indicates the poor overlap of preexisting electronic NP resources and the added value of an aggregated dataset. This is also illustrated in Figure 5 B., where the number of organisms for which the 2D structure of β-sitosterol (KZJWDPNRJALLNS) has been reported for each intersection is shown. NAPRALERT has by far the highest number of entries (2085 in total), while other electronic NP resource complement this well (UNPD, for example, has 532 organisms where β-sitosterol is reported that are not overlapping with the ones reported in NAPRALERT). Interestingly, β-sitosterol is documented in only 3 organisms in the DNP, highlighting the importance of a better systematic reporting of ubiquitous metabolites and the interest of multiple data sources agglomeration.

#### Biologically-interpreted Chemical Tree

LOTUS chemical diversity can be visualized via a TMAP (Figure 6), a visualization method allowing the structural organization of very large chemical datasets as a two-dimensional tree [Probst et al 2020 TMAP]. Using Faerun, an interactive HTML file is generated to display metadata and molecule structures (using the Java-Script library SmilesDrawer) [Probst et al 2018 FUN + Probst et al 2018 SmilesDrawer]. It is to be noticed that planar structures for all compounds (with MAP4 encoding) were used for the TMAP generation since LOTUS contains a mix of planar and 3D structures [Capecchi et al 2020]. As a result, the same planar structure may be present multiple times in the TMAP if this structure was reported with different stereochemistry and/or as a planar structure. Since structures are organized in the tree according to their molecular fingerprint, a good clustering of compounds of the same NPClassifier chemical class can be observed (Figure 6 A.). For clarity reasons, the eight most represented chemical classes of LOTUS plus the Quassinoids and Carotenoids (C40, β-β) classes are mapped, with an example of a quassinoid (yellow star) and a carotenoid (green star) with their corresponding location in the TMAP.

![
**TMAP visualizations of the chemical diversity present in LOTUS** where each dot corresponds to a chemical structure. Dots are colored according to the NPClassifier chemical class they belong to (A), the most frequently reported biological family (Simaroubaceae or other) (B), and to the specificity of chemical classes to the containing biological organisms (C). A quassinoid (yellow star) and a carotenoid (C40, β-β) (green star) are mapped as an example in all visualizations. The biological specificity at a given taxonomic level for a given structure is defined as the number of occurrences of the most frequently reported taxon divided by the number of biosources reported. For clarity in the plotting, the average specificity of the structure’s NPClassifier class is displayed and values below 0.8 have been set to 0.8. An average chemical class biological specificity of 1 at the family level means that compounds in that NPClassifier class were only reported in one family. An interactive HTML visualization of the LOTUS TMAP is available at https://osf.io/kqa8b/.
](images/tmap.svg "tmap"){#fig:tmap}

To explore relationships between chemistry and biology, it is possible to map biological information such as the most reported biological family (Figure 6 B.) and the chemical class biological specificity (Figure 6 C.) on the TMAP. These visualizations allow to highlight chemical classes specific to a given taxon, such as the Quassinoids in the Simaroubaceae family example. It is striking to see how well in that case the chemical class (Quassinoids) and the most reported biological family (Simaroubaceae) overlap, highlighting a very specific chemical class (specificity of 0.97 at the biological family level). On the other hand, it is also possible to identify classes that are widely spread among living organisms, such as the Carotenoids (C40, β-β) class (specificity of 0.79 at the biological family level).

#### Chemically-interpreted Biological Tree

A summary of the biological and chemical diversity covered by LOTUS is illustrated in Figure 7. To limit biases due to underreporting while keeping a reasonable display size, only families with at least 50 reported compounds were kept for this illustration. Organisms were classified according to the OTL taxonomy and structures according to NPClassifier. The tips were labeled according to the biological family and colored according to their biological kingdom. The bars represent structure specificity of the most characteristic chemical class of the given biological family (the higher the more specific), calculated as the square of the number of structures reported in the chemical class within the given family, over the product of the number of reported structures in the chemical class and the number of reported structures in the biological family.

![
**Chemical and biological diversity present in LOTUS**. The tree corresponds to the biological taxonomy, with the kingdom as label color. The size of the leaves node corresponds to the number of genera reported in the family. The outer bars correspond to the most specific chemical class found in the biological family. The height of the bar is proportional to a specificity score corresponding to the square of the number of structures reported in the chemical class within the given family, over the product of the number of reported structures with the number of reported structures in the biological family. The bar’s color corresponds to the chemical pathway (NPClassifier classification system) of the most specific chemical class. 
](images/magicTree.svg "magicTree"){#fig:magicTree}

In Figure 7, it is possible to spot highly specific compound classes such as trinervitane terpenoids in the Termitidae, rhizoxin macrolides in the Rhizopodaceae, or typical quassinoids and limonoids from the Simaroubaceae and Meliaceae, respectively. More generic tendencies can also be observed. For example, within the fungal kingdom, Basidiomycotina appear to have a higher biosynthetic specificity toward terpenoids than other fungi, which mostly focus on polyketides production. When observed at a finer scale (down to the structure level), such chemotaxonomic representation can give valuable insights. For example, among all chemical structures, only two were found in all biological kingdoms, namely heptadecanoic acid (KEMQGTRYUADPNZ-UHFFFAOYSA-N) and β-carotene (OENHQHLEOONYIE-JLTXGRSLSA-N). Looking at the repartition of the β-sitosterol (KZJWDPNRJALLNS-VJSFXXLFSA-N) within the overall biological tree, SI-5 plots its presence/absence vs those of its superior chemical classifications, namely the stigmastane, steroid and terpenoid derivatives, over the same tree used in Figure 7. The comparison of these five chemically-interpreted biological trees clearly highlights the increasing speciation of the β-sitosterol biosynthetic pathway in the Archaeplastida kingdom, while the superior classes were distributed across all kingdoms. As illustrated, the possibility of data interrogation at multiple precision levels (from fully defined 3D structures to broader chemical classes) is of great interest, e.g., for taxonomic and evolution studies.


As shown recently in the context of spectral annotation (Dührkop et al., 2020), lowering the precision level of the annotation allows a broader coverage together with greater confidence. Genetic studies investigating the pathways involved and the organisms carrying the responsible biosynthetic genes responsible would be of interest to confirm the previous observations. These forms of data interpretation exemplify the importance of reporting not only new structures but also novel occurrences of known structures in organisms as comprehensive chemotaxonomic studies are pivotal for a better understanding of the metabolomes of living organisms.


## Conclusion & Perspectives {.page_break_before}

At its current development stage, and despite its diligent assembly, data compiled in LOTUS remains imperfect and, by its nature, at least partially biased. As discussed above, in the context of bioactive NP research and due to global editorial practices, publications tend to emphasize new compounds for which interesting bioactivity has been measured. In contrast, ubiquitous compounds are poorly documented. For the time being, until more rigorous data becomes available, this provides a partial yet relatively comprehensive view on the actual metabolome of all studied organisms, including the more thoroughly studied model organisms. The comprehensive editing opportunities provided within LOTUS and by the associated Wikidata distribution platform open new opportunities for collaborative community engagement, which the authors believe is the most efficient means of correcting any remaining bias.


Moreover, the dissemination of referenced structure-organism pairs through Wikidata, together with the harmonized data format enables query of an increasingly globalized NP research knowledge from a radically novel perspective. All researchers involved in NP research and special aspects of metabolism can benefit from this opportunity, whether the interest is in ecology and evolution, chemical ecology, drug discovery, biosynthesis pathway elucidation, chemotaxonomy, or similar research fields.


The introduction of LOTUS also provides a new opportunity to advance the FAIR guiding principles for scientific data management and stewardship established in 2016 (Wilkinson et al., 2016). Researchers widely acknowledge the availability of data that has been unavailable, and at least in part even been discouraged, e.g., due to page limitations and mechanisms involved in the calculation of journal impact factors) by the classical print and static PDF publication formats. In particular raw data such as experimental readings, spectroscopic data, instrumental measurements, statistical, and other calculations are valued by all, but disseminated by only very few. The immense value of raw data and the desire to advance the public dissemination has recently been documented in detail for NMR spectroscopic data by a large consortium of NP researchers (McAlpine et al., 2019). While first steps in this direction are currently being taken (for example: Harvard Dataverse repositories for individual research groups [Pauli group at UIC: dataverse.harvard.edu/dataverse/gfpuic] and projects [dataverse.harvard.edu/dataverse/cenapt]; an NIH-funded raw NMR data repository [www.np-mrd.org/]), one missing element in this multifaceted development is the adequate incentivization of community efforts; to generate the vital flow of contributed data, the effort associated with preparing and submitting content to open repositories should be better acknowledged in academia, government, regulatory, and industrial environments. Moreover, data reuse should also be acknowledged (Cousijn et al., 2018, 2019; Pierce et al., 2019).


The possibilities for expansion and future applications of Wikidata-hosted LOTUS data are significant. For example, properly formatted spectral data, e.g. data obtained by mass spectrometry or nuclear magnetic resonance, can be linked to the Wikidata entries for the respective chemical compounds. MassBank (Horai et al., 2010) and SPLASH (Wohlgemuth et al., 2016) identifiers are already reported in Wikidata, and this existing information can be used to report MassBank records for single organisms such as Arabidopsis thaliana compounds (https://w.wiki/335H). Such possibilities will help to bridge experimental data results obtained during the early stages of NP research with data that has been reported and formatted previously in different content. This will open exciting new perspectives in the area of structural dereplication and NP annotation. The authors have previously demonstrated that taxonomically-informed metabolite annotation can critically improve the NP annotation process (Rutz et al., 2019). Alternative approaches linking structural annotation to biological organisms have also shown tremendous improvements (Hoffmann et al., 2021). The availability of LOTUS as an open repository that links chemical objects to both their spectral information and biological occurrences will facilitate such applications significantly.


As shown in SI-5, observing the chemical and biological diversity at various granularity levels can offer new insights. Regarding the chemical objects involved, it will be important to implement chemical taxonomy annotations for the Wikidata entries. However, this is a rather complex task, for which stability and coverage issues will have to be addressed first. Existing chemical taxonomies such as ChEBI, ClassyFire, or NPClassifier are evolving steadily, and it will be important to constantly update the tools used to make further annotations. Repositioning NP within their greater biosynthetic context is another major challenge. The fact that LOTUS is disseminated on Wikidata should facilitate its integration within biological pathway DB such as WikiPathways and contribute to this complex task (Martens et al., 2020; Slenter et al., 2018).
In the field of ecology, molecular traits are gaining increased attention (Kessler and Kalske, 2018; Sedio, 2017). Conceptually, LOTUS can help associate classical plant traits (e.g., leaf surface area, photosynthetic capacities, etc.) with Wikidata biological organisms entries, and thus, allow the integration and comparison with chemicals that are associated with the organisms. Likewise, the association of biogeography data documented in repositories such as GBIF could be further exploited in Wikidata to pursue the exciting but understudied topic of “chemodiverse hotspots” (Defossez et al., 2021).


Other NP-related information is of great interest but unfortunately remains poorly formatted. For example, traditional medicine (including ethnomedicine and ethnobotany) is the historical and empiric approach of mankind to discover and use bioactive products from Nature, primarily plants. The amount of knowledge generated in human history on the use of medicinal substances represents fascinating yet underutilized information. To this end, LOTUS represents a resource for new concepts by which such information could be valued and conserved in the digital era, as it provides a blueprint for appropriate formatting and sharing of such data (Allard et al., 2018; Cordell, 2017b, 2017a).


The various facets discussed above represent future developments that the combination of LOTUS and the Wikidata knowledge base can feasibly accommodate. Behind the scenes, all underlying resources represent data in a multidimensional space and can be extracted as individual graphs that can be interconnected. The craft of appropriate federated queries allows users to navigate these spaces and graphs and fully exploit their potential (Kratochvíl, 2018; Waagmeester et al., 2020). The development of interfaces such as RDFFrames (Mohamed et al., 2020) will also facilitate the use of the wide arsenal of existing machine learning approaches to automate reasoning on these knowledge graphs - efficiently complementing human interpretation, validation, and thinking.


Overall, the LOTUS project is intended to give the pharmacognosy et al. research community access to a greater quantity and better quality of data and, thereby, ultimately pave the way towards a global open electronic NP resource. This reflects the authors’ belief that the integration of NP research results requires a truly open and FAIR knowledge-based in the form of an electronic NP resource structure, and this approach will fuel a virtuous (rather than a vicious) cycle of research habits aiming at a better understanding of Life and its chemistry.


## Methods {.page_break_before}

### Data Curation

#### Gathering

Before their inclusion, the overall quality of source was manually assessed to estimate the quality of referenced structure-organism pairs and the lack of ambiguities in the links between data and references. This led to the identification of thirty-six electronic NP resources as valuable LOTUS input. Data from the proprietary Dictionary of Natural Products (DNP v 29.1) was also used for comparison purposes only and is not publicly disseminated. FooDB was also curated but not publicly disseminated since its license did not allow sharing in Wikidata. SI-1 gives all necessary details regarding electronic NP resources access and characteristics.


Manual inspection of each electronic NP resource revealed that the structure, organism, and reference fields were widely variable in format and contents, thus requiring standardization to be comparable. The initial stage consisted of writing tailored scripts that are capable of harmonizing and categorizing knowledge from each source (Fig. 2). This transformative process led to three categories: fields relevant to the chemical structure described, to the producing biological organism, and the reference describing the occurrence of the chemical structure in the producing biological organism. This process resulted in categorized columns for each source, providing an initial harmonized format for each table.


For all thirty-eight sources, if a single file or multiple files were accessible via a download option including FTP, data was gathered that way. For some sources, data was scraped (cf. SI-1). All scraping scripts written to automatically retrieve entries can be found in the lotusProcessor repository in the src/1_gathering directory (under each respective subdirectory). Data extraction scripts for the DNP are available and should allow users with a license to further exploit the data (src/1_gathering/db/dnp). The chemical structure fields, organism fields, and reference fields were manually categorized into three, two, and ten subcategories, respectively. For chemical structures, ”InChI”, “SMILES”, and “chemical name” (not necessarily IUPAC). For organisms, “clean” and “dirty”, meaning lot text not referred to the canonical name was present or the organism was not described by its canonical name. For the references, the original reference was kept in the “original” field. When the format allowed it, references were divided into: “authors”, ”doi”, ”external”, ”isbn”, ”journal”, “original”, ”publishing details”, “pubmed”, “title”, “split”. The generic “external” field was used for all external cross-references to other websites or electronic NP resources (for example, “also in knapsack”). The last subcategory, “split”, corresponds to a still non-atomic field after the removal of parts of the original reference. Other field titles are self-explanatory. The producing organism field was kept as a single field.


#### Harmonization

To perform the harmonization of all previously gathered sources, sixteen columns were chosen as described above. Upon electronic NP resources harmonization, resulting subcategories were divided and subject to further cleaning. The “chemical structure” fields were divided into files according to their subcategories (“InChI”, “names” and “SMILES”). A file containing all initial structures from all three subcategories was also generated. The same procedure was followed for organisms and references.

#### Cleaning

To obtain an unambiguously referenced structure-organism pair for Wikidata dissemination, the initial sixteen columns were translated and cleaned into three fields: the reported structure, the organism canonical name, and the reference. The structure was reported as InChI, together with its SMILES and InChIKey translation. The biological organism field was reported as three minimal necessary and sufficient fields, namely its canonical name and the taxonID and taxonomic DB corresponding to the latter. The reference was reported as four minimal fields, namely reference title, DOI, PMCID, and PMID, one being sufficient. For the forthcoming translation processes, automated solutions were used when available. However, for specific cases (common or vernacular names of the biological organisms, Traditional Chinese Medicine (TCM) names, and conversion between digital reference identifiers), no solution existed, thus requiring the use of tailored dictionaries. The initial entries (containing one or multiple producing organisms per structure, with one or multiple accepted names per organism) were cleaned into over 2M referenced structure-organism pairs.

##### Chemical Structures

To retrieve as much information as possible from the original structure field(s) of each of the sources, the following procedure was followed. Allowed structural fields for the sources were divided into two types: structural (InChI, SMILES) or nominal (chemical name, not necessarily IUPAC). If multiple fields were present, structural identifiers were preferred over structure names. Among structural identifiers, when both identifiers led to different structures, InChI was preferred over SMILES. SMILES were translated to InChI using the RDKit (2020.03.3) implementation in Python 3.8 (src/2_curating/2_editing/structure/1_translating/smiles.py). They were first converted to ROMOL objects which were then converted to InChI. When no structural identifier was available, the nominal identifier was translated to InChI first thanks to OPSIN (Lowe et al., 2011), a fast Java-based translation open-source solution. If no translation was obtained, chemical names were then submitted to the CTS (Wohlgemuth et al., 2010), once in lower case only, once with the first letter capitalized. If again no translation was obtained, candidates were then submitted to the Chemical Identifier Resolver via the cts_convert function from the webchem package (Szöcs et al., 2020). Before the translation process, some typical chemical structure-related greek characters (such as α, ß) were replaced by their textual equivalents (alpha, beta) to obtain better results. All pre-translation steps are included in the preparing_name function and are available in src/r/preparing_name.R.


The chemical sanitization step sought to standardize the representation of chemical structures coming from different sources. It consisted of three main stages (standardizing, fragment removal, and uncharging) achieved via the MolVS package. The initial standardizer function consists of six stages (RDKit Sanitization, RDKit Hs removal, Metals Disconnection, Normalization, Acids Reionization, and Stereochemistry recalculation) detailed in the molvs documentation. In a second step, the FragmentRemover functionality was applied using a list of SMARTS to detect and remove common counterions and crystallization reagents sometimes occurring in the input DB. Finally, the Uncharger function was employed to neutralize molecules when appropriate. 


MarvinSuite was used for traditional and IUPAC names translation, Marvin 20.19, ChemAxon. When stereochemistry was not fully defined, (+) and (-) symbols were removed from names. All details are available in the following script: src/2_curating/2_editing/structure/4_enriching/naming.R.
Chemical classification of all resulting structures was done using classyfireR (Feunang et al., 2016) and NPClassifier API.


From the 269,436 initial InChI, 233,652 (87%) sanitized structures were obtained, of which 176,235 (75%) had complete stereochemistry defined. 197,392 (73%) were uploaded to Wikidata.
From the 244,724 initial SMILES, 205,244 (84%) sanitized structures were obtained, of which 8,542 (4%) had complete stereochemistry defined. 173,590 (71%) were uploaded to Wikidata.
From the 49,315 initial chemical names, 27,780 (56%) sanitized structures were obtained, of which 2,324 (8%) had complete stereochemistry defined. 23,098 (47%) were uploaded to Wikidata.
In total, 164,308 structures with fully defined stereochemistry were uploaded as “chemical compounds” (Q11173), and 106,043 structures without fully defined stereochemistry were uploaded as “group of stereoisomers” (Q59199015) .


##### Biological Organisms

The cleaning process at the biological organism’s level had three objectives: convert the original organism string to (a) taxon name(s), atomize fields containing multiple taxon names, and deduplicate synonyms. The original organism strings were treated with Global Names Finder (GNF) and Global Names Verify (GNV), both tools coming from the Global Names Architecture (GNA) a system of web services that helps people to register, find, index, check and organize biological scientific names and interconnect on-line information about species. GNF allows scientific name recognition within raw text blocks and searches for found scientific names among public taxonomic DB. GNV takes names or lists of names and verifies them against various biodiversity data sources. Canonical names, their taxonID, and the taxonomic DB they were found in were retrieved. When a single entry led to multiple canonical names (accepted synonyms), all of them were kept. Because both GNF and GNV recognize scientific names and not common ones, common names were translated before a second resubmission.

###### Dictionaries

To perform the translations from common biological organism name to latin scientific name, specialized dictionaries included in DrDuke, FooDB, PhenolExplorer were aggregated together with the translation dictionary of GBIF Backbone Taxonomy. The script used for this was src/1_gathering/translation/common.R. When the canonical translation of a common name contained a specific epithet that was not initially present, the translation pair was discarded (for example, “Aloe” translated in “Aloe vera” was discarded). Common names corresponding to a generic name were also discarded (for example “Kiwi” corresponding to the synonym of an Apteryx spp. (https://www.gbif.org/species/4849989)). When multiple translations were given for a single common name, the following procedure was followed: the canonical name was split into species name, genus name, and possible subnames. For each common name, genus names and species names were counted. If both the species and genus names were consistent at more than 50%, they were considered consistent overall and, therefore, kept (for example, “Aberrant Bush Warbler” had “Horornis flavolivaceus” and “Horornis flavolivaceus intricatus” as translation; as both the generic (“Horornis”) and the specific (“flavolivaceus”) epithets were consistent at 100%, both (“Horornis flavolivaceus”) were kept). When only the generic epithet had more than 50% consistency, it was kept (for example, “Angelshark” had “Squatina australis” and “Squatina squatina” as translation, so only “Squatina” was kept). Some unspecific common names were removed (see https://osf.io/gqhcn/) and only common names with more than three characters were kept. This resulted in 181,891 translation pairs further used for the conversion from common names to scientific names. For TCM names, translation dictionaries from TCMID, TMMC, and coming from the Chinese Medicine Board of Australia were aggregated. The script used for this was src/1_gathering/translation/tcm.R. Some unspecific common names were removed (see https://osf.io/zs7ky/). Careful attention was given to the Latin genitive translations and custom dictionaries were written (see https://osf.io/c3ja4/, https://osf.io/u75e9/). Organ names of the producing organism were removed to avoid wrong translation (see https://osf.io/94fa2/). This resulted in 7070 translation pairs. Both common and TCM translation pairs were then ordered by decreasing string length, first translating the longer names to avoid part of them being translated incorrectly.

###### Translation

To ensure compatibility between obtained taxonID with Wikidata, the taxonomic DB 3 (ITIS), 4 (NCBI), 5 (Index Fungorum), 6 (GRIN Taxonomy for Plants), 8 (The Interim Register of Marine and Nonmarine Genera), 9 (World Register of Marine Species), 11 (GBIF Backbone Taxonomy), 12 (Encyclopedia of Life), 118 (AmphibiaWeb), 128 (ARKive), 132 (ZooBank), 147 (Database of Vascular Plants of Canada (VASCAN)), 148 (Phasmida Species File), 150 (USDA NRCS PLANTS Database), 155 (FishBase), 158 (EUNIS), 163 (IUCN Red List of Threatened Species), 164 (BioLib.cz), 165 (Tropicos - Missouri Botanical Garden), 167 (The International Plant Names Index), 169 (uBio NameBank), 174 (The Mammal Species of The World), 175 (BirdLife International), 179 (Open Tree of Life), 180 (iNaturalist) and 187 (The eBird/Clements Checklist of Birds of the World) were chosen. All other available taxonomic DB are listed at http://index.globalnames.org/datasource. To retrieve as much information as possible from the original organism field of each of the sources, the following procedure was followed: First, a scientific name recognition step, allowing us to retrieve canonical names was carried (src/2_curating/2_editing/organisms/subscripts/1_cleaningOriginal.R). Then, a subtraction step of the obtained canonical names from the original field was applied, to avoid unwanted translation of parts of canonical names. For example, Bromus mango contains “mango” as a specific epithet, which is also the common name for Mangifera indica. After this subtraction step, the remaining names were translated from vernacular (common) and TCM names to scientific names, with help of the dictionaries. For performance reasons, this cleaning step was written in Kotlin and used coroutines to allow efficient parallelization of that process (src/2_curating/2_editing/organisms/2_translating_organism_kotlin/). They were subsequently submitted again to scientific name recognition (src/2_curating/2_editing/organisms/3_cleaningTranslated.R). 
After full resolution of canonical names, all obtained names were submitted to rotl (Michonneau et al., 2016) to obtain a unified taxonomy.


From the 86,076 initial “clean” organism fields, 43,421 (50%) canonical names were obtained, of which 31,965 (37%) were uploaded to Wikidata.
From the 294 initial “dirty” organism fields, 241 (82%) canonical names were obtained, of which 198 (67%) were uploaded to Wikidata.

##### References

The Rcrossref package interfacing with the Crossref API was used to translate references from their original subcategory (“original”, “publishingDetails”, “split”, “title”) to a DOI, the title of its corresponding article, the journal it was published in, its date of publication and the name of the first author. The first twenty candidates were kept and ranked according to the score returned by Crossref, which is a tf-idf score. For DOI and PMID, only a single candidate was kept. All parameters are available in src/functions/reference.R. All DOIs were also translated with this method, to eventually discard any DOI not leading to an object. PMIDs were translated, thanks to the entrez_summary function of the rentrez package. Scripts used for all subcategories of references are available in the directory src/2_curating/2_editing/reference/1_translating/. Once all translations were made, results coming from each subcategory were integrated, (src/2_curating/2_editing/reference/2_integrating.R) and the producing organism related to the reference was added for further treatment. Because the crossref score was not informative enough, at least one other metric was chosen to complement it. The first metric was related to the presence of the producing organism’s generic name in the title of the returned article. If the title contained the generic name of the organism, a score of 1 was given, else 0. Regarding the subcategories “doi”, “pubmed” and “title”, for which the same subcategory was retrieved via crossref or rentrez, distances between the input’s string and the candidates’ one were calculated. Optimal string alignment (restricted Damerau-Levenshtein distance) was used as a method. Among “publishing details”, “original” and “split” categories, three additional metrics were used: If the journal name was present in the original field, a score of 1 was given, else 0. If the name of the first author was present in the original field, a score of 1 was given, else 0. Those three scores were then summed together. All candidates were first ordered according to their crossref score, then by the complement score for related subcategories, then again according to their title-producing organism score, and finally according to their translation distance score. After this reranking step, only the first candidate was kept. Finally, the Pubmed PMCID dictionary (PMC-ids.csv.gz) was used to perform the translations between DOI, PMID, and PMCID. (src/2_curating/2_editing/reference/3_cleaning.R)


From the 36,692 initial “original” references, 21,928 (60%) references with sufficient quality were obtained, of which 15,674 (71%) had the organism name in their title. 9,882 (27%) were uploaded to Wikidata.
From the 21,306 initial “pubmed” references, 9,015 (42%) references with sufficient quality were obtained, of which 5,695 (63%) had the organism name in their title. 3,013 (14%) were uploaded to Wikidata.
From the 35,348 initial “doi” references, 19,682 (56%) references with sufficient quality were obtained, of which 15,522 (79%) had the organism name in their title. 11,847 (34%) were uploaded to Wikidata.
From the 29,584 initial “title” references, 17,410 (59%) references with sufficient quality were obtained, of which 12,732 (73%) had the organism name in their title. 9,638 (33%) were uploaded to Wikidata.
From the 11,322 initial “split” references, 5,856 (52%) references with sufficient quality were obtained, of which 3,221 (55%) had the organism name in their title. 2,255 (20%) were uploaded to Wikidata.
From the 3,310 initial “publishingDetails” references, 119 (4%) references with sufficient quality were obtained, of which 59 (50%) had the organism name in their title. 16 (0%) were uploaded to Wikidata.


#### Realignment

In order to fetch back the referenced structure-organism pairs links in the original data, the cleaned structures, cleaned organisms, and cleaned references were re-aligned with the initial entries. This resulted in over 6.2M referenced structure-organism pairs. Those pairs were not unique, with redundancies among electronic NP resources and different original categories leading to the same final pair (for example, entry reporting InChI=1/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in Crataegus oxyacantha or InChI=1S/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in Crataegus stevenii both led to OVSQVDMCBVZWGM-DTGCRPNFSA-N in Crataegus monogyna). After deduplication, over 2M unique structure-organism pairs were obtained.


After the curation of all three objects, all of them were put together again. Therefore, the original aligned table containing the original pairs was joined with each curation result. Only entries containing a structure, an organism, and a reference after curation were kept. Each curated object was divided into minimal data (for Wikidata upload) and metadata. A dictionary containing original and curated object translations was written for each object to avoid those translations being made again during the next curation step (src/2_curating/3_integrating.R).

#### Validation

The pairs obtained after curation were of different quality. Globally, structure and organism translation was satisfactory whereas reference translation was not. Therefore, to assess the validity of the obtained results, a randomized set of 420 referenced structure-organism pairs was sampled in each reference subcategory and validated or rejected manually. Entries were sampled with at least 55 of each reference subcategory present (to get a representative idea of each subcategory) (src/3_analysing/1_sampling.R). An entry was only validated if: i) the structure (as any structural descriptor that could be linked to the final sanitized InChIKey) was described in the reference ii) the producing organism (as any organism descriptor that could be linked to the accepted canonical name) was described in the reference and iii) the reference was describing the occurrence of the chemical structure in the biological organism. Results obtained on the manually analyzed set were categorized according to the initial reference subcategory and are detailed in SI-2. To improve these results, further cleaning of the references was needed. This was done by accepting entries whose reference was coming from a DOI, a PMID, or from a title which restricted Damerau-Levenshtein distance between original and translated was lower than ten or if it was coming from one of the three main journals where occurrences are published (i.e., Journal of Natural Products, Phytochemistry, or Journal of Agricultural and Food Chemistry). For “split”, “publishingDetails” and “original” subcategories, the year of publication of the obtained reference, its journal, and the name of the first author were searched in the original entry and if at least two of them were present, the entry was kept. Entries were then further filtered to keep the ones where the reference title contained the first element of the detected canonical name. Except for COCONUT, exceptions to this filter were made for all DOI-based references. To validate those filtering criteria, an additional set of 100 structure-organism pairs were manually analyzed. F0.5 score was used as a metric. F0.5 score is a modified F1 score where precision has twice more weight than recall.


The F-score was calculated with ß = 0.5, as in Equation 1. 


$$F_{\beta}\ =\ \left(1+\beta^2\right)\cdot\frac{precision\cdot recall}{\left(\beta^2\ \cdot\ precision\right)\ +\ recall}$$ {#eq:validation}

Based on this first manually validated dataset, filtering criteria (src/r/filter.R) were established to maximize precision and recall. Another 100 entries were sampled, this time respecting the whole set ratios. After manual validation, 97% of true positives were reached on the second set. A summary of the validation results is given in SI-2. Once validated, the filtering criteria were established to the whole curated set to filter entries chosen for dissemination (src/3_analysing/2_validating.R).

#### Unit Testing

To provide robustness of the whole process and code, unit tests and partial data full-tests were written. They can run on the developer machine but also on the CI/CD system (GitLab) upon each commit to the codebase.
Those tests assess that the functions are providing results coherent with what is expected especially for edge cases detected during the development. The Kotlin code has tests based on JUnit and code quality control checks based on Ktlint, Detekt and Ben Mane's version plugin.

### Data Dissemination

#### Wikidata

All the data produced for this work has been made available on Wikidata under a Creative Commons 0 license according to Wikidata:Licensing. This license is a "No-right-reserved" license that allows most reuses.

#### Lotus.NaturalProducts.Net (LNPN)

The web interface is implemented following the same protocol as described in the COCONUT publication (Sorokina et al., 2021) i.e. the data are stored in a MongoDB repository, the backend runs with Kotlin and Java, using the Spring framework, and the frontend is written in React.js, and completely Dockerized. In addition to the diverse search functions available through this web interface, an API is also implemented, allowing a programmatic LNPN querying. The complete API usage is described on the “Documentation” page of the website. LNPN is part of the NaturalProducts.net portal, an initiative aimed at gathering diverse open NP resources in one place.

### Data Interaction

#### Data Retrieval

Bulk retrieval of a frozen (2021-02-23) version of LOTUS data is also available at https://osf.io/zupqj/. 
WikidataLotusExporter allows the download of all chemical compounds with a "found in taxon" property. That way, it does not only get the data produced by this work, but any that would have existed beforehand or that would have been added directly on Wikidata by our users. It makes a copy of all the entities (compounds, taxa, references) into a local triplestore that can be queried with SPARQL as is or converted to a TSV file for inclusion in other projects. It is currently adapted to export directly into the SSOT thus allowing direct reuse by the processing/curation pipeline.

#### Data Addition

##### Wikidata

Data is loaded by the Kotlin importer available in the WikidataLotusImporter repository under a GPL V3 license and imported into Wikidata. The importer processes the curated outputs grouping references, organisms, and compounds together. It then checks if they already exist in Wikidata (using SPARQL or a direct connection to Wikidata depending on the kind of data). It then uses update or insert, also called upsert, the entities as needed. The script currently takes the tabular file of the documented structure-organism pairs resulting from the LOTUS curation process as input. It is currently being adapted to use directly the SSOT and avoid an unnecessary conversion step. To import references, it first double checks for the presence of duplicated DOIs and utilizes the Crossref REST API to retrieve metadata associated with the DOI, the support for other citation sources such as Europe PMC is in progress. The structure-related fields are only subject to limited processing: basic formatting of the molecular formula by subscripting of the numbers. Due to limitations in Wikidata, the molecule names are dropped if they are longer than 250 characters and likewise the InChI strings cannot be stored if they are longer than 1500 characters.


Uploaded taxonomical DB identifiers are currently restricted to ITIS, GBIF, NCBI Taxon, Index Fungorum, IRMNG, WORMS, VASCAN, and iNaturalist. The taxa levels are currently limited to family, subfamily, tribe, subtribe, genus, species, variety. The importer checks for the existence of each item based on their InChIKey and upserts the compound with the found in taxon statement and the associated organisms and references.


##### LNPN

From the onset, LNPN has been importing data directly from the frozen tabular data of the LOTUS dataset (https://osf.io/hgjdb/). In future versions, LNPN will directly feed on the SSOT.

##### Data Edition

The bot framework WikidataLotusImporter was adapted such that, in addition to batch upload capabilities, it can also edit erroneously created entries on Wikidata. As massive edits have a large potential to disrupt otherwise good data, progressive deployment of this script is used, starting by editing progressively 1, 10, then 100 entries that are manually checked. Upon validation of 100 entries, the full script is run and check its behavior checked at regular intervals. An example of a corrected entry is as follows: https://www.wikidata.org/w/index.php?title=Q105349871&type=revision&diff=1365519277&oldid=1356145998

##### Curation interface

A web-based (Kotlin, Spring Boot for the back-end, and TypeScript with Vue for the front-end) curation interface is currently in construction. It will allow mass-editing of entries and navigate quick navigation in the SSOT for the curation of new and existing entries. This new interface is intended to become open to the public to foster the curation of entries by further means, driven by the users. In line with the overall LOTUS approach, any modification made in this curation interface will be mirrored on Wikidata and LNPN.


## Code Availability {.page_break_before}

### General Repository

All programs written for this work can be found in the following group: [https://gitlab.com/lotus7](https://gitlab.com/lotus7). 

#### Processing

The source data curation system is available at [https://gitlab.com/lotus7/lotusProcessor](https://gitlab.com/lotus7/lotusProcessor). This program takes the source data as input and outputs curated data, ready for dissemination. The first step involves checking if the source data has already been processed. If not, all three elements (biological organism, chemical structures, and references) are submitted to various steps of translation and curation, before validation for dissemination.

#### Wikidata

##### Import

The Wikidata importer is available at [https://gitlab.com/lotus7/wikidataLotusImporter](https://gitlab.com/lotus7/wikidataLotusImporter). This program takes the processed data resulting from the lotusProcessor subprocess as input and uploads it to Wikidata. It performs a SPARQL query to check which objects already exist. If needed, it creates the missing objects. It then updates the content of each object. Finally, it updates the chemical compound page with a "found in taxon" statement complemented with a "stated in" reference.

##### Export
The Wikidata exporter is available at [https://gitlab.com/lotus7/wikidataLotusExporter](https://gitlab.com/lotus7/wikidataLotusExporter). This program takes the structured data in Wikidata corresponding to chemical compounds found in taxa with a reference associated as input and exports it in both RDF and tabular formats for further use. Two subsequent options are (a) that the end-user can directly use the exported data.; or (b) that the exported data, which can be new or modified since the last iteration, is used as new source data in lotusProcessor.

### LNPN 

The LNPN website and processing system is available at [https://github.com/mSorok/LOTUSweb](https://github.com/mSorok/LOTUSweb). This system takes the processed data resulting from the lotusProcessor as input and uploads it on [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net). The repository is not part of the main GitLab group as it benefits from already established pipelines developed by CS and MS. The website allows searches from different points of view, complemented with taxonomies for both the chemical and biological sides. Many chemical molecular properties and molecular descriptors that otherwise are unavailable in Wikidata are also provided.

### Code Freezing

All repository hyperlinks in the manuscript point to the preprint branches by default. The links contain all programs and code before submission (2021-02-23) and will eventually be updated to a publication branch using modifications resulting from the peer-reviewing process. As the code evolves, readers are invited to refer to the main branch of each repository for the most up-to-date code.
A frozen version (2021-02-23) of all programs and code is also available in the LOTUS OSF repository ([https://osf.io/pmgux/](https://osf.io/pmgux/)).

### Programs and packages

#### R
The [R](https://www.r-project.org/) version used was 4.0.4 (2021-02-15) -- "Lost Library Book" [@rcoreteamLanguageEnvironmentStatistical2020] and R-packages used were, in alphabetical order:
ChemmineR (3.42.1) [@doi:10/b46rs7], chorddiag (0.1.2) [@florChorddiagInteractiveChord2020], ClassyfireR (0.3.6) [@doi:10/gc5tqv], data.table (1.13.6) [@dowleDataTableExtension2020], DBI (1.1.1) [@rspecialinterestgroupondatabasesr-sig-dbDBIDatabaseInterface2021], gdata (2.18.0) [@warnesGdataVariousProgramming2017], ggalluvial (0.12.3) [@doi:10/gg4ghf], ggfittext (0.9.1) [@wilkinsGgfittextFitText2020], ggnewscale (0.4.5) [@campitelliGgnewscaleMultipleFill2021], ggraph (2.0.4) [@pedersenGgraphImplementationGrammar2020], ggstar (1.0.1) [@xuGgstarStarLayer2021], ggtree (2.4.1) [@doi:10/f9qv8x], ggtreeExtra (1.0.1) [@doi:10.21203/rs.3.rs-155672/v2], Hmisc (4.4-2) [@jrHmiscHarrellMiscellaneous2020], jsonlite (1.7.2) [@wikidata:Q106204620], pbmcapply (1.5.0) [@kuangPbmcapplyTrackingProgress2019], plotly (4.9.3) [@sievertInteractiveWebBasedData2020], rcrossref(1.1.0) [@chamberlainRcrossrefClientVarious2020], readxl (1.3.1) [@wickhamReadxlReadExcel2019], rentrez (1.2.3) [@doi:10/gh5ptg], rotl (3.0.11) [@doi:10/f9jgkm], rvest (0.3.6) [@wickhamRvestEasilyHarvest2020], splitstackshape (1.4.8) [@mahtoSplitstackshapeStackReshape2019], RSQLite (2.2.3) [@mullerRSQLiteSQLiteInterface2021], stringdist (0.9.6.3) [@doi:10/ghfr9v], stringi (1.5.3) [@gagolewskiPackageStringiCharacter2020], tidyverse (1.3.0) [@doi:10/ggddkj], treeio (1.14.3) [@doi:10/ggwr93], UpSetR (1.4.0) [@gehlenborgUpSetRMoreScalable2019], vroom (1.3.2) [@hesterVroomReadWrite2020], webchem (1.1.1) [@doi:10/gh5ptf], XML (3.99-05) [@langXMLToolsParsing2020], xml2 (1.3.2) [@wickhamXml2ParseXML2020]

#### Python
The [Python](https://www.python.org/) version used was 3.8.6, and the Python packages utilized were, in alphabetical order: 
faerun (0.3.2) [@doi:10/gcmhc5], map4 (1.0) [@doi:10/gjjz4c], matplotlib (3.1.3) [@doi:10.1109/MCSE.2007.55], [Molvs](https://github.com/mcs07/MolVS) (0.1.1), pandas (1.1.4) [@doi:10/gh6dqz], rdkit (2020.09.2) [@RDKitOpensourceCheminformatics], scipy (1.5.0) [@doi:10/ggj45f], tmap (1.0.4) [@doi:10/gjjz4f]. 

#### Kotlin
Kotlin packages used were as follows: 
Common: Kotlin 1.4.21 up to 1.4.30, Univocity 2.9.0, OpenJDK 15, Kotlin serialization 1.0.1, konnector 0.1.27, Log4J 2.14.0
Wikidata Importer Bot:, WikidataTK 0.11.1, CDK 2.3 [@doi:10/gbppnb], RDF4J 3.6.0, Ktor 1.5.0, KotlinXCli 0.3.1, Wikidata data processing: Shadow 5.0.0
Quality control and testing: Ktlint 9.4.1, Kotlinter 3.3.0, Detekt 1.15.0, Ben Mane's version plugin 0.36.0, Junit 5.7.0

#### Additional executable files
[GNFinder](https://github.com/gnames/gnfinder/releases/tag/v0.11.1) v.0.11.1, [GNVerify](https://github.com/gnames/gnverify/releases/tag/v0.1.0) v.0.1.0, [OPSIN](https://github.com/dan2097/opsin/releases/tag/2.5.0) v.2.5.0 [@doi:10/b2zkr9]


## Data Availability

A snapshot of the obtained data at the time of submission is available at the following OSF repository (datasets): https://osf.io/pmgux/.


## Acknowledgments {.page_break_before}

JLW and PMA are thankful to the Swiss National Science Foundation for supporting part of this project through the SNF Sinergia grant CRSII5_189921. JB and AR are really thankful to JetBrains for the Free educational license of IntelliJ and the excellent support received on Youtrack. JB, JGG, and GFP gratefully acknowledge the support of this work by grant U41 AT008706 and supplemental funding to P50 AT000155 from NCCIH and ODS of the NIH. MS and CS are supported by the German Research Foundation within the framework ChemBioSys (Project-ID 239748522, SFB 1127). The work on the Wikidata IDSM/Sachem endpoint was supported by an ELIXIR CZ research infrastructure project grant (MEYS Grant No: LM2018131) including access to computing and storage facilities. The authors would like to thank [Dmitry Mozzherin](https://github.com/dimus) for his work done for the Global Names Architecture and related improvements. EW and DM acknowledge the Scholia grant from the Alfred P. Sloan Foundation under grant number G-2019-11458. The authors would also like to thank contributors of all DB used in this work.

## Competing interests

The authors declare no competing interest.

## Author contributions

![](images/author_contribution.svg){author-contribution}

## Supporting Information {.page_break_before}

### SI 1 - Data Sources List

<font size="1.5">
<table>
<caption>Table SI-1: Data Sources List</caption>
<colgroup>
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 10%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
<col style="text-align: center; vertical-align: middle; width: 9%;" />
</colgroup>
<thead>
<tr class="header">
<th>database</th>
<th>type</th>
<th>initial retrieved unique entries</th>
<th>cleaned documented structure-organism pairs</th>
<th>pairs validated for wikidata export</th>
<th>website</th>
<th>article</th>
<th>retrieval</th>
<th>license</th>
<th>contact</th>
<th>varia</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><p>afrotryp</p></td>
<td><p>open</p></td>
<td><p>312</p></td>
<td><p>135</p></td>
<td><p>28</p></td>
<td><p>NA</p></td>
<td><p><a href="https://link.springer.com/article/10.1007/s00044-016-1764-y?shared-article-renderer">afrotryp_article</a></p></td>
<td><p><a href="https://static-content.springer.com/esm/art%3A10.1007%2Fs00044-016-1764-y/MediaObjects/44_2016_1764_MOESM1_ESM.doc">afrotryp_download</a></p></td>
<td><p><a href="docs/licenses/afrotryp.md">afrotryp_license</a></p></td>
<td><p><a href="https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c1/new">Fidele Ntie-Kang</a> or <a href="https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c2/new">Ngozi Justina Nwodo</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>alkamid</p></td>
<td><p>open</p></td>
<td><p>4434</p></td>
<td><p>2582</p></td>
<td><p>2092</p></td>
<td><p><a href="http://alkamid.ugent.be">alkamid_website</a></p></td>
<td><p><a href="http://dx.doi.org/10.1016/j.jep.2012.05.038">alkamid_article</a></p></td>
<td><p><a href="/src/1_gathering/db/alkamid/scraping.R">alkamid_script</a></p></td>
<td><p><a href="docs/licenses/alkamid.md">alkamid_license</a></p></td>
<td><p><a href="mailto:Bart.DeSpiegeleer@UGent.be">Bart De Spiegeleer</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>biofacquim</p></td>
<td><p>open</p></td>
<td><p>531</p></td>
<td><p>683</p></td>
<td><p>534</p></td>
<td><p><a href="https://biofacquim.herokuapp.com">biofacquim_website</a> (old version)</p></td>
<td><p><a href="https://www.mdpi.com/2218-273X/9/1/31">biofacquim_article_old</a> <a href="https://f1000research.com/articles/8-2071">biofacquim_article_new</a></p></td>
<td><p><a href="https://ndownloader.figshare.com/files/20050244">biofacquim_download</a></p></td>
<td><p><a href="docs/licenses/biofacquim.md">biofacquim_license</a></p></td>
<td><p><a href="mailto:medinajl@unam.mx">José Medina-Franco</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>biophytmol</p></td>
<td><p>open</p></td>
<td><p>546</p></td>
<td><p>632</p></td>
<td><p>356</p></td>
<td><p><a href="http://ab-openlab.csir.res.in/biophytmol/">biophytmol_website</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2">biophytmol_article</a></p></td>
<td><p><a href="/src/1_gathering/db/biophytmol/scraping.R">biophytmol_script</a></p></td>
<td><p><a href="docs/licenses/biophytmol.md">biophytmol_license</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2/email/correspondent/c1/new">Anshu Bhardwaj</a></p></td>
<td><p>website often down</p></td>
</tr>
<tr class="odd">
<td><p>carotenoiddb</p></td>
<td><p>open</p></td>
<td><p>2922</p></td>
<td><p>1204</p></td>
<td><p>639</p></td>
<td><p><a href="http://carotenoiddb.jp/About.html">carotenoiddb_website</a></p></td>
<td><p><a href="https://doi.org/10.1093/database/bax004">carotenoiddb_article</a></p></td>
<td><p><a href="/src/1_gathering/db/carotenoiddb/scraping.R">carotenoiddb_script</a></p></td>
<td><p><a href="docs/licenses/carotenoiddb.md">carotenoiddb_license</a></p></td>
<td><p><a href="mailto:yzjunko@gmail.com">yzjunko@gmail.com</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>coconut</p></td>
<td><p>open</p></td>
<td><p>583623</p></td>
<td><p>345147</p></td>
<td><p>34437</p></td>
<td><p><a href="https://coconut.naturalproducts.net">coconut_website</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9">coconut_article</a></p></td>
<td><p><a href="https://zenodo.org/record/3778405/files/COCONUT.sdf.zip?download=1">coconut_download</a></p></td>
<td><p><a href="docs/licenses/coconut.md">coconut_license</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9/email/correspondent/c1/new">Maria Sorokina</a></p></td>
<td><p><a href="https://zenodo.org/record/3778405#.XsylyC1Ph0Y">coconut_zenodo</a></p></td>
</tr>
<tr class="odd">
<td><p>cyanometdb</p></td>
<td><p>open</p></td>
<td><p>1930</p></td>
<td><p>1812</p></td>
<td><p>1754</p></td>
<td><p>NA</p></td>
<td><p><a href="https://doi.org/10.1016/j.watres.2021.117017">cyanometdb_article</a></p></td>
<td><p><a href="https://ars.els-cdn.com/content/image/1-s2.0-S0043135421002153-mmc4.zip">cyanometdb_download</a></p></td>
<td><p><a href="docs/licenses/cyanometdb.md">cyanometdb_license</a></p></td>
<td><p><a href="mailto:elisabeth.janssen@eawag.ch">elisabeth.janssen@eawag.ch</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>datawarrior</p></td>
<td><p>open</p></td>
<td><p>589</p></td>
<td><p>1052</p></td>
<td><p>102</p></td>
<td><p><a href="http://www.openmolecules.org/datawarrior/index.html">datawarrior_website</a></p></td>
<td><p><a href="https://doi.org/10.1021/ci500588j">datawarrior_article</a></p></td>
<td><p><a href="http://www.openmolecules.org/datawarrior/download.html">datawarrior_download</a></p></td>
<td><p><a href="docs/licenses/datawarrior.md">datawarrior_license</a></p></td>
<td><p><a href="mailto:thomas.sander@idorsia.com">thomas.sander@idorsia.com</a></p></td>
<td><p>no real link to the dataset inside it</p></td>
</tr>
<tr class="odd">
<td><p>dianatdb</p></td>
<td><p>open</p></td>
<td><p>290</p></td>
<td><p>406</p></td>
<td><p>27</p></td>
<td><p><a href="http://rdu.iquimica.unam.mx/handle/20.500.12214/1186">dianatdb_website</a></p></td>
<td><p><a href="https://doi.org/10.1039/D0RA10453A">dianatdb_article</a></p></td>
<td><p><a href="http://rdu.iquimica.unam.mx/bitstream/20.500.12214/1186/3/2020_DiaNatDB_336.xlsx">dianatdb_download</a></p></td>
<td><p><a href="docs/licenses/dianatdb.md">dianatdb_license</a></p></td>
<td><p><a href="mailto:amadariaga@iquimica.unam.mx">amadariaga@iquimica.unam.mx</a> or <a href="mailto:kmtzm@unam.mx">kmtzm@unam.mx</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>dnp</p></td>
<td><p>commercial</p></td>
<td><p>210832</p></td>
<td><p>258864</p></td>
<td><p>NA</p></td>
<td><p><a href="http://dnp.chemnetbase.com/faces/chemical/ChemicalSearch.xhtml">dnp_website</a></p></td>
<td><p>NA</p></td>
<td><p><a href="/src/1_gathering/db/dnp/extracting.py">dnp_script</a></p></td>
<td><p>TODO</p></td>
<td><p><a href="mailto:support@taylorfrancis.com">support@taylorfrancis.com</a></p></td>
<td><p>commercial</p></td>
</tr>
<tr class="odd">
<td><p>drduke</p></td>
<td><p>open</p></td>
<td><p>90675</p></td>
<td><p>9052</p></td>
<td><p>4266</p></td>
<td><p><a href="https://phytochem.nal.usda.gov/phytochem/help/index/about">drduke_website</a></p></td>
<td><p>NA</p></td>
<td><p><a href="https://data.nal.usda.gov/system/files/Duke-Source-CSV.zip">drduke_download</a></p></td>
<td><p><a href="docs/licenses/drduke.md">drduke_license</a></p></td>
<td><p><a href="mailto:agref@usda.gov">agref@usda.gov</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>foodb</p></td>
<td><p>restricted</p></td>
<td><p>82415</p></td>
<td><p>376</p></td>
<td><p>NA</p></td>
<td><p><a href="https://foodb.ca/about#cite">foodb_website</a></p></td>
<td><p>NA</p></td>
<td><p><a href="https://foodb.ca/public/system/downloads/foodb_2020_4_7_csv.tar.gz">foodb_download</a></p></td>
<td><p><a href="docs/licenses/foodb.md">foodb_license</a></p></td>
<td><p><a href="mailto:jreid3@ualberta.ca">jreid3@ualberta.ca</a> (Jennifer)</p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>inflamnat</p></td>
<td><p>open</p></td>
<td><p>665</p></td>
<td><p>656</p></td>
<td><p>282</p></td>
<td><p>NA</p></td>
<td><p><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.8b00560">inflamnat_article</a></p></td>
<td><p><a href="https://pubs.acs.org/doi/suppl/10.1021/acs.jcim.8b00560/suppl_file/ci8b00560_si_001.xlsx">inflamnat_download</a></p></td>
<td><p><a href="docs/licenses/inflamnat.md">inflamnat_license</a></p></td>
<td><p><a href="mailto:xiaoweilie@ynu.edu.cn">xiaoweilie@ynu.edu.cn</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>knapsack</p></td>
<td><p>open</p></td>
<td><p>116284</p></td>
<td><p>142466</p></td>
<td><p>63094</p></td>
<td><p><a href="http://www.knapsackfamily.com/knapsack_core/top.php">knapsack_website</a></p></td>
<td><p><a href="https://link.springer.com/chapter/10.1007/3-540-29782-0_13">knapsack_article</a></p></td>
<td><p><a href="/src/1_gathering/db/knapsack/scraping.R">knapsack_script</a></p></td>
<td><p><a href="docs/licenses/knapsack.md">knapsack_license</a></p></td>
<td><p><a href="mailto:skanaya@gtc.naist.jp">skanaya@gtc.naist.jp</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>metabolights</p></td>
<td><p>open</p></td>
<td><p>29864</p></td>
<td><p>29498</p></td>
<td><p>4920</p></td>
<td><p><a href="https://www.ebi.ac.uk/metabolights/about">metabolights_website</a></p></td>
<td><p><a href="https://doi.org/10.1093/nar/gkz1019">metabolights_article</a></p></td>
<td><p><a href="ftp://ftp.ebi.ac.uk/pub/databases/metabolights/eb-eye/eb-eye_metabolights_complete.xml">metabolights_download</a></p></td>
<td><p><a href="docs/licenses/metabolights.md">metabolights_license</a></p></td>
<td><p>???</p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>mibig</p></td>
<td><p>open</p></td>
<td><p>1310</p></td>
<td><p>1132</p></td>
<td><p>548</p></td>
<td><p><a href="https://mibig.secondarymetabolites.org">mibig_website</a></p></td>
<td><p><a href="https://doi.org/10.1093/nar/gkz882">mibig_article</a></p></td>
<td><p><a href="https://dl.secondarymetabolites.org/mibig/mibig_json_2.0.tar.gz">mibig_download</a></p></td>
<td><p><a href="docs/licenses/mibig.md">mibig_license</a></p></td>
<td><p><a href="mailto:tiwe@biosustain.dtu.dk">Tilmann Weber</a> or <a href="mailto:marnix.medema@wur.nl">Marnix Medema</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>mitishamba</p></td>
<td><p>open</p></td>
<td><p>1073</p></td>
<td><p>1174</p></td>
<td><p>369</p></td>
<td><p><a href="http://mitishamba.uonbi.ac.ke">mitishamba_website</a></p></td>
<td><p><a href="http://erepository.uonbi.ac.ke/handle/11295/92273">mitishamba_"article"</a></p></td>
<td><p><a href="/src/1_gathering/db/mitishamba/scraping.R">mitishamba_script</a></p></td>
<td><p><a href="docs/licenses/mitishamba.md">mitishamba_license</a></p></td>
<td><p>???</p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>nanpdb</p></td>
<td><p>open</p></td>
<td><p>5752</p></td>
<td><p>6452</p></td>
<td><p>5327</p></td>
<td><p><a href="http://african-compounds.org/nanpdb/">nanpdb_website</a></p></td>
<td><p><a href="https://doi.org/10.1021/acs.jnatprod.7b00283">nanpdb_article</a></p></td>
<td><p><a href="/src/1_gathering/db/nanpdb/scraping.R">nanpdb_script</a></p></td>
<td><p><a href="docs/licenses/nanpdb.md">nanpdb_license</a></p></td>
<td><p><a href="mailto:ntiekfidele@gmail.com">ntiekfidele@gmail.com</a> <a href="mailto:stefan.guenther@pharmazie.uni-freiburg.de">stefan.guenther@pharmazie.uni-freiburg.de</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>napralert</p></td>
<td><p>commercial</p></td>
<td><p>681401</p></td>
<td><p>380559</p></td>
<td><p>263586</p></td>
<td><p><a href="https://www.napralert.org">napralert_website</a></p></td>
<td><p><a href="https://doi.org/10.1016/B978-008045382-8.00060-5">napralert_article</a></p></td>
<td><p>TODO</p></td>
<td><p><a href="docs/licenses/napralert.md">napralert_license</a></p></td>
<td><p><a href="mailto:napralert@uic.edu">napralert@uic.edu</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>npass</p></td>
<td><p>open</p></td>
<td><p>290539</p></td>
<td><p>28991</p></td>
<td><p>21913</p></td>
<td><p><a href="http://bidd2.nus.edu.sg/NPASS/about.php">npass_website</a></p></td>
<td><p><a href="https://academic.oup.com/nar/article/46/D1/D1217/4584628">npass_article</a></p></td>
<td><p><a href="http://bidd.group/NPASS/downloadnpass.html">npass_download</a></p></td>
<td><p><a href="docs/licenses/npass.md">npass_license</a></p></td>
<td><p><a href="mailto:phacyz@nus.edu.sg">phacyz@nus.edu.sg</a> <a href="mailto:jiangyy@sz.tsinghua.edu.cn">jiangyy@sz.tsinghua.edu.cn</a> <a href="mailto:iaochen@163.com">iaochen@163.com</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>npatlas</p></td>
<td><p>open</p></td>
<td><p>29006</p></td>
<td><p>49861</p></td>
<td><p>44456</p></td>
<td><p><a href="https://www.npatlas.org/joomla/">npatlas_website</a></p></td>
<td><p><a href="https://doi.org/10.1021/acscentsci.9b00806">npatlas_article</a></p></td>
<td><p><a href="https://www.npatlas.org/custom/versions/np_atlas_2020_06/NPAtlas_download.tsv">npatlas_download</a></p></td>
<td><p><a href="docs/licenses/npatlas.md">npatlas_license</a></p></td>
<td><p><a href="mailto:rliningt@sfu.ca">rliningt@sfu.ca</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>npcare</p></td>
<td><p>open</p></td>
<td><p>7763</p></td>
<td><p>4665</p></td>
<td><p>2525</p></td>
<td><p><a href="http://silver.sejong.ac.kr/npcare/">npcare_website</a></p></td>
<td><p><a href="https://doi.org/10.1186/s13321-016-0188-5">npcare_article</a></p></td>
<td><p><a href="http://silver.sejong.ac.kr/npcare/csv/npcare.zip">npcare_download</a></p></td>
<td><p><a href="docs/licenses/npcare.md">npcare_license</a></p></td>
<td><p><a href="mailto:choihwanho@gmail.com">choihwanho@gmail.com</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>npedia</p></td>
<td><p>open</p></td>
<td><p>82</p></td>
<td><p>99</p></td>
<td><p>23</p></td>
<td><p><a href="http://www.cbrg.riken.jp/npedia/?LANG=en">npedia_website</a></p></td>
<td><p><a href="https://www.jstage.jst.go.jp/article/ciqs/2006/0/2006_0_JL6/_article">npedia_article</a></p></td>
<td><p><a href="/src/1_gathering/db/npedia/scraping.R">npedia_script</a></p></td>
<td><p><a href="docs/licenses/npedia.md">npedia_license</a></p></td>
<td><p><a href="mailto:hisyo@riken.jp">hisyo@riken.jp</a> <a href="mailto:npd@riken.jp">npd@riken.jp</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>nubbe</p></td>
<td><p>open</p></td>
<td><p>2189</p></td>
<td><p>2618</p></td>
<td><p>2463</p></td>
<td><p><a href="https://nubbe.iq.unesp.br/portal/nubbe-search.html">nubbe_website</a></p></td>
<td><p><a href="https://www.nature.com/articles/s41598-017-07451-x">nubbe_article</a></p></td>
<td><p><a href="/src/1_gathering/db/nubbe/scraping.R">MISSING_nubbe_script</a></p></td>
<td><p><a href="docs/licenses/nubbe.md">nubbe_license</a></p></td>
<td><p><a href="https://www.nature.com/articles/s41598-017-07451-x/email/correspondent/c1/new">Vanderlan S. Bolzani</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>pamdb</p></td>
<td><p>open</p></td>
<td><p>3061</p></td>
<td><p>3198</p></td>
<td><p>64</p></td>
<td><p><a href="http://pseudomonas.umaryland.edu/PAMDB.htm">pamdb_website</a></p></td>
<td><p><a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkx1061">pamdb_article</a></p></td>
<td><p><a href="http://pseudomonas.umaryland.edu/PaDl/PaMet.xlsx">pamdb_download</a></p></td>
<td><p><a href="docs/licenses/pamdb.md">pamdb_license</a></p></td>
<td><p><a href="mailto:awilks@rx.umaryland.edu">awilks@rx.umaryland.edu</a> <a href="mailto:aoglesby@rx.umaryland.edu">aoglesby@rx.umaryland.edu</a> <a href="mailto:mkane@rx.umaryland.edu">aoglesby@rx.umaryland.edu</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>phenolexplorer</p></td>
<td><p>open</p></td>
<td><p>8968</p></td>
<td><p>11779</p></td>
<td><p>6755</p></td>
<td><p><a href="http://phenol-explorer.eu/cite_us">phenolexplorer_website</a></p></td>
<td><p><a href="http://database.oxfordjournals.org/content/2013/bat070.long">phenolexplorer_article</a></p></td>
<td><p><a href="http://phenol-explorer.eu/downloads">phenolexplorer_download</a></p></td>
<td><p><a href="docs/licenses/phenolexplorer.md">phenolexplorer_license</a></p></td>
<td><p><a href="mailto:scalberta@iarc.fr">scalberta@iarc.fr</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>phytohub</p></td>
<td><p>open</p></td>
<td><p>2363</p></td>
<td><p>1425</p></td>
<td><p>52</p></td>
<td><p><a href="http://phytohub.eu/about">phytohub_website</a></p></td>
<td><p><a href="https://hal.archives-ouvertes.fr/hal-01697081/file/2017_Giacomoni_ICPH_Québec.pdf">phytohub_"article"</a></p></td>
<td><p><a href="/src/1_gathering/db/phytohub/scraping.R">phytohub_script</a></p></td>
<td><p><a href="docs/licenses/phytohub.md">phytohub_license</a></p></td>
<td><p><a href="mailto:claudine.manach@inra.fr">claudine.manach@inra.fr</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>procardb</p></td>
<td><p>open</p></td>
<td><p>6606</p></td>
<td><p>9864</p></td>
<td><p>70</p></td>
<td><p><a href="https://bioinfo.imtech.res.in/servers/procardb/">procardb_website</a></p></td>
<td><p><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6">procardb_article</a></p></td>
<td><p><a href="/src/1_gathering/db/procardb/scraping.R">procardb_script</a></p></td>
<td><p><a href="docs/licenses/procardb.md">procardb_license</a></p></td>
<td><p><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c1/new">Anil Kumar Pinnaka</a> <a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c2/new">Ashwani Kumar</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>respect</p></td>
<td><p>open</p></td>
<td><p>2759</p></td>
<td><p>601</p></td>
<td><p>282</p></td>
<td><p><a href="http://spectra.psc.riken.jp/menta.cgi/respect/licence/licence">respect_website</a></p></td>
<td><p><a href="https://www.sciencedirect.com/science/article/pii/S003194221200310X">respect_article</a></p></td>
<td><p><a href="http://spectra.psc.riken.jp/menta.cgi/static/respect/respect.zip">respect_download</a></p></td>
<td><p><a href="docs/licenses/respect.md">respect_license</a></p></td>
<td><p><a href="mailto:ksaito@psc.riken.jp">ksaito@psc.riken.jp</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>sancdb</p></td>
<td><p>open</p></td>
<td><p>861</p></td>
<td><p>981</p></td>
<td><p>772</p></td>
<td><p><a href="https://sancdb.rubi.ru.ac.za/cite_us/">sancdb_website</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8">sancdb_article</a></p></td>
<td><p><a href="/src/1_gathering/db/sancdb/scraping.R">sancdb_script</a></p></td>
<td><p><a href="docs/licenses/sancdb.md">sancdb_license</a></p></td>
<td><p><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8/email/correspondent/c1/new">Özlem Tastan Bishop</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>streptomedb</p></td>
<td><p>open</p></td>
<td><p>71638</p></td>
<td><p>37816</p></td>
<td><p>19037</p></td>
<td><p><a href="http://132.230.56.4/streptomedb3/">streptomedb_website</a></p></td>
<td><p><a href="https://doi.org/10.1093/nar/gkv1319">streptomedb_article</a></p></td>
<td><p><a href="http://132.230.56.4/streptomedb/download/">streptomedb_download</a></p></td>
<td><p><a href="docs/licenses/streptomedb.md">streptomedb_license</a></p></td>
<td><p><a href="mailto:stefan.guenther@pharmazie.uni-freiburg.de">stefan.guenther@pharmazie.uni-freiburg.de</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>swmd</p></td>
<td><p>open</p></td>
<td><p>1075</p></td>
<td><p>1616</p></td>
<td><p>1368</p></td>
<td><p><a href="http://www.swmd.co.in/index.php">swmd_website</a></p></td>
<td><p><a href="https://dx.doi.org/10.6026%2F97320630005361">swmd_article</a></p></td>
<td><p><a href="/src/1_gathering/db/swmd/scraping.R">swmd_script</a></p></td>
<td><p><a href="docs/licenses/swmd.md">swmd_license</a></p></td>
<td><p><a href="mailto:Dicky.John@gmail.com">Dicky.John@gmail.com</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>tmdb</p></td>
<td><p>open</p></td>
<td><p>2116</p></td>
<td><p>841</p></td>
<td><p>14</p></td>
<td><p><a href="http://pcsb.ahau.edu.cn:8080/TCDB/index.jsp">tmdb_website</a></p></td>
<td><p><a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1">tmdb_article</a></p></td>
<td><p><a href="/src/1_gathering/db/tmdb/scraping.R">tmdb_script</a></p></td>
<td><p><a href="docs/licenses/tmdb.md">tmdb_license</a></p></td>
<td><p><a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c1/new">Xiao-Chun Wan</a> <a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c2/new">Guan-Hu Bao</a></p></td>
<td><p>currently down</p></td>
</tr>
<tr class="even">
<td><p>tmmc</p></td>
<td><p>open</p></td>
<td><p>15033</p></td>
<td><p>5727</p></td>
<td><p>2658</p></td>
<td><p><a href="http://informatics.kiom.re.kr/compound/index.jsp">tmmc_website</a></p></td>
<td><p><a href="https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5">tmmc_article</a></p></td>
<td><p><a href="http://informatics.kiom.re.kr/compound/download/compound.xlsx">tmmc_download</a></p></td>
<td><p><a href="docs/licenses/tmmc.md">tmmc_license</a></p></td>
<td><p><a href="https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5/email/correspondent/c1/new">Jeong-Ju Lee</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>tppt</p></td>
<td><p>open</p></td>
<td><p>27182</p></td>
<td><p>28355</p></td>
<td><p>941</p></td>
<td><p><a href="https://www.agroscope.admin.ch/agroscope/en/home/publications/apps/tppt.html">tppt_website</a></p></td>
<td><p><a href="https://pubs.acs.org/doi/10.1021/acs.jafc.8b01639">tppt_article</a></p></td>
<td><p><a href="https://www.agroscope.admin.ch/dam/agroscope/de/dokumente/publikationen/tppt-xls.xlsx.download.xlsx/TPPT_database.xlsx">tppt_download</a></p></td>
<td><p><a href="doc/licenses/tppt.md">tppt_license</a></p></td>
<td><p><a href="mailto:thomas.bucheli@agroscope.admin.ch">thomas.bucheli@agroscope.admin.ch</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="even">
<td><p>unpd</p></td>
<td><p>open</p></td>
<td><p>340319</p></td>
<td><p>550980</p></td>
<td><p>353433</p></td>
<td><p><a href="https://web.archive.org/web/20160314180639/http://pkuxxj.pku.edu.cn/UNPD/introduction.php">unpd_website</a></p></td>
<td><p><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0062839">unpd_article</a></p></td>
<td><p>TODO</p></td>
<td><p>TODO</p></td>
<td><p><a href="mailto:lirongc@pku.edu.cn">lirongc@pku.edu.cn</a> <a href="mailto:xiaojxu@pku.edu.cn">xiaojxu@pku.edu.cn</a></p></td>
<td><p>NA</p></td>
</tr>
<tr class="odd">
<td><p>wakankensaku</p></td>
<td><p>open</p></td>
<td><p>367</p></td>
<td><p>49</p></td>
<td><p>41</p></td>
<td><p><a href="https://wakankensaku.inm.u-toyama.ac.jp/wiki/Main_Page">wakankensaku_website</a></p></td>
<td><p>???</p></td>
<td><p><a href="/src/1_gathering/db/wakankensaku/scraping.R">wakankensaku_script</a></p></td>
<td><p>TODO</p></td>
<td><p>???</p></td>
<td><p>NA</p></td>
</tr>
</tbody>
</table> 


### SI 2 - Summary of the Validation Statistics {.page_break_before}

|          | **First validation dataset (n =420)** <!-- $colspan="8" --> | | | | | | | | **Second validation dataset (n = 100)** <!-- $colspan="2" -->  | |
|:----------------------------------------:|:--------------------:|:--------------------:|:--------------------:|:--------------------:|:---------------------:|:--------------------:|:--------------------:|:--------------------:|:--------------------:|:--------------------:|
| **Reference Type ** | **True positives**   | **False positives** |**False negatives** |**True negatives** |**Relative abundance** |**Precision** | **Recall** |**F~0.5~ score** |**True positives** |**False negatives** |
| **Original** | 80 | 6 | 7 | 11 | 0.31 | 0.93 | 0.92 | 0.92 | 38 | 1 |
| **Pubmed** | 37 | 1 | 5 | 6 | 0.30 | 0.97 | 0.88 | 0.92 | 5 | 1 |
| **DOI** | 115 | 6 | 0 | 6 | 0.19 | 0.95 | 1.00 | 0.97 | 43 | 1 |
| **Title** | 38 | 2 | 0 | 16 | 0.12 | 0.95 | 1.00 | 0.97 | 7 | 0 |
| **Split** | 8 | 0 | 15 | 27 | 0.08 | 1.00 | 0.35 | 0.52 | 4 | 0 |
| **Publishing details** | 1 | 0 | 1 | 32 | 0.01 | 1.00 | 0.50 | 0.67 | 0 | 0 |
| **Total** | 279 | 15 | 28 | 98 | 1.00 | - | - | - | **97** | **3** |
| **Correctd total** | - | - | - | - | - | **0.96** | **0.89** | **0.91** | - | - |

Table: Summary of the Validation Statistics {#tbl:si-2 tag="SI-2"}


### SI 3 - Wikidata SPARQL Queries {.page_break_before}

**TODO** Comment them!

#### Query 1 - *Arabidopsis thaliana*

This query answers to the following question:
`What are the compounds present in Mouse-ear cress (Arabidopsis thaliana)?`
Link: [https://w.wiki/32y8](https://w.wiki/32y8)
```SQL
SELECT DISTINCT ?compound ?compoundLabel WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
  }
  ?compound wdt:P31 ?classes.
  ?taxon wdt:P225 "Arabidopsis thaliana".
  {
    ?compound p:P703 ?stmt.
    ?stmt ps:P703 ?taxon;
      prov:wasDerivedFrom ?ref.
    ?ref pr:P248 ?art.
    ?art wdt:P356 ?art_doi.
  }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
```

#### Query 2 - β-sitosterol planar structure {.page_break_before}

This query answers to the following question:
`Which organisms are known to contain compounds sharing the planar structure of β-sitosterol?`
Link: [https://w.wiki/334q](https://w.wiki/334q)
```SQL
SELECT DISTINCT ?taxon ?taxonname WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
  }
  ?compound wdt:P31 ?classes;
    wdt:P235 ?inchikey.
  FILTER(REGEX(?inchikey, "KZJWDPNRJALLNS", "i"))
  {
    ?compound p:P703 ?stmt.
    ?stmt ps:P703 ?taxon.
    {
      ?stmt prov:wasDerivedFrom ?ref.
      ?ref pr:P248 ?art.
      ?art wdt:P356 ?art_doi.
    }
  }
  ?taxon wdt:P225 ?taxonname .
}
```

#### Query 3 - β-sitosterol stereoisomers {.page_break_before}

This query answers to the following question:
`Which organisms are known to contain compounds sharing the planar structure of β-sitosterol?`
Link: [https://w.wiki/334s](https://w.wiki/334s)
```SQL
# taxa with chemical compounds related to (but different from) beta-sitosterol
SELECT ?compound ?compoundLabel ?InChIKey ?taxonname 
WITH {
  SELECT ?compound ?InChIKey WHERE {
    wd:Q121802 wdt:P235 ?queryKey . # beta-sitosterol
    ?compound wdt:P235 ?InChIKey .
    FILTER ( ?InChIKey != ?queryKey )
    FILTER (regex(str(?InChIKey), concat("^", substr($queryKey,1,14), "-")))
  }
} AS %compounds
WHERE {
  INCLUDE %compounds
  ?compound wdt:P703/wdt:P225 ?taxonname .
  ?compound rdfs:label ?compoundLabel.
  FILTER(LANG(?compoundLabel) = "en").
}
ORDER BY ASC(?InChIKey)
}
```

#### Query 4 - Pigments {.page_break_before}

This query answers to the following question:
`Which pigments are found in which taxa, according to which reference?`
Link: [https://w.wiki/38Rt](https://w.wiki/38Rt)
```SQL
# Pigments and the taxa they were found in, along with references
SELECT DISTINCT ?compound ?compoundLabel ?taxon ?taxonname ?DOI 
WITH {
  SELECT ?compound WHERE {
    ?compound wdt:P31*/wdt:P279* wd:Q161179.  # get pigments
  }
} AS %compounds
WITH {
  SELECT ?compound ?P703statement WHERE {
    INCLUDE %compounds                        
            ?compound p:P703 ?P703statement.  # check for "found in taxon" statements
  }
} AS %P703statement
WITH {
  SELECT ?compound ?taxon ?DOI WHERE {
    INCLUDE %P703statement
            ?P703statement ps:P703 ?taxon ;   # get the respective taxa
            prov:wasDerivedFrom / pr:P248 [   # get the reference supporting that statement
              wdt:P356 ?DOI                   # get the DOI for the reference
            ] .
  }
} AS %taxa
WHERE {
  {
    INCLUDE %taxa

            ?taxon wdt:P225 ?taxonname .      # get the taxon name
  }
  ?compound rdfs:label ?compoundLabel .       # get compound labels
  FILTER (LANG(?compoundLabel) = "en") .      # filter for English
}
ORDER BY ASC(?compoundLabel)
LIMIT 1000
```

#### Query 5 - Sister taxon compounds {.page_break_before}

This query answers to the following question:
`What are examples of organisms where compounds were reported to be produced by an organism sharing the same parent taxon, but not the organism itself?`
Link: [https://w.wiki/3359](https://w.wiki/3359)
```SQL
SELECT ?compound ?compoundLabel ?taxonname_with_compound ?taxonname_without_compound

?parent_taxon 

WITH
{ 
  SELECT DISTINCT ?compound ?taxon_with_compound ?parent_taxon 
  WHERE {
    VALUES ?classes {
      wd:Q11173
      wd:Q59199015
    }
    ?compound wdt:P31 ?classes;
              wdt:P235 ?inchikey.
    SERVICE bd:sample { ?compound wdt:P703 ?taxon_with_compound . bd:serviceParam bd:sample.limit 1000 }   
    ?taxon_with_compound wdt:P171 ?parent_taxon .
  }
} AS %taxon_with_compound 
WITH
{ 
  SELECT ?taxon_without_compound ?parent_taxon ?compound 
  WHERE {
    INCLUDE %taxon_with_compound
    ?taxon_without_compound wdt:P171 ?parent_taxon .
    FILTER (?taxon_with_compound != ?taxon_without_compound)
  }
} AS %taxon2 
WHERE {
  INCLUDE %taxon_with_compound
  INCLUDE %taxon2
  FILTER NOT EXISTS {  ?compound wdt:P703 ?taxon_without_compound .}
  ?taxon_with_compound wdt:P225 ?taxonname_with_compound .
  ?taxon_without_compound wdt:P225 ?taxonname_without_compound .
  ?compound rdfs:label ?compoundLabel.
  FILTER(LANG(?compoundLabel) = "en").
}
```

#### Query 6 - *Zephyranthes* sister taxon compounds {.page_break_before}

This query answers to the following question:
`Which *Zephyranthes* species lack compounds known from at least two sister species?`
Link: [https://w.wiki/335x](https://w.wiki/335x)
```SQL
PREFIX target: <http://www.wikidata.org/entity/Q191364> # Zephyranthes
SELECT DISTINCT ?compound ?compoundLabel ?taxon_with_compound ?another_taxon_with_compound ?taxon_without_compound
WITH
{ 
  SELECT DISTINCT ?compound ?taxon_YES_1 ?taxon_YES_2 
  WHERE {
    ?taxon_YES_1 wdt:P171 target: .
    ?taxon_YES_2 wdt:P171 target: .
    FILTER (?taxon_YES_2 != ?taxon_YES_1)
    ?compound wdt:P703 ?taxon_YES_1 . 
    ?compound wdt:P703 ?taxon_YES_2
  }
} AS %taxa_with_compound 
WITH
{ 
  SELECT DISTINCT ?taxon_NO ?compound 
  WHERE {
    INCLUDE %taxa_with_compound
    ?taxon_NO wdt:P171 target: .
    FILTER (?taxon_YES_1 != ?taxon_NO)
  }
} AS %taxon_without_compond 
WHERE {
  INCLUDE %taxa_with_compound
  INCLUDE %taxon_without_compond
  FILTER NOT EXISTS { ?compound wdt:P703 ?taxon_NO .}
  ?taxon_YES_1 wdt:P225 ?taxon_with_compound .
  ?taxon_YES_2 wdt:P225 ?another_taxon_with_compound .
  ?taxon_NO wdt:P225 ?taxon_without_compound .
  ?compound rdfs:label ?compoundLabel.
  FILTER(LANG(?compoundLabel) = "en").
}
```

#### Query 7 - Antibiotic-like compounds {.page_break_before}

This query answers to the following question:
`How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in.`
Link: [https://w.wiki/32Qb](https://w.wiki/32Qb)
```SQL
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX idsm: <https://idsm.elixir-czech.cz/sparql/endpoint/>

SELECT ?parent_taxon ?parent_taxon_name (COUNT(DISTINCT ?compound) AS ?count) WHERE {
  SERVICE idsm:wikidata {
    SERVICE <https://query.wikidata.org/bigdata/namespace/wdq/sparql> {
      ?antibiotic wdt:P279* / wdt:P2868 / wdt:P486 "D000900".
      ?antibiotic wdt:P233 ?smiles.
    }
    
    ?compound sachem:similarCompoundSearch [
      sachem:query ?smiles;
      sachem:cutoff "0.9"^^xsd:double ].
  }

  hint:Prior hint:runFirst true. # hint to evaluate the idsm service first 

  ?compound wdt:P703 ?taxon.
  ?taxon wdt:P171 ?parent_taxon.
  OPTIONAL { ?parent_taxon wdt:P225 ?parent_taxon_name. }
}
GROUP BY ?parent_taxon ?parent_taxon_name
ORDER BY DESC (?count)
```

#### Query 8 - Triples {.page_break_before}

This query answers to the following question:
`Which compounds are found in a biological organism, according to which references?`
Link: [https://w.wiki/335C](https://w.wiki/335C)
```SQL
SELECT DISTINCT ?compound ?compound_inchi ?compound_inchikey ?taxon ?taxon_name ?reference ?reference_doi  WHERE {
  ?compound wdt:P235 ?compound_inchikey;
             p:P703 [
               ps:P703 ?taxon ;
                       prov:wasDerivedFrom / pr:P248 [
                         wdt:P356 ?reference_doi
                       ]
             ] .
  ?compound wdt:P234 ?compound_inchi.
  ?taxon wdt:P225 ?taxon_name.
  ?reference wdt:P356 ?reference_doi.
}
LIMIT 1000
```

#### Query 9 - Indolic scaffold {.page_break_before}

This query answers to the following question:
`Which organisms contain indolic scaffold? Results are grouped by the parent taxon and ordered.`
Link: [https://w.wiki/32KZ](https://w.wiki/32KZ)
```SQL
PREFIX sachem: <http://bioinfo.uochb.cas.cz/rdf/v1.0/sachem#>
PREFIX wd: <http://www.wikidata.org/entity/>
PREFIX p: <http://www.wikidata.org/prop/>
PREFIX idsm: <https://idsm.elixir-czech.cz/sparql/endpoint/>

SELECT ?parent_taxon ?parent_taxon_name (COUNT(?compound) AS ?count) WHERE {
  SERVICE idsm:wikidata {
    ?compound sachem:substructureSearch
      [ sachem:query "NCCC1=CNC2=C1C=CC=C2" ]
  }

  hint:Prior hint:runFirst true. # hint to evaluate the idsm service first

  ?compound p:P703 ?statement.
  ?compound wdt:P235 ?inchikey.
  ?statement ps:P703 ?taxon.
  ?taxon wdt:P171 ?parent_taxon.
  OPTIONAL { ?taxon wdt:P171 ?parent_taxon. }
  OPTIONAL { ?taxon wdt:P225 ?taxon_name. }
  OPTIONAL { ?parent_taxon wdt:P225 ?parent_taxon_name. }
  OPTIONAL { ?taxon wdt:P846 ?taxon_id_gbif. }
  OPTIONAL { ?taxon wdt:P685 ?taxon_id_ncbi. }
}
GROUP BY ?parent_taxon ?parent_taxon_name
ORDER BY DESC (?count)
```

#### Query 10 - Senior NP chemists {.page_break_before}

This query answers to the following question:
`How many structure-organism pairs have been referenced by these authors? (Here, two senior natural products chemists and co-authors of this paper are compared to the late Ferdinand Bohlmann).`
Link: [https://w.wiki/32$m](https://w.wiki/32$m)
```SQL
#defaultView:BarChart
SELECT ?authors_namesLabel (COUNT(DISTINCT(?compound)) AS ?count) WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
    wd:Q79529 # chemical substance
    wd:Q17339814 # group of chemical substances
    wd:Q47154513 # structural class of chemical compounds
  }
  ?compound wdt:P31 ?classes. # instance of
  #?taxon wdt:P225 'Arabidopsis thaliana'
  {
    ?compound p:P703 ?stmt. # found in taxon
    ?stmt ps:P703 ?taxon. # found in taxon
    ?stmt prov:wasDerivedFrom ?ref.
    ?ref pr:P248 ?art. # stated in
  }
  VALUES ?authors_names {
    wd:Q56084663 # JLW
    wd:Q40259636 # GFP
    wd:Q1405133 # A german chemist of the 20th century ... Ferdinand Bohlmann
  }
  ?art wdt:P50 ?authors_names.
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
GROUP BY ?authors_namesLabel
ORDER BY DESC (?count)
```


### SI 4 - Wikidata Entry Creation Tutorial {.page_break_before}

[
<i class="fas fa-scroll fa-lg"></i> **Tutorial for manual creation**<br>
available at [https://osf.io/7dk8h/](https://osf.io/7dk8h/) and [https://oolonek.github.io/dendron/notes/235ba226-b0da-4c23-bbb7-c46c4a65d2f1.html](https://oolonek.github.io/dendron/notes/235ba226-b0da-4c23-bbb7-c46c4a65d2f1.html)
]{.banner .lightblue}

#### Manual addition of a documented structure-organism pair to Wikidata

##### Select a documented structure-organism pair

Throughout this demonstration, we are going to use the following example:
> [trigocherrin A](https://pubchem.ncbi.nlm.nih.gov/compound/101556657) is found in [_Trigonostemon cherrieri_](http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search), as stated in [Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri](https://doi.org/10.1021/OL2030907).

##### Fetch the information for the documented structure-organism pair

###### Structure

Search PubChem for your compound, here [trigocherrin A](https://pubchem.ncbi.nlm.nih.gov/#query=%22trigocherrin%20A%22). This leads to [https://pubchem.ncbi.nlm.nih.gov/compound/101556657](https://pubchem.ncbi.nlm.nih.gov/compound/101556657).

![tutorial-image-01](images/tutorial/2021-02-25-14-15-54.png)

From there, you can fetch the compound's name, InChIKey and InChI as well as its Canonical and Isomeric SMILES.
Here we keep, respectively:

```
* trigocherrin A
* QOVGHDRCAGYGEB-FFZYJECLSA-N
* InChI=1S/C38H36Cl2O12/c1-18(2)35(44)27-19(3)37-25-16-24(31(39)40)28(48-32(43)22-12-8-6-9-13-22)36(25,45)33(47-21(5)42)34(17-46-20(4)41)29(49-34)26(37)30(35)51-38(50-27,52-37)23-14-10-7-11-15-23/h6-16,19,26-30,33,44-45H,1,17H2,2-5H3/t19-,26+,27?,28+,29+,30-,33-,34+,35+,36-,37+,38?/m1/s1
* CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O
* C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O
```

###### Organism

You can check if your organism name is correctly spelled using the Global Names resolver service: [http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search](http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&commit=Search).

![tutorial-image-02](images/tutorial/2021-02-27-18-40-20.png)

Alternatively, you can use [gnfinder](https://github.com/gnames/gnfinder) in your command line interface to check for the spelling of your organism string.
 
```
echo "Trigonostemion cherrieri" | gnfinder find -c -l eng

{
  "metadata": {
    "date": "2021-02-27T18:44:41.640982+01:00",
    "gnfinderVersion": "v0.11.1",
    "withBayes": true,
    "tokensAround": 0,
    "language": "eng",
    "detectLanguage": false,
    "totalWords": 2,
    "totalCandidates": 1,
    "totalNames": 1
  },
  "names": [
    {
      "cardinality": 2,
      "verbatim": "Trigonostemion cherrieri",
      "name": "Trigonostemion cherrieri",
      "odds": 77581.46698350731,
      "start": 0,
      "end": 24,
      "annotationNomenType": "NO_ANNOT",
      "annotation": "",
      "verification": {
        "bestResult": {
          "dataSourceId": 1,
          "dataSourceTitle": "Catalogue of Life",
          "taxonId": "1575885",
          "matchedName": "Trigonostemon cherrieri Veillon",
          "matchedCardinality": 2,
          "matchedCanonicalSimple": "Trigonostemon cherrieri",
          "matchedCanonicalFull": "Trigonostemon cherrieri",
          "classificationPath": "Plantae|Tracheophyta|Magnoliopsida|Malpighiales|Euphorbiaceae|Trigonostemon|Trigonostemon cherrieri",
          "classificationRank": "kingdom|phylum|class|order|family|genus|species",
          "classificationIds": "3939764|3942634|3942724|3942777|3942795|4210752|1575885",
          "editDistance": 1,
          "stemEditDistance": 1,
          "matchType": "FuzzyCanonicalMatch"
        },
        "dataSourcesNum": 13,
        "dataSourceQuality": "HasCuratedSources",
        "retries": 1
      }
    }
  ]
}
```

For misspellings like _Trigonstemion cherrieri_, gnfinder can help resolve them, in this case to _Trigonostemon cherrieri_.


###### Reference

Make sure that you have the correct [Digital Object Identifier (DOI)](https://www.doi.org/doi_handbook/Glossary.html#doi) for it.
For "[Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri](https://doi.org/10.1021/OL2030907)", this is  **10.1021/ol2030907**. Note that DOIs are uppercase-normalized in Wikidata.

##### Check for the presence of your compound in Wikidata <a name="check_compound"></a>

Using the compound's InChIKey (i.e. ```QOVGHDRCAGYGEB-FFZYJECLSA-N``` for trigocherrin A), run a SPARQL query to check if your compound is present in Wikidata or not:


```SQL
SELECT ?item ?itemLabel WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
    wd:Q79529 # chemical substance
    wd:Q17339814 # group of chemical substances
    wd:Q47154513 # structural class of chemical compounds
  }
  ?item wdt:P31 ?classes. # instance of
  ?item wdt:P235 'QOVGHDRCAGYGEB-FFZYJECLSA-N'
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
```
[Try this query](https://w.wiki/32zU). You can adapt it by replacing the InChIKey with the one for your compound.

Alternatively you can use the following Scholia link (replace by your compounds InChIKey) [https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N](https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N) 

If your compound is already present on Wikidata, you can directly skip to the [Add the biological source information](#add_bio) section below.

##### Add your data manually to Wikidata <a name="add_your_data_manually"></a>

First, if you do not have a Wikidata account already, it is advisable that you create one via [https://www.wikidata.org/wiki/Special:CreateAccount](https://www.wikidata.org/wiki/Special:CreateAccount). While an account is not strictly required for manual edits, having one will be useful if you want to contribute more than once, and it helps in getting your contributions recognized. Note that Wikidata accounts are integrated with accounts across the Wikimedia ecosystem, so if you already have an account on, say, any Wikipedia or on Wikispecies, then you can use the same credentials on Wikidata.

If you are unfamiliar with how Wikidata works, you can start by reading the Wikidata introduction page [https://www.wikidata.org/wiki/Wikidata:Introduction](https://www.wikidata.org/wiki/Wikidata:Introduction) and have a look at the Wikidata Tours page [https://www.wikidata.org/wiki/Wikidata:Tours](https://www.wikidata.org/wiki/Wikidata:Tours).

Now that you are all set up, you can go to Wikidata's page for creating new items, [https://www.wikidata.org/wiki/Special:NewItem](https://www.wikidata.org/wiki/Special:NewItem):


![tutorial-image-03](images/tutorial/2021-02-25-14-18-03.png)

An empty page with a new Wikidata identifier is created 

![tutorial-image-04](images/tutorial/2021-02-25-14-18-50.png)

###### Add the chemical compound information <a name="add_chemical"></a>

Create a new statement for ```is an instance of```

![tutorial-image-05](images/tutorial/2021-02-25-14-20-08.png)

and select chemical compound (i.e. [Q11173](https://www.wikidata.org/wiki/Q11173)):

![tutorial-image-06](images/tutorial/2021-02-25-14-20-58.png)

Click ```publish``` to save your changes and make them public.

Since you created a new item about an instance of a chemical compound, the user interface will automatically propose to you a set of additional statements commonly found on items about chemical compounds.

![tutorial-image-07](images/tutorial/2021-02-25-14-22-44.png)

You can then go on and fill these in.

Here, we start with the InChIKey. 
Note the little flag which will automatically tell you if you have some problems with the recently created statements.

![tutorial-image-08](images/tutorial/2021-02-25-14-24-34.png)

Here, Wikidata tells us that if we add an InChiKey, we will need to also add an InChI. Logical, but good to have a reminder !

Let's go ahead and add the InChI string.

Likewise, the addition of an isomeric SMILES string will require us to add a Canonical SMILES.

Note that you might have to copy and paste the SMILES string from PubChem to a plain text editor and then back to Wikidata because of some formatting issues when copy pasting directly from PubChem.

```
C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O

CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O
```

###### Add the biological source information <a name="add_bio"></a>

Now let's add the ```found in taxon``` property ([P703](https://www.wikidata.org/wiki/Property:P703)).

Just click on ```Add a new statement``` and type in the first letters of the property you want to add:

![tutorial-image-09](images/tutorial/2021-02-25-14-31-56.png)

Again, type in the first letters of the taxon, and if the organism is present, it will autocomplete. Here is how this looks like for _Trigonostemon cherrieri_:

![tutorial-image-10](images/tutorial/2021-02-25-14-33-14.png)

Click ```publish``` to save your changes and make them public.


If your target taxon is not yet present on Wikidata and you are sure you have a valid taxon name that is spelled correctly, then you can go to [https://www.wikidata.org/wiki/Special:NewItem](https://www.wikidata.org/wiki/Special:NewItem), as described in the [Add your data manually to Wikidata](#add_your_data_manually) section. For items about taxa, the ```instance of``` statement should have a value ```taxon``` (i.e. [Q16521](https://www.wikidata.org/wiki/Q16521)). As for chemical compounds, the user interface will then suggest to you further statements to add. For taxa, these include taxon name, parent taxon and taxon rank.


###### Add the reference documenting the structure-organism pair <a name="add_ref"></a>

Finally, since we report documented structure-organisms pairs, we need to add the reference for this newly created ```compound found in taxon``` relationship.
This happens on the item about the compound, just below the ```found in taxon``` statement. Click on the ```0 references``` link and then on ```add reference```:

![tutorial-image-11](images/tutorial/2021-02-25-14-35-33.png)

Here, we use the ```stated in``` property ([P248](https://www.wikidata.org/wiki/Property:P248)):

![tutorial-image-12](images/tutorial/2021-02-25-14-36-33.png)

Now, type in the first letters or word of the scientific publication documenting the natural product occurence, autocompletion happens again. Note that multiple publications might have the same title, and that there could be minor differences in punctuation or special characters between the information you and Wikidata have about the same reference. If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check, as outlined in the [Check whether your target reference is already on Wikidata](#check_ref) section.

![tutorial-image-13](images/tutorial/2021-02-25-14-38-54.png)

Click ```publish``` to save your changes and make them public.

![tutorial-image-14](images/tutorial/2021-02-25-14-39-48.png)

###### Check whether your target reference is already on Wikidata <a name="check_ref"></a>

If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check. For our DOI ```10.1021/ol2030907```, the URL [https://scholia.toolforge.org/doi/___10.1021/ol2030907___](https://scholia.toolforge.org/doi/10.1021/ol2030907) will lead you to a [Scholia](https://www.wikidata.org/wiki/Wikidata:Scholia) page about this publication: [https://scholia.toolforge.org/work/Q83059010](https://scholia.toolforge.org/work/Q83059010). Scholia visualizes information from Wikidata, so if it has an entry for your target reference, then so does Wikidata, and both of them will use the same identifier (in this case [Q83059010](https://www.wikidata.org/wiki/Q83059010)). If you prefer to resolve your DOI to Wikidata directly, you can do so by using the uppercase-normalized DOI in the following URL pattern: [https://hub.toolforge.org/P356:**10.1021/OL2030907**](https://hub.toolforge.org/P356:10.1021/OL2030907), which will lead you to the respective Wikidata page, in this case [Q83059010](https://www.wikidata.org/wiki/Q83059010).

If you think that no Wikidata entry exists for your target reference, you can use the DOI in the URL pattern [https://sourcemd.toolforge.org/index_old.php?id=**10.1021/ol2030907**&doit=Check+source](https://sourcemd.toolforge.org/index_old.php?id=10.1021/ol2030907&doit=Check+source), which will trigger a check with both Crossref and Wikidata, and if no Wikidata entry can be found, the metadata from Crossref will be fetched and presented to you for creating the respective Wikidata item semi-automatically. Using such semi-automated workflows does require and account that is a minimum number of days old and has made a minimum number of edits on Wikidata.

If you are interested the annotation of article with topics in Scholia here is a tutorial [https://laurendupuis.github.io/Scholia_tutorial/](https://laurendupuis.github.io/Scholia_tutorial/)


##### Voilà !

You have added your first documented structure-organism relationship to Wikidata and made a valuable contribution to the community.
You can add further statements, e.g. ```molecular formula```, or ```SPLASH code``` for linking to spectral data.

The Wikidata entry [https://www.wikidata.org/wiki/Q105674316](https://www.wikidata.org/wiki/Q105674316) has been started using these instructions.

You can run a SPARQL query and check that everything went smoothly by modifying the InChiKey line in the following [SPARQL query](https://w.wiki/32zb):

```SQL
SELECT ?item ?itemLabel ?taxonLabel ?artLabel WHERE {
  VALUES ?classes {
    wd:Q11173 # chemical compound
    wd:Q59199015 # group of stereoisomers
    wd:Q79529 # chemical substance
    wd:Q17339814 # group of chemical substances
    wd:Q47154513 # structural class of chemical compounds
  }
  ?item wdt:P31 ?classes. # instance of
  ?item wdt:P235 'QOVGHDRCAGYGEB-FFZYJECLSA-N' # InChiKey
  {
    ?item p:P1582 ?stmt. # natural product of taxon
    ?stmt ps:P1582 ?taxon. # natural product of taxon
    OPTIONAL {
      ?stmt prov:wasDerivedFrom ?ref. 
      ?ref pr:P248 ?art. # stated in
    }
  }
  UNION
  {
    ?item p:P703 ?stmt. # found in taxon
    ?stmt ps:P703 ?taxon. # found in taxon
    OPTIONAL {
      ?stmt prov:wasDerivedFrom ?ref.
      ?ref pr:P248 ?art. # stated in
    }
  }
  SERVICE wikibase:label { bd:serviceParam wikibase:language "[AUTO_LANGUAGE],en". }
}
```


### SI 5 - Complement to Figure 7{.page_break_before}

![**Complement to Figure 7**: distribution of β-sitosterol and related chemical parents among families with at least 50 reported compounds present in LOTUS. Script used for the generation of each tree in the figure is the same ([src/4_visualizing/plot_magicTree.R](https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/4_visualizing/plot_magicTree.R)) as for Figure @fig:magicTree as both figures are related.
](images/magicTree_additional.svg "magicTree_additional"){#fig:magicTree-additional tag="SI-5"}


## Citations {.page_break_before}

This list will maagically disappear later on... just here for safety for now!

Afendi [@doi:10/cstrnq]

All natural [@doi:10/bkfd8h]

Allard [@doi:10/ggb79p]

Bandyukova [@doi:10/b4w5q4]

Bierer [@doi:10/ggb7x6]

Bisson IMPS [@doi:10.1021/acs.jmedchem.5b01009]

Bisson NMR [@doi:10.1039/c6np00022c]

Boonen [@doi:10/gh5ptc]

Brunson [@doi:10/gg4ghf]

Campitelli [@campitelliGgnewscaleMultipleFill2021]

Cao [@doi:10/b46rs7]

Capecchi [@doi:10/gjjz4c]

Chamberlain [@chamberlainRcrossrefClientVarious2020]

Choi [@doi:10/f9rhrj]

Cordell 2001 [@doi:10.1002/ptr.890]

Cordell [@doi:10/gh533j]

Cordell [@doi:10.1177/1934578X1701200849]

Cousijn [@doi:10/gh533h]

Cousijn [@doi:10/gfnxpj]

Davis [@doi:10/gh5ps5]

Defossez [@doi:10/ghssx4]

Derese [@deresesolomonMitishambaDatabaseWeb2015]

Dowle [@dowleDataTableExtension2020]

Dührkop [@doi:10/gh5482]

Feunang [@doi:10/gc5tqv]

Flor [@florChorddiagInteractiveChord2020]

Gagolewski [@gagolewskiPackageStringiCharacter2020]

GBIF.org [@GBIFOrg2020]

Gehlenborg [@gehlenborgUpSetRMoreScalable2019]

Giacomoni [@https://hal.archives-ouvertes.fr/hal-01697081]

Graham [@doi:10.1016/b978-008045382-8.00060-5]

Gu [@doi:10/f43xwq]

Günthardt [@doi:10/gd3bm6]

Hatherley [@doi:10/f7gjs6]

Haug [@doi:10.1093/nar/gkz1019]

Heller [@doi:10/6bg]

Helmy [@doi:10/gc4h7x]

Hester [@hesterVroomReadWrite2020]

Horai [@doi:10/d8pdcr]

Huang [@doi:10/gh5ps9]

Hunter [@doi:10.1109/MCSE.2007.55]

Ibezim [@doi:10/gh5ptd]

Jones [@doi:10/gjgdzm]

Jr [@jrHmiscHarrellMiscellaneous2020]

Kessler [@doi:10/gftqcm]

Kim [@doi:10.26434/chemrxiv.12885494.v1]

Kim [@doi:10/gfxmsx]

Kim [@doi:10/f7hnsd]

Klementz [@doi:10/f8cnnk]

Kratochvíl [@doi:10/gf5bv5]

Kratochvíl [@doi:10/gdn52j]

Kuang [@kuangPbmcapplyTrackingProgress2019]

Lang [@langXMLToolsParsing2020]

Lin [@doi:10/ggwrtj]

Loo [@doi:10/ghfr9v]

Lowe [@doi:10/b2zkr9]

Madariaga-Mazón [@doi:10/ghxp29]

Mahto [@mahtoSplitstackshapeStackReshape2019]

McAlpine [@doi:10.1039/c7np00064b]

Wakankensaku [@MainPageWAKANKENSAKU]

Martens [@doi:10/gh6dq2]

Michonneau [@doi:10/f9jgkm]

Mohamed [@wikidata:Q106204599]

Müller [@mullerRSQLiteSQLiteInterface2021]

Nielsen [@wikidata:Q28942417]

Noteborn [@doi:10/b3z2g7]

Ntie-Kang [@doi:10/ggfdw7]

Nupur [@doi:10/f8wjvh]

Olivon [@doi:10/gcg6xw]

Ooms [@wikidata:Q106204620]

Pedersen [@pedersenGgraphImplementationGrammar2020]

Peroni [@doi:10/gf2tnn]

Pierce [@doi:10/gf3j9t]

Pilon [@doi:10/gh5ps7]

Pilón-Jiménez [@doi:10/gh5ptb]

Probst [@doi:10/gjjz4f]

Probst [@doi:10/gcmhc5]

Probst [@doi:10/gjjz4d]

R [@rcoreteamLanguageEnvironmentStatistical2020]

R DBI [@rspecialinterestgroupondatabasesr-sig-dbDBIDatabaseInterface2021]

Rasberry [@doi:10.3897/rio.5.e35820]

Rdkit [@RDKitOpensourceCheminformatics]

Reback [@doi:10/gh6dqz]

Rees [@doi:10/ggkjbt]

Rothwell [@doi:10/gh5ps6]

Rutz [@doi:10/fm73]

Sander [@doi:10/f637rz]

Sawada [@doi:10/f36bkc]

Sedio [@doi:10/ghttk6]

Sharma [@doi:10/f6ss69]

Shinbo [@doi:10.1007/3-540-29782-0_13]

Sievert [@sievertInteractiveWebBasedData2020]

Slenter [@doi:10.1093/nar/gkx1064]

Sorokina [@doi:10/ghssx5]

Sorokina [@doi:10/ggwwcq]

Sorokina [@doi:10/gh6dqw]

Szöcs [@doi:10/gh5ptf]

Tomiki [@doi:10/gh5ps8]

Tsugawa [@doi:10/gfwm27]

DrDuke [@DepartmentAgricultureAgricultural1992]

van Santen [@doi:10/gh5ptm]

Virtanen [@doi:10/ggj45f]

Waagmeester [@doi:10/ggqqc6]

Wang [@doi:10/ggwr93]

Warnes [@warnesGdataVariousProgramming2017]

Weininger [@doi:10/fg6q5f]

Wickham [@wickhamReadxlReadExcel2019]

Wickham [@doi:10/ggddkj]

Wickham [@wickhamRvestEasilyHarvest2020]

Wickham [@wickhamXml2ParseXML2020]

Wilkins [@wilkinsGgfittextFitText2020]

Wilkinson [@doi:10/bdd4]

Willighagen [@doi:10/gbppnb]

Winter [@doi:10/gh5ptg]

Wohlgemuth [@doi:10.1093/bioinformatics/btq476]

Wohlgemuth [@doi:10/gh533g]

Wolfender [@doi:10/gh6dqv]

Xu [@xuGgstarStarLayer2021]

Xu [@doi:10.21203/rs.3.rs-155672/v2]

Yabuzaki [@doi:10/f9x5p2]

Yu [@doi:10/f9qv8x]

Yue [@doi:10/f6hnfz]

Zeng [@doi:10/gh5ptk]

Zhang [@doi:10/gfs7x6]


## References {.page_break_before}

<!-- Explicitly insert bibliography here -->
<div id="refs"></div>
