## Introduction {.page_break_before}

### Evolution of Electronic Natural Product Resources

Natural Products (NP) research is a transdisciplinary field with wide-ranging interests, from fundamental structural aspects of naturally-occurring molecular entities to their effects on living organisms, and extending to the study of chemically-mediated interactions within entire ecosystems. Technological and methodological developments are constantly reshaping NP research, as is typical for a field with a rich legacy of experimental practices (Allard et al., 2018). Knowledge is enabled by and embedded in historical precedent. Technological and conceptual infrastructures are inherently biased simply by emerging in a temporal "state-of-the-art" when published (often recast as anachronistic). To this extent, contemporary bioinformatic approaches enable the (re-)interpretation and (re-)annotation of (existing) datasets that document molecular aspects of biodiversity (Mongia, 2021; Jarmusch et al., 2020). The advancement of NP science has always relied on the development of new technologies. In the past century alone, the rate at which unambiguous identification of new NP entities from biological matrices can be achieved has been reduced from years to days, and in the past few decades, the scale at which new NP discoveries are being reported has increased exponentially, This has resulted in the explosion on new journals, both in print, and increasingly in electronic form. Without a means to process these disparate NP data points, information is fragmented, and knowledge impaired. The efficiency of annotating previously reported NP occurrences or characterizing new entities relies on NP spectral and structural databases (DB), and their maintenance status is a critical factor in this dependency (Tsugawa, 2018). Technological changes over time strongly influence both the quantity and quality of recorded NP data, and editorial practices strongly influence their canonization in academic text corpora (Nissen et al., 2016). A number of generic publication biases are recognized (Lee et al., 2013), they can take the following forms in NP research. The first example is the positivity bias existing in the publication process. Indeed, most NP publishers encourage the reporting of novel and/or bioactive structures only. This does not allow for proper documentation of previously reported and/or inactive compounds (in a given bioassay) and thus severely hinders the comprehensive metabolome coverage of the studied organisms. A second bias can appear at the DB assembly stages, in favor of a specific compound class or a given taxonomic range for example. It therefore seems valuable to establish ways to facilitate reporting of NP occurrences beyond their "publishable" status and across biological or chemical taxa.

Despite the ambiguous definition of “natural” (“All natural,” 2007), and assuming it is reasonable to define a NP as a chemical entity found in a living organism, a future-oriented electronic NP resource should contain, at a minimum, a list of referenced pairs consisting of chemical entities and organisms, where the literature references fully establish the linkages between them.

While well-structured and freely accessible DB exist, they are often concerned primarily with chemical structures (e.g. PubChem (Kim et al., 2019), with over 100M entries) or biological organisms (e.g. GBIF (“GBIF.org,” 2020), with over 1,400M entries), and their scarce interlinkages limit their application for documentation of NP occurrence(s). Currently, no open, cross-kingdom, comprehensive electronic NP resource links NP and their producing organisms, along with referral to the underlying experimental work.

One key shortcoming of currently available electronic NP resources is the provision of structure-organism pairs lacking reference to the original work that describes them. This breaks the crucial evidentiary link required for tracing information back to the original data and assessing its quality. Even valuable efforts for compiling NP data made by commercial products, such as the Dictionary of Natural Products (DNP), lack proper documentation of these critical links.

Pioneering efforts to address such shortcomings led to the establishment of KNApSAck (Shinbo et al., 2006), which is likely the first public, curated electronic NP resource of referenced structure-organism pairs. KNApSAck currently contains 50,000+ structures and 100,000+ structure-organism pairs. However, the organism field is not standardized, and access to the data is not straightforward. Another effort led to the NAPRALERT dataset (Graham and Farnsworth, 2010), which was compiled over five decades from the NP literature, gathering and annotating data derived from over 200,000 primary literature sources. The dataset contains 200,000+ distinct compound names and structural elements, along with 500,000+ records of distinct structure-organism pairs. In total, NAPRALERT represents over 900,000 such records, due to equivalent structure-organism pairs reported in different citations. NAPRALERT currently employs an access model that provides only limited free searches of the dataset. Finally, the NPAtlas (Santen et al., 2019) is a more recent project aimed at complying with the FAIR (Findability, Accessibility, Interoperability, and Reuse) guidelines for digital assets (Wilkinson et al., 2016) and offering web access. While the NPAtlas encourages submission of new compounds with their biological source, it focuses on microbial NP and ignores a wide range of biosynthetically active organisms found in the Eukaryota.

Building on experience with the recently published COlleCtion of Open NatUral producTs (COCONUT) (Sorokina et al., 2021), the LOTUS initiative seeks to address the aforementioned shortcomings. At its current stage of development, LOTUS disseminates 500,000+ referenced structure-organism pairs. After extensive data curation and harmonization, each documented structure-organism pair was standardized at the chemical, biological, and reference levels. These efforts and experiences represent an intensive preliminary curatorial phase, and the first major step towards providing a high quality, computer-interpretable knowledge base capable of transforming NP research data management from a classical (siloed) database approach to an optimally-shared resource.

### Accommodating Principles of FAIRness and TRUSTworthiness for Natural Products Knowledge Management

In awareness of the multi-faceted pitfalls associated with implementing, using, and maintaining classical scientific DBs (Helmy et al., 2016), and to enhance current and future sharing options, the LOTUS initiative selected the Wikidata platform for disseminating its resources. Since its creation, Wikidata has focused on cross-disciplinary and multilingual support. Wikidata is curated and governed collaboratively by a global community of volunteers, about 20,000 of which are contributing monthly. Wikidata currently contains more than 1 billion statements in the form of subject-predicate-object triples. Triples are machine-interpretable and can be enriched with qualifiers and references. Within Wikidata, data triples correspond to ~90 million entries, which can be grouped into classes as diverse as countries, songs, disasters, or chemical compounds. The statements are closely integrated with Wikipedia and serve as the source for its infoboxes. Various workflows have been established for reporting such classes, particularly those of interest to life sciences, such as genes, proteins, diseases, drugs, or biological taxa (Waagmeester et al., 2020).

Building on the principles and experiences described above, the present report introduces the development and implementation of the LOTUS workflow for NP occurrences’ curation and dissemination, which applies both FAIR and TRUST (Transparency, Responsibility, User focus, Sustainability, and Technology) principles (Lin et al., 2020). LOTUS’ data upload and retrieval procedures ensure optimal accessibility by the research community, allowing any researcher to contribute, edit and reuse the data with a clear and open CC0 license (Creative Commons 0).

Despite many advantages, Wikidata hosting has some notable, yet manageable drawbacks. While its SPARQL query language offers a powerful way to query available data, it can also appear intimidating to the less experienced user. Furthermore, some typical queries of molecular electronic NP resources such as structural or spectral searches are not yet available in Wikidata. To bridge this gap, LOTUS is hosted in parallel at [https://lotus.naturalproducts.net](https://lotus.naturalproducts.net) (LNPN) within the naturalproducts.net ecosystem. The Natural Products Online is a portal for the open-source and open-data resources for NP research. In addition to the generalistic COCONUT and LNPN databases, the portal will enable hosting of arbitrary and skinned collections, themed in particular by species or taxonomic clade, by geographic location or by institution, together with a range of cheminformatics tools for NP research. LNPN is periodically updated with the latest LOTUS data. This dual hosting provides an integrated, community-curated, and vast knowledge base (via Wikidata), as well as a NP community-oriented product with tailored search modes (via LNPN).

The LOTUS initiative and its multiple data interaction options establish the basis for transparent and sustainable ways to access, share, and create knowledge on NP occurrence. As a new electronic resource, LOTUS represents an important advancement in the design and deployment of a comprehensive and collaborative NP knowledge base. More broadly, the LOTUS initiative fosters cross-fertilization of the fields of chemistry, biology and associated disciplines.

This article is structured as follows: the first section presents an overview of the LOTUS initiative at its current stage of development. As part of this overview, the central curation and dissemination elements of the LOTUS initiative are explained in detail. The second section addresses the interaction between LOTUS and its end-users, including data retrieval, addition, and editing. The final section is dedicated to the interpretation of LOTUS data and illustrates the dimensions and qualities of the current LOTUS dataset from chemical and biological perspectives.
