<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="Adriano Rutz" />
  <meta name="author" content="Maria Sorokina" />
  <meta name="author" content="Jakub Galgonek" />
  <meta name="author" content="Daniel Mietchen" />
  <meta name="author" content="Egon Willighagen" />
  <meta name="author" content="Arnaud Gaudry" />
  <meta name="author" content="James G. Graham" />
  <meta name="author" content="Ralf Stephan" />
  <meta name="author" content="Roderic Page" />
  <meta name="author" content="Jiří Vondrášek" />
  <meta name="author" content="Christoph Steinbeck" />
  <meta name="author" content="Guido F. Pauli" />
  <meta name="author" content="Jean-Luc Wolfender" />
  <meta name="author" content="Jonathan Bisson" />
  <meta name="author" content="Pierre-Marie Allard" />
  <meta name="dcterms.date" content="2021-05-15" />
  <meta name="keywords" content="to do" />
  <title>The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    pre > code.sourceCode { white-space: pre; position: relative; }
    pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
    pre > code.sourceCode > span:empty { height: 1.2em; }
    code.sourceCode > span { color: inherit; text-decoration: inherit; }
    div.sourceCode { margin: 1em 0; }
    pre.sourceCode { margin: 0; }
    @media screen {
    div.sourceCode { overflow: auto; }
    }
    @media print {
    pre > code.sourceCode { white-space: pre-wrap; }
    pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
    }
    pre.numberSource code
      { counter-reset: source-line 0; }
    pre.numberSource code > span
      { position: relative; left: -4em; counter-increment: source-line; }
    pre.numberSource code > span > a:first-child::before
      { content: counter(source-line);
        position: relative; left: -1em; text-align: right; vertical-align: baseline;
        border: none; display: inline-block;
        -webkit-touch-callout: none; -webkit-user-select: none;
        -khtml-user-select: none; -moz-user-select: none;
        -ms-user-select: none; user-select: none;
        padding: 0 4px; width: 4em;
        color: #aaaaaa;
      }
    pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
    div.sourceCode
      {   }
    @media screen {
    pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
    }
    code span.al { color: #ff0000; font-weight: bold; } /* Alert */
    code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
    code span.at { color: #7d9029; } /* Attribute */
    code span.bn { color: #40a070; } /* BaseN */
    code span.bu { } /* BuiltIn */
    code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
    code span.ch { color: #4070a0; } /* Char */
    code span.cn { color: #880000; } /* Constant */
    code span.co { color: #60a0b0; font-style: italic; } /* Comment */
    code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
    code span.do { color: #ba2121; font-style: italic; } /* Documentation */
    code span.dt { color: #902000; } /* DataType */
    code span.dv { color: #40a070; } /* DecVal */
    code span.er { color: #ff0000; font-weight: bold; } /* Error */
    code span.ex { } /* Extension */
    code span.fl { color: #40a070; } /* Float */
    code span.fu { color: #06287e; } /* Function */
    code span.im { } /* Import */
    code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
    code span.kw { color: #007020; font-weight: bold; } /* Keyword */
    code span.op { color: #666666; } /* Operator */
    code span.ot { color: #007020; } /* Other */
    code span.pp { color: #bc7a00; } /* Preprocessor */
    code span.sc { color: #4070a0; } /* SpecialChar */
    code span.ss { color: #bb6688; } /* SpecialString */
    code span.st { color: #4070a0; } /* String */
    code span.va { color: #19177c; } /* Variable */
    code span.vs { color: #4070a0; } /* VerbatimString */
    code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/main/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta name="citation_title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta property="og:title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta property="twitter:title" content="The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata" />
  <meta name="dc.date" content="2021-05-15" />
  <meta name="citation_publication_date" content="2021-05-15" />
  <meta name="dc.language" content="en-US" />
  <meta name="citation_language" content="en-US" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="Adriano Rutz" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-0443-9902" />
  <meta name="twitter:creator" content="@adafede" />
  <meta name="citation_author" content="Maria Sorokina" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-9359-7149" />
  <meta name="twitter:creator" content="@ms_sorok" />
  <meta name="citation_author" content="Jakub Galgonek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-7038-544X" />
  <meta name="twitter:creator" content="@JGalgonek" />
  <meta name="citation_author" content="Daniel Mietchen" />
  <meta name="citation_author_institution" content="School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, Virginia 22904, United States" />
  <meta name="citation_author_orcid" content="0000-0001-9488-1870" />
  <meta name="twitter:creator" content="@EvoMRI" />
  <meta name="citation_author" content="Egon Willighagen" />
  <meta name="citation_author_institution" content="Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands" />
  <meta name="citation_author_orcid" content="0000-0001-7542-0286" />
  <meta name="twitter:creator" content="@egonwillighagen" />
  <meta name="citation_author" content="Arnaud Gaudry" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-3648-7362" />
  <meta name="citation_author" content="James G. Graham" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0002-7114-8921" />
  <meta name="citation_author" content="Ralf Stephan" />
  <meta name="citation_author_institution" content="Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada" />
  <meta name="citation_author_orcid" content="0000-0002-4650-631X" />
  <meta name="citation_author" content="Roderic Page" />
  <meta name="citation_author_institution" content="IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom" />
  <meta name="citation_author_orcid" content="0000-0002-7101-9767" />
  <meta name="twitter:creator" content="@rdmpage" />
  <meta name="citation_author" content="Jiří Vondrášek" />
  <meta name="citation_author_institution" content="Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic" />
  <meta name="citation_author_orcid" content="0000-0002-6066-973X" />
  <meta name="citation_author" content="Christoph Steinbeck" />
  <meta name="citation_author_institution" content="Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany" />
  <meta name="citation_author_orcid" content="0000-0001-6966-0814" />
  <meta name="twitter:creator" content="@csteinbeck" />
  <meta name="citation_author" content="Guido F. Pauli" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1022-4326" />
  <meta name="citation_author" content="Jean-Luc Wolfender" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0002-0125-952X" />
  <meta name="citation_author" content="Jonathan Bisson" />
  <meta name="citation_author_institution" content="Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_institution" content="Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States" />
  <meta name="citation_author_orcid" content="0000-0003-1640-9989" />
  <meta name="twitter:creator" content="@Bjonnh" />
  <meta name="citation_author" content="Pierre-Marie Allard" />
  <meta name="citation_author_institution" content="School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland" />
  <meta name="citation_author_institution" content="Department of Biology, University of Fribourg, Chemin du Musée 10, 1700 Fribourg, Switzerland" />
  <meta name="citation_author_orcid" content="0000-0003-3389-2191" />
  <meta name="twitter:creator" content="@NatprodCbn" />
  <link rel="canonical" href="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="og:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta property="twitter:url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_fulltext_html_url" content="https://lotusnprod.github.io/lotus-manuscript/" />
  <meta name="citation_pdf_url" content="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://lotusnprod.github.io/lotus-manuscript/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://lotusnprod.github.io/lotus-manuscript/v/e5909db04d96d7702d8044aa199673cef788cc1b/" />
  <meta name="manubot_html_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/e5909db04d96d7702d8044aa199673cef788cc1b/" />
  <meta name="manubot_pdf_url_versioned" content="https://lotusnprod.github.io/lotus-manuscript/v/e5909db04d96d7702d8044aa199673cef788cc1b/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <meta property="og:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/e5909db04d96d7702d8044aa199673cef788cc1b/content/images/thumbnail.png" />
  <meta property="twitter:image" content="https://github.com/lotusnprod/lotus-manuscript/raw/e5909db04d96d7702d8044aa199673cef788cc1b/content/images/thumbnail.png" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
  
  <!-- pandoc-eqnos: equation style -->
  <style>
    .eqnos { display: inline-block; position: relative; width: 100%; }
    .eqnos br { display: none; }
    .eqnos-number { position: absolute; right: 0em; top: 50%; line-height: 0; }
  </style>
</head>
<body>
<!--
  D3 Plugin (third-party) 
  
  Allows you to create complex, dynamic,interactive, data-driven visualizations
  in SVG. See https://d3js.org/.
-->

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/d3/6.2.0/d3.min.js"
  integrity="sha512-C2RveGuPIWqkaLAluvoxyiaN1XYNe5ss11urhZWZYBUA9Ydgj+hfGKPcxCzTwut1/fmjEZR7Ac35f2aycT8Ogw=="
  crossorigin="anonymous"
></script>
<header id="title-block-header">
<h1 class="title">The LOTUS Initiative for Open Natural Products Research: Knowledge Management through Wikidata</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://lotusnprod.github.io/lotus-manuscript/v/e5909db04d96d7702d8044aa199673cef788cc1b/">permalink</a>)
was automatically generated
from <a href="https://github.com/lotusnprod/lotus-manuscript/tree/e5909db04d96d7702d8044aa199673cef788cc1b">lotusnprod/lotus-manuscript@e5909db</a>
on May 15, 2021.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><p><strong>Adriano Rutz</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-0443-9902">0000-0003-0443-9902</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/adafede">adafede</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/adafede">adafede</a><br>
<small>
School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
</small></p></li>
<li><p><strong>Maria Sorokina</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9359-7149">0000-0001-9359-7149</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/mSorok">mSorok</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/ms_sorok">ms_sorok</a><br>
<small>
Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
</small></p></li>
<li><p><strong>Jakub Galgonek</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7038-544X">0000-0002-7038-544X</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/galgonek">galgonek</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/JGalgonek">JGalgonek</a><br>
<small>
Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
</small></p></li>
<li><p><strong>Daniel Mietchen</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-9488-1870">0000-0001-9488-1870</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/Daniel-Mietchen">Daniel-Mietchen</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/EvoMRI">EvoMRI</a><br>
<small>
School of Data Science, University of Virginia, Dell 1 Building, Charlottesville, Virginia 22904, United States
</small></p></li>
<li><p><strong>Egon Willighagen</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-7542-0286">0000-0001-7542-0286</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/egonw">egonw</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/egonwillighagen">egonwillighagen</a><br>
<small>
Dept of Bioinformatics-BiGCaT, NUTRIM, Maastricht University, Universiteitssingel 50, NL-6229 ER, Maastricht, The Netherlands
</small></p></li>
<li><p><strong>Arnaud Gaudry</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-3648-7362">0000-0002-3648-7362</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/ArnaudGaudry">ArnaudGaudry</a><br>
<small>
School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
</small></p></li>
<li><p><strong>James G. Graham</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7114-8921">0000-0002-7114-8921</a><br>
<small>
Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
</small></p></li>
<li><p><strong>Ralf Stephan</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-4650-631X">0000-0002-4650-631X</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/rwst">rwst</a><br>
<small>
Ontario Institute for Cancer Research (OICR), 661 University Ave Suite 510, Toronto, Canada
</small></p></li>
<li><p><strong>Roderic Page</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-7101-9767">0000-0002-7101-9767</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/rdmpage">rdmpage</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/rdmpage">rdmpage</a><br>
<small>
IBAHCM, MVLS, University of Glasgow, Glasgow, United Kingdom
</small></p></li>
<li><p><strong>Jiří Vondrášek</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-6066-973X">0000-0002-6066-973X</a><br>
<small>
Institute of Organic Chemistry and Biochemistry of the CAS, Flemingovo náměstí 2, 166 10, Prague 6, Czech Republic
</small></p></li>
<li><p><strong>Christoph Steinbeck</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0001-6966-0814">0000-0001-6966-0814</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/steinbeck">steinbeck</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/csteinbeck">csteinbeck</a><br>
<small>
Institute for Inorganic and Analytical Chemistry, Friedrich-Schiller-University Jena, Lessingstr. 8, 07732 Jena, Germany
</small></p></li>
<li><p><strong>Guido F. Pauli</strong> <br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1022-4326">0000-0003-1022-4326</a><br>
<small>
Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
</small></p></li>
<li><p><strong>Jean-Luc Wolfender</strong> <sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0002-0125-952X">0000-0002-0125-952X</a><br>
<small>
School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland
</small></p></li>
<li><p><strong>Jonathan Bisson</strong> <sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-1640-9989">0000-0003-1640-9989</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/bjonnh">bjonnh</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/Bjonnh">Bjonnh</a><br>
<small>
Center for Natural Product Technologies and WHO Collaborating Centre for Traditional Medicine (WHO CC/TRM), Pharmacognosy Institute; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States; Department of Pharmaceutical Sciences; College of Pharmacy, University of Illinois at Chicago, 833 South Wood Street, Chicago, Illinois 60612, United States
</small></p></li>
<li><p><strong>Pierre-Marie Allard</strong> <sup><a href="#correspondence">✉</a></sup><br>
<img src="images/orcid.svg" class="inline_icon" alt="ORCID icon" />
<a href="https://orcid.org/0000-0003-3389-2191">0000-0003-3389-2191</a>
· <img src="images/github.svg" class="inline_icon" alt="GitHub icon" />
<a href="https://github.com/oolonek">oolonek</a>
· <img src="images/twitter.svg" class="inline_icon" alt="Twitter icon" />
<a href="https://twitter.com/NatprodCbn">NatprodCbn</a><br>
<small>
School of Pharmaceutical Sciences, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Institute of Pharmaceutical Sciences of Western Switzerland, University of Geneva, CMU - Rue Michel-Servet 1, CH-1211 Geneva 4, Switzerland; Department of Biology, University of Fribourg, Chemin du Musée 10, 1700 Fribourg, Switzerland
</small></p></li>
</ul>
<div id="correspondence">
<p>✉ — correspondence preferred via <a href="https://gitlab.com/groups/lotus7/-/issues">GitLab Issues</a>.
Otherwise, address correspondence to <a href="mailto:jean-luc.wolfender@unige.ch" class="email">jean-luc.wolfender@unige.ch</a>, <a href="mailto:bjo@uic.edu" class="email">bjo@uic.edu</a>, and <a href="mailto:pierre-marie.allard@unifr.ch" class="email">pierre-marie.allard@unifr.ch</a>.</p>
</div>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<p>Contemporary bioinformatic and chemoinformatic capabilities hold promise to reshape knowledge management, analysis, and interpretation of data in natural products research. However, current reliance on a disparate set of non-standardized, insular and specialized databases presents a series of challenges to data access, either within the discipline or to integration and interoperability between related domains. The fundamental elements of exchange are referenced structure-organism pairs that establish relationships between distinct molecular structures and the living organisms from which they were identified. Consolidating and sharing such information via an open platform has strong transformative potential for natural products research, and beyond. This is the ultimate goal of the newly established LOTUS initiative, which has now completed the first steps towards the harmonization, curation, validation, and open dissemination of 700,000+ referenced structure-organism pairs. LOTUS data is hosted on Wikidata and regularly mirrored on <a href="https://lotus.naturalproducts.net">https://lotus.naturalproducts.net</a>. Data sharing within the Wikidata framework broadens data access and interoperability, opening new possibilities for community curation and evolving publication models. Furthermore, embedding LOTUS data into the vast Wikidata knowledge graph will facilitate new biological and chemical insights. The LOTUS initiative represents an important advancement in the design and deployment of a comprehensive and collaborative natural products knowledge base.</p>
<figure>
<img src="images/graphicalAbstract.svg" class="graphical-abstract" alt="" />
</figure>
<h2 class="page_break_before" id="introduction">Introduction</h2>
<h3 id="evolution-of-electronic-natural-product-resources">Evolution of Electronic Natural Product Resources</h3>
<p>Natural Products (NP) research is a transdisciplinary field with wide-ranging interests, from fundamental structural aspects of naturally-occurring molecular entities to their effects on living organisms, and extending to the study of chemically-mediated interactions within entire ecosystems. Technological and methodological developments are constantly reshaping NP research, as is typical for a field with a rich legacy of experimental practices <span class="citation" data-cites="QBTqPbZC">(Allard et al., <a href="#ref-QBTqPbZC" role="doc-biblioref">2018</a>)</span>. Knowledge is enabled by and embedded in historical precedent. Technological and conceptual infrastructures are inherently biased simply by emerging in a temporal “state-of-the-art” when published (often recast as anachronistic). To this extent, contemporary bioinformatic approaches enable the (re-)interpretation and (re-)annotation of (existing) datasets that document molecular aspects of biodiversity <span class="citation" data-cites="8sNOa0nm vgWLelZQ">(Jarmusch et al., <a href="#ref-vgWLelZQ" role="doc-biblioref">2020</a>; Mongia and Mohimani, <a href="#ref-8sNOa0nm" role="doc-biblioref">2021</a>)</span>. The advancement of NP science has always relied on the development of new technologies. In the past century alone, the rate at which unambiguous identification of new NP entities from biological matrices can be achieved has been reduced from years to days, and in the past few decades, the scale at which new NP discoveries are being reported has increased exponentially. Without a means to process these disparate NP data points, information is fragmented, and scientific progress impaired <span class="citation" data-cites="iFXju5sG">(Balietti et al., <a href="#ref-iFXju5sG" role="doc-biblioref">2015</a>)</span>. The efficiency of annotating previously reported NP occurrences or characterizing new entities relies on NP spectral and structural databases (DB), and their maintenance status is a critical factor in this dependency <span class="citation" data-cites="TgDlrv2t">(Tsugawa, <a href="#ref-TgDlrv2t" role="doc-biblioref">2018</a>)</span>. Technological changes over time strongly influence both the quantity and quality of recorded NP data, and editorial practices strongly influence their canonization in academic text corpora <span class="citation" data-cites="17CeNW62K">(Nissen et al., <a href="#ref-17CeNW62K" role="doc-biblioref">2016</a>)</span>. While a number of generic publication biases are recognized <span class="citation" data-cites="WNY3oEAs">(Lee et al., <a href="#ref-WNY3oEAs" role="doc-biblioref">2013</a>)</span>, they can take the following forms in NP research. The first example is the positivity bias existing in the publication process: most NP publishers encourage the reporting of novel and/or bioactive structures only. This does not allow for proper documentation of previously reported and/or inactive compounds (in a given bioassay) and thus severely hinders the comprehensive metabolome coverage of the studied organisms. A second bias can appear at the DB assembly stages, in favor of a specific compound class or a given taxonomic range for example. It therefore seems valuable to establish ways to facilitate reporting of NP occurrences beyond their “publishable” status and across biological or chemical taxa.</p>
<p>Despite the ambiguous definition of “natural” <span class="citation" data-cites="hd3vl6GL">(“All natural,” <a href="#ref-hd3vl6GL" role="doc-biblioref">2007</a>)</span>, and assuming it is reasonable to define a NP as a chemical entity <em>found in</em> a living organism, a future-oriented electronic NP resource should contain, at a minimum, a list of referenced pairs consisting of chemical entities and organisms, where the literature references fully establish the linkages between them.</p>
<p>While well-structured and freely accessible DB exist, they are often concerned primarily with chemical structures (e.g. <a href="https://pubchem.ncbi.nlm.nih.gov/">PubChem</a> <span class="citation" data-cites="1BsHixMtu">(Kim et al., <a href="#ref-1BsHixMtu" role="doc-biblioref">2019</a>)</span>, with over 100M entries) or biological organisms (e.g. <a href="https://www.gbif.org/">GBIF</a> <span class="citation" data-cites="aBMvYmj0">(“GBIF.org,” <a href="#ref-aBMvYmj0" role="doc-biblioref">2020</a>)</span>, with over 1,400M entries), and their scarce interlinkages limit their application for documentation of NP occurrence(s). Currently, no open, cross-kingdom, comprehensive, computer-interpretable electronic NP resource links NP and their producing organisms, along with referral to the underlying experimental work. One key shortcoming of currently available electronic NP resources is the provision of structure-organism pairs lacking reference to the original work that describes them. This breaks the crucial evidentiary link required for tracing information back to the original data and assessing its quality. Even valuable efforts for compiling NP data made by commercial products, such as the <a href="https://dnp.chemnetbase.com/">Dictionary of Natural Products</a> (DNP), lack proper documentation of these critical links.</p>
<p>Pioneering efforts to address such shortcomings led to the establishment of <a href="http://www.knapsackfamily.com/KNApSAcK/">KNApSAck</a> <span class="citation" data-cites="2fEiIH9c">(Shinbo et al., <a href="#ref-2fEiIH9c" role="doc-biblioref">2006</a>)</span>, which is likely the first public, curated electronic NP resource of referenced structure-organism pairs. KNApSAck currently contains 50,000+ structures and 100,000+ structure-organism pairs. However, the organism field is not standardized, and access to the data is not straightforward. Another effort led to the NAPRALERT dataset <span class="citation" data-cites="KmGsdzOE">(Graham and Farnsworth, <a href="#ref-KmGsdzOE" role="doc-biblioref">2010</a>)</span>, which was compiled over five decades from the NP literature, gathering and annotating data derived from over 200,000 primary literature sources. The dataset contains 200,000+ distinct compound names and structural elements, along with 500,000+ records of distinct structure-organism pairs. In total, NAPRALERT represents over 900,000 such records, due to equivalent structure-organism pairs reported in different citations. NAPRALERT currently employs an access model that provides only limited free searches of the dataset. Finally, the <a href="https://www.npatlas.org/">NPAtlas</a> <span class="citation" data-cites="1H1kBOCRr">(van Santen et al., <a href="#ref-1H1kBOCRr" role="doc-biblioref">2019</a>)</span> is a more recent project aimed at complying with the FAIR (<u>F</u>indability, <u>A</u>ccessibility, <u>I</u>nteroperability, and <u>R</u>euse) guidelines for digital assets <span class="citation" data-cites="6DjakjNS">(Wilkinson et al., <a href="#ref-6DjakjNS" role="doc-biblioref">2016</a>)</span> and offering web access. While the NPAtlas encourages submission of new compounds with their biological source, it focuses on microbial NP and ignores a wide range of biosynthetically active organisms found in the Archaeplastida.</p>
<p>Building on experience with the recently published <a href="https://coconut.naturalproducts.net/">COlleCtion of Open NatUral producTs</a> (COCONUT) <span class="citation" data-cites="16ijqhOBz">(Sorokina et al., <a href="#ref-16ijqhOBz" role="doc-biblioref">2021</a>)</span>, the LOTUS initiative seeks to address the aforementioned shortcomings. At its current stage of development, LOTUS disseminates 700,000+ referenced structure-organism pairs. After extensive data curation and harmonization, each documented structure-organism pair was standardized at the chemical, biological, and reference levels. These efforts and experiences represent an intensive preliminary curatorial phase, and the first major step towards providing a high quality, computer-interpretable knowledge base capable of transforming NP research data management from a classical (siloed) database approach to an optimally-shared resource.</p>
<h3 id="accommodating-principles-of-fairness-and-trustworthiness-for-natural-products-knowledge-management">Accommodating Principles of FAIRness and TRUSTworthiness for Natural Products Knowledge Management</h3>
<p>In awareness of the multi-faceted pitfalls associated with implementing, using, and maintaining classical scientific DBs <span class="citation" data-cites="18PujMmok">(Helmy et al., <a href="#ref-18PujMmok" role="doc-biblioref">2016</a>)</span>, and to enhance current and future sharing options, the LOTUS initiative selected the <a href="https://www.wikidata.org/wiki/Wikidata:Main_Page">Wikidata</a> platform for disseminating its resources. Since its creation, Wikidata has focused on cross-disciplinary and multilingual support. Wikidata is curated and governed collaboratively by a global community of volunteers, about 20,000 of which are contributing monthly. Wikidata currently contains more than 1 billion statements in the form of subject-predicate-object triples. Triples are machine-interpretable and can be enriched with qualifiers and references. Within Wikidata, data triples correspond to ~90 million entries, which can be grouped into classes as diverse as countries, songs, disasters, or chemical compounds. The statements are closely integrated with <a href="https://en.wikipedia.org/wiki/Main_Page">Wikipedia</a> and serve as the source for its infoboxes. Various workflows have been established for reporting such classes, particularly those of interest to life sciences, such as genes, proteins, diseases, drugs, or biological taxa <span class="citation" data-cites="HUodcdz1">(Waagmeester et al., <a href="#ref-HUodcdz1" role="doc-biblioref">2020</a>)</span>.</p>
<p>Building on the principles and experiences described above, the present report introduces the development and implementation of the LOTUS workflow for NP occurrences’ curation and dissemination, which applies both FAIR and TRUST (<u>T</u>ransparency, <u>R</u>esponsibility, <u>U</u>ser focus, <u>S</u>ustainability, and <u>T</u>echnology) principles <span class="citation" data-cites="MMp6cQ1x">(Lin et al., <a href="#ref-MMp6cQ1x" role="doc-biblioref">2020</a>)</span>. LOTUS’ data upload and retrieval procedures ensure optimal accessibility by the research community, allowing any researcher to contribute, edit and reuse the data with a clear and open CC0 license (<a href="https://creativecommons.org/publicdomain/zero/1.0/">Creative Commons 0</a>).</p>
<p>Despite many advantages, Wikidata hosting has some notable, yet manageable drawbacks. While its SPARQL query language offers a powerful way to query available data, it can also appear intimidating to the less experienced user. Furthermore, some typical queries of molecular electronic NP resources such as structural or spectral searches are not yet available in Wikidata. To bridge this gap, LOTUS is hosted in parallel at <a href="https://lotus.naturalproducts.net">https://lotus.naturalproducts.net</a> (LNPN) within the naturalproducts.net ecosystem. The <a href="https://naturalproducts.net/">Natural Products Online</a> is a portal for the open-source and open-data resources for NP research. In addition to the generalistic COCONUT and LNPN databases, the portal will enable hosting of arbitrary and skinned collections, themed in particular by species or taxonomic clade, by geographic location or by institution, together with a range of cheminformatics tools for NP research. LNPN is periodically updated with the latest LOTUS data. This dual hosting provides an integrated, community-curated, and vast knowledge base (<em>via</em> Wikidata), as well as a NP community-oriented product with tailored search modes (<em>via</em> LNPN).</p>
<p>The LOTUS initiative and its multiple data interaction options establish the basis for transparent and sustainable ways to access, share, and create knowledge on NP occurrence. As a new electronic resource, LOTUS represents an important advancement in the design and deployment of a comprehensive and collaborative NP knowledge base. More broadly, the LOTUS initiative fosters cross-fertilization of the fields of chemistry, biology and associated disciplines.</p>
<p>This article is structured as follows: the first section presents an overview of the LOTUS initiative at its current stage of development. As part of this overview, the central curation and dissemination elements of the LOTUS initiative are explained in detail. The second section addresses the interaction between LOTUS and its end-users, including data retrieval, addition, and editing. The final section is dedicated to the interpretation of LOTUS data and illustrates the dimensions and qualities of the current LOTUS dataset from chemical and biological perspectives.</p>
<h2 id="results-discussion">Results &amp; Discussion</h2>
<h3 id="blueprint-of-the-lotus-initiative">Blueprint of the LOTUS Initiative</h3>
<p>Building on the standards established by the related Wikidata project (<a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry">Chemistry</a>, <a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Taxonomy">Taxonomy</a>, and <a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Source_MetaData">Source Metadata</a>), a NP chemistry-oriented subproject was created (<a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products">Chemistry/Natural products</a>). Its central data consists of three minimal sufficient objects:</p>
<ul>
<li>A <em>chemical structure object</em>, with associated Simplified Molecular Input Line Entry System (SMILES) <span class="citation" data-cites="t7Cw5tqt">(Weininger, <a href="#ref-t7Cw5tqt" role="doc-biblioref">1988</a>)</span>, International Chemical Identifier (InChI) <span class="citation" data-cites="AP6h5CYA">(Heller et al., <a href="#ref-AP6h5CYA" role="doc-biblioref">2013</a>)</span>, and InChIKey (a hashed version of the InChI).</li>
<li>A <em>biological organism object</em>, with associated taxon name, the taxonomic DB where it was described, and the taxon ID in the respective DB.</li>
<li>A <em>reference object</em> describing the structure-organism pair, with the associated article title and a Digital Object Identifier (DOI), a PubMed (PMID), or PubMed Central (PMCID) ID.</li>
</ul>
<p>As data formats are largely inhomogeneous among existing electronic NP resources, fields related to chemical structure, biological organism, and references are variable and essentially not standardized. Therefore, LOTUS implements multiple stages of harmonization, cleaning and validation (Figure <a href="#fig:workflow">1</a>, stages 1 to 3). LOTUS employs a Single Source of Truth (SSOT, <a href="https://en.wikipedia.org/wiki/Single_source_of_truth">Single_source_of_truth</a>) to ensure data reliability and continuous availability of the latest curated version of LOTUS data in both Wikidata and LNPN (Figure <a href="#fig:workflow">1</a>, stages 4). The SSOT approach consists of a <a href="https://www.postgresql.org/">PostgreSQL</a> DB that structures links and data schemes such that every data element has a single place. To accommodate the addition of data directly from new data sources or at the Wikidata level, the LOTUS processing pipeline is tailored to efficiently include and diffuse novel or curated data. This iterative workflow relies both on data addition and retrieval actions as described in the <a href="#user-interaction-with-lotus-data">Data Interaction</a> section. The overall process leading to referenced and curated structure-organisms pairs is illustrated in Figure <a href="#fig:workflow">1</a> and detailed below.</p>
<div id="fig:workflow" class="fignos">
<figure>
<img src="images/workflow.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 1:</span> <strong>Blueprint of the LOTUS initiative.</strong> Data undergo a four-stage process: (1) Harmonization, (2) Cleaning, (3) Validation, and (4) Dissemination. The process was designed to incorporate future contributions (5), either by the addition of new data from within Wikidata (a) or new sources (b) or via curation of existing data (c).</figcaption>
</figure>
</div>
<p>By design, this iterative process fosters community participation, essential to efficiently document NP occurrences. All stages of the workflow are described on the git sites of the LOTUS initiative at <a href="https://gitlab.com/lotus7">https://gitlab.com/lotus7</a> and <a href="https://github.com/mSorok/LOTUSweb">https://github.com/mSorok/LOTUSweb</a>. At the time of writing, 700,000+ LOTUS entries contained a curated chemical structure, biological organism, and reference and were available on both Wikidata and LNPN. As the LOTUS data volume is expected to increase over time, a frozen (as of 2021-05-05) tabular version of this dataset with its associated metadata is made available at <a href="https://osf.io/hgjdb/">https://osf.io/hgjdb/</a>.</p>
<h3 id="data-harmonization">Data Harmonization</h3>
<p>Multiple data sources were processed as described hereafter. All publicly accessible electronic NP resources included in COCONUT that contain referenced structure-organism pairs were considered as initial input. The data were complemented with COCONUT’s own structure-organism documented pairs <span class="citation" data-cites="R06ZE1xj">(Sorokina and Steinbeck, <a href="#ref-R06ZE1xj" role="doc-biblioref">2020</a><a href="#ref-R06ZE1xj" role="doc-biblioref">a</a>)</span>, as well as the following additional electronic NP resources: Dr. Duke <span class="citation" data-cites="1HQ8C1kF4">(“U.S. Department of Agriculture, Agricultural Research Service. Dr. Duke’s Phytochemical and Ethnobotanical Databases.” <a href="#ref-1HQ8C1kF4" role="doc-biblioref">1992–2016</a>)</span>, Cyanometdb <span class="citation" data-cites="OhONtrcV">(Jones et al., <a href="#ref-OhONtrcV" role="doc-biblioref">2021</a>)</span>, Datawarrior <span class="citation" data-cites="bOvsdNwM">(Sander et al., <a href="#ref-bOvsdNwM" role="doc-biblioref">2015</a>)</span>, a subset of NAPRALERT, Wakankensaku <span class="citation" data-cites="1CVDln27l">(“Main Page - WAKANKENSAKU,” <a href="#ref-1CVDln27l" role="doc-biblioref">n.d.</a>)</span>, and DiaNat-DB <span class="citation" data-cites="dyiEPToF">(Madariaga-Mazón et al., <a href="#ref-dyiEPToF" role="doc-biblioref">2021</a>)</span>.</p>
<p>The contacts of the electronic NP resources not explicitly licensed as open were individually reached for permission to access and reuse data. A detailed list of data sources and related information is available as <a href="#si-1-data-sources-list">SI-1</a>. All necessary scripts for data gathering and harmonization can be found in the <a href="https://gitlab.com/lotus7/lotusProcessor">lotusProcessor</a> repository in the <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/1_gathering">src/1_gathering</a> directory. All subsequent and future iterations that include additional data sources, either updated information from the same data sources or new data, will involve a comparison of the new with previously gathered data at the SSOT level to ensure that the data is only curated once.</p>
<h3 id="data-cleaning-vaidation">Data Cleaning &amp; Vaidation</h3>
<p>As shown in Figure <a href="#fig:workflow">1</a>, data curation consisted of three stages: harmonization, cleaning, and validation. Thereby, after the harmonization stage, each of the three central objects of the referenced pairs - chemical compounds, biological organisms, and reference - were cleaned. Given the data size (2.5M+ initial entries), manual validation was unfeasible. Curating the references turned out to be a particularly challenging part of the process. Whereas organisms are typically reported by at least their vernacular or scientific denomination, and structures via their SMILES, InChI, InChIKey, or image (not covered in this work), references suffer from largely insufficient reporting standards. Despite relatively poor standardization of the initial reference field, proper referencing remains an indispensable way to establish the validity of structure-organism pairs. Better reporting practices, supported by new tools such as <a href="https://scholia.toolforge.org/">Scholia</a> <span class="citation" data-cites="jqMOfVhn ZG9WyHw3">(Nielsen et al., <a href="#ref-jqMOfVhn" role="doc-biblioref">2017</a>; Rasberry et al., <a href="#ref-ZG9WyHw3" role="doc-biblioref">2019</a>)</span>, and relying on Wikidata, <a href="https://fatcat.wiki/">Fatcat</a>, or <a href="https://www.semanticscholar.org/">Semantic Scholar</a> should improve reference-related information retrieval in the future.</p>
<p>In addition to curating the entries during data processing, 420 referenced structure-organism pairs were selected for manual validation. An entry was considered as valid if: <em>i)</em> the structure (in the form of any structural descriptor that could be linked to the final sanitized InChIKey) was described in the reference <em>ii)</em> the containing organism (as any organism descriptor that could be linked to the accepted canonical name) was described in the reference and <em>iii)</em> the reference was describing the occurrence of the chemical structure in the biological organism. This process allowed us to establish rules for automatic filtering and validation of the entries. The filtering was then applied to all entries. To confirm the efficacy of the filtering process, a new subset of 100 diverse, automatically curated, and automatically validated entries was manually checked, yielding a rate of 97% of true positives. The detailed results of the two manual validation steps are reported in Supporting Information <a href="#si-2-summary-of-the-validation-statistics">SI-2</a>. The resulting data is also available in the dataset shared at <a href="https://osf.io/c8hgb/">https://osf.io/c8hgb/</a>.</p>
<div id="tbl:examples" class="tablenos">
<table>
<caption><span>Table 1:</span> Example of a referenced structure-organism pair before and after curation </caption>
<colgroup>
<col style="width: 18%" />
<col style="width: 27%" />
<col style="width: 27%" />
<col style="width: 27%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;"></th>
<th style="text-align: center;">Structure</th>
<th style="text-align: center;">Organism</th>
<th style="text-align: center;">Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong>Before curation</strong></td>
<td style="text-align: center;">Cyathocaline</td>
<td style="text-align: center;">Stem bark of Cyathocalyx zeylanica CHAMP. ex HOOK. f. &amp; THOMS. (Annonaceae)</td>
<td style="text-align: center;">Wijeratne E. M. K., de Silva L. B., Kikuchi T., Tezuka Y., Gunatilaka A. A. L., Kingston D. G. I., J. Nat. Prod., 58, 459-462 (1995).</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>After curation</strong></td>
<td style="text-align: center;">VFIIVOHWCNHINZ-UHFFFAOYSA-N</td>
<td style="text-align: center;">Cyathocalyx zeylanicus</td>
<td style="text-align: center;">10.1021/NP50117A020</td>
</tr>
</tbody>
</table>
</div>
<p>Table <a href="#tbl:examples">1</a> shows an example of a referenced structure-organism pair before and after curation. This process resolved the structure to an InChIKey, the organism to a valid taxonomic name, and the reference to a DOI, thereby completing the essential referenced structure-organism pair. Challenging examples encountered during the development of the curation process were compiled in an edge case table (<a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/preprint/tests/tests.tsv">tests/tests.tsv</a>) to allow for automated unit testing. These tests allow a continuous revalidation of any change made to the code, ensuring that corrected errors will not reappear.</p>
<div id="fig:alluvial" class="fignos">
<figure>
<img src="images/alluvial.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 2:</span> <strong>Alluvial plot of the data transformation flow within LOTUS during the automated curation and validation processes.</strong> The figure also reflects the relative proportions of the data stream in terms of the contributions from the various sources (“source” block, left), the composition of the harmonized subcategories (“original subcategory” block, middle), and the validated data after curation (“cleaned category” block, right). Automatically validated entries are represented in green, rejected entries in blue.</figcaption>
</figure>
</div>
<p>The alluvial plot in Figure <a href="#fig:alluvial">2</a> illustrates the individual contribution of each <em>source</em> and <em>original</em> <em>subcategory</em> that led to the <em>cleaned</em> <em>categories</em>: structure, organism, and reference. For example, the graph highlights the essential contribution of the DOI category of references contained in NAPRALERT towards the current set of validated references in LOTUS. The combination of the results of the automated curation pipeline and the manually curated entries led to the establishment of four categories (manually validated, manually rejected, automatically validated, and automatically rejected) of the documented structure-organism pairs that formed the processed part of the SSOT. Out of a total of 2.5M+ pairs, the manual and automatic validation retained 700,000+ pairs (~30 %), which were then selected for dissemination on Wikidata. The disseminated data contains 250,000+ unique chemical structures, 30,000+ distinct organisms, and 75,000+ references.</p>
<h3 id="data-dissemination">Data Dissemination</h3>
<p>Research worldwide can benefit the most when all results of published scientific studies are fully accessible immediately upon publication. This concept is considered the foundation of scientific investigation and a prerequisite for effectively directing new research efforts based on prior information. To achieve this, research results have to be made publicly available and reusable. As computers are now the main investigation tool for a growing number of scientists, all research data including those in publications should be disseminated in computer-readable format, following the FAIR principles. LOTUS uses Wikidata as a repository for referenced structure-organism pairs, as this allows documented research data to be integrated with a large, pre-existing and extensible body of chemical and biological knowledge.</p>
<p>The dynamic nature of Wikidata fosters the continuous curation of deposited data through the user community. Independence from individual and institutional funding represents another major advantage of Wikidata. The Wikidata knowledge base and the option to use elaborate SPARQL queries allow the exploration of the dataset from a sheer unlimited number of angles. The openness of Wikidata also offers unprecedented opportunities for community curation, which will support, if not guarantee, a dynamic and evolving data repository. At the same time, certain limitations of this approach can be anticipated. Despite (or possibly due to) their power, SPARQL queries are complex and often require a relatively in-depth understanding of the models and data structure. This involves a steep learning curve which tends to discourage end-users. Furthermore, traditional ways to query electronic NP resources such as structural or spectral searches are currently not within the scope of Wikidata and, thus, are addressed in LNPN. Using the pre-existing COCONUT template, LNPN hosting allows the user to perform structural searches by drawing a molecule, thereby addressing the current lack of structural search possibilities in Wikidata. Since metabolite profiling by LC-MS is now routinely used for the chemical composition assessment of natural extracts, future versions of LOTUS and COCONUT are envisioned to be augmented by predicted MS spectra and hosted at <a href="https://naturalproducts.net/">https://naturalproducts.net/</a> to allow mass, fragment, and spectral-based queries. To facilitate queries focused on specific taxa (e.g., “return all molecules found in the Asteraceae family”), a unified taxonomy is paramount. As the taxonomy of living organisms is a complex and always evolving field, all the taxon identifiers from all accepted taxonomic DB for a given taxon name were kept. This implies that for a given name, multiple taxonomies from different taxonomic DB are allowed. Initiatives such as the <a href="https://tree.opentreeoflife.org/">Open Tree of Life</a> (OTL) <span class="citation" data-cites="1F083MXGq">(Rees and Cranston, <a href="#ref-1F083MXGq" role="doc-biblioref">2017</a>)</span> will help to gradually reduce these discrepancies, the Wikidata platform should support such developments. OTL also benefits from regular expert curation and new data. As the taxonomic identifier property for this resource did not exist in Wikidata, its creation was requested and obtained. The property is now available as “Open Tree of Life ID” (<a href="https://www.wikidata.org/wiki/Property:P9157">P9157</a>).</p>
<p>Following the previously described curation process, all validated entries have been made available through Wikidata and LNPN. LNPN will be regularly mirroring Wikidata LOTUS through the SSOT as described in Figure <a href="#fig:workflow">1</a>.</p>
<h3 id="user-interaction-with-lotus-data">User Interaction with LOTUS Data</h3>
<p>The possibilities to interact with the LOTUS data are numerous. The following gives examples on how to retrieve, add, and edit LOTUS data.</p>
<h4 id="data-retrieval">Data Retrieval</h4>
<p>LOTUS data can be queried and retrieved either directly in Wikidata or on LNPN, both of which have distinct advantages. While Wikidata offers modularity at the cost of potentially complex access to the data, LNPN has a Graphical User Interface with capabilities of drawing chemical structure, simplified structural or biological filtering, and advanced chemical descriptors, albeit with a more rigid structure. For bulk download, a frozen version of LOTUS data (timestamp of 2021-05-05) is also available at <a href="https://osf.io/c8hgb/">https://osf.io/c8hgb/</a>. More refined approaches to the direct interrogation of the up-to-date LOTUS data both in Wikidata and LNPN are detailed in the following.</p>
<h5 id="wikidata">Wikidata</h5>
<p>The easiest way to search for NP occurrence information in Wikidata is by typing the name of a chemical structure directly into the “Search Wikidata’’ field. For example, by typing “erysodine”, the user will land on the page of this compound (<a href="https://www.wikidata.org/wiki/Q27265641">Q27265641</a>). Scrolling down to the “found in taxon” statement will allow the user to view the biological organisms reported to contain this NP (Figure <a href="#fig:wdpage">3</a>). Clicking the reference link under each taxon name links to the publication(s) documenting the occurrence.</p>
<div id="fig:wdpage" class="fignos">
<figure>
<img src="images/erysodine.png" style="width:100.0%" alt="" /><figcaption><span>Figure 3:</span> <strong>Illustration of the “found in taxon” statement section on the Wikidata page</strong> of erysodine <a href="https://www.wikidata.org/wiki/Q27265641">Q27265641</a> showing a selection of containing taxa and the references documenting these occurrences.</figcaption>
</figure>
</div>
<p>The typical approach of more elaborated queries consists of writing SPARQL queries using the <a href="https://query.wikidata.org/">Wikidata Query Service</a> or a direct connection to the SPARQL endpoint. Below are some examples from simple to more elaborated queries, demonstrating what can be done using this approach. The full-text queries with explanations are included in <a href="#si-3-wikidata-sparql-queries">SI-3</a>.</p>
<div id="tbl:queries" class="tablenos">
<table>
<caption><span>Table 2:</span> Potential questions about referenced structure-organism relationships and the corresponding Wikidata SPARQL query that provides an answer. </caption>
<colgroup>
<col style="width: 63%" />
<col style="width: 36%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Question</th>
<th style="text-align: center;">Wikidata SPARQL query</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">What are the compounds present in Mouse-ear cress (<em>Arabidopsis thaliana</em>)?</td>
<td style="text-align: center;"><a href="https://w.wiki/3HMX">https://w.wiki/3HMX</a></td>
</tr>
<tr class="even">
<td style="text-align: center;">Which organisms are known to contain <em>β</em>-sitosterol?</td>
<td style="text-align: center;"><a href="https://w.wiki/3HLy">https://w.wiki/3HLy</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;">Which organisms are known to contain stereoisomers of <em>β</em>-sitosterol?</td>
<td style="text-align: center;"><a href="https://w.wiki/3Jgs">https://w.wiki/3Jgs</a></td>
</tr>
<tr class="even">
<td style="text-align: center;">Which pigments are found in which taxa, according to which reference?</td>
<td style="text-align: center;"><a href="https://w.wiki/3H3o">https://w.wiki/3H3o</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;">What are examples of organisms where compounds were found in an organism sharing the same parent taxon, but not the organism itself?</td>
<td style="text-align: center;"><a href="https://w.wiki/3HM6">https://w.wiki/3HM6</a></td>
</tr>
<tr class="even">
<td style="text-align: center;">Which <em>Zephyranthes</em> species lack compounds known from at least two species in the genus?</td>
<td style="text-align: center;"><a href="https://w.wiki/3HJf">https://w.wiki/3HJf</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;">How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in.</td>
<td style="text-align: center;"><a href="https://w.wiki/3HMA">https://w.wiki/3HMA</a></td>
</tr>
<tr class="even">
<td style="text-align: center;">Which are the available referenced structure-organism pairs? (example limited to 1000 results)</td>
<td style="text-align: center;"><a href="https://w.wiki/3JpE">https://w.wiki/3JpE</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;">Which organisms contain indolic scaffolds? Count occurences, group and order the results by the parent taxon.</td>
<td style="text-align: center;"><a href="https://w.wiki/3HMD">https://w.wiki/3HMD</a></td>
</tr>
<tr class="even">
<td style="text-align: center;">How many structure-organism pairs have been referenced by certain authors? (Here, two senior natural products chemists and co-authors of this paper are compared to the late Ferdinand Bohlmann).</td>
<td style="text-align: center;"><a href="https://w.wiki/3HML">https://w.wiki/3HML</a></td>
</tr>
</tbody>
</table>
</div>
<p>The queries presented in Table <a href="#tbl:queries">2</a> are only a few examples, and many other ways of interrogating LOTUS can be formulated. Generic queries can also be used, for example, for hypothesis generation when starting a research project. For instance, a generic SPARQL query - listed in Table <a href="#tbl:queries">2</a> as “Which are the available referenced structure-organism pairs?” - retrieves all structures, identified by their InChIKey (<a href="https://www.wikidata.org/wiki/Property:P235">P235</a>), which contain found in taxon (<a href="https://www.wikidata.org/wiki/Property:P703">P703</a>) statements that are stated in (<a href="https://www.wikidata.org/wiki/Property:P248">P248</a>) a bibliographic reference: <a href="https://w.wiki/3JpE">https://w.wiki/3JpE</a>. Data can then be exported in various formats, such as classical tabular formats, json, or html tables (see Download tab on the lower right of the query frame). At the time of writing (2021-05-05), this query returned 797,123 entries; a frozen query result is available at <a href="https://osf.io/thqaw/">https://osf.io/thqaw/</a>.</p>
<p>Targeted queries allowing to interrogate LOTUS data from the perspective of one of the three objects forming the referenced structure-organism pairs can be also built. Users can, for example, retrieve a list of all structures reported from a given organism, such as all structures reported from <em>Arabidopsis thaliana</em> (<a href="https://www.wikidata.org/wiki/Q158695">Q158695</a>) (<a href="https://w.wiki/3HLn">https://w.wiki/3HLn</a>). Alternatively, all organisms containing a given chemical can be queried via its structure, such as in the search for all organisms where <em>β</em>-sitosterol (<a href="https://www.wikidata.org/wiki/Q121802">Q121802</a>) was found in (<a href="https://w.wiki/3HLy">https://w.wiki/3HLy</a>). For programmatic access, the <a href="https://gitlab.com/lotus7/wikidataLotusExporter">WikidataLotusExporter</a> repository also allows data retrieval in RDF format and as TSV tables.</p>
<p>As indicated, certain types of queries that are customary in existing molecular electronic resources, such as structure or similarity searches, are not directly available in Wikidata as SPARQL does not natively support a simple integration of such queries. To address this issue, Galgonek et al. developed an in-house SPARQL engine that allows utilization of Sachem, a high-performance chemical DB cartridge for PostgreSQL for fingerprint-guided substructure and similarity search <span class="citation" data-cites="mjVvWLK2">(Kratochvíl et al., <a href="#ref-mjVvWLK2" role="doc-biblioref">2018</a>)</span>. The engine is used by the Integrated Database of Small Molecules (IDSM) that operates, among other things, several dedicated endpoints allowing structural search in selected small-molecule datasets via SPARQL <span class="citation" data-cites="13RtkyRi5">(Kratochvíl et al., <a href="#ref-13RtkyRi5" role="doc-biblioref">2019</a>)</span>. To allow substructure and similarity searches via SPARQL also on compounds from Wikidata, a <a href="https://idsm.elixir-czech.cz/sparql/endpoint/wikidata">dedicated IDSM/Sachem endpoint</a> was created for the LOTUS project. The endpoint indexes isomeric (<a href="https://www.wikidata.org/wiki/Property:P2017">P2017</a>) and canonical (<a href="https://www.wikidata.org/wiki/Property:P233">P233</a>) SMILES code available in Wikidata. To ensure that data is kept up-to-date, SMILES codes are automatically downloaded from Wikidata daily. The endpoint allows users to run <a href="https://www.w3.org/TR/sparql11-federated-query/">federated queries</a> and, thereby, proceed to structure-oriented searches on the LOTUS data hosted at Wikidata. For example, the SPARQL query <a href="https://w.wiki/3HMD">https://w.wiki/3HMD</a> returns a list of all organisms that produce NP with an indolic scaffold. The output is aggregated at the parent taxa level of the containing organisms and ranked by the number of scaffold occurrences.</p>
<h5 id="lotus.naturalproducts.net-lnpn">Lotus.NaturalProducts.Net (LNPN)</h5>
<p>In the search field of the LNPN interface (<a href="https://lotus.naturalproducts.net/">https://lotus.naturalproducts.net/</a>), simple queries can be achieved by typing the molecule name (e.g., <a href="https://lotus.naturalproducts.net/search/simple/protopine">protopine</a>) or pasting a SMILES, InChI, InChIKey string, or a Wikidata identifier. All compounds reported from a given organism can be found by entering the organism name at the species or any higher taxa level (e.g. <a href="https://lotus.naturalproducts.net/search/simple/Tabernanthe%20iboga"><em>Tabernanthe iboga</em></a>). Compound search by chemical class is also possible.</p>
<p>Alternatively, a structure can be directly drawn in the structure search interface (<a href="https://lotus.naturalproducts.net/search/structure">https://lotus.naturalproducts.net/search/structure</a>), where the user can also decide on the nature of the structure search (exact, similarity, substructure search). Refined search mode combining multiple search criteria, in particular physicochemical properties, is available in the advanced search interface (<a href="https://lotus.naturalproducts.net/search/advanced">https://lotus.naturalproducts.net/search/advanced</a>).</p>
<p>Within LNPN, LOTUS bulk data can be retrieved as SDF or SMILES files, or as a complete MongoDB dump via <a href="https://lotus.naturalproducts.net/download">https://lotus.naturalproducts.net/download</a>. Extensive documentation describing the search possibilities and data entries is available at <a href="https://lotus.naturalproducts.net/documentation">https://lotus.naturalproducts.net/documentation</a>. LNPN can also be queried via the application programming interface (API) as described in the documentation.</p>
<h4 id="data-addition-and-evolution">Data Addition and Evolution</h4>
<p>One major advantage of the LOTUS architecture is that every user has the option to contribute to the NP occurrences documentation effort by adding new or editing existing data. As all LOTUS data applies the SSOT mechanism, reprocessing of previously treated elements is avoided. However, at the moment, the SSOT channels are not open to the public for direct write access to maintain data coherence and evolution of the SSOT scheme. For now, the users can employ the following approaches to add or modify data in LOTUS.</p>
<h5 id="sources">Sources</h5>
<p>LOTUS data management involves regular re-importing of both current and new data sources. New and edited information from these electronic NP resources will be checked against the SSOT. If absent or different, data will be passed through the curation pipeline and subsequently stored in the SSOT. Accordingly, by contributing to external electronic NP resources, any researcher has a means of providing new data for LOTUS, keeping in mind the inevitable delay between data addition and subsequent inclusion into LOTUS.</p>
<h5 id="wikidata-1">Wikidata</h5>
<p>The currently favored approach to add new data to LOTUS is to edit Wikidata entries directly. Newly edited data will then be imported into the SSOT repository. There are several ways to interact with Wikidata which depend on the technical skills of the user and the volume of data to be imported/modified.</p>
<h6 id="manual-upload">Manual Upload</h6>
<p>Any researcher interested in reporting NP occurrences can manually add the data directly in Wikidata, without any particular technical knowledge requirement. The only prerequisite is a Wikidata account and following the <a href="https://www.wikidata.org/wiki/Wikidata:Tours">general object editing guidelines</a>. Regarding the addition of NP-centered objects (i.e., documented structure-organisms pairs), users shall refer to the <a href="https://www.wikidata.org/wiki/Wikidata:WikiProject_Chemistry/Natural_products">WikiProject Chemistry/Natural products</a> group page.</p>
<p>A tutorial for the manual creation and upload of a documented structure-organism pair to Wikidata is available in <a href="#si-3-wikidata-sparql-queries">SI-3</a>. While direct Wikidata upload is possible, contributors are encouraged to use the LOTUS curation pipeline as a preliminary step to strengthen the initial data quality. The added data will therefore benefit from the curation and validation stages implemented in the LOTUS processing pipeline.</p>
<h6 id="batch-and-automated-upload">Batch and Automated Upload</h6>
<p>Through the initial curation process described previously, 700,000+ referenced structure-organism pairs were validated for Wikidata upload. To automate this process, a set of programs were written to automatically process the curated outputs, group references, organisms, and compounds, check if they are already present in Wikidata (using SPARQL and direct Wikidata querying), and insert or update the entities as needed (i.e., upserting). These scripts can be used for future batch upload of properly curated and referenced structure-organism pairs to Wikidata. Programs for data addition to Wikidata can be found in the repository <a href="https://gitlab.com/lotus7/wikidataLotusImporter">WikidataLotusImporter</a>. The following <a href="https://xtools.wmflabs.org/ec/www.wikidata.org/NPImporterBot">Xtools page</a> offers an overview of the latest activity performed by our <a href="https://www.wikidata.org/wiki/User:NPImporterBothttps://www.wikidata.org/wiki/User:NPImporterBot">NPimporterBot</a>, using those programs.</p>
<h4 id="data-edition">Data Edition</h4>
<p>Even if correct at a given time point, scientific advances can invalidate or update previously uploaded data. Thus, the possibility to continuously edit the data is desirable and guarantees data quality and sustainability. Community-maintained knowledge bases such as Wikidata encourage such a process. Wikidata presents the advantage of allowing both manual and automated correction. Field-specific robots such as <a href="https://www.wikidata.org/wiki/User:SuccuBot">SuccuBot</a>, <a href="https://www.wikidata.org/wiki/User:KrBot">KrBot</a>, <a href="https://www.wikidata.org/wiki/User:Pi_bot">Pi_bot</a>, and <a href="https://www.wikidata.org/wiki/User:ProteinBoxBot">ProteinBoxBot</a> or our <a href="https://www.wikidata.org/wiki/User:NPImporterBot">NPimporterBot</a> went through an approval process. The robots are capable of performing thousands of edits without the need for human input. This automation helps reduce the amount of incorrect data that would otherwise require manual editing. However, manual curation by human experts remains irreplaceable as a standard. Users who value this approach and are interested in contributing are invited to follow the manual curation tutorial in <a href="#si-4--wikidata-entry-creation-tutorial">SI-4</a>.</p>
<p>The <a href="https://scholia.toolforge.org/">Scholia platform</a> provides a visual interface to display the links among Wikidata objects such as researchers, topics, species or chemicals. It now provides an interesting way to view the chemical compounds found in a given biological organism (see here for the metabolome view of <a href="https://scholia.toolforge.org/taxon/Q311710"><em>Eurycoma longifolia</em></a>). If Scholia currently does not offer a direct editing interface for scientific references, it still allows users to proceed to convenient batch editing via <a href="https://www.wikidata.org/wiki/Help:QuickStatements">Quick Statements</a>. The adaptation of such a framework to edit the documented structure-pairs in the LOTUS initiative could thus facilitate the capture of future expert curation, especially manual efforts that cannot be replaced by automated scripts.</p>
<h3 id="data-interpretation">Data Interpretation</h3>
<p>To illustrate the nature and dimensions of the LOTUS dataset, some selected examples of data interpretation are shown. First, the repartition of chemical structures among four important NP reservoirs: plants, fungi, animals and bacteria (Table <a href="#tbl:repartition">3</a>). Then, the distribution of biological organisms according to the number of related chemical structures and likewise the distribution of chemical structures across biological organisms are illustrated (Figure <a href="#fig:distribution">4</a>). Furthermore, the individual electronic NP resources participation in LOTUS data is resumed using the UpSet plot depiction, which allows the visualization of intersections in data sets (Figure <a href="#fig:upset">5</a>). Across these figures we take again the two previous examples, i.e, <em>β</em>-sitosterol as chemical structure and <em>Arabidopsis thaliana</em> as biological organism because of their well-documented statuses. Finally, a biologically-interpreted chemical tree and a chemically-interpreted biological tree are presented (Figure <a href="#fig:tmap">6</a> and Figure <a href="#fig:magicTree">7</a>). The examples illustrate the overall chemical and biological coverage of LOTUS by linking family-specific classes of chemical structures to their taxonomic position. Table <a href="#tbl:repartition">3</a>, Figures <a href="#fig:distribution">4</a>, <a href="#fig:tmap">6</a>, and <a href="#fig:magicTree">7</a> were generated using the frozen data (2021-05-05 timestamp), which is available for download at <a href="https://osf.io/c8hgb/">https://osf.io/c8hgb/</a>. Figure <a href="#fig:upset">5</a> required a dataset containing information from DNP and the complete data used for its generation is therefore not available for public distribution. All scripts used for the generation of the figures (including <a href="#si-5-complement-to-figure-7">SI-5</a>) are available in the <a href="https://gitlab.com/lotus7/lotusProcessorv">lotusProcessor</a> repository in the <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/4_visualizing">src/4_visualizing</a> directory for reproducibility.</p>
<h4 id="repartition-of-chemical-structures-across-reported-biological-organisms-in-lotus">Repartition of Chemical Structures across reported Biological Organisms in LOTUS</h4>
<p>The Table 3 summarizes the repartition of chemical structures (or planar structures) and their chemical classes (according to NPClassifier <span class="citation" data-cites="lUVqn8aT">(kim et al., <a href="#ref-lUVqn8aT" role="doc-biblioref">2020</a>)</span>) across the biological organisms reported in LOTUS. For this, the biological organisms were grouped at four artificial taxa levels (plants, fungi, animals and bacteria). These were built by combining the two highest taxonomic levels in the OTL taxonomy, namely Domain and Kingdom levels.</p>
<div id="tbl:repartition" class="tablenos">
<table style="width:100%;">
<caption><span>Table 3:</span> Repartition and specificity of chemical structures across four important NP reservoirs: plants, fungi, animals and bacteria. Chemical classes were attributed with NPClassifier. </caption>
<colgroup>
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
<col style="width: 11%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;"><strong>Group</strong></th>
<th style="text-align: center;"><strong>Organisms</strong></th>
<th style="text-align: center;"><strong>Pairs</strong></th>
<th style="text-align: center;"><strong>Chemical Structures</strong></th>
<th style="text-align: center;"><strong>Specific Chemical Structures</strong></th>
<th style="text-align: center;"><strong>Planar Chemical Structures</strong></th>
<th style="text-align: center;"><strong>Specific Planar Chemical Structures</strong></th>
<th style="text-align: center;"><strong>Chemical classes</strong></th>
<th style="text-align: center;"><strong>Specific Chemical classes</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong>Plantae</strong></td>
<td style="text-align: center;">25,348</td>
<td style="text-align: center;">458,426</td>
<td style="text-align: center;">174,505</td>
<td style="text-align: center;">169,269</td>
<td style="text-align: center;">83,274</td>
<td style="text-align: center;">79,693</td>
<td style="text-align: center;">523</td>
<td style="text-align: center;">62</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Fungi</strong></td>
<td style="text-align: center;">3,511</td>
<td style="text-align: center;">49,976</td>
<td style="text-align: center;">34,968</td>
<td style="text-align: center;">32,111</td>
<td style="text-align: center;">18,241</td>
<td style="text-align: center;">16,300</td>
<td style="text-align: center;">400</td>
<td style="text-align: center;">20</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Animalia</strong></td>
<td style="text-align: center;">2,398</td>
<td style="text-align: center;">30,839</td>
<td style="text-align: center;">22,061</td>
<td style="text-align: center;">18,657</td>
<td style="text-align: center;">11,989</td>
<td style="text-align: center;">9,535</td>
<td style="text-align: center;">432</td>
<td style="text-align: center;">16</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Bacteria</strong></td>
<td style="text-align: center;">1,313</td>
<td style="text-align: center;">24,586</td>
<td style="text-align: center;">19,958</td>
<td style="text-align: center;">18,367</td>
<td style="text-align: center;">11,766</td>
<td style="text-align: center;">10,449</td>
<td style="text-align: center;">348</td>
<td style="text-align: center;">42</td>
</tr>
</tbody>
</table>
</div>
<h4 id="distributions-of-organisms-per-structure-and-structures-per-organism">Distributions of Organisms per Structure and Structures per Organism</h4>
<p>Readily achievable outcomes from LOTUS show that the depth of exploration of the world of NP is rather limited: as depicted in Figure <a href="#fig:distribution">4</a>, on average, three organisms are reported per chemical structure and eleven structures per organism. Notably, half of all structures have been reported from a single organism and half of all studied organisms are reported to contain five or fewer structures. Metabolomic studies suggest that these numbers are heavily underrated <span class="citation" data-cites="IhdADmDO 1EloBUtuI">(Noteborn et al., <a href="#ref-IhdADmDO" role="doc-biblioref">2000</a>; Wang et al., <a href="#ref-1EloBUtuI" role="doc-biblioref">2020</a>)</span> and suggest that a better reporting of the metabolites during a phytochemical investigation could greatly improve coverage. This incomplete coverage may be partially explained by the habit in classical NP journals to accept only new and/or bioactive chemical structures for publication.</p>
<div id="fig:distribution" class="fignos">
<figure>
<img src="images/distribution.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 4:</span> <strong>Distribution of “structures per organism” and “organisms per structure”</strong>. The number of organisms linked to the planar structure of <em>β</em>-sitosterol (KZJWDPNRJALLNS) and the number of chemical structures in <em>Arabidopsis thaliana are</em> two exemplary highlights.</figcaption>
</figure>
</div>
<h4 id="contribution-of-individual-electronic-np-resources-to-lotus">Contribution of Individual Electronic NP Resources to LOTUS</h4>
<p>The added value of the LOTUS Initiative to assemble multiple electronic NP resources is illustrated in Figure <a href="#fig:upset">5</a> : Panel A shows the contributions of the individual electronic NP resources to the ensemble of chemical structures found in one of the most studied vascular plants, <em>Arabidopsis thaliana</em> (“Mouse-ear cress”; <a href="https://www.wikidata.org/wiki/Q147096">Q147096</a>). Panel B shows the ensemble of taxa reported to contain the ubiquitous higher plant triterpenoid, <em>β</em>-sitosterol, represented by its planar structure (<a href="https://www.wikidata.org/wiki/Q121802">Q121802</a> and <a href="https://www.wikidata.org/wiki/Q63409374">Q63409374</a>).</p>
<div id="fig:upset" class="fignos">
<figure>
<img src="images/upset_combined.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 5:</span> <strong>UpSet plots of the individual contribution of electronic NP resources</strong> to the planar structures found in <em>Arabidopsis thaliana</em> (A) vs. organisms reported to contain the planar structure of <em>β</em>-sitosterol (KZJWDPNRJALNS) (B). UpSet plots are evolved Venn diagrams, allowing to represent intersections between multiple sets. The horizontal bars on the lower left represent the number of corresponding entries per electronic NP resource. The dots and their connecting line represent the intersection between source and consolidate sets. The vertical bars indicate the number of entries at the intersection. For example, 479 organisms containing the planar structure of <em>β</em>-sitosterol are present in both UNPD and NAPRALERT, whereas each of them individually report 1,164 and 2,085 organisms containing the planar structure of <em>β</em>-sitosterol, respectively.</figcaption>
</figure>
</div>
<p>Figure <a href="#fig:upset">5</a> A. also shows that according to NPClassifier, the chemical pathway distribution across electronic NP resources is unconserved. Note that NPClassifier and ClassyFire <span class="citation" data-cites="1GFQixHzv">(Djoumbou Feunang et al., <a href="#ref-1GFQixHzv" role="doc-biblioref">2016</a>)</span> chemical taxonomies are both available as metadata in the frozen LOTUS export (<a href="https://osf.io/c8hgb/">https://osf.io/c8hgb/</a>) and LNPN. Both classification tools return a chemical taxonomy for individual structures, thus allowing their grouping at higher hierarchical levels, in the same way as it is done for biological taxonomies. The UpSet plot in Figure <a href="#fig:upset">5</a> indicates the poor overlap of preexisting electronic NP resources and the added value of an aggregated dataset. This is particularly well illustrated in Figure <a href="#fig:upset">5</a> B., where the number of organisms for which the planar structure of <em>β</em>-sitosterol (KZJWDPNRJALLNS) has been reported is shown for each intersection. NAPRALERT has by far the highest number of entries (2,085 in total), while other electronic NP resources complement this well: e.g., UNPD has 532 reported organisms with <em>β</em>-sitosterol that do not overlap with those reported in NAPRALERT. Of note, <em>β</em>-sitosterol is documented in only 3 organisms in the DNP, highlighting the importance of a better systematic reporting of ubiquitous metabolites and the interest of multiple data sources agglomeration.</p>
<h4 class="page_break_before" id="a-biologically-interpreted-chemical-tree">A Biologically-interpreted Chemical Tree</h4>
<p>The chemical diversity captured in LOTUS is here displayed via a TMAP (Figure <a href="#fig:tmap">6</a>), a visualization method allowing the structural organization of large chemical datasets as a minimum spanning tree <span class="citation" data-cites="1DcpY38I1">(Probst and Reymond, <a href="#ref-1DcpY38I1" role="doc-biblioref">2020</a>)</span>. Using Faerun, an interactive HTML file is generated to display metadata and molecule structures by embedding the SmilesDrawer library <span class="citation" data-cites="VMUBsFFG eLVYiiAg">(Probst and Reymond, <a href="#ref-VMUBsFFG" role="doc-biblioref">2018</a><a href="#ref-VMUBsFFG" role="doc-biblioref">a</a>, <a href="#ref-eLVYiiAg" role="doc-biblioref">2018</a><a href="#ref-eLVYiiAg" role="doc-biblioref">b</a>)</span>. Planar structures were used for all compounds to generate the TMAP using MAP4 encoding <span class="citation" data-cites="2HOkPhmh">(Capecchi et al., <a href="#ref-2HOkPhmh" role="doc-biblioref">2020</a>)</span>. As the tree organizes structures according to their molecular fingerprint, an anticipated coherence between the clustering of compounds and the mapped NPClassifier chemical class is indeed observed (Figure <a href="#fig:tmap">6</a> A.). For clarity, the eight most represented chemical classes of LOTUS plus the quassinoids and carotenoids (C40, <em>β</em>-<em>β</em>) classes are mapped, with an example of a quassinoid (yellow star) and a carotenoid (green star) and their corresponding location in the TMAP.</p>
<div id="fig:tmap" class="fignos">
<figure>
<img src="images/tmap.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 6:</span> <strong>TMAP visualizations of the chemical diversity present in LOTUS.</strong> Each dot corresponds to a chemical structure. Dots are colored according to the NPClassifier chemical class they belong to (A); the most frequently reported biological family (Simaroubaceae or other; B); the specificity of chemical classes found in biological organisms (C). A quassinoid (yellow star) and a carotenoid (C40, <em>β</em>-<em>β</em>; green star) are mapped as examples in all visualizations. The biological specificity score at a given taxonomic level for a given chemical class is calculated as the number of structure-organism pairs within the taxon where the chemical class occurs the most, divided by the total number of pairs. A chemical class biological specificity of 1 at the family level means that compounds in that chemical class were only reported in one family. An interactive HTML visualization of the LOTUS TMAP is available at <a href="https://osf.io/kqa8b/">https://osf.io/kqa8b/</a>.</figcaption>
</figure>
</div>
<p>To explore relationships between chemistry and biology, it is possible to map biological information such as the most reported biological family (Figure <a href="#fig:tmap">6</a> B.) and the chemical class biological specificity (Figure <a href="#fig:tmap">6</a> C.) on the TMAP.
The biological specificity score at a given taxonomic level for a given chemical class is calculated as the number of structure-organism pairs within the taxon where the chemical class occurs the most, divided by the total number of pairs. See Equation <a href="#eq:spec_chem">1</a>:</p>
<p><span id="eq:spec_chem" class="eqnos"><span class="math inline">\({\text {Specificity score}_{\text {chem}}=\frac{\mid \text{Pairs in chemical class} \cap \text {Pairs in taxon where chemical class occurs the most}\mid}{\mid \text {Pairs in chemical class}\mid}}\)</span><span class="eqnos-number">(1)</span></span> </p>
<p>This visualization allows to highlight chemical classes specific to a given taxon, such as the quassinoids in the Simaroubaceae family. In this case it is striking to see how well the compounds of a given chemical class (quassinoids) (Figure <a href="#fig:tmap">6</a> A.) and the most reported plant family per compound (Simaroubaceae) (Figure <a href="#fig:tmap">6</a> B.) overlap. This is also evidenced on Figure <a href="#fig:tmap">6</a> C. with a chemical class specificity of 0.94 at the biological family level for quassinoids. In this plot, it is also possible to identify chemical classes that are widely spread among living organisms, such as the carotenoids (C40, <em>β</em>-<em>β</em>), which exhibit a specificity of 0.11 at the biological family level. This means that among all the carotenoids (C40, <em>β</em>-<em>β</em>) - organism pairs, about one tenth belong to the most common family.</p>
<h4 id="a-chemically-interpreted-biological-tree">A Chemically-interpreted Biological Tree</h4>
<p>An alternative view of the biological and chemical diversity covered by LOTUS is illustrated in Figure <a href="#fig:magicTree">7</a>. Here chemical compounds are not organized but biological organisms are placed in their taxonomy. To limit bias due to underreporting in the literature and keep a reasonable display size, only families with at least 50 reported compounds were included. Organisms were classified according to the OTL taxonomy, and structures according to NPClassifier. The tips were labeled according to the biological family and colored according to their biological kingdom. The bars represent structure specificity of the most characteristic chemical class of the given biological family (the higher the more specific). This specificity score was calculated as in Equation <a href="#eq:spec_bio">2</a>:</p>
<p><span id="eq:spec_bio" class="eqnos"><span class="math inline">\({\text{Specificity score}_{bio}=\frac{\mid\text {Structures in chemical class}\cap\text{Structures in taxon}\mid^2}{\mid \text{Structures in chemical class}\mid\boldsymbol{\cdot}\mid\text{Structures in taxon} \mid}}\)</span><span class="eqnos-number">(2)</span></span> </p>
<div id="fig:magicTree" class="fignos">
<figure>
<img src="images/magicTree.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 7:</span> <strong>LOTUS provides new means of exploring and representing chemical and biological diversity.</strong> The tree generated from current LOTUS data builds on biological taxonomy and employs the kingdom as label color (only families containing 50+ chemical structures were considered). The size of the leaf nodes corresponds to the number of genera reported in the family. The outer bars correspond to the most specific chemical class found in the biological family. The height of the bar is proportional to a specificity score corresponding to the square of the number of structures reported in the chemical class within the given biological family, over the product of the number of reported structures in the chemical class with the number of reported structures in the biological family. The bar color corresponds to the chemical pathway of the most specific chemical class in the NPClassifier classification system.</figcaption>
</figure>
</div>
<p>Figure <a href="#fig:magicTree">7</a> makes it possible to spot highly specific compound classes such as trinervitane terpenoids in the Termitidae, the rhizoxin macrolides in the Rhizopodaceae, or the quassinoids and limonoids typical, respectively, of Simaroubaceae and Meliaceae. Similarly, tendencies of more generic occurrence of NP can be observed. For example, within the fungal kingdom, Basidiomycotina appear to have a higher biosynthetic specificity toward terpenoids than other fungi, which mostly focus on polyketides production. When observed at a finer scale, down to the structure level, such chemotaxonomic representation can give valuable insights. For example, among all chemical structures, only two were found in all biological kingdoms, namely heptadecanoic acid (KEMQGTRYUADPNZ-UHFFFAOYSA-N) and <em>β</em>-carotene (OENHQHLEOONYIE-JLTXGRSLSA-N). Looking at the repartition of <em>β</em>-sitosterol (KZJWDPNRJALLNS-VJSFXXLFSA-N) within the overall biological tree, <a href="#si-5-complement-to-figure-7">SI-5</a> plots its presence/absence vs. those of its superior chemical classifications, namely the stigmastane, steroid, and terpenoid derivatives, over the same tree used in Figure <a href="#fig:magicTree">7</a>. The comparison of these five chemically-interpreted biological trees clearly highlights the increasing speciation of the <em>β</em>-sitosterol biosynthetic pathway in the Archaeplastida kingdom, while the superior classes are distributed across all kingdoms. A zoomable and vectorized version of Figure <a href="#fig:magicTree">7</a> is available for detailed inspection.</p>
<p>As illustrated, the possibility of data interrogation at multiple precision levels, from fully defined 3D structures to broader chemical classes, is of great interest, e.g., for taxonomic and evolution studies. This makes LOTUS a unique tool for the advancement of chemotaxonomy, pioneered by Augustin Pyramus de Candolle and pursued by other notable researchers (Robert Hegnauer, Otto R. Gottlieb) <span class="citation" data-cites="15tHEF73t 18uusIb9e xVEzsOX1">(de Candolle, <a href="#ref-xVEzsOX1" role="doc-biblioref">1816</a>; Gottlieb, <a href="#ref-15tHEF73t" role="doc-biblioref">1982</a>; Hegnauer, <a href="#ref-18uusIb9e" role="doc-biblioref">1986</a><a href="#ref-18uusIb9e" role="doc-biblioref">a</a>)</span>. Six decades after Hegnauer’s publication of “Die Chemotaxonomie der Pflanzen” <span class="citation" data-cites="AgCpbRSb">(Hegnauer, <a href="#ref-AgCpbRSb" role="doc-biblioref">1986</a><a href="#ref-AgCpbRSb" role="doc-biblioref">b</a>)</span> much remains to be done for the advancement of this field of study and the LOTUS initiatives aims to provide a solid basis for researchers willing to pursue these exciting explorations at the interface of chemistry, biology and evolution.</p>
<p>As shown recently in the context of spectral annotation <span class="citation" data-cites="gYzuBVpk">(Dührkop et al., <a href="#ref-gYzuBVpk" role="doc-biblioref">2020</a>)</span>, lowering the precision level of the annotation allows a broader coverage along with greater confidence. Genetic studies investigating the pathways involved and the organisms carrying the responsible biosynthetic genes would be of interest to confirm the previous observations. These forms of data interpretation exemplify the importance of reporting not only new structures, but also novel occurrences of known structures in organisms as comprehensive chemotaxonomic studies are pivotal for a better understanding of the metabolomes of living organisms.</p>
<p>The integration of multiple knowledge sources, e.g. genetics for NP producing gene clusters <span class="citation" data-cites="F5HJbaQH">(Kautsar et al., <a href="#ref-F5HJbaQH" role="doc-biblioref">2019</a>)</span> combined to taxonomies and occurrences DB, also opens new opportunities to understand if an organism is responsible for the <em>biosynthesis</em> of a NP or merely <em>contains</em> it. This understanding is of utmost importance for the chemotaxonomic field and will help to understand to which extent microorganisms (endosymbionts) play a role in host development and its NP expression <span class="citation" data-cites="BSCQ82T">(SAIKKONEN, <a href="#ref-BSCQ82T" role="doc-biblioref">2004</a>)</span>.</p>
<h2 id="conclusion-perspectives">Conclusion &amp; Perspectives</h2>
<h3 id="advancing-np-knowledge">Advancing NP Knowledge</h3>
<p>At its current development stage, data harmonized and curated in the LOTUS initiative remain imperfect and, by the very nature of research, at least partially biased (see introduction). In the context of bioactive NP research, and due to global editorial practices, it should not be ignored that many publications tend to emphasize new compounds and/or those for which interesting bioactivity has been measured. Near-ubiquitous (primarily plant-based) compounds tend to be overrepresented in the NP literature, yet the implication of their wide distribution in nature and associated patterns of broad, non-specific activity are often underappreciated <span class="citation" data-cites="WwEEYuQ9">(Bisson et al., <a href="#ref-WwEEYuQ9" role="doc-biblioref">2015</a>)</span>. Ideally, all characterized compounds independent of structural novelty and/or bioactivity potential should be documented. New data is fundamental for the advancement of NP research. The LOTUS initiative provides a framework for rigorous review and incorporation of new records, and already presents a valuable overview of the distribution of NP occurrences studied to date. While these data may already represent a reasonable approximation of the chemical content of some well-studied organisms such as <em>Arabidopsis thaliana</em>, they remain very patchy for most other members of the tree of life. The comprehensive editing opportunities provided within LOTUS and through the associated Wikidata distribution platform open new opportunities for collaborative community engagement. The authors believe that (inter)active community participation is the most efficient means of achieving a more comprehensive documentation of NP occurrences. In addition to facilitating the introduction of new data, it also provides a forum for critical review of existing data, as well as harmonization and verification of existing NP datasets as they come online.</p>
<h3 id="fostering-fairness-and-trustworthiness">Fostering FAIRness and TRUSTworthiness</h3>
<p>The LOTUS harmonized data and dissemination of referenced structure-organism pairs through Wikidata, enables novel forms of queries and transformational perspectives in NP research. As LOTUS follows the guidelines of FAIRness and TRUSTworthiness, all researchers across disciplines can benefit from this opportunity, whether the interest is in ecology and evolution, chemical ecology, drug discovery, biosynthesis pathway elucidation, chemotaxonomy, or other research fields that connect with NP.</p>
<p>The introduction of LOTUS even provides a new opportunity to advance the FAIR guiding principles for scientific data management and stewardship originally established in 2016 <span class="citation" data-cites="6DjakjNS">(Wilkinson et al., <a href="#ref-6DjakjNS" role="doc-biblioref">2016</a>)</span>. Researchers worldwide uniformly acknowledge the limitations caused by the intrinsic unavailability of essential (raw) data <span class="citation" data-cites="3cUZctro">(Bisson et al., <a href="#ref-3cUZctro" role="doc-biblioref">2016</a>)</span>. The lack of progress is, at least in part, due to elements in the dissemination channels of the classical print and static PDF publication formats that complicate or sometimes even discourage data sharing, e.g., due to page limitations and economically motivated mechanisms, including those involved in the focus on and calculation of journal impact factors. In particular raw data such as experimental readings, spectroscopic data, instrumental measurements, statistical, and other calculations are valued by all, but disseminated by only very few. The immense value of raw data and the desire to advance the public dissemination has recently been documented in detail for NMR spectroscopic data by a large consortium of NP researchers <span class="citation" data-cites="R6AOorG5">(McAlpine et al., <a href="#ref-R6AOorG5" role="doc-biblioref">2019</a>)</span>. However, to generate the vital flow of contributed data, the effort associated with preparing and submitting content to open repositories as well as data reuse should be better acknowledged in academia, government, regulatory, and industrial environments <span class="citation" data-cites="Ebuu5mfF IN34cxcI xf44BZw9">(Cousijn et al., <a href="#ref-Ebuu5mfF" role="doc-biblioref">2019</a>, <a href="#ref-IN34cxcI" role="doc-biblioref">2018</a>; Pierce et al., <a href="#ref-xf44BZw9" role="doc-biblioref">2019</a>)</span>.</p>
<h3 id="opening-new-perspectives-for-spectral-data">Opening New Perspectives for Spectral Data</h3>
<p>The possibilities for expansion and future applications of the Wikidata-based LOTUS initiative are significant. For example, properly formatted spectral data, e.g., data obtained by mass spectrometry (MS) or nuclear magnetic resonance (NMR), can be linked to the Wikidata entries for the respective chemical compounds. MassBank <span class="citation" data-cites="jTCd8yG0">(Horai et al., <a href="#ref-jTCd8yG0" role="doc-biblioref">2010</a>)</span> and SPLASH <span class="citation" data-cites="WRJsojww">(Wohlgemuth et al., <a href="#ref-WRJsojww" role="doc-biblioref">2010</a>)</span> identifiers are already reported in Wikidata, and this existing information can be used to report MassBank or SPLASH records for example for <em>Arabidopsis thaliana</em> compounds (<a href="https://w.wiki/3Jor">https://w.wiki/3Jor</a>). Such possibilities will help to bridge experimental data results obtained during the early stages of NP research with data that had been reported and formatted in different contexts. This opens exciting perspectives for structural dereplication, NP annotation, and metabolomic analysis. The authors have previously demonstrated that taxonomically-informed metabolite annotation is critical for the improvement of the NP annotation process <span class="citation" data-cites="1EaGvQl0B">(Rutz et al., <a href="#ref-1EaGvQl0B" role="doc-biblioref">2019</a>)</span>. Alternative approaches linking structural annotation to biological organisms have also shown substantial improvements <span class="citation" data-cites="OceiRpiP">(Hoffmann et al., <a href="#ref-OceiRpiP" role="doc-biblioref">2021</a>)</span>. The LOTUS initiative offers new opportunities for linking chemical objects to both their biological occurrences and spectral information and should significantly facilitate such applications.</p>
<h3 id="integrating-chemodiversity-biodiversity-and-human-health">Integrating Chemodiversity, Biodiversity, and Human Health</h3>
<p>As shown in <a href="#si-5-complement-to-figure-7">SI-5</a>, observing the chemical and biological diversity at various granularities can offer new insights. Regarding the chemical objects involved, it will be important to document the taxonomies of chemical annotations for the Wikidata entries. However, this is a rather complex task, for which stability and coverage issues will have to be addressed first. Existing chemical taxonomies such as ChEBI, ClassyFire, or NPClassifier are evolving steadily, and it will be important to constantly update the tools used to make further annotations. Repositioning NP within their greater biosynthetic context is another major challenge - and active field of research. The fact that the LOTUS disseminates data through Wikidata will help facilitate its integration with biological pathway knowledge bases such as <a href="https://www.wikipathways.org/index.php/WikiPathways">WikiPathways</a> and contribute to this complex task <span class="citation" data-cites="pzgOjFLZ 1CVZzR5SC">(Martens et al., <a href="#ref-pzgOjFLZ" role="doc-biblioref">2021</a>; Slenter et al., <a href="#ref-1CVZzR5SC" role="doc-biblioref">2018</a>)</span>.</p>
<p>In the field of ecology, for example, molecular traits are gaining increased attention <span class="citation" data-cites="1Hc8Xn2Ff 1Ck6coooY">(Kessler and Kalske, <a href="#ref-1Hc8Xn2Ff" role="doc-biblioref">2018</a>; Sedio, <a href="#ref-1Ck6coooY" role="doc-biblioref">2017</a>)</span>. Conceptually, LOTUS can help associate classical plant traits (e.g., leaf surface area, photosynthetic capacities, etc.) with Wikidata biological organisms entries, and, thus, allow their integration and comparison with chemicals that are associated with the organisms. Likewise, the association of biogeography data documented in repositories such as GBIF could be further exploited in Wikidata to pursue the exciting but understudied topic of “chemodiverse hotspots” <span class="citation" data-cites="ToSx7lEy">(Defossez et al., <a href="#ref-ToSx7lEy" role="doc-biblioref">2021</a>)</span>.</p>
<p>Further NP-related information of great interest remains poorly formatted. One example of such a shortcoming relates to traditional medicine, including ethnomedicine and ethnobotany, which is the historical and empiric approach of mankind to discover and use bioactive products from Nature, primarily plants. The amount of knowledge generated in human history on the use of medicinal substances represents fascinating yet underutilized information. Notably, the body of literature on the pharmacology and toxicology of NP is compound-centric, increases steadily, and relatively scattered, but still highly relevant (not necessarily: sufficient) for exploring the role and potential utility of NP for human health. To this end, the LOTUS initiative represents a resource for new concepts by which such information could be valued and conserved in the digital era, as LOTUS provides a blueprint for appropriate formatting and sharing of such data <span class="citation" data-cites="QBTqPbZC MGL5MqgM NtzQIgT9">(Allard et al., <a href="#ref-QBTqPbZC" role="doc-biblioref">2018</a>; Geoffrey A. Cordell, <a href="#ref-MGL5MqgM" role="doc-biblioref">2017</a><a href="#ref-MGL5MqgM" role="doc-biblioref">a</a>, <a href="#ref-NtzQIgT9" role="doc-biblioref">2017</a><a href="#ref-NtzQIgT9" role="doc-biblioref">b</a>)</span>. This underscores the transformative value of the LOTUS initiative for the advancement of Traditional Medicine and its drug discovery potential in health systems worldwide.</p>
<h3 id="summary-outlook">Summary &amp; Outlook</h3>
<p>The various facets discussed above connect with ongoing and future developments that the tandem of the LOTUS initiative and its Wikidata integration can accommodate through a broader knowledge base. The information of the LOTUS initiative is already readily accessible by third party projects build on top of Wikidata such as the SLING project (<a href="https://github.com/ringgaard/sling">https://github.com/ringgaard/sling</a>, see entry for <a href="https://ringgaard.com/kb/Q413364">gliotoxin</a>) or the Plant Humanities Lab project (<a href="https://lab.plant-humanities.org/">https://lab.plant-humanities.org/</a>, see entry for <a href="https://search.plant-humanities.org/?eid=wd%3AQ3772280"><em>Ilex guayusa</em></a>). Behind the scenes, all underlying resources represent data in a multidimensional space and can be extracted as individual graphs that can be interconnected. The craft of appropriate federated queries allows users to navigate these graphs and fully exploit their potential <span class="citation" data-cites="mjVvWLK2 HUodcdz1">(Kratochvíl et al., <a href="#ref-mjVvWLK2" role="doc-biblioref">2018</a>; Waagmeester et al., <a href="#ref-HUodcdz1" role="doc-biblioref">2020</a>)</span>. The development of interfaces such as RDFFrames <span class="citation" data-cites="VNMRDIw3">(Mohamed et al., <a href="#ref-VNMRDIw3" role="doc-biblioref">2020</a>)</span> will also facilitate the use of the wide arsenal of existing machine learning approaches to automate reasoning on these knowledge graphs.</p>
<p>Overall, the LOTUS initiative aims to make more and better data available. This project paves the way for the establishment of an open and expandable electronic NP resource. The design and efforts of the LOTUS initiative reflect our conviction that the integration of NP research results is long-needed and requires a truly open and FAIR knowledge base. We believe that the LOTUS initiative has the potential to fuel a virtuous cycle of research habits and, as a result, contribute to a better understanding of <em>Life and its chemistry</em>.</p>
<h2 class="page_break_before" id="methods">Methods</h2>
<h3 id="data-curation">Data Curation</h3>
<h4 id="gathering">Gathering</h4>
<p>Before their inclusion, the overall quality of the source was manually assessed to estimate the quality of referenced structure-organism pairs and the lack of ambiguities in the links between data and references. This led to the identification of thirty-six electronic NP resources as valuable LOTUS input. Data from the proprietary Dictionary of Natural Products (DNP v 29.1) was also used for comparison purposes only and is not publicly disseminated. <a href="https://foodb.ca/">FooDB</a> was also curated but not publicly disseminated since its license did not allow sharing in Wikidata. <a href="#si-1-data-sources-list">SI-1</a> gives all necessary details regarding electronic NP resources access and characteristics.</p>
<p>Manual inspection of each electronic NP resource revealed that the structure, organism, and reference fields were widely variable in format and contents, thus requiring standardization to be comparable. The initial stage consisted of writing tailored scripts that are capable of harmonizing and categorizing knowledge from each source (Fig. 2). This transformative process led to three categories: fields relevant to the chemical structure described, to the producing biological organism, and the reference describing the occurrence of the chemical structure in the producing biological organism. This process resulted in categorized columns for each source, providing an initial harmonized format for each table.</p>
<p>For all thirty-eight sources, if a single file or multiple files were accessible via a download option including FTP, data was gathered that way. For some sources, data was scraped (cf. <a href="#si-1-data-sources-list">SI-1</a>). All scraping scripts written to automatically retrieve entries can be found in the <a href="https://gitlab.com/lotus7/lotusProcessor">lotusProcessor</a> repository in the <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/1_gathering">src/1_gathering</a> directory (under each respective subdirectory). Data extraction scripts for the DNP are available and should allow users with a license to further exploit the data (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/1_gathering/db/dnp">src/1_gathering/db/dnp</a>). The chemical structure fields, organism fields, and reference fields were manually categorized into three, two, and ten subcategories, respectively. For chemical structures, ”InChI”, “SMILES”, and “chemical name” (not necessarily IUPAC). For organisms, “clean” and “dirty”, meaning lot text not referred to the canonical name was present or the organism was not described by its canonical name. For the references, the original reference was kept in the “original” field. When the format allowed it, references were divided into: “authors”, ”doi”, ”external”, ”isbn”, ”journal”, “original”, ”publishing details”, “pubmed”, “title”, “split”. The generic “external” field was used for all external cross-references to other websites or electronic NP resources (for example, “also in knapsack”). The last subcategory, “split”, corresponds to a still non-atomic field after the removal of parts of the original reference. Other field titles are self-explanatory. The producing organism field was kept as a single field.</p>
<h4 id="harmonization">Harmonization</h4>
<p>To perform the harmonization of all previously gathered sources, sixteen columns were chosen as described above. Upon electronic NP resources harmonization, resulting subcategories were divided and subject to further cleaning. The “chemical structure” fields were divided into files according to their subcategories (“InChI”, “names” and “SMILES”). A file containing all initial structures from all three subcategories was also generated. The same procedure was followed for organisms and references.</p>
<h4 id="cleaning">Cleaning</h4>
<p>To obtain an unambiguously referenced structure-organism pair for Wikidata dissemination, the initial sixteen columns were translated and cleaned into three fields: the reported structure, the organism canonical name, and the reference. The structure was reported as InChI, together with its SMILES and InChIKey translation. The biological organism field was reported as three minimal necessary and sufficient fields, namely its canonical name and the taxonID and taxonomic DB corresponding to the latter. The reference was reported as four minimal fields, namely reference title, DOI, PMCID, and PMID, one being sufficient. For the forthcoming translation processes, automated solutions were used when available. However, for specific cases (common or vernacular names of the biological organisms, Traditional Chinese Medicine (TCM) names, and conversion between digital reference identifiers), no solution existed, thus requiring the use of tailored dictionaries. The initial entries (containing one or multiple producing organisms per structure, with one or multiple accepted names per organism) were cleaned into over 2M referenced structure-organism pairs.</p>
<h5 id="chemical-structures">Chemical Structures</h5>
<p>To retrieve as much information as possible from the original structure field(s) of each of the sources, the following procedure was followed. Allowed structural fields for the sources were divided into two types: structural (InChI, SMILES) or nominal (chemical name, not necessarily IUPAC). If multiple fields were present, structural identifiers were preferred over structure names. Among structural identifiers, when both identifiers led to different structures, InChI was preferred over SMILES. SMILES were translated to InChI using the RDKit (2020.03.3) implementation in Python 3.8 (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/structure/1_translating/smiles.py">src/2_curating/2_editing/structure/1_translating/smiles.py</a>). They were first converted to <a href="https://www.rdkit.org/docs/cppapi/ROMol_8h.html#details">ROMol</a> objects which were then converted to InChI. When no structural identifier was available, the nominal identifier was translated to InChI first thanks to <a href="https://github.com/dan2097/opsin">OPSIN</a> <span class="citation" data-cites="pf43Zseq">(Lowe et al., <a href="#ref-pf43Zseq" role="doc-biblioref">2011</a>)</span>, a fast Java-based translation open-source solution. If no translation was obtained, chemical names were then submitted to the CTS <span class="citation" data-cites="WRJsojww">(Wohlgemuth et al., <a href="#ref-WRJsojww" role="doc-biblioref">2010</a>)</span>, once in lower case only, once with the first letter capitalized. If again no translation was obtained, candidates were then submitted to the <a href="https://cactus.nci.nih.gov">Chemical Identifier Resolver</a> via the cts_convert function from the webchem package <span class="citation" data-cites="kM2kraS2">(Szöcs et al., <a href="#ref-kM2kraS2" role="doc-biblioref">2020</a>)</span>. Before the translation process, some typical chemical structure-related greek characters (such as <em>α</em>, <em>ß</em>) were replaced by their textual equivalents (alpha, beta) to obtain better results. All pre-translation steps are included in the preparing_name function and are available in <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/r/preparing_name.R">src/r/preparing_name.R</a>.</p>
<p>The chemical sanitization step sought to standardize the representation of chemical structures coming from different sources. It consisted of three main stages (standardizing, fragment removal, and uncharging) achieved via the MolVS package. The initial standardizer function consists of six stages (RDKit Sanitization, RDKit Hs removal, Metals Disconnection, Normalization, Acids Reionization, and Stereochemistry recalculation) detailed in the <a href="https://molvs.readthedocs.io/en/latest/guide/standardize.html">molvs documentation</a>. In a second step, the FragmentRemover functionality was applied using a list of SMARTS to detect and remove common counterions and crystallization reagents sometimes occurring in the input DB. Finally, the Uncharger function was employed to neutralize molecules when appropriate.</p>
<p>MarvinSuite was used for traditional and IUPAC names translation, Marvin 20.19, <a href="https://www.chemaxon.com">ChemAxon</a>. When stereochemistry was not fully defined, (+) and (-) symbols were removed from names. All details are available in the following script: <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/structure/4_enriching/naming.R">src/2_curating/2_editing/structure/4_enriching/naming.R</a>.
Chemical classification of all resulting structures was done using classyfireR <span class="citation" data-cites="1GFQixHzv">(Djoumbou Feunang et al., <a href="#ref-1GFQixHzv" role="doc-biblioref">2016</a>)</span> and <a href="https://ccms-ucsd.github.io/GNPSDocumentation/api/#structure-np-classifier">NPClassifier API</a>.</p>
<p>From the 269,436 initial InChI, 233,652 (87%) sanitized structures were obtained, of which 176,235 (75%) had complete stereochemistry defined. 197,392 (73%) were uploaded to Wikidata.
From the 244,724 initial SMILES, 205,244 (84%) sanitized structures were obtained, of which 8,542 (4%) had complete stereochemistry defined. 173,590 (71%) were uploaded to Wikidata.
From the 49,315 initial chemical names, 27,780 (56%) sanitized structures were obtained, of which 2,324 (8%) had complete stereochemistry defined. 23,098 (47%) were uploaded to Wikidata.
In total, 164,308 structures with fully defined stereochemistry were uploaded as “chemical compounds” (<a href="https://www.wikidata.org/wiki/Q11173">Q11173</a>), and 106,043 structures without fully defined stereochemistry were uploaded as “group of stereoisomers” (<a href="https://www.wikidata.org/wiki/Q59199015">Q59199015</a>).</p>
<h5 id="biological-organisms">Biological Organisms</h5>
<p>The cleaning process at the biological organism’s level had three objectives: convert the original organism string to (a) taxon name(s), atomize fields containing multiple taxon names, and deduplicate synonyms. The original organism strings were treated with <a href="https://github.com/gnames/gnfinder">Global Names Finder</a> (GNF) and <a href="https://github.com/gnames/gnverifier">Global Names Verifier</a> (GNV), both tools coming from the <a href="http://globalnames.org">Global Names Architecture</a> (GNA) a system of web services that helps people to register, find, index, check and organize biological scientific names and interconnect on-line information about species. GNF allows scientific name recognition within raw text blocks and searches for found scientific names among public taxonomic DB. GNV takes names or lists of names and verifies them against various biodiversity data sources. Canonical names, their taxonID, and the taxonomic DB they were found in were retrieved. When a single entry led to multiple canonical names (accepted synonyms), all of them were kept. Because both GNF and GNV recognize scientific names and not common ones, common names were translated before a second resubmission.</p>
<h6 id="dictionaries">Dictionaries</h6>
<p>To perform the translations from common biological organism name to latin scientific name, specialized dictionaries included in DrDuke, FooDB, PhenolExplorer were aggregated together with the translation dictionary of <a href="https://www.gbif.org/dataset/d7dddbf4-2cf0-4f39-9b2a-bb099caae36c">GBIF Backbone Taxonomy</a>. The script used for this was <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/1_gathering/translation/common.R">src/1_gathering/translation/common.R</a>. When the canonical translation of a common name contained a specific epithet that was not initially present, the translation pair was discarded (for example, “Aloe” translated in “<em>Aloe vera</em>” was discarded). Common names corresponding to a generic name were also discarded (for example “Kiwi” corresponding to the synonym of an <em>Apteryx</em> spp. (<a href="https://www.gbif.org/species/4849989">https://www.gbif.org/species/4849989</a>)). When multiple translations were given for a single common name, the following procedure was followed: the canonical name was split into species name, genus name, and possible subnames. For each common name, genus names and species names were counted. If both the species and genus names were consistent at more than 50%, they were considered consistent overall and, therefore, kept (for example, “Aberrant Bush Warbler” had “<em>Horornis flavolivaceus</em>” and “<em>Horornis flavolivaceus intricatus</em>” as translation; as both the generic (“<em>Horornis</em>”) and the specific (“<em>flavolivaceus</em>”) epithets were consistent at 100%, both (“<em>Horornis flavolivaceus</em>”) were kept). When only the generic epithet had more than 50% consistency, it was kept (for example, “Angelshark” had “<em>Squatina australis</em>” and “<em>Squatina squatina</em>” as translation, so only “<em>Squatina</em>” was kept). Some unspecific common names were removed (see <a href="https://osf.io/gqhcn/">https://osf.io/gqhcn/</a>) and only common names with more than three characters were kept. This resulted in 181,891 translation pairs further used for the conversion from common names to scientific names. For TCM names, translation dictionaries from TCMID, TMMC, and coming from the Chinese Medicine Board of Australia were aggregated. The script used for this was <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/1_gathering/translation/tcm.R">src/1_gathering/translation/tcm.R</a>. Some unspecific common names were removed (see <a href="https://osf.io/zs7ky/">https://osf.io/zs7ky/</a>). Careful attention was given to the Latin genitive translations and custom dictionaries were written (see <a href="https://osf.io/c3ja4/">https://osf.io/c3ja4/</a>, <a href="https://osf.io/u75e9/">https://osf.io/u75e9/</a>). Organ names of the producing organism were removed to avoid wrong translation (see <a href="https://osf.io/94fa2/">https://osf.io/94fa2/</a>). This resulted in 7,070 translation pairs. Both common and TCM translation pairs were then ordered by decreasing string length, first translating the longer names to avoid part of them being translated incorrectly.</p>
<h6 id="translation">Translation</h6>
<p>To ensure compatibility between obtained taxonID with Wikidata, the taxonomic DB 3 (<a href="https://www.itis.gov/">ITIS</a>), 4 (<a href="https://www.ncbi.nlm.nih.gov/taxonomy">NCBI</a>), 5 (<a href="http://www.indexfungorum.org/">Index Fungorum</a>), 6 (<a href="http://wgb.cimmyt.org/gringlobal/taxon/abouttaxonomy.aspx?language=en&amp;chapter=scope">GRIN Taxonomy for Plants</a>), 8 (<a href="https://www.irmng.org/">The Interim Register of Marine and Nonmarine Genera</a>), 9 (<a href="http://www.marinespecies.org/">World Register of Marine Species</a>), 11 (<a href="https://www.gbif.org/">GBIF Backbone Taxonomy</a>), 12 (<a href="https://eol.org/">Encyclopedia of Life</a>), 118 (<a href="https://amphibiaweb.org/">AmphibiaWeb</a>), 128 (<a href="https://www.wildscreen.org/arkive-closure/">ARKive</a>), 132 (<a href="http://zoobank.org/">ZooBank</a>), 147 (<a href="http://data.canadensys.net/vascan/search">Database of Vascular Plants of Canada (VASCAN)</a>), 148 (<a href="http://phasmida.speciesfile.org/">Phasmida Species File</a>), 150 (<a href="https://plants.sc.egov.usda.gov/">USDA NRCS PLANTS Database</a>), 155 (<a href="https://www.fishbase.se/">FishBase</a>), 158 (<a href="https://eunis.eea.europa.eu/">EUNIS</a>), 163 (<a href="https://www.iucnredlist.org/">IUCN Red List of Threatened Species</a>), 164 (<a href="https://www.biolib.cz/">BioLib.cz</a>), 165 (<a href="https://www.tropicos.org/">Tropicos - Missouri Botanical Garden</a>), 167 (<a href="https://www.ipni.org/">The International Plant Names Index</a>), 169 (<a href="http://ubio.org/">uBio NameBank</a>), 174 (<a href="https://www.departments.bucknell.edu/biology/resources/msw3/">The Mammal Species of The World</a>), 175 (<a href="http://www.birdlife.org/">BirdLife International</a>), 179 (<a href="https://tree.opentreeoflife.org/">Open Tree of Life</a>), 180 (<a href="https://www.inaturalist.org/">iNaturalist</a>) and 187 (<a href="https://ebird.org/science/use-ebird-data/the-ebird-taxonomy">The eBird/Clements Checklist of Birds of the World</a>) were chosen. All other available taxonomic DB are listed at <a href="http://index.globalnames.org/datasource">http://index.globalnames.org/datasource</a>. To retrieve as much information as possible from the original organism field of each of the sources, the following procedure was followed: First, a scientific name recognition step, allowing us to retrieve canonical names was carried (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/organisms/subscripts/1_cleaningOriginal.R">src/2_curating/2_editing/organisms/subscripts/1_cleaningOriginal.R</a>). Then, a subtraction step of the obtained canonical names from the original field was applied, to avoid unwanted translation of parts of canonical names. For example, Bromus mango contains “mango” as a specific epithet, which is also the common name for Mangifera indica. After this subtraction step, the remaining names were translated from vernacular (common) and TCM names to scientific names, with help of the dictionaries. For performance reasons, this cleaning step was written in Kotlin and used coroutines to allow efficient parallelization of that process (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/organisms/2_translating_organism_kotlin/">src/2_curating/2_editing/organisms/2_translating_organism_kotlin/</a>). They were subsequently submitted again to scientific name recognition (<a href="http://src/2_curating/2_editing/organisms/3_cleaningTranslated.R">src/2_curating/2_editing/organisms/3_cleaningTranslated.R</a>).</p>
<p>After full resolution of canonical names, all obtained names were submitted to rotl (Michonneau et al., 2016) to obtain a unified taxonomy.</p>
<p>From the 86,076 initial “clean” organism fields, 43,421 (50%) canonical names were obtained, of which 31,965 (37%) were uploaded to Wikidata.
From the 294 initial “dirty” organism fields, 241 (82%) canonical names were obtained, of which 198 (67%) were uploaded to Wikidata.</p>
<h5 class="page_break_before" id="references">References</h5>
<p>The <a href="https://cran.r-project.org/web/packages/rcrossref/">Rcrossref</a> package <span class="citation" data-cites="UHC5x6VA">(Chamberlain et al., <a href="#ref-UHC5x6VA" role="doc-biblioref">2020</a>)</span> interfacing with the <a href="https://www.crossref.org">Crossref</a> API was used to translate references from their original subcategory (“original”, “publishingDetails”, “split”, “title”) to a DOI, the title of its corresponding article, the journal it was published in, its date of publication and the name of the first author. The first twenty candidates were kept and ranked according to the score returned by Crossref, which is a <a href="https://lucene.apache.org/core/8_8_0/core/org/apache/lucene/search/similarities/TFIDFSimilarity.html">tf-idf</a> score. For DOI and PMID, only a single candidate was kept. All parameters are available in <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/functions/reference.R">src/functions/reference.R</a>. All DOIs were also translated with this method, to eventually discard any DOI not leading to an object. PMIDs were translated, thanks to the entrez_summary function of the <a href="https://cran.r-project.org/web/packages/rentrez/">rentrez</a> package <span class="citation" data-cites="13ZrxSjw4">(Winter, <a href="#ref-13ZrxSjw4" role="doc-biblioref">2017</a>)</span>. Scripts used for all subcategories of references are available in the directory <a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/reference/1_translating/">src/2_curating/2_editing/reference/1_translating/</a>. Once all translations were made, results coming from each subcategory were integrated, (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/reference/2_integrating.R">src/2_curating/2_editing/reference/2_integrating.R</a>) and the producing organism related to the reference was added for further treatment. Because the crossref score was not informative enough, at least one other metric was chosen to complement it. The first metric was related to the presence of the producing organism’s generic name in the title of the returned article. If the title contained the generic name of the organism, a score of 1 was given, else 0. Regarding the subcategories “doi”, “pubmed” and “title”, for which the same subcategory was retrieved via crossref or rentrez, distances between the input’s string and the candidates’ one were calculated. Optimal string alignment (restricted <a href="https://en.wikipedia.org/wiki/Damerau%E2%80%93Levenshtein_distance">Damerau-Levenshtein distance</a>) was used as a method. Among “publishing details”, “original” and “split” categories, three additional metrics were used: If the journal name was present in the original field, a score of 1 was given, else 0. If the name of the first author was present in the original field, a score of 1 was given, else 0. Those three scores were then summed together. All candidates were first ordered according to their crossref score, then by the complement score for related subcategories, then again according to their title-producing organism score, and finally according to their translation distance score. After this reranking step, only the first candidate was kept. Finally, the Pubmed PMCID dictionary (<a href="https://ftp.ncbi.nlm.nih.gov/pub/pmc/PMC-ids.csv.gz">PMC-ids.csv.gz</a>) was used to perform the translations between DOI, PMID, and PMCID. (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/2_editing/reference/3_cleaning.R">src/2_curating/2_editing/reference/3_cleaning.R</a>)</p>
<p>From the 36,692 initial “original” references, 21,928 (60%) references with sufficient quality were obtained, of which 15,674 (71%) had the organism name in their title. 9,882 (27%) were uploaded to Wikidata.
From the 21,306 initial “pubmed” references, 9,015 (42%) references with sufficient quality were obtained, of which 5,695 (63%) had the organism name in their title. 3,013 (14%) were uploaded to Wikidata.
From the 35,348 initial “doi” references, 19,682 (56%) references with sufficient quality were obtained, of which 15,522 (79%) had the organism name in their title. 11,847 (34%) were uploaded to Wikidata.
From the 29,584 initial “title” references, 17,410 (59%) references with sufficient quality were obtained, of which 12,732 (73%) had the organism name in their title. 9,638 (33%) were uploaded to Wikidata.
From the 11,322 initial “split” references, 5,856 (52%) references with sufficient quality were obtained, of which 3,221 (55%) had the organism name in their title. 2,255 (20%) were uploaded to Wikidata.
From the 3,310 initial “publishingDetails” references, 119 (4%) references with sufficient quality were obtained, of which 59 (50%) had the organism name in their title. 16 (&lt;0.1%) were uploaded to Wikidata.</p>
<h4 id="realignment">Realignment</h4>
<p>In order to fetch back the referenced structure-organism pairs links in the original data, the cleaned structures, cleaned organisms, and cleaned references were re-aligned with the initial entries. This resulted in over 6.2M referenced structure-organism pairs. Those pairs were not unique, with redundancies among electronic NP resources and different original categories leading to the same final pair (for example, entry reporting InChI=1/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in <em>Crataegus oxyacantha</em> or InChI=1S/C21H20O12/c22-6-13-15(27)17(29)18(30)21(32-13)33-20-16(28)14-11(26)4-8(23)5-12(14)31-19(20)7-1-2-9(24)10(25)3-7/h1-5,13,15,17-18,21-27,29-30H,6H2/t13-,15+,17+,18-,21+/m1/s1 in <em>Crataegus stevenii</em> both led to OVSQVDMCBVZWGM-DTGCRPNFSA-N in <em>Crataegus monogyna</em>). After deduplication, over 2M unique structure-organism pairs were obtained.</p>
<p>After the curation of all three objects, all of them were put together again. Therefore, the original aligned table containing the original pairs was joined with each curation result. Only entries containing a structure, an organism, and a reference after curation were kept. Each curated object was divided into minimal data (for Wikidata upload) and metadata. A dictionary containing original and curated object translations was written for each object to avoid those translations being made again during the next curation step (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/2_curating/3_integrating.R">src/2_curating/3_integrating.R</a>).</p>
<h4 id="validation">Validation</h4>
<p>The pairs obtained after curation were of different quality. Globally, structure and organism translation was satisfactory whereas reference translation was not. Therefore, to assess the validity of the obtained results, a randomized set of 420 referenced structure-organism pairs was sampled in each reference subcategory and validated or rejected manually. Entries were sampled with at least 55 of each reference subcategory present (to get a representative idea of each subcategory) (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/3_analysing/1_sampling.R">src/3_analysing/1_sampling.R</a>). An entry was only validated if: <em>i)</em> the structure (as any structural descriptor that could be linked to the final sanitized InChIKey) was described in the reference <em>ii)</em> the producing organism (as any organism descriptor that could be linked to the accepted canonical name) was described in the reference and <em>iii)</em> the reference was describing the occurrence of the chemical structure in the biological organism. Results obtained on the manually analyzed set were categorized according to the initial reference subcategory and are detailed in <a href="#si-2-summary-of-the-validation-statistics">SI-2</a>. To improve these results, further cleaning of the references was needed. This was done by accepting entries whose reference was coming from a DOI, a PMID, or from a title which restricted Damerau-Levenshtein distance between original and translated was lower than ten or if it was coming from one of the three main journals where occurrences are published (i.e., <em>Journal of Natural Products</em>, <em>Phytochemistry</em>, or <em>Journal of Agricultural and Food Chemistry</em>). For “split”, “publishingDetails” and “original” subcategories, the year of publication of the obtained reference, its journal, and the name of the first author were searched in the original entry and if at least two of them were present, the entry was kept. Entries were then further filtered to keep the ones where the reference title contained the first element of the detected canonical name. Except for COCONUT, exceptions to this filter were made for all DOI-based references. To validate those filtering criteria, an additional set of 100 structure-organism pairs were manually analyzed. F0.5 score was used as a metric. F0.5 score is a modified F1 score where precision has twice more weight than recall.</p>
<p>The F-score was calculated with ß = 0.5, as in Equation <a href="#eq:validation">3</a>:</p>
<p><span id="eq:validation" class="eqnos"><span class="math inline">\({F_{\beta}\ =\ \left(1+\beta^2\right)\cdot\frac{precision\cdot recall}{\left(\beta^2\ \cdot\ precision\right)\ +\ recall}}\)</span><span class="eqnos-number">(3)</span></span> </p>
<p>Based on this first manually validated dataset, filtering criteria (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/r/filter.R">src/r/filter.R</a>) were established to maximize precision and recall. Another 100 entries were sampled, this time respecting the whole set ratios. After manual validation, 97% of true positives were reached on the second set. A summary of the validation results is given in <a href="#si-2-summary-of-the-validation-statistics">SI-2</a>. Once validated, the filtering criteria were established to the whole curated set to filter entries chosen for dissemination (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/3_analysing/2_validating.R">src/3_analysing/2_validating.R</a>).</p>
<h4 id="unit-testing">Unit Testing</h4>
<p>To provide robustness of the whole process and code, unit tests and partial data full-tests were written. They can run on the developer machine but also on the CI/CD system (GitLab) upon each commit to the codebase.</p>
<p>Those tests assess that the functions are providing results coherent with what is expected especially for edge cases detected during the development. The Kotlin code has tests based on JUnit and code quality control checks based on Ktlint, Detekt and Ben Mane’s version plugin.</p>
<h3 id="data-dissemination-1">Data Dissemination</h3>
<h4 id="wikidata-2">Wikidata</h4>
<p>All the data produced for this work has been made available on Wikidata under a Creative Commons 0 license according to <a href="https://www.wikidata.org/wiki/Wikidata:Licensing">Wikidata:Licensing</a>. This license is a “No-right-reserved” license that allows most reuses.</p>
<h4 id="lotus.naturalproducts.net-lnpn-1">Lotus.NaturalProducts.Net (LNPN)</h4>
<p>The web interface is implemented following the same protocol as described in the COCONUT publication <span class="citation" data-cites="1EoPmmMQv">(Sorokina and Steinbeck, <a href="#ref-1EoPmmMQv" role="doc-biblioref">2020</a><a href="#ref-1EoPmmMQv" role="doc-biblioref">b</a>)</span> i.e. the data are stored in a MongoDB repository, the backend runs with Kotlin and Java, using the Spring framework, and the frontend is written in React.js, and completely Dockerized. In addition to the diverse search functions available through this web interface, an API is also implemented, allowing programmatic LNPN querying. The complete API usage is described on the “Documentation” page of the website. LNPN is part of the NaturalProducts.net portal, an initiative aimed at gathering diverse open NP resources in one place.</p>
<h3 id="data-interaction">Data Interaction</h3>
<h4 id="data-retrieval-1">Data Retrieval</h4>
<p>Bulk retrieval of a frozen (2021-05-05) version of LOTUS data is also available at <a href="https://osf.io/c8hgb/">https://osf.io/c8hgb/</a>.</p>
<p><a href="https://gitlab.com/lotus7/wikidataLotusExporter">WikidataLotusExporter</a> allows the download of all chemical compounds with a “found in taxon” property. That way, it does not only get the data produced by this work, but any that would have existed beforehand or that would have been added directly on Wikidata by our users. It makes a copy of all the entities (compounds, taxa, references) into a local triplestore that can be queried with SPARQL as is or converted to a TSV file for inclusion in other projects. It is currently adapted to export directly into the SSOT thus allowing direct reuse by the processing/curation pipeline.</p>
<h4 id="data-addition">Data Addition</h4>
<h5 id="wikidata-3">Wikidata</h5>
<p>Data is loaded by the Kotlin importer available in the <a href="https://gitlab.com/lotus7/wikidataLotusImporter">WikidataLotusImporter</a> repository under a GPL V3 license and imported into Wikidata. The importer processes the curated outputs grouping references, organisms, and compounds together. It then checks if they already exist in Wikidata (using SPARQL or a direct connection to Wikidata depending on the kind of data). It then uses <em>update</em> or <em>insert</em>, also called <em>upsert</em>, the entities as needed. The script currently takes the tabular file of the documented structure-organism pairs resulting from the LOTUS curation process as input. It is currently being adapted to use directly the SSOT and avoid an unnecessary conversion step. To import references, it first double checks for the presence of duplicated DOIs and utilizes the <a href="https://www.crossref.org/education/retrieve-metadata/rest-api/">Crossref REST API</a> to retrieve metadata associated with the DOI, the support for other citation sources such as Europe PMC is in progress. The structure-related fields are only subject to limited processing: basic formatting of the molecular formula by subscripting of the numbers. Due to limitations in Wikidata, the molecule names are dropped if they are longer than 250 characters and likewise the InChI strings cannot be stored if they are longer than 1500 characters.</p>
<p>Uploaded taxonomical DB identifiers are currently restricted to ITIS, GBIF, NCBI Taxon, Index Fungorum, IRMNG, WORMS, VASCAN, and iNaturalist. The taxa levels are currently limited to family, subfamily, tribe, subtribe, genus, species, variety. The importer checks for the existence of each item based on their InChIKey and upserts the compound with the <em>found in taxon</em> statement and the associated organisms and references.</p>
<h5 id="lnpn">LNPN</h5>
<p>From the onset, LNPN has been importing data directly from the frozen tabular data of the LOTUS dataset (<a href="https://osf.io/hgjdb/">https://osf.io/hgjdb/</a>). In future versions, LNPN will directly feed on the SSOT.</p>
<h5 id="data-edition-1">Data Edition</h5>
<p>The bot framework <a href="https://gitlab.com/lotus7/wikidataLotusImporter">WikidataLotusImporter</a> was adapted such that, in addition to batch upload capabilities, it can also edit erroneously created entries on Wikidata. As massive edits have a large potential to disrupt otherwise good data, progressive deployment of this script is used, starting by editing progressively 1, 10, then 100 entries that are manually checked. Upon validation of 100 entries, the full script is run and check its behavior checked at regular intervals. An example of a corrected entry is as follows: <a href="https://www.wikidata.org/w/index.php?title=Q105349871&amp;type=revision&amp;diff=1365519277&amp;oldid=1356145998">https://www.wikidata.org/w/index.php?title=Q105349871&amp;type=revision&amp;diff=1365519277&amp;oldid=1356145998</a></p>
<h5 id="curation-interface">Curation interface</h5>
<p>A web-based (Kotlin, Spring Boot for the back-end, and TypeScript with Vue for the front-end) curation interface is currently in construction. It will allow mass-editing of entries and navigate quick navigation in the SSOT for the curation of new and existing entries. This new interface is intended to become open to the public to foster the curation of entries by further means, driven by the users. In line with the overall LOTUS approach, any modification made in this curation interface will be mirrored after validation on Wikidata and LNPN.</p>
<h2 id="code-availability">Code Availability</h2>
<h3 id="general-repository">General Repository</h3>
<p>All programs written for this work can be found in the following group: <a href="https://gitlab.com/lotus7">https://gitlab.com/lotus7</a>.</p>
<h4 id="processing">Processing</h4>
<p>The source data curation system is available at <a href="https://gitlab.com/lotus7/lotusProcessor">https://gitlab.com/lotus7/lotusProcessor</a>. This program takes the source data as input and outputs curated data, ready for dissemination. The first step involves checking if the source data has already been processed. If not, all three elements (biological organism, chemical structures, and references) are submitted to various steps of translation and curation, before validation for dissemination.</p>
<h4 id="wikidata-4">Wikidata</h4>
<h5 id="import">Import</h5>
<p>The Wikidata importer is available at <a href="https://gitlab.com/lotus7/wikidataLotusImporter">https://gitlab.com/lotus7/wikidataLotusImporter</a>. This program takes the processed data resulting from the lotusProcessor subprocess as input and uploads it to Wikidata. It performs a SPARQL query to check which objects already exist. If needed, it creates the missing objects. It then updates the content of each object. Finally, it updates the chemical compound page with a “found in taxon” statement complemented with a “stated in” reference.</p>
<h5 id="export">Export</h5>
<p>The Wikidata exporter is available at <a href="https://gitlab.com/lotus7/wikidataLotusExporter">https://gitlab.com/lotus7/wikidataLotusExporter</a>. This program takes the structured data in Wikidata corresponding to chemical compounds found in taxa with a reference associated as input and exports it in both RDF and tabular formats for further use. Two subsequent options are (a) that the end-user can directly use the exported data.; or (b) that the exported data, which can be new or modified since the last iteration, is used as new source data in lotusProcessor.</p>
<h3 id="lnpn-1">LNPN</h3>
<p>The LNPN website and processing system is available at <a href="https://github.com/mSorok/LOTUSweb">https://github.com/mSorok/LOTUSweb</a>. This system takes the processed data resulting from the lotusProcessor as input and uploads it on <a href="https://lotus.naturalproducts.net">https://lotus.naturalproducts.net</a>. The repository is not part of the main GitLab group as it benefits from already established pipelines developed by CS and MS. The website allows searches from different points of view, complemented with taxonomies for both the chemical and biological sides. Many chemical molecular properties and molecular descriptors that otherwise are unavailable in Wikidata are also provided.</p>
<h3 id="code-freezing">Code Freezing</h3>
<p>All repository hyperlinks in the manuscript point to the preprint branches by default. The links contain all programs and code before submission (2021-02-23) and will eventually be updated to a publication branch using modifications resulting from the peer-reviewing process. As the code evolves, readers are invited to refer to the main branch of each repository for the most up-to-date code.
A frozen version (2021-02-23) of all programs and code is also available in the LOTUS OSF repository (<a href="https://osf.io/pmgux/">https://osf.io/pmgux/</a>).</p>
<h3 id="programs-and-packages">Programs and packages</h3>
<h4 id="r">R</h4>
<p>The <a href="https://www.r-project.org/">R</a> version used for the project was 4.0.5 (2021-03-31) – “Shake and Throw”, and R-packages used were, in alphabetical order::
ChemmineR (3.42.1) <span class="citation" data-cites="zukFZ2aq">(Cao et al., <a href="#ref-zukFZ2aq" role="doc-biblioref">2008</a>)</span>, chorddiag (0.1.2) <span class="citation" data-cites="156ifmHvV">(Flor, <a href="#ref-156ifmHvV" role="doc-biblioref">2020</a>)</span>, ClassyfireR (0.3.6) <span class="citation" data-cites="1GFQixHzv">(Djoumbou Feunang et al., <a href="#ref-1GFQixHzv" role="doc-biblioref">2016</a>)</span>, data.table (1.13.6) <span class="citation" data-cites="hR0edX6b">(Dowle and Srinivasan, <a href="#ref-hR0edX6b" role="doc-biblioref">2020</a>)</span>, DBI (1.1.1) <span class="citation" data-cites="7JFonzW0">(R Special Interest Group on Databases (R-SIG-DB) et al., <a href="#ref-7JFonzW0" role="doc-biblioref">2021</a>)</span>, gdata (2.18.0) <span class="citation" data-cites="7AFWc7sN">(Warnes et al., <a href="#ref-7AFWc7sN" role="doc-biblioref">2017</a>)</span>, ggalluvial (0.12.3) <span class="citation" data-cites="13wsG3vmM">(Brunson, <a href="#ref-13wsG3vmM" role="doc-biblioref">2020</a>)</span>, ggfittext (0.9.1) <span class="citation" data-cites="ejxxPICa">(Wilkins, <a href="#ref-ejxxPICa" role="doc-biblioref">2020</a>)</span>, ggnewscale (0.4.5) <span class="citation" data-cites="7J4ZPUL7">(Campitelli, <a href="#ref-7J4ZPUL7" role="doc-biblioref">2021</a>)</span>, ggraph (2.0.4) <span class="citation" data-cites="gdTufCbt">(Pedersen, <a href="#ref-gdTufCbt" role="doc-biblioref">2020</a>)</span>, ggstar (1.0.1) <span class="citation" data-cites="17NxHx9HG">(Xu, <a href="#ref-17NxHx9HG" role="doc-biblioref">2021</a>)</span>, ggtree (2.4.1) <span class="citation" data-cites="YdA5Ivd8">(Yu et al., <a href="#ref-YdA5Ivd8" role="doc-biblioref">2016</a>)</span>, ggtreeExtra (1.0.1) <span class="citation" data-cites="121McWIls">(Xu et al., <a href="#ref-121McWIls" role="doc-biblioref">2021</a>)</span>, Hmisc (4.4-2) <span class="citation" data-cites="dkabTMGA">(Jr et al., <a href="#ref-dkabTMGA" role="doc-biblioref">2020</a>)</span>, jsonlite (1.7.2) <span class="citation" data-cites="Jdw1vygr">(Ooms, <a href="#ref-Jdw1vygr" role="doc-biblioref">2014</a>)</span>, pbmcapply (1.5.0) <span class="citation" data-cites="XiPNsL5P">(Kuang et al., <a href="#ref-XiPNsL5P" role="doc-biblioref">2019</a>)</span>, plotly (4.9.3) <span class="citation" data-cites="kVXkIA2g">(Sievert, <a href="#ref-kVXkIA2g" role="doc-biblioref">2020</a>)</span>, rcrossref(1.1.0) <span class="citation" data-cites="UHC5x6VA">(Chamberlain et al., <a href="#ref-UHC5x6VA" role="doc-biblioref">2020</a>)</span>, readxl (1.3.1) <span class="citation" data-cites="xN7NElEK">(Wickham and Bryan, <a href="#ref-xN7NElEK" role="doc-biblioref">2019</a>)</span>, rentrez (1.2.3) <span class="citation" data-cites="13ZrxSjw4">(Winter, <a href="#ref-13ZrxSjw4" role="doc-biblioref">2017</a>)</span>, rotl (3.0.11) <span class="citation" data-cites="hPmhpanN">(Michonneau et al., <a href="#ref-hPmhpanN" role="doc-biblioref">2016</a>)</span>, rvest (0.3.6) <span class="citation" data-cites="1ELrikmXF">(Wickham, <a href="#ref-1ELrikmXF" role="doc-biblioref">2020</a>)</span>, splitstackshape (1.4.8) <span class="citation" data-cites="NfiSMvAq">(Mahto, <a href="#ref-NfiSMvAq" role="doc-biblioref">2019</a>)</span>, RSQLite (2.2.3) <span class="citation" data-cites="LnilgHS9">(Müller et al., <a href="#ref-LnilgHS9" role="doc-biblioref">2021</a>)</span>, stringdist (0.9.6.3) <span class="citation" data-cites="LmFzkeBM">(Loo, <a href="#ref-LmFzkeBM" role="doc-biblioref">2014</a>)</span>, stringi (1.5.3) <span class="citation" data-cites="12mR83H1T">(Gagolewski, <a href="#ref-12mR83H1T" role="doc-biblioref">2020</a>)</span>, tidyverse (1.3.0) <span class="citation" data-cites="EWll2pyF">(Wickham et al., <a href="#ref-EWll2pyF" role="doc-biblioref">2019</a>)</span>, treeio (1.14.3) <span class="citation" data-cites="1EloBUtuI">(Wang et al., <a href="#ref-1EloBUtuI" role="doc-biblioref">2020</a>)</span>, UpSetR (1.4.0) <span class="citation" data-cites="T2D9o6RC">(Gehlenborg, <a href="#ref-T2D9o6RC" role="doc-biblioref">2019</a>)</span>, vroom (1.3.2) <span class="citation" data-cites="otfsIdc0">(Hester and Wickham, <a href="#ref-otfsIdc0" role="doc-biblioref">2020</a>)</span>, webchem (1.1.1) <span class="citation" data-cites="kM2kraS2">(Szöcs et al., <a href="#ref-kM2kraS2" role="doc-biblioref">2020</a>)</span>, XML (3.99-05) <span class="citation" data-cites="TceFUyOp">(Lang, <a href="#ref-TceFUyOp" role="doc-biblioref">2020</a>)</span>, xml2 (1.3.2) <span class="citation" data-cites="2keqmgUs">(Wickham et al., <a href="#ref-2keqmgUs" role="doc-biblioref">2020</a>)</span></p>
<h4 id="python">Python</h4>
<p>The <a href="https://www.python.org/">Python</a> version used was 3.8.6, and the Python packages utilized were, in alphabetical order:
faerun (0.3.2) <span class="citation" data-cites="VMUBsFFG">(Probst and Reymond, <a href="#ref-VMUBsFFG" role="doc-biblioref">2018</a><a href="#ref-VMUBsFFG" role="doc-biblioref">a</a>)</span>, map4 (1.0) <span class="citation" data-cites="2HOkPhmh">(Capecchi et al., <a href="#ref-2HOkPhmh" role="doc-biblioref">2020</a>)</span>, matplotlib (3.1.3) <span class="citation" data-cites="1026Gxdsi">(Hunter, <a href="#ref-1026Gxdsi" role="doc-biblioref">2007</a>)</span>, <a href="https://github.com/mcs07/MolVS">Molvs</a> (0.1.1), pandas (1.1.4) <span class="citation" data-cites="zEc5CK8P">(Reback et al., <a href="#ref-zEc5CK8P" role="doc-biblioref">2020</a>)</span>, rdkit (2020.09.2) <span class="citation" data-cites="1BXLY6HF1">(“RDKit: Open-source cheminformatics,” <a href="#ref-1BXLY6HF1" role="doc-biblioref">n.d.</a>)</span>, scipy (1.5.0) <span class="citation" data-cites="8Miti2Gz">(Virtanen et al., <a href="#ref-8Miti2Gz" role="doc-biblioref">2020</a>)</span>, tmap (1.0.4) <span class="citation" data-cites="1DcpY38I1">(Probst and Reymond, <a href="#ref-1DcpY38I1" role="doc-biblioref">2020</a>)</span>.</p>
<h4 id="kotlin">Kotlin</h4>
<p>Kotlin packages used were as follows:
Common: Kotlin 1.4.21 up to 1.4.30, Univocity 2.9.0, OpenJDK 15, Kotlin serialization 1.0.1, konnector 0.1.27, Log4J 2.14.0
Wikidata Importer Bot:, WikidataTK 0.11.1, CDK 2.3 <span class="citation" data-cites="xpBWVpji">(Willighagen et al., <a href="#ref-xpBWVpji" role="doc-biblioref">2017</a>)</span>, RDF4J 3.6.0, Ktor 1.5.0, KotlinXCli 0.3.1, Wikidata data processing: Shadow 5.0.0
Quality control and testing: Ktlint 9.4.1, Kotlinter 3.3.0, Detekt 1.15.0, Ben Mane’s version plugin 0.36.0, Junit 5.7.0</p>
<h4 id="additional-executable-files">Additional executable files</h4>
<p><a href="https://github.com/gnames/gnfinder/releases/tag/v0.12.1">GNFinder</a> v.0.12.1, <a href="https://github.com/gnames/gnverifier/releases/tag/v0.3.1">GNVerifier</a> v.0.3.1, <a href="https://github.com/dan2097/opsin/releases/tag/2.5.0">OPSIN</a> v.2.5.0 <span class="citation" data-cites="pf43Zseq">(Lowe et al., <a href="#ref-pf43Zseq" role="doc-biblioref">2011</a>)</span></p>
<h2 id="data-availability">Data Availability</h2>
<p>A snapshot of the obtained data at the time of submission is available at the following OSF repository (data): <a href="https://osf.io/pmgux/">https://osf.io/pmgux/</a>. The <a href="https://lotus.nprod.net">https://lotus.nprod.net</a> website is intended to gather news and features related to the LOTUS initiative in the future.</p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>JLW and PMA are thankful to the Swiss National Science Foundation for supporting part of this project through the SNF Sinergia grant CRSII5_189921. JB and AR are really thankful to JetBrains for the Free educational license of IntelliJ and the excellent support received on Youtrack. JB, JGG, and GFP gratefully acknowledge the support of this work by grant U41 AT008706 and supplemental funding to P50 AT000155 from NCCIH and ODS of the NIH. MS and CS are supported by the German Research Foundation within the framework ChemBioSys (Project-ID 239748522, SFB 1127). The work on the Wikidata IDSM/Sachem endpoint was supported by an ELIXIR CZ research infrastructure project grant (MEYS Grant No: LM2018131) including access to computing and storage facilities. The authors would like to thank <a href="https://github.com/dimus">Dmitry Mozzherin</a> for his work done for the Global Names Architecture and related improvements. The authors would also like to thank Layla Michán for starting to add pigment information on Wikidata. EW and DM acknowledge the Scholia grant from the Alfred P. Sloan Foundation under grant number G-2019-11458. The authors would also like to thank contributors of all electronic NP resources used in this work and the NP community at large.</p>
<h2 id="competing-interests">Competing interests</h2>
<p>The authors declare no competing interest.</p>
<h2 class="page_break_before" id="author-contributions">Author contributions</h2>
<figure>
<img src="images/author_contribution.svg" class="author-contribution" alt="" />
</figure>
<h2 class="page_break_before" id="supporting-information">Supporting Information</h2>
<h3 id="si-1-data-sources-list">SI 1 Data Sources List</h3>
<div style="font-size:10pt">
<div id="tbl:si-1" class="tablenos">
<table>
<caption><span>Table SI-1:</span> Data Sources List </caption>
<colgroup>
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
<col style="width: 9%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">database</th>
<th style="text-align: center;">type</th>
<th style="text-align: center;">initial retrieved unique observations</th>
<th style="text-align: center;">cleaned documented structure-organism pairs</th>
<th style="text-align: center;">pairs validated for wikidata export</th>
<th style="text-align: center;">website</th>
<th style="text-align: center;">article</th>
<th style="text-align: center;">retrieval</th>
<th style="text-align: center;">license</th>
<th style="text-align: center;">contact</th>
<th style="text-align: center;">varia</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">afrotryp</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">312</td>
<td style="text-align: center;">135</td>
<td style="text-align: center;">28</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://link.springer.com/article/10.1007/s00044-016-1764-y?shared-article-renderer">article</a> <span class="citation" data-cites="mpr5sPI7">(Ibezim et al., <a href="#ref-mpr5sPI7" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;"><a href="https://static-content.springer.com/esm/art%3A10.1007%2Fs00044-016-1764-y/MediaObjects/44_2016_1764_MOESM1_ESM.doc">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/afrotryp.md">license</a></td>
<td style="text-align: center;"><a href="https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c1/new">Fidele Ntie-Kang</a> or <a href="https://link.springer.com/article/10.1007/s00044-016-1764-y/email/correspondent/c2/new">Ngozi Justina Nwodo</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">alkamid</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">4,434</td>
<td style="text-align: center;">2,582</td>
<td style="text-align: center;">2,076</td>
<td style="text-align: center;"><a href="http://alkamid.ugent.be">website</a></td>
<td style="text-align: center;"><a href="http://dx.doi.org/10.1016/j.jep.2012.05.038">article</a> <span class="citation" data-cites="Go2AAIYt">(Boonen et al., <a href="#ref-Go2AAIYt" role="doc-biblioref">2012</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/alkamid/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/alkamid.md">license</a></td>
<td style="text-align: center;"><a href="Bart.DeSpiegeleer@UGent.be">Bart De Spiegeleer</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">biofacquim</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">531</td>
<td style="text-align: center;">683</td>
<td style="text-align: center;">534</td>
<td style="text-align: center;"><a href="https://biofacquim.herokuapp.com">website</a> (old version)</td>
<td style="text-align: center;"><a href="https://www.mdpi.com/2218-273X/9/1/31">article_old</a> <a href="https://f1000research.com/articles/8-2071">article_new</a> <span class="citation" data-cites="Ue7W1SPi">(Pilón-Jiménez et al., <a href="#ref-Ue7W1SPi" role="doc-biblioref">2019</a>)</span></td>
<td style="text-align: center;"><a href="https://ndownloader.figshare.com/files/20050244">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/biofacquim.md">license</a></td>
<td style="text-align: center;"><a href="medinajl@unam.mx">José Medina-Franco</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">biophytmol</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">546</td>
<td style="text-align: center;">628</td>
<td style="text-align: center;">353</td>
<td style="text-align: center;"><a href="http://ab-openlab.csir.res.in/biophytmol/">website</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2">article</a> <span class="citation" data-cites="GFBBPIgV">(Sharma et al., <a href="#ref-GFBBPIgV" role="doc-biblioref">2014</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/biophytmol/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/biophytmol.md">license</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-014-0046-2/email/correspondent/c1/new">Anshu Bhardwaj</a></td>
<td style="text-align: center;">website often down</td>
</tr>
<tr class="odd">
<td style="text-align: center;">carotenoiddb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">2,922</td>
<td style="text-align: center;">1,199</td>
<td style="text-align: center;">639</td>
<td style="text-align: center;"><a href="http://carotenoiddb.jp/About.html">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1093/database/bax004">article</a> <span class="citation" data-cites="FzUbylmA">(Yabuzaki, <a href="#ref-FzUbylmA" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/carotenoiddb/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/carotenoiddb.md">license</a></td>
<td style="text-align: center;"><a href="yzjunko@gmail.com">yzjunko@gmail.com</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">coconut</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">583,623</td>
<td style="text-align: center;">345,328</td>
<td style="text-align: center;">34,429</td>
<td style="text-align: center;"><a href="https://coconut.naturalproducts.net">website</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9">article</a> <span class="citation" data-cites="R06ZE1xj">(Sorokina and Steinbeck, <a href="#ref-R06ZE1xj" role="doc-biblioref">2020</a><a href="#ref-R06ZE1xj" role="doc-biblioref">a</a>)</span></td>
<td style="text-align: center;"><a href="https://zenodo.org/record/3778405/files/COCONUT.sdf.zip?download=1">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/coconut.md">license</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-020-00424-9/email/correspondent/c1/new">Maria Sorokina</a></td>
<td style="text-align: center;"><a href="https://zenodo.org/record/3778405#.XsylyC1Ph0Y">zenodo</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;">cyanometdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">1,930</td>
<td style="text-align: center;">1,844</td>
<td style="text-align: center;">1,774</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://doi.org/10.1016/j.watres.2021.117017">article</a> <span class="citation" data-cites="OhONtrcV">(Jones et al., <a href="#ref-OhONtrcV" role="doc-biblioref">2021</a>)</span></td>
<td style="text-align: center;"><a href="https://ars.els-cdn.com/content/image/1-s2.0-S0043135421002153-mmc4.zip">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/cyanometdb.md">license</a></td>
<td style="text-align: center;"><a href="elisabeth.janssen@eawag.ch">elisabeth.janssen@eawag.ch</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">datawarrior</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">589</td>
<td style="text-align: center;">1,062</td>
<td style="text-align: center;">102</td>
<td style="text-align: center;"><a href="http://www.openmolecules.org/datawarrior/index.html">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1021/ci500588j">article</a> <span class="citation" data-cites="bOvsdNwM">(Sander et al., <a href="#ref-bOvsdNwM" role="doc-biblioref">2015</a>)</span></td>
<td style="text-align: center;"><a href="http://www.openmolecules.org/datawarrior/download.html">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/datawarrior.md">icense</a></td>
<td style="text-align: center;"><a href="thomas.sander@idorsia.com">thomas.sander@idorsia.com</a></td>
<td style="text-align: center;">no real link to the dataset inside it</td>
</tr>
<tr class="odd">
<td style="text-align: center;">dianatdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">290</td>
<td style="text-align: center;">404</td>
<td style="text-align: center;">27</td>
<td style="text-align: center;"><a href="http://rdu.iquimica.unam.mx/handle/20.500.12214/1186">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1039/D0RA10453A">article</a> <span class="citation" data-cites="dyiEPToF">(Madariaga-Mazón et al., <a href="#ref-dyiEPToF" role="doc-biblioref">2021</a>)</span></td>
<td style="text-align: center;"><a href="http://rdu.iquimica.unam.mx/bitstream/20.500.12214/1186/3/2020_DiaNatDB_336.xlsx">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/dianatdb.md">license</a></td>
<td style="text-align: center;"><a href="amadariaga@iquimica.unam.mx">amadariaga@iquimica.unam.mx</a> or <a href="kmtzm@unam.mx">kmtzm@unam.mx</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">dnp</td>
<td style="text-align: center;">commercial</td>
<td style="text-align: center;">210,832</td>
<td style="text-align: center;">258,328</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="http://dnp.chemnetbase.com/faces/chemical/ChemicalSearch.xhtml">website</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/dnp/extracting.py">script</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="support@taylorfrancis.com">support@taylorfrancis.com</a></td>
<td style="text-align: center;">commercial</td>
</tr>
<tr class="odd">
<td style="text-align: center;">drduke</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">90,675</td>
<td style="text-align: center;">9,072</td>
<td style="text-align: center;">4,266</td>
<td style="text-align: center;"><a href="https://phytochem.nal.usda.gov/phytochem/help/index/about">website</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://data.nal.usda.gov/system/files/Duke-Source-CSV.zip">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/drduke.md">license</a></td>
<td style="text-align: center;"><a href="agref@usda.gov">agref@usda.gov</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">foodb</td>
<td style="text-align: center;">restricted</td>
<td style="text-align: center;">82,415</td>
<td style="text-align: center;">361</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://foodb.ca/about#cite">website</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://foodb.ca/public/system/downloads/foodb_2020_4_7_csv.tar.gz">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/foodb.md">license</a></td>
<td style="text-align: center;"><a href="jreid3@ualberta.ca">jreid3@ualberta.ca</a> (Jennifer)</td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">inflamnat</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">665</td>
<td style="text-align: center;">656</td>
<td style="text-align: center;">282</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://pubs.acs.org/doi/10.1021/acs.jcim.8b00560">article</a> <span class="citation" data-cites="lB6krLxn">(Zhang et al., <a href="#ref-lB6krLxn" role="doc-biblioref">2018</a>)</span></td>
<td style="text-align: center;"><a href="https://pubs.acs.org/doi/suppl/10.1021/acs.jcim.8b00560/suppl_file/ci8b00560_si_001.xlsx">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/inflamnat.md">license</a></td>
<td style="text-align: center;"><a href="xiaoweilie@ynu.edu.cn">xiaoweilie@ynu.edu.cn</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">knapsack</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">116,284</td>
<td style="text-align: center;">143,062</td>
<td style="text-align: center;">62,727</td>
<td style="text-align: center;"><a href="http://www.knapsackfamily.com/knapsack_core/top.php">website</a></td>
<td style="text-align: center;"><a href="https://link.springer.com/chapter/10.1007/3-540-29782-0_13">article</a> <span class="citation" data-cites="2fEiIH9c pIM4qxZA">(Afendi et al., <a href="#ref-pIM4qxZA" role="doc-biblioref">2012</a>; Shinbo et al., <a href="#ref-2fEiIH9c" role="doc-biblioref">2006</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/knapsack/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/knapsack.md">license</a></td>
<td style="text-align: center;"><a href="skanaya@gtc.naist.jp">skanaya@gtc.naist.jp</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">metabolights</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">32,928</td>
<td style="text-align: center;">32,484</td>
<td style="text-align: center;">4,919</td>
<td style="text-align: center;"><a href="https://www.ebi.ac.uk/metabolights/about">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1093/nar/gkz1019">article</a> <span class="citation" data-cites="1D8ShigK1">(Haug et al., <a href="#ref-1D8ShigK1" role="doc-biblioref">2019</a>)</span></td>
<td style="text-align: center;"><a href="ftp://ftp.ebi.ac.uk/pub/databases/metabolights/eb-eye/eb-eye_metabolights_complete.xml">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/metabolights.md">license</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">mibig</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">1,310</td>
<td style="text-align: center;">1,142</td>
<td style="text-align: center;">548</td>
<td style="text-align: center;"><a href="https://mibig.secondarymetabolites.org">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1093/nar/gkz882">article</a> <span class="citation" data-cites="F5HJbaQH">(Kautsar et al., <a href="#ref-F5HJbaQH" role="doc-biblioref">2019</a>)</span></td>
<td style="text-align: center;"><a href="https://dl.secondarymetabolites.org/mibig/mibig_json_2.0.tar.gz">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/mibig.md">license</a></td>
<td style="text-align: center;"><a href="tiwe@biosustain.dtu.dk">Tilmann Weber</a> or <a href="marnix.medema@wur.nl">Marnix Medema</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">mitishamba</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">1,073</td>
<td style="text-align: center;">1,159</td>
<td style="text-align: center;">368</td>
<td style="text-align: center;"><a href="http://mitishamba.uonbi.ac.ke">website</a></td>
<td style="text-align: center;"><a href="http://erepository.uonbi.ac.ke/handle/11295/92273">“article”</a> <span class="citation" data-cites="fAKz2qgk">(Derese, Solomon et al., <a href="#ref-fAKz2qgk" role="doc-biblioref">31st August to 3rd September 2015</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/mitishamba/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/mitishamba.md">license</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">nanpdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">5,752</td>
<td style="text-align: center;">6,447</td>
<td style="text-align: center;">5,303</td>
<td style="text-align: center;"><a href="http://african-compounds.org/nanpdb/">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1021/acs.jnatprod.7b00283">article</a> <span class="citation" data-cites="6xTnsq5S">(Ntie-Kang et al., <a href="#ref-6xTnsq5S" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/nanpdb/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/nanpdb.md">license</a></td>
<td style="text-align: center;"><a href="ntiekfidele@gmail.com">ntiekfidele@gmail.com</a> <a href="stefan.guenther@pharmazie.uni-freiburg.de">stefan.guenther@pharmazie.uni-freiburg.de</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">napralert</td>
<td style="text-align: center;">commercial</td>
<td style="text-align: center;">681,401</td>
<td style="text-align: center;">380,860</td>
<td style="text-align: center;">261,545</td>
<td style="text-align: center;"><a href="https://www.napralert.org">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1016/B978-008045382-8.00060-5">article</a> <span class="citation" data-cites="KmGsdzOE">(Graham and Farnsworth, <a href="#ref-KmGsdzOE" role="doc-biblioref">2010</a>)</span></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/napralert.md">license</a></td>
<td style="text-align: center;"><a href="napralert@uic.edu">napralert@uic.edu</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">npass</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">290,539</td>
<td style="text-align: center;">34,403</td>
<td style="text-align: center;">23,060</td>
<td style="text-align: center;"><a href="http://bidd2.nus.edu.sg/NPASS/about.php">website</a></td>
<td style="text-align: center;"><a href="https://academic.oup.com/nar/article/46/D1/D1217/4584628">article</a> <span class="citation" data-cites="4RKmYOPE">(Zeng et al., <a href="#ref-4RKmYOPE" role="doc-biblioref">2018</a>)</span></td>
<td style="text-align: center;"><a href="http://bidd.group/NPASS/downloadnpass.html">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/npass.md">license</a></td>
<td style="text-align: center;"><a href="phacyz@nus.edu.sg">phacyz@nus.edu.sg</a> <a href="jiangyy@sz.tsinghua.edu.cn">jiangyy@sz.tsinghua.edu.cn</a> <a href="iaochen@163.com">iaochen@163.com</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">npatlas</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">29,006</td>
<td style="text-align: center;">49,968</td>
<td style="text-align: center;">44,466</td>
<td style="text-align: center;"><a href="https://www.npatlas.org/joomla/">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1021/acscentsci.9b00806">article</a> <span class="citation" data-cites="1H1kBOCRr">(van Santen et al., <a href="#ref-1H1kBOCRr" role="doc-biblioref">2019</a>)</span></td>
<td style="text-align: center;"><a href="https://www.npatlas.org/custom/versions/np_atlas_2020_06/NPAtlas_download.tsv">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/npatlas.md">license</a></td>
<td style="text-align: center;"><a href="rliningt@sfu.ca">rliningt@sfu.ca</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">npcare</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">7,763</td>
<td style="text-align: center;">4,650</td>
<td style="text-align: center;">2,525</td>
<td style="text-align: center;"><a href="http://silver.sejong.ac.kr/npcare/">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1186/s13321-016-0188-5">article</a> <span class="citation" data-cites="1DPlqketM">(Choi et al., <a href="#ref-1DPlqketM" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;"><a href="http://silver.sejong.ac.kr/npcare/csv/npcare.zip">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/npcare.md">license</a></td>
<td style="text-align: center;"><a href="choihwanho@gmail.com">choihwanho@gmail.com</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">npedia</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">82</td>
<td style="text-align: center;">99</td>
<td style="text-align: center;">23</td>
<td style="text-align: center;"><a href="http://www.cbrg.riken.jp/npedia/?LANG=en">website</a></td>
<td style="text-align: center;"><a href="https://www.jstage.jst.go.jp/article/ciqs/2006/0/2006_0_JL6/_article">article</a> <span class="citation" data-cites="mtVu8pJx">(富木 et al., <a href="#ref-mtVu8pJx" role="doc-biblioref">2006</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/npedia/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/npedia.md">license</a></td>
<td style="text-align: center;"><a href="hisyo@riken.jp">hisyo@riken.jp</a> <a href="npd@riken.jp">npd@riken.jp</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">nubbe</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">2,189</td>
<td style="text-align: center;">2,614</td>
<td style="text-align: center;">2,389</td>
<td style="text-align: center;"><a href="https://nubbe.iq.unesp.br/portal/nubbe-search.html">website</a></td>
<td style="text-align: center;"><a href="https://www.nature.com/articles/s41598-017-07451-x">article</a> <span class="citation" data-cites="1EBJWvXBS">(Pilon et al., <a href="#ref-1EBJWvXBS" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/nubbe.md">license</a></td>
<td style="text-align: center;"><a href="https://www.nature.com/articles/s41598-017-07451-x/email/correspondent/c1/new">Vanderlan S. Bolzani</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">pamdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">3,061</td>
<td style="text-align: center;">3,198</td>
<td style="text-align: center;">64</td>
<td style="text-align: center;"><a href="http://pseudomonas.umaryland.edu/PAMDB.htm">website</a></td>
<td style="text-align: center;"><a href="https://academic.oup.com/nar/article-lookup/doi/10.1093/nar/gkx1061">article</a> <span class="citation" data-cites="yJZjE7TJ">(Huang et al., <a href="#ref-yJZjE7TJ" role="doc-biblioref">2018</a>)</span></td>
<td style="text-align: center;"><a href="http://pseudomonas.umaryland.edu/PaDl/PaMet.xlsx">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/pamdb.md">license</a></td>
<td style="text-align: center;"><a href="awilks@rx.umaryland.edu">awilks@rx.umaryland.edu</a> <a href="aoglesby@rx.umaryland.edu">aoglesby@rx.umaryland.edu</a> <a href="mkane@rx.umaryland.edu">mkane@rx.umaryland.edu</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">phenolexplorer</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">8,968</td>
<td style="text-align: center;">12,027</td>
<td style="text-align: center;">6,536</td>
<td style="text-align: center;"><a href="http://phenol-explorer.eu/cite_us">website</a></td>
<td style="text-align: center;"><a href="http://database.oxfordjournals.org/content/2013/bat070.long">article</a> <span class="citation" data-cites="9GVjNRag">(Rothwell et al., <a href="#ref-9GVjNRag" role="doc-biblioref">2013</a>)</span></td>
<td style="text-align: center;"><a href="http://phenol-explorer.eu/downloads">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/phenolexplorer.md">license</a></td>
<td style="text-align: center;"><a href="scalberta@iarc.fr">scalberta@iarc.fr</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">phytohub</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">2,363</td>
<td style="text-align: center;">1,451</td>
<td style="text-align: center;">50</td>
<td style="text-align: center;"><a href="http://phytohub.eu/about">website</a></td>
<td style="text-align: center;"><a href="https://hal.archives-ouvertes.fr/hal-01697081/file/2017_Giacomoni_ICPH_Québec.pdf">“article”</a> <span class="citation" data-cites="1APrBLQ3X">(Giacomoni et al., <a href="#ref-1APrBLQ3X" role="doc-biblioref">2017</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/phytohub/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/phytohub.md">license</a></td>
<td style="text-align: center;"><a href="claudine.manach@inra.fr">claudine.manach@inra.fr</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">procardb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">6,606</td>
<td style="text-align: center;">9,933</td>
<td style="text-align: center;">70</td>
<td style="text-align: center;"><a href="https://bioinfo.imtech.res.in/servers/procardb/">website</a></td>
<td style="text-align: center;"><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6">article</a> <span class="citation" data-cites="Pf1Puxxg">(Nupur et al., <a href="#ref-Pf1Puxxg" role="doc-biblioref">2016</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/procardb/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/procardb.md">license</a></td>
<td style="text-align: center;"><a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c1/new">Anil Kumar Pinnaka</a> <a href="https://bmcmicrobiol.biomedcentral.com/articles/10.1186/s12866-016-0715-6/email/correspondent/c2/new">Ashwani Kumar</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">respect</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">2,759</td>
<td style="text-align: center;">607</td>
<td style="text-align: center;">263</td>
<td style="text-align: center;"><a href="http://spectra.psc.riken.jp/menta.cgi/respect/licence/licence">website</a></td>
<td style="text-align: center;"><a href="https://www.sciencedirect.com/science/article/pii/S003194221200310X">article</a> <span class="citation" data-cites="12Ui1Rc6x">(Sawada et al., <a href="#ref-12Ui1Rc6x" role="doc-biblioref">2012</a>)</span></td>
<td style="text-align: center;"><a href="http://spectra.psc.riken.jp/menta.cgi/static/respect/respect.zip">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/respect.md">license</a></td>
<td style="text-align: center;"><a href="ksaito@psc.riken.jp">ksaito@psc.riken.jp</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">sancdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">861</td>
<td style="text-align: center;">987</td>
<td style="text-align: center;">774</td>
<td style="text-align: center;"><a href="https://sancdb.rubi.ru.ac.za/cite_us/">website</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8">article</a> <span class="citation" data-cites="ZRFusbLt">(Hatherley et al., <a href="#ref-ZRFusbLt" role="doc-biblioref">2015</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/sancdb/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/sancdb.md">license</a></td>
<td style="text-align: center;"><a href="https://jcheminf.biomedcentral.com/articles/10.1186/s13321-015-0080-8/email/correspondent/c1/new">Özlem Tastan Bishop</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">streptomedb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">71,638</td>
<td style="text-align: center;">38,343</td>
<td style="text-align: center;">19,021</td>
<td style="text-align: center;"><a href="http://132.230.56.4/streptomedb3/">website</a></td>
<td style="text-align: center;"><a href="https://doi.org/10.1093/nar/gkv1319">article</a> <span class="citation" data-cites="15rL9Psbo">(Klementz et al., <a href="#ref-15rL9Psbo" role="doc-biblioref">2016</a>)</span></td>
<td style="text-align: center;"><a href="http://132.230.56.4/streptomedb/download/">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/streptomedb.md">license</a></td>
<td style="text-align: center;"><a href="stefan.guenther@pharmazie.uni-freiburg.de">stefan.guenther@pharmazie.uni-freiburg.de</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">swmd</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">1,075</td>
<td style="text-align: center;">1,616</td>
<td style="text-align: center;">1,377</td>
<td style="text-align: center;"><a href="http://www.swmd.co.in/index.php">website</a></td>
<td style="text-align: center;"><a href="https://dx.doi.org/10.6026%2F97320630005361">article</a> <span class="citation" data-cites="tPji342r">(Davis and Vasanthi, <a href="#ref-tPji342r" role="doc-biblioref">2011</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/swmd/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/swmd.md">license</a></td>
<td style="text-align: center;"><a href="Dicky.John@gmail.com">Dicky.John@gmail.com</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">tmdb</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">2,116</td>
<td style="text-align: center;">841</td>
<td style="text-align: center;">17</td>
<td style="text-align: center;"><a href="http://pcsb.ahau.edu.cn:8080/TCDB/index.jsp">website</a></td>
<td style="text-align: center;"><a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1">article</a> <span class="citation" data-cites="16mSGZn7C">(Yue et al., <a href="#ref-16mSGZn7C" role="doc-biblioref">2014</a>)</span></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/tmdb/scraping.R">script</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/tmdb.md">license</a></td>
<td style="text-align: center;"><a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c1/new">Xiao-Chun Wan</a> <a href="https://bmcplantbiol.biomedcentral.com/articles/10.1186/s12870-014-0243-1/email/correspondent/c2/new">Guan-Hu Bao</a></td>
<td style="text-align: center;">currently down</td>
</tr>
<tr class="even">
<td style="text-align: center;">tmmc</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">15,033</td>
<td style="text-align: center;">5,771</td>
<td style="text-align: center;">2,662</td>
<td style="text-align: center;"><a href="http://informatics.kiom.re.kr/compound/index.jsp">website</a></td>
<td style="text-align: center;"><a href="https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5">article</a> <span class="citation" data-cites="a48CrnhC">(Kim et al., <a href="#ref-a48CrnhC" role="doc-biblioref">2015</a>)</span></td>
<td style="text-align: center;"><a href="http://informatics.kiom.re.kr/compound/download/compound.xlsx">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/tmmc.md">license</a></td>
<td style="text-align: center;"><a href="https://bmccomplementmedtherapies.biomedcentral.com/articles/10.1186/s12906-015-0758-5/email/correspondent/c1/new">Jeong-Ju Lee</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">tppt</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">27,182</td>
<td style="text-align: center;">28,412</td>
<td style="text-align: center;">941</td>
<td style="text-align: center;"><a href="https://www.agroscope.admin.ch/agroscope/en/home/publications/apps/tppt.html">website</a></td>
<td style="text-align: center;"><a href="https://pubs.acs.org/doi/10.1021/acs.jafc.8b01639">article</a> <span class="citation" data-cites="fP0lai2j">(Günthardt et al., <a href="#ref-fP0lai2j" role="doc-biblioref">2018</a>)</span></td>
<td style="text-align: center;"><a href="https://www.agroscope.admin.ch/dam/agroscope/de/dokumente/publikationen/tppt-xls.xlsx.download.xlsx/TPPT_database.xlsx">download</a></td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/blob/main/docs/licenses/tppt.md">license</a></td>
<td style="text-align: center;"><a href="thomas.bucheli@agroscope.admin.ch">thomas.bucheli@agroscope.admin.ch</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">unpd</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">340,319</td>
<td style="text-align: center;">552,307</td>
<td style="text-align: center;">352,814</td>
<td style="text-align: center;"><a href="https://web.archive.org/web/20160314180639/http://pkuxxj.pku.edu.cn/UNPD/introduction.php">website</a></td>
<td style="text-align: center;"><a href="https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0062839">article</a> <span class="citation" data-cites="W6BmnntI">(Gu et al., <a href="#ref-W6BmnntI" role="doc-biblioref">2013</a>)</span></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="lirongc@pku.edu.cn">lirongc@pku.edu.cn</a> <a href="xiaojxu@pku.edu.cn">xiaojxu@pku.edu.cn</a></td>
<td style="text-align: center;">-</td>
</tr>
<tr class="odd">
<td style="text-align: center;">wakankensaku</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">367</td>
<td style="text-align: center;">49</td>
<td style="text-align: center;">41</td>
<td style="text-align: center;"><a href="https://wakankensaku.inm.u-toyama.ac.jp/wiki/Main_Page">website</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/main/src/1_gathering/db/wakankensaku/scraping.R">script</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
</tr>
<tr class="even">
<td style="text-align: center;">wikidata</td>
<td style="text-align: center;">open</td>
<td style="text-align: center;">nextStep</td>
<td style="text-align: center;">nextStep</td>
<td style="text-align: center;">nextStep</td>
<td style="text-align: center;"><a href="https://www.wikidata.org/wiki/Wikidata:Main_Page">website</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><a href="https://w.wiki/3JpE">download</a></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
</tr>
</tbody>
</table>
</div>
</div>
<h3 class="page_break_before" id="si-2-summary-of-the-validation-statistics">SI 2 Summary of the Validation Statistics</h3>
<div id="tbl:si-2" class="tablenos">
<table style="width:100%;">
<caption><span>Table SI-2:</span> Summary of the Validation Statistics </caption>
<colgroup>
<col style="width: 15%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
<col style="width: 8%" />
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;"></th>
<th style="text-align: center;"><strong>First validation dataset (n =420)</strong> <!-- $colspan="8" --></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"></th>
<th style="text-align: center;"><strong>Second validation dataset (n = 100)</strong> <!-- $colspan="2" --></th>
<th style="text-align: center;"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><strong>Reference Type </strong></td>
<td style="text-align: center;"><strong>True positives</strong></td>
<td style="text-align: center;"><strong>False positives</strong></td>
<td style="text-align: center;"><strong>False negatives</strong></td>
<td style="text-align: center;"><strong>True negatives</strong></td>
<td style="text-align: center;"><strong>Relative abundance</strong></td>
<td style="text-align: center;"><strong>Precision</strong></td>
<td style="text-align: center;"><strong>Recall</strong></td>
<td style="text-align: center;"><strong>F<sub>0.5</sub> score</strong></td>
<td style="text-align: center;"><strong>True positives</strong></td>
<td style="text-align: center;"><strong>False negatives</strong></td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Original</strong></td>
<td style="text-align: center;">80</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">7</td>
<td style="text-align: center;">11</td>
<td style="text-align: center;">0.31</td>
<td style="text-align: center;">0.93</td>
<td style="text-align: center;">0.92</td>
<td style="text-align: center;">0.92</td>
<td style="text-align: center;">38</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Pubmed</strong></td>
<td style="text-align: center;">37</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">0.30</td>
<td style="text-align: center;">0.97</td>
<td style="text-align: center;">0.88</td>
<td style="text-align: center;">0.92</td>
<td style="text-align: center;">5</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>DOI</strong></td>
<td style="text-align: center;">115</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">6</td>
<td style="text-align: center;">0.19</td>
<td style="text-align: center;">0.95</td>
<td style="text-align: center;">1.00</td>
<td style="text-align: center;">0.97</td>
<td style="text-align: center;">43</td>
<td style="text-align: center;">1</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Title</strong></td>
<td style="text-align: center;">38</td>
<td style="text-align: center;">2</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">16</td>
<td style="text-align: center;">0.12</td>
<td style="text-align: center;">0.95</td>
<td style="text-align: center;">1.00</td>
<td style="text-align: center;">0.97</td>
<td style="text-align: center;">7</td>
<td style="text-align: center;">0</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Split</strong></td>
<td style="text-align: center;">8</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">15</td>
<td style="text-align: center;">27</td>
<td style="text-align: center;">0.08</td>
<td style="text-align: center;">1.00</td>
<td style="text-align: center;">0.35</td>
<td style="text-align: center;">0.52</td>
<td style="text-align: center;">4</td>
<td style="text-align: center;">0</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Publishing details</strong></td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">32</td>
<td style="text-align: center;">0.01</td>
<td style="text-align: center;">1.00</td>
<td style="text-align: center;">0.50</td>
<td style="text-align: center;">0.67</td>
<td style="text-align: center;">0</td>
<td style="text-align: center;">0</td>
</tr>
<tr class="even">
<td style="text-align: center;"><strong>Total</strong></td>
<td style="text-align: center;">279</td>
<td style="text-align: center;">15</td>
<td style="text-align: center;">28</td>
<td style="text-align: center;">98</td>
<td style="text-align: center;">1.00</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><strong>97</strong></td>
<td style="text-align: center;"><strong>3</strong></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><strong>Corrected total</strong></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
<td style="text-align: center;"><strong>0.96</strong></td>
<td style="text-align: center;"><strong>0.89</strong></td>
<td style="text-align: center;"><strong>0.91</strong></td>
<td style="text-align: center;">-</td>
<td style="text-align: center;">-</td>
</tr>
</tbody>
</table>
</div>
<h3 class="page_break_before" id="si-3-wikidata-sparql-queries">SI 3 Wikidata SPARQL Queries</h3>
<h4 id="query-1---arabidopsis-thaliana">Query 1 - <em>Arabidopsis thaliana</em></h4>
This query answers to the following question:
<pre>
What are the compounds found in Mouse-ear cress (<i>Arabidopsis thaliana</i>)?
</pre>
<p>Link: <a href="https://w.wiki/3HMX">https://w.wiki/3HMX</a></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb1-1"><a href="#cb1-1"></a># What are <span class="kw">the</span> compounds found <span class="kw">in</span> Mouse<span class="op">-</span>ear cress (Arabidopsis thaliana)?</span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?<span class="kw">structure</span> ?structureLabel ?structure_inchikey <span class="kw">WHERE</span> {</span>
<span id="cb1-3"><a href="#cb1-3"></a>  ?<span class="kw">structure</span> wdt<span class="ch">:P235</span> ?structure_inchikey; # get <span class="kw">the</span> inchikey</span>
<span id="cb1-4"><a href="#cb1-4"></a>    p<span class="ch">:P703</span> ?statement.                     # statement found <span class="kw">in</span> taxon</span>
<span id="cb1-5"><a href="#cb1-5"></a>  ?statement ps<span class="ch">:P703</span> wd<span class="ch">:Q158695</span>;           # Wikidata <span class="kw">identifier</span> <span class="kw">of</span> your taxa <span class="kw">of</span> interest (here Arabidopsis thaliana). </span>
<span id="cb1-6"><a href="#cb1-6"></a>                                           # You can remove <span class="kw">the</span> Qxxxxxx <span class="kw">and</span> hit Ctrl<span class="op">+</span>space, <span class="kw">type</span> <span class="kw">the</span> <span class="fu">first</span> letters <span class="kw">and</span> it should autocomplete</span>
<span id="cb1-7"><a href="#cb1-7"></a>    prov<span class="ch">:wasDerivedFrom</span> ?<span class="fu">ref</span>.              # get results <span class="kw">with</span> a reference</span>
<span id="cb1-8"><a href="#cb1-8"></a>  ?<span class="fu">ref</span> pr<span class="ch">:P248</span> ?art.                       # get <span class="kw">the</span> reference <span class="kw">ID</span></span>
<span id="cb1-9"><a href="#cb1-9"></a>  ?art wdt<span class="ch">:P356</span> ?art_doi.                  # get <span class="kw">the</span> reference DOI</span>
<span id="cb1-10"><a href="#cb1-10"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb1-11"><a href="#cb1-11"></a>}</span></code></pre></div>
<h4 class="page_break_before" id="query-2---β-sitosterol">Query 2 - <em>β</em>-sitosterol</h4>
This query answers to the following question:
<pre>
Which organisms are known to contain compounds sharing the planar structure of <i>β</i>-sitosterol?
</pre>
<p>Link: <a href="https://w.wiki/3HLy">https://w.wiki/3HLy</a></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb2-1"><a href="#cb2-1"></a># Which organisms are known <span class="kw">to</span> contain β-sitosterol?</span>
<span id="cb2-2"><a href="#cb2-2"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?taxon ?taxonName <span class="kw">WHERE</span> {</span>
<span id="cb2-3"><a href="#cb2-3"></a>  {</span>
<span id="cb2-4"><a href="#cb2-4"></a>    wd<span class="ch">:Q121802</span> p<span class="ch">:P703</span> ?stmt.          # limits <span class="kw">the</span> search <span class="kw">to</span> β-sitosterol</span>
<span id="cb2-5"><a href="#cb2-5"></a>    ?stmt ps<span class="ch">:P703</span> ?taxon.             # found <span class="kw">in</span> taxon</span>
<span id="cb2-6"><a href="#cb2-6"></a>    {</span>
<span id="cb2-7"><a href="#cb2-7"></a>      ?stmt prov<span class="ch">:wasDerivedFrom</span> ?<span class="fu">ref</span>.</span>
<span id="cb2-8"><a href="#cb2-8"></a>      ?<span class="fu">ref</span> pr<span class="ch">:P248</span> ?art.              # stated <span class="kw">in</span> </span>
<span id="cb2-9"><a href="#cb2-9"></a>      ?art wdt<span class="ch">:P356</span> ?art_doi.         # DOI <span class="kw">of</span> <span class="kw">the</span> reference</span>
<span id="cb2-10"><a href="#cb2-10"></a>    }</span>
<span id="cb2-11"><a href="#cb2-11"></a>  }</span>
<span id="cb2-12"><a href="#cb2-12"></a>  ?taxon wdt<span class="ch">:P225</span> ?taxonName.         # scientific name <span class="kw">of</span> <span class="kw">the</span> taxon</span>
<span id="cb2-13"><a href="#cb2-13"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb2-14"><a href="#cb2-14"></a>}</span></code></pre></div>
<h4 class="page_break_before" id="query-3---β-sitosterol-stereoisomers">Query 3 - <em>β</em>-sitosterol stereoisomers</h4>
This query answers to the following question:
<pre>
Which organisms are known to contain stereoisomers of <i>β</i>-sitosterol?
</pre>
<p>Link: <a href="https://w.wiki/3Jgs">https://w.wiki/3Jgs</a></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb3-1"><a href="#cb3-1"></a># Which organisms are known <span class="kw">to</span> contain stereoisomers <span class="kw">of</span> β-sitosterol?</span>
<span id="cb3-2"><a href="#cb3-2"></a><span class="kw">SELECT</span> ?compound ?compoundLabel ?InChIKey ?taxonname </span>
<span id="cb3-3"><a href="#cb3-3"></a><span class="kw">WITH</span> {</span>
<span id="cb3-4"><a href="#cb3-4"></a>  <span class="kw">SELECT</span> ?compound ?InChIKey <span class="kw">WHERE</span> {</span>
<span id="cb3-5"><a href="#cb3-5"></a>    wd<span class="ch">:Q121802</span> wdt<span class="ch">:P235</span> ?queryKey .         # β-sitosterol</span>
<span id="cb3-6"><a href="#cb3-6"></a>    ?compound wdt<span class="ch">:P235</span> ?InChIKey .          # get <span class="kw">the</span> inchikey <span class="kw">of</span> β-sitosterol</span>
<span id="cb3-7"><a href="#cb3-7"></a>    <span class="kw">FILTER</span> (regex(str(?InChIKey), <span class="fu">concat</span>(<span class="ot">&quot;^&quot;</span>, <span class="fu">substr</span>($queryKey,<span class="dv">1</span>,<span class="dv">14</span>), <span class="ot">&quot;-&quot;</span>))) # results containing <span class="kw">the</span> <span class="fu">first</span> <span class="dv">14</span> <span class="dt">character</span> <span class="kw">of</span> <span class="kw">the</span> inchikey <span class="kw">of</span> β-sitosterol</span>
<span id="cb3-8"><a href="#cb3-8"></a>    <span class="kw">FILTER</span> ( ?InChIKey <span class="op">!=</span> ?queryKey )       # but <span class="kw">not</span> containing <span class="kw">the</span> inchikey <span class="kw">of</span> β-sitosterol</span>
<span id="cb3-9"><a href="#cb3-9"></a>  }</span>
<span id="cb3-10"><a href="#cb3-10"></a>} <span class="kw">AS</span> %compounds</span>
<span id="cb3-11"><a href="#cb3-11"></a><span class="kw">WHERE</span> {</span>
<span id="cb3-12"><a href="#cb3-12"></a>  INCLUDE %compounds</span>
<span id="cb3-13"><a href="#cb3-13"></a>  ?compound wdt<span class="ch">:P703</span><span class="op">/</span>wdt<span class="ch">:P225</span> ?taxonname .  # found <span class="kw">in</span> taxon <span class="op">/</span> taxon scientific name</span>
<span id="cb3-14"><a href="#cb3-14"></a>  ?compound rdfs<span class="ch">:label</span> ?compoundLabel.      # compound name</span>
<span id="cb3-15"><a href="#cb3-15"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb3-16"><a href="#cb3-16"></a>}</span>
<span id="cb3-17"><a href="#cb3-17"></a><span class="kw">ORDER</span> <span class="kw">BY</span> <span class="kw">ASC</span>(?InChIKey)                     # <span class="kw">sort</span> <span class="kw">by</span> inchikey</span></code></pre></div>
<h4 class="page_break_before" id="query-4---pigments">Query 4 - Pigments</h4>
This query answers to the following question:
<pre>
Which pigments are found in which taxa, according to which reference?
</pre>
<p>Link: <a href="https://w.wiki/3H3o">https://w.wiki/3H3o</a></p>
<div class="sourceCode" id="cb4"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb4-1"><a href="#cb4-1"></a># Which pigments are found <span class="kw">in</span> which taxa, according <span class="kw">to</span> which reference?</span>
<span id="cb4-2"><a href="#cb4-2"></a># special thanks goes <span class="kw">to</span> @biocolores (Layla Michán) <span class="cf">for</span> updating this information!</span>
<span id="cb4-3"><a href="#cb4-3"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?compound ?compoundLabel ?taxon ?taxonname ?DOI </span>
<span id="cb4-4"><a href="#cb4-4"></a><span class="kw">WITH</span> {</span>
<span id="cb4-5"><a href="#cb4-5"></a>  <span class="kw">SELECT</span> ?compound <span class="kw">WHERE</span> {</span>
<span id="cb4-6"><a href="#cb4-6"></a>    ?compound wdt<span class="ch">:P31</span><span class="op">*/</span>wdt<span class="ch">:P279</span><span class="op">*</span> wd<span class="ch">:Q161179</span>.  # get pigments</span>
<span id="cb4-7"><a href="#cb4-7"></a>  }</span>
<span id="cb4-8"><a href="#cb4-8"></a>} <span class="kw">AS</span> %compounds</span>
<span id="cb4-9"><a href="#cb4-9"></a><span class="kw">WITH</span> {</span>
<span id="cb4-10"><a href="#cb4-10"></a>  <span class="kw">SELECT</span> ?compound ?P703statement <span class="kw">WHERE</span> {</span>
<span id="cb4-11"><a href="#cb4-11"></a>    INCLUDE %compounds</span>
<span id="cb4-12"><a href="#cb4-12"></a>            ?compound p<span class="ch">:P703</span> ?P703statement.  # <span class="kw">check</span> <span class="cf">for</span> <span class="ot">&quot;found in taxon&quot;</span> statements</span>
<span id="cb4-13"><a href="#cb4-13"></a>  }</span>
<span id="cb4-14"><a href="#cb4-14"></a>} <span class="kw">AS</span> %P703statement</span>
<span id="cb4-15"><a href="#cb4-15"></a><span class="kw">WITH</span> {</span>
<span id="cb4-16"><a href="#cb4-16"></a>  <span class="kw">SELECT</span> ?compound ?taxon ?DOI <span class="kw">WHERE</span> {</span>
<span id="cb4-17"><a href="#cb4-17"></a>    INCLUDE %P703statement</span>
<span id="cb4-18"><a href="#cb4-18"></a>            ?P703statement ps<span class="ch">:P703</span> ?taxon ;   # get <span class="kw">the</span> respective taxa</span>
<span id="cb4-19"><a href="#cb4-19"></a>            prov<span class="ch">:wasDerivedFrom</span> <span class="op">/</span> pr<span class="ch">:P248</span> [   # get <span class="kw">the</span> reference supporting that statement</span>
<span id="cb4-20"><a href="#cb4-20"></a>              wdt<span class="ch">:P356</span> ?DOI                   # get <span class="kw">the</span> DOI <span class="cf">for</span> <span class="kw">the</span> reference</span>
<span id="cb4-21"><a href="#cb4-21"></a>            ] .</span>
<span id="cb4-22"><a href="#cb4-22"></a>  }</span>
<span id="cb4-23"><a href="#cb4-23"></a>} <span class="kw">AS</span> %taxa</span>
<span id="cb4-24"><a href="#cb4-24"></a><span class="kw">WHERE</span> {</span>
<span id="cb4-25"><a href="#cb4-25"></a>  {</span>
<span id="cb4-26"><a href="#cb4-26"></a>    INCLUDE %taxa</span>
<span id="cb4-27"><a href="#cb4-27"></a></span>
<span id="cb4-28"><a href="#cb4-28"></a>            ?taxon wdt<span class="ch">:P225</span> ?taxonname .      # get <span class="kw">the</span> taxon name</span>
<span id="cb4-29"><a href="#cb4-29"></a>  }</span>
<span id="cb4-30"><a href="#cb4-30"></a>  ?compound rdfs<span class="ch">:label</span> ?compoundLabel .       # get compound labels</span>
<span id="cb4-31"><a href="#cb4-31"></a>  <span class="kw">FILTER</span> (LANG(?compoundLabel) <span class="op">=</span> <span class="ot">&quot;en&quot;</span>) .      # <span class="kw">filter</span> <span class="cf">for</span> English</span>
<span id="cb4-32"><a href="#cb4-32"></a>}</span>
<span id="cb4-33"><a href="#cb4-33"></a><span class="kw">ORDER</span> <span class="kw">BY</span> <span class="kw">ASC</span>(?compoundLabel)</span>
<span id="cb4-34"><a href="#cb4-34"></a><span class="kw">LIMIT</span> <span class="dv">10000</span></span></code></pre></div>
<h4 class="page_break_before" id="query-5---sister-taxon-compounds">Query 5 - Sister taxon compounds</h4>
This query answers to the following question:
<pre>
What are examples of organisms where compounds were found in an organism sharing the same parent taxon, but not the organism itself?
</pre>
<p>Link: <a href="https://w.wiki/3HM6">https://w.wiki/3HM6</a></p>
<div class="sourceCode" id="cb5"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb5-1"><a href="#cb5-1"></a># What are examples <span class="kw">of</span> organisms <span class="kw">where</span> compounds were found <span class="kw">in</span> an organism sharing <span class="kw">the</span> same <span class="kw">parent</span> taxon, but <span class="kw">not</span> <span class="kw">the</span> organism itself? </span>
<span id="cb5-2"><a href="#cb5-2"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?compound ?compoundLabel ?taxonname_with_compound ?taxonname_without_compound ?parent_taxon <span class="kw">WITH</span>{ </span>
<span id="cb5-3"><a href="#cb5-3"></a>  <span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?compound ?taxon_with_compound ?parent_taxon </span>
<span id="cb5-4"><a href="#cb5-4"></a>  <span class="kw">WHERE</span> {</span>
<span id="cb5-5"><a href="#cb5-5"></a>    ?compound wdt<span class="ch">:P235</span> ?inchikey.</span>
<span id="cb5-6"><a href="#cb5-6"></a>    SERVICE bd<span class="ch">:sample</span> { ?compound wdt<span class="ch">:P703</span> ?taxon_with_compound . bd<span class="ch">:serviceParam</span> bd<span class="ch">:sample</span>.<span class="kw">limit</span> <span class="dv">1000</span> }   </span>
<span id="cb5-7"><a href="#cb5-7"></a>    ?taxon_with_compound wdt<span class="ch">:P171</span> ?parent_taxon .</span>
<span id="cb5-8"><a href="#cb5-8"></a>  }</span>
<span id="cb5-9"><a href="#cb5-9"></a>} <span class="kw">AS</span> %taxon_with_compound </span>
<span id="cb5-10"><a href="#cb5-10"></a><span class="kw">WITH</span></span>
<span id="cb5-11"><a href="#cb5-11"></a>{ </span>
<span id="cb5-12"><a href="#cb5-12"></a>  <span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?taxon_without_compound ?parent_taxon ?compound </span>
<span id="cb5-13"><a href="#cb5-13"></a>  <span class="kw">WHERE</span> {</span>
<span id="cb5-14"><a href="#cb5-14"></a>    INCLUDE %taxon_with_compound</span>
<span id="cb5-15"><a href="#cb5-15"></a>    ?taxon_without_compound wdt<span class="ch">:P171</span> ?parent_taxon .</span>
<span id="cb5-16"><a href="#cb5-16"></a>    <span class="kw">FILTER</span> (?taxon_with_compound <span class="op">!=</span> ?taxon_without_compound)</span>
<span id="cb5-17"><a href="#cb5-17"></a>  }</span>
<span id="cb5-18"><a href="#cb5-18"></a>} <span class="kw">AS</span> %taxon2 </span>
<span id="cb5-19"><a href="#cb5-19"></a><span class="kw">WHERE</span> {</span>
<span id="cb5-20"><a href="#cb5-20"></a>  INCLUDE %taxon_with_compound</span>
<span id="cb5-21"><a href="#cb5-21"></a>  INCLUDE %taxon2</span>
<span id="cb5-22"><a href="#cb5-22"></a>  <span class="kw">FILTER</span> <span class="kw">NOT</span> <span class="kw">EXISTS</span> {  ?compound wdt<span class="ch">:P703</span> ?taxon_without_compound .}</span>
<span id="cb5-23"><a href="#cb5-23"></a>  ?taxon_with_compound wdt<span class="ch">:P225</span> ?taxonname_with_compound .</span>
<span id="cb5-24"><a href="#cb5-24"></a>  ?taxon_without_compound wdt<span class="ch">:P225</span> ?taxonname_without_compound .</span>
<span id="cb5-25"><a href="#cb5-25"></a>  ?compound rdfs<span class="ch">:label</span> ?compoundLabel.</span>
<span id="cb5-26"><a href="#cb5-26"></a>  <span class="kw">FILTER</span>(LANG(?compoundLabel) <span class="op">=</span> <span class="ot">&quot;en&quot;</span>).</span>
<span id="cb5-27"><a href="#cb5-27"></a>}</span></code></pre></div>
<h4 class="page_break_before" id="query-6---zephyranthes-sister-taxon-compounds">Query 6 - <em>Zephyranthes</em> sister taxon compounds</h4>
<p>This query answers to the following question:</p>
<pre>
Which <i>Zephyranthes</i> species lack compounds known from at least two species in the genus?
</pre>
<p>Link: <a href="https://w.wiki/3HJf">https://w.wiki/3HJf</a></p>
<div class="sourceCode" id="cb6"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb6-1"><a href="#cb6-1"></a># Which Zephyranthes species lack compounds known <span class="kw">from</span> <span class="kw">at</span> <span class="fu">least</span> two species <span class="kw">in</span> <span class="kw">the</span> genus? </span>
<span id="cb6-2"><a href="#cb6-2"></a>PREFIX target: <span class="op">&lt;</span>http:<span class="op">//</span>www.wikidata.org<span class="op">/</span>entity<span class="op">/</span>Q191364<span class="op">&gt;</span> # Zephyranthes</span>
<span id="cb6-3"><a href="#cb6-3"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?compound ?compoundLabel ?taxon_with_compound ?another_taxon_with_compound ?taxon_without_compound</span>
<span id="cb6-4"><a href="#cb6-4"></a><span class="kw">WITH</span></span>
<span id="cb6-5"><a href="#cb6-5"></a>{ </span>
<span id="cb6-6"><a href="#cb6-6"></a>  <span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?compound ?taxon_YES_1 ?taxon_YES_2 </span>
<span id="cb6-7"><a href="#cb6-7"></a>  <span class="kw">WHERE</span> {</span>
<span id="cb6-8"><a href="#cb6-8"></a>    ?compound wdt<span class="ch">:P703</span> ?taxon_YES_1 .</span>
<span id="cb6-9"><a href="#cb6-9"></a>    ?compound wdt<span class="ch">:P703</span> ?taxon_YES_2 .</span>
<span id="cb6-10"><a href="#cb6-10"></a>    ?taxon_YES_1 wdt<span class="ch">:P171</span> target: .</span>
<span id="cb6-11"><a href="#cb6-11"></a>    ?taxon_YES_2 wdt<span class="ch">:P171</span> target: .</span>
<span id="cb6-12"><a href="#cb6-12"></a>    <span class="kw">FILTER</span> (?taxon_YES_2 <span class="op">!=</span> ?taxon_YES_1)</span>
<span id="cb6-13"><a href="#cb6-13"></a>  }</span>
<span id="cb6-14"><a href="#cb6-14"></a>} <span class="kw">AS</span> %taxa_with_compound </span>
<span id="cb6-15"><a href="#cb6-15"></a><span class="kw">WITH</span></span>
<span id="cb6-16"><a href="#cb6-16"></a>{ </span>
<span id="cb6-17"><a href="#cb6-17"></a>  <span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?taxon_NO ?compound </span>
<span id="cb6-18"><a href="#cb6-18"></a>  <span class="kw">WHERE</span> {</span>
<span id="cb6-19"><a href="#cb6-19"></a>    INCLUDE %taxa_with_compound</span>
<span id="cb6-20"><a href="#cb6-20"></a>    ?taxon_NO wdt<span class="ch">:P171</span> target: .</span>
<span id="cb6-21"><a href="#cb6-21"></a>    <span class="kw">FILTER</span> (?taxon_YES_1 <span class="op">!=</span> ?taxon_NO)</span>
<span id="cb6-22"><a href="#cb6-22"></a>  }</span>
<span id="cb6-23"><a href="#cb6-23"></a>} <span class="kw">AS</span> %taxon_without_compond </span>
<span id="cb6-24"><a href="#cb6-24"></a><span class="kw">WHERE</span> {</span>
<span id="cb6-25"><a href="#cb6-25"></a>  INCLUDE %taxa_with_compound</span>
<span id="cb6-26"><a href="#cb6-26"></a>  INCLUDE %taxon_without_compond</span>
<span id="cb6-27"><a href="#cb6-27"></a>  <span class="kw">FILTER</span> <span class="kw">NOT</span> <span class="kw">EXISTS</span> { ?compound wdt<span class="ch">:P703</span> ?taxon_NO .}</span>
<span id="cb6-28"><a href="#cb6-28"></a>  <span class="kw">VALUES</span> ?classes {</span>
<span id="cb6-29"><a href="#cb6-29"></a>    wd<span class="ch">:Q11173</span></span>
<span id="cb6-30"><a href="#cb6-30"></a>    wd<span class="ch">:Q59199015</span></span>
<span id="cb6-31"><a href="#cb6-31"></a>  }</span>
<span id="cb6-32"><a href="#cb6-32"></a>  ?taxon_YES_1 wdt<span class="ch">:P225</span> ?taxon_with_compound .</span>
<span id="cb6-33"><a href="#cb6-33"></a>  ?taxon_YES_2 wdt<span class="ch">:P225</span> ?another_taxon_with_compound .</span>
<span id="cb6-34"><a href="#cb6-34"></a>  ?taxon_NO wdt<span class="ch">:P225</span> ?taxon_without_compound .</span>
<span id="cb6-35"><a href="#cb6-35"></a>  ?compound wdt<span class="ch">:P31</span><span class="op">*/</span>wdt<span class="ch">:P279</span><span class="op">*</span> ?classes .</span>
<span id="cb6-36"><a href="#cb6-36"></a>  ?compound rdfs<span class="ch">:label</span> ?compoundLabel.</span>
<span id="cb6-37"><a href="#cb6-37"></a>  <span class="kw">FILTER</span>(LANG(?compoundLabel) <span class="op">=</span> <span class="ot">&quot;en&quot;</span>).</span>
<span id="cb6-38"><a href="#cb6-38"></a>}</span></code></pre></div>
<h4 class="page_break_before" id="query-7---antibiotic-like-compounds">Query 7 - Antibiotic-like compounds</h4>
<p>This query answers to the following question:</p>
<pre>
How many compounds are structurally similar to compounds labeled as antibiotics? Results are grouped by the parent taxon of the organism they were found in.
</pre>
<p>Link: <a href="https://w.wiki/3HMA">https://w.wiki/3HMA</a></p>
<div class="sourceCode" id="cb7"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb7-1"><a href="#cb7-1"></a># How many compounds are structurally similar <span class="kw">to</span> compounds labeled <span class="kw">as</span> antibiotics? </span>
<span id="cb7-2"><a href="#cb7-2"></a># Results are grouped <span class="kw">by</span> <span class="kw">the</span> <span class="kw">parent</span> taxon <span class="kw">of</span> <span class="kw">the</span> organism they were found <span class="kw">in</span>.</span>
<span id="cb7-3"><a href="#cb7-3"></a>PREFIX sachem: <span class="op">&lt;</span>http:<span class="op">//</span>bioinfo.uochb.cas.cz<span class="op">/</span>rdf<span class="op">/</span>v1<span class="fl">.0</span><span class="op">/</span>sachem#<span class="op">&gt;</span> # prefixes <span class="kw">needed</span> <span class="cf">for</span> structural similarity search</span>
<span id="cb7-4"><a href="#cb7-4"></a>PREFIX idsm: <span class="op">&lt;</span>https:<span class="op">//</span>idsm.elixir<span class="op">-</span>czech.cz<span class="op">/</span>sparql<span class="op">/</span>endpoint<span class="op">/&gt;</span></span>
<span id="cb7-5"><a href="#cb7-5"></a><span class="kw">SELECT</span> ?parent_taxon ?parent_taxon_name (<span class="fu">COUNT</span>(<span class="kw">DISTINCT</span> ?compound) <span class="kw">AS</span> ?<span class="fu">count</span>) <span class="kw">WHERE</span> {</span>
<span id="cb7-6"><a href="#cb7-6"></a>  SERVICE idsm<span class="ch">:wikidata</span> {</span>
<span id="cb7-7"><a href="#cb7-7"></a>    SERVICE <span class="op">&lt;</span>https:<span class="op">//</span><span class="kw">query</span>.wikidata.org<span class="op">/</span>bigdata<span class="op">/</span>namespace<span class="op">/</span>wdq<span class="op">/</span>sparql<span class="op">&gt;</span> {</span>
<span id="cb7-8"><a href="#cb7-8"></a>      ?antibiotic ((wdt<span class="ch">:P279</span><span class="op">*</span>)<span class="op">/</span>wdt<span class="ch">:P2868</span><span class="op">/</span>wdt<span class="ch">:P486</span>) <span class="ot">&quot;D000900&quot;</span>; # get antibiotics</span>
<span id="cb7-9"><a href="#cb7-9"></a>        wdt<span class="ch">:P233</span> ?smiles.                                     # get SMILES</span>
<span id="cb7-10"><a href="#cb7-10"></a>    }</span>
<span id="cb7-11"><a href="#cb7-11"></a>    ?compound sachem<span class="ch">:similarCompoundSearch</span> _<span class="ch">:b40</span>.</span>
<span id="cb7-12"><a href="#cb7-12"></a>    _<span class="ch">:b40</span> sachem<span class="ch">:query</span> ?smiles;</span>
<span id="cb7-13"><a href="#cb7-13"></a>      sachem<span class="ch">:cutoff</span> <span class="ot">&quot;0.9&quot;</span>^^xsd<span class="ch">:double</span>.                        # similarity cut<span class="op">-</span><span class="kw">off</span> <span class="kw">at</span> <span class="fl">0.9</span> similarity</span>
<span id="cb7-14"><a href="#cb7-14"></a>  }</span>
<span id="cb7-15"><a href="#cb7-15"></a>  hint<span class="ch">:Prior</span> hint<span class="ch">:runFirst</span> <span class="ot">&quot;true&quot;</span>^^xsd<span class="ch">:boolean</span>.</span>
<span id="cb7-16"><a href="#cb7-16"></a>  ?compound wdt<span class="ch">:P703</span> ?taxon.                                  # found <span class="kw">in</span> taxon</span>
<span id="cb7-17"><a href="#cb7-17"></a>  ?taxon wdt<span class="ch">:P171</span> ?parent_taxon.                              # get <span class="kw">the</span> <span class="kw">parent</span> taxon </span>
<span id="cb7-18"><a href="#cb7-18"></a>  OPTIONAL { ?parent_taxon wdt<span class="ch">:P225</span> ?parent_taxon_name. }</span>
<span id="cb7-19"><a href="#cb7-19"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb7-20"><a href="#cb7-20"></a>}</span>
<span id="cb7-21"><a href="#cb7-21"></a><span class="kw">GROUP</span> <span class="kw">BY</span> ?parent_taxon ?parent_taxon_name                     # <span class="fu">count</span> per <span class="kw">parent</span> taxon </span>
<span id="cb7-22"><a href="#cb7-22"></a><span class="kw">ORDER</span> <span class="kw">BY</span> <span class="kw">DESC</span> (?<span class="fu">count</span>)</span></code></pre></div>
<h4 class="page_break_before" id="query-8---triples">Query 8 - Triples</h4>
This query answers to the following question:
<pre>
Which are the available referenced structure-organism pairs? (example limited to 1000 results)
</pre>
<p>Link: <a href="https://w.wiki/3JpE">https://w.wiki/3JpE</a></p>
<div class="sourceCode" id="cb8"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb8-1"><a href="#cb8-1"></a><span class="kw">SELECT</span> <span class="kw">DISTINCT</span> ?<span class="kw">structure</span> ?structure_inchikey ?taxon ?taxon_name ?reference ?reference_doi <span class="kw">WHERE</span> {</span>
<span id="cb8-2"><a href="#cb8-2"></a>  ?<span class="kw">structure</span> wdt<span class="ch">:P235</span> ?structure_inchikey;       # get <span class="kw">the</span> inchikey</span>
<span id="cb8-3"><a href="#cb8-3"></a>    p<span class="ch">:P703</span>[                                      # statement found <span class="kw">in</span> taxon</span>
<span id="cb8-4"><a href="#cb8-4"></a>     ps<span class="ch">:P703</span> ?taxon;                             # get <span class="kw">the</span> taxon</span>
<span id="cb8-5"><a href="#cb8-5"></a>     (prov<span class="ch">:wasDerivedFrom</span><span class="op">/</span>pr<span class="ch">:P248</span>) ?reference ]. # get <span class="kw">the</span> reference</span>
<span id="cb8-6"><a href="#cb8-6"></a>  ?taxon wdt<span class="ch">:P225</span> ?taxon_name.                   # get <span class="kw">the</span> taxon scientific name</span>
<span id="cb8-7"><a href="#cb8-7"></a>  ?reference wdt<span class="ch">:P356</span> ?reference_doi.            # get <span class="kw">the</span> reference DOI</span>
<span id="cb8-8"><a href="#cb8-8"></a>}</span>
<span id="cb8-9"><a href="#cb8-9"></a><span class="kw">LIMIT</span> <span class="dv">1000</span></span></code></pre></div>
<h4 class="page_break_before" id="query-9---indolic-scaffold">Query 9 - Indolic scaffold</h4>
This query answers to the following question:
<pre>
Which organisms contain indolic scaffolds? Count occurences, group and order the results by the parent taxon.
</pre>
<p>Link: <a href="https://w.wiki/3HMD">https://w.wiki/3HMD</a></p>
<div class="sourceCode" id="cb9"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb9-1"><a href="#cb9-1"></a># Which organisms contain indolic scaffold? <span class="kw">Group</span> <span class="kw">and</span> <span class="kw">order</span> <span class="kw">the</span> results <span class="kw">by</span> <span class="kw">the</span> <span class="kw">parent</span> taxon.</span>
<span id="cb9-2"><a href="#cb9-2"></a>PREFIX sachem: <span class="op">&lt;</span>http:<span class="op">//</span>bioinfo.uochb.cas.cz<span class="op">/</span>rdf<span class="op">/</span>v1<span class="fl">.0</span><span class="op">/</span>sachem#<span class="op">&gt;</span> # prefixes <span class="kw">needed</span> <span class="cf">for</span> structural similarity search</span>
<span id="cb9-3"><a href="#cb9-3"></a>PREFIX wd: <span class="op">&lt;</span>http:<span class="op">//</span>www.wikidata.org<span class="op">/</span>entity<span class="op">/&gt;</span></span>
<span id="cb9-4"><a href="#cb9-4"></a>PREFIX p: <span class="op">&lt;</span>http:<span class="op">//</span>www.wikidata.org<span class="op">/</span>prop<span class="op">/&gt;</span></span>
<span id="cb9-5"><a href="#cb9-5"></a>PREFIX idsm: <span class="op">&lt;</span>https:<span class="op">//</span>idsm.elixir<span class="op">-</span>czech.cz<span class="op">/</span>sparql<span class="op">/</span>endpoint<span class="op">/&gt;</span></span>
<span id="cb9-6"><a href="#cb9-6"></a></span>
<span id="cb9-7"><a href="#cb9-7"></a><span class="kw">SELECT</span> ?parent_taxon ?parent_taxon_name (<span class="fu">COUNT</span>(<span class="kw">DISTINCT</span> ?compound) <span class="kw">AS</span> ?<span class="fu">count</span>)  <span class="kw">WHERE</span> {</span>
<span id="cb9-8"><a href="#cb9-8"></a>  SERVICE idsm<span class="ch">:wikidata</span> {</span>
<span id="cb9-9"><a href="#cb9-9"></a>    ?compound sachem<span class="ch">:substructureSearch</span></span>
<span id="cb9-10"><a href="#cb9-10"></a>              [ sachem<span class="ch">:query</span> <span class="ot">&quot;NCCC1=CNC2=C1C=CC=C2&quot;</span> ] # indolic scaffold</span>
<span id="cb9-11"><a href="#cb9-11"></a>  }</span>
<span id="cb9-12"><a href="#cb9-12"></a>  hint<span class="ch">:Prior</span> hint<span class="ch">:runFirst</span> <span class="kw">true</span>.                      # hint <span class="kw">to</span> evaluate <span class="kw">the</span> idsm service <span class="fu">first</span></span>
<span id="cb9-13"><a href="#cb9-13"></a>  ?compound p<span class="ch">:P703</span> ?statement;                        # found <span class="kw">in</span> taxon</span>
<span id="cb9-14"><a href="#cb9-14"></a>            wdt<span class="ch">:P235</span> ?inchikey.                       # get <span class="kw">the</span> inchikey</span>
<span id="cb9-15"><a href="#cb9-15"></a>  ?statement ps<span class="ch">:P703</span> ?taxon.                          # get <span class="kw">the</span> taxon</span>
<span id="cb9-16"><a href="#cb9-16"></a>  ?taxon wdt<span class="ch">:P171</span> ?parent_taxon.                      # get <span class="kw">the</span> <span class="kw">parent</span> taxon</span>
<span id="cb9-17"><a href="#cb9-17"></a>  ?parent_taxon wdt<span class="ch">:P225</span> ?parent_taxon_name.          # get <span class="kw">the</span> <span class="kw">parent</span> taxon name</span>
<span id="cb9-18"><a href="#cb9-18"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb9-19"><a href="#cb9-19"></a>}</span>
<span id="cb9-20"><a href="#cb9-20"></a><span class="kw">GROUP</span> <span class="kw">BY</span> ?parent_taxon ?parent_taxon_name</span>
<span id="cb9-21"><a href="#cb9-21"></a><span class="kw">ORDER</span> <span class="kw">BY</span> <span class="kw">DESC</span> (?<span class="fu">count</span>)</span></code></pre></div>
<h4 class="page_break_before" id="query-10---senior-np-chemists">Query 10 - Senior NP chemists</h4>
This query answers to the following question:
<pre>
How many structure-organism pairs have been referenced by certain authors? (Here, two senior natural products chemists and co-authors of this paper are compared to the late Ferdinand Bohlmann).
</pre>
<p>Link: <a href="https://w.wiki/3HML">https://w.wiki/3HML</a></p>
<div class="sourceCode" id="cb10"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb10-1"><a href="#cb10-1"></a># How many <span class="kw">structure</span><span class="op">-</span>organism pairs have been referenced <span class="kw">by</span> certain authors? </span>
<span id="cb10-2"><a href="#cb10-2"></a># Here, two senior <span class="kw">natural</span> products chemists are compared <span class="kw">to</span> <span class="kw">the</span> late Ferdinand Bohlmann</span>
<span id="cb10-3"><a href="#cb10-3"></a>#defaultView<span class="ch">:BarChart</span></span>
<span id="cb10-4"><a href="#cb10-4"></a><span class="kw">SELECT</span> ?authors_namesLabel (<span class="fu">COUNT</span>(<span class="kw">DISTINCT</span>(?compound)) <span class="kw">AS</span> ?<span class="fu">count</span>) <span class="kw">WHERE</span> {</span>
<span id="cb10-5"><a href="#cb10-5"></a>  ?compound p<span class="ch">:P703</span> ?stmt.           # statement found <span class="kw">in</span> taxon</span>
<span id="cb10-6"><a href="#cb10-6"></a>    ?stmt prov<span class="ch">:wasDerivedFrom</span> ?<span class="fu">ref</span>. # get <span class="kw">the</span> reference</span>
<span id="cb10-7"><a href="#cb10-7"></a>    ?<span class="fu">ref</span> pr<span class="ch">:P248</span> ?art.              # stated <span class="kw">in</span></span>
<span id="cb10-8"><a href="#cb10-8"></a>  <span class="kw">VALUES</span> ?authors_names {</span>
<span id="cb10-9"><a href="#cb10-9"></a>    wd<span class="ch">:Q56084663</span>                    # JLW</span>
<span id="cb10-10"><a href="#cb10-10"></a>    wd<span class="ch">:Q40259636</span>                    # GFP</span>
<span id="cb10-11"><a href="#cb10-11"></a>    wd<span class="ch">:Q1405133</span>                     # A german chemist <span class="kw">of</span> <span class="kw">the</span> 20th century <span class="op">..</span>. Ferdinand Bohlmann</span>
<span id="cb10-12"><a href="#cb10-12"></a>  }</span>
<span id="cb10-13"><a href="#cb10-13"></a>  ?art wdt<span class="ch">:P50</span> ?authors_names.      # <span class="kw">limit</span> <span class="kw">to</span> <span class="kw">references</span> containing <span class="kw">the</span> author names</span>
<span id="cb10-14"><a href="#cb10-14"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb10-15"><a href="#cb10-15"></a>}</span>
<span id="cb10-16"><a href="#cb10-16"></a><span class="kw">GROUP</span> <span class="kw">BY</span> ?authors_namesLabel</span>
<span id="cb10-17"><a href="#cb10-17"></a><span class="kw">ORDER</span> <span class="kw">BY</span> <span class="kw">DESC</span> (?<span class="fu">count</span>)</span></code></pre></div>
<h3 class="page_break_before" id="si-4-wikidata-entry-creation-tutorial">SI 4 Wikidata Entry Creation Tutorial</h3>
<p><span class="banner lightblue"><i class="fas fa-scroll fa-lg"></i> <strong>Tutorial for manual creation</strong><br>
available at <a href="https://osf.io/7dk8h/">https://osf.io/7dk8h/</a> and <a href="https://oolonek.github.io/dendron/notes/235ba226-b0da-4c23-bbb7-c46c4a65d2f1.html">https://oolonek.github.io/dendron/notes/235ba226-b0da-4c23-bbb7-c46c4a65d2f1.html</a></span></p>
<h4 id="manual-addition-of-a-documented-structure-organism-pair-to-wikidata">Manual addition of a documented structure-organism pair to Wikidata</h4>
<h5 id="select-a-documented-structure-organism-pair">Select a documented structure-organism pair</h5>
<p>Throughout this demonstration, we are going to use the following example:
&gt; <a href="https://pubchem.ncbi.nlm.nih.gov/compound/101556657">trigocherrin A</a> is found in <a href="http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&amp;commit=Search"><em>Trigonostemon cherrieri</em></a>, as stated in <a href="https://doi.org/10.1021/OL2030907">Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri</a>.</p>
<h5 id="fetch-the-information-for-the-documented-structure-organism-pair">Fetch the information for the documented structure-organism pair</h5>
<h6 id="structure">Structure</h6>
<p>Search PubChem for your compound, here <a href="https://pubchem.ncbi.nlm.nih.gov/#query=%22trigocherrin%20A%22">trigocherrin A</a>. This leads to <a href="https://pubchem.ncbi.nlm.nih.gov/compound/101556657">https://pubchem.ncbi.nlm.nih.gov/compound/101556657</a>.</p>
<figure>
<img src="images/tutorial/2021-02-25-14-15-54.png" alt="" /><figcaption>tutorial-image-01</figcaption>
</figure>
<p>From there, you can fetch the compound’s name, InChIKey and InChI as well as its Canonical and Isomeric SMILES.
Here we keep, respectively:</p>
<pre><code>* trigocherrin A
* QOVGHDRCAGYGEB-FFZYJECLSA-N
* InChI=1S/C38H36Cl2O12/c1-18(2)35(44)27-19(3)37-25-16-24(31(39)40)28(48-32(43)22-12-8-6-9-13-22)36(25,45)33(47-21(5)42)34(17-46-20(4)41)29(49-34)26(37)30(35)51-38(50-27,52-37)23-14-10-7-11-15-23/h6-16,19,26-30,33,44-45H,1,17H2,2-5H3/t19-,26+,27?,28+,29+,30-,33-,34+,35+,36-,37+,38?/m1/s1
* CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O
* C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O</code></pre>
<h6 id="organism">Organism</h6>
<p>You can check if your organism name is correctly spelled using the Global Names resolver service: <a href="http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&amp;commit=Search">http://gni.globalnames.org/name_strings?search_term=trigonostemon+cherrieri&amp;commit=Search</a>.</p>
<figure>
<img src="images/tutorial/2021-02-27-18-40-20.png" alt="" /><figcaption>tutorial-image-02</figcaption>
</figure>
<p>Alternatively, you can use <a href="https://github.com/gnames/gnfinder">gnfinder</a> in your command line interface to check for the spelling of your organism string.</p>
<pre><code>echo &quot;Trigonostemion cherrieri&quot; | gnfinder find -c -l eng

{
  &quot;metadata&quot;: {
    &quot;date&quot;: &quot;2021-02-27T18:44:41.640982+01:00&quot;,
    &quot;gnfinderVersion&quot;: &quot;v0.11.1&quot;,
    &quot;withBayes&quot;: true,
    &quot;tokensAround&quot;: 0,
    &quot;language&quot;: &quot;eng&quot;,
    &quot;detectLanguage&quot;: false,
    &quot;totalWords&quot;: 2,
    &quot;totalCandidates&quot;: 1,
    &quot;totalNames&quot;: 1
  },
  &quot;names&quot;: [
    {
      &quot;cardinality&quot;: 2,
      &quot;verbatim&quot;: &quot;Trigonostemion cherrieri&quot;,
      &quot;name&quot;: &quot;Trigonostemion cherrieri&quot;,
      &quot;odds&quot;: 77581.46698350731,
      &quot;start&quot;: 0,
      &quot;end&quot;: 24,
      &quot;annotationNomenType&quot;: &quot;NO_ANNOT&quot;,
      &quot;annotation&quot;: &quot;&quot;,
      &quot;verification&quot;: {
        &quot;bestResult&quot;: {
          &quot;dataSourceId&quot;: 1,
          &quot;dataSourceTitle&quot;: &quot;Catalogue of Life&quot;,
          &quot;taxonId&quot;: &quot;1575885&quot;,
          &quot;matchedName&quot;: &quot;Trigonostemon cherrieri Veillon&quot;,
          &quot;matchedCardinality&quot;: 2,
          &quot;matchedCanonicalSimple&quot;: &quot;Trigonostemon cherrieri&quot;,
          &quot;matchedCanonicalFull&quot;: &quot;Trigonostemon cherrieri&quot;,
          &quot;classificationPath&quot;: &quot;Plantae|Tracheophyta|Magnoliopsida|Malpighiales|Euphorbiaceae|Trigonostemon|Trigonostemon cherrieri&quot;,
          &quot;classificationRank&quot;: &quot;kingdom|phylum|class|order|family|genus|species&quot;,
          &quot;classificationIds&quot;: &quot;3939764|3942634|3942724|3942777|3942795|4210752|1575885&quot;,
          &quot;editDistance&quot;: 1,
          &quot;stemEditDistance&quot;: 1,
          &quot;matchType&quot;: &quot;FuzzyCanonicalMatch&quot;
        },
        &quot;dataSourcesNum&quot;: 13,
        &quot;dataSourceQuality&quot;: &quot;HasCuratedSources&quot;,
        &quot;retries&quot;: 1
      }
    }
  ]
}</code></pre>
<p>For misspellings like <em>Trigonstemion cherrieri</em>, gnfinder can help resolve them, in this case to <em>Trigonostemon cherrieri</em>.</p>
<h6 id="reference">Reference</h6>
<p>Make sure that you have the correct <a href="https://www.doi.org/doi_handbook/Glossary.html#doi">Digital Object Identifier (DOI)</a> for it.
For “<a href="https://doi.org/10.1021/OL2030907">Trigocherrin A, the first natural chlorinated daphnane diterpene orthoester from Trigonostemon cherrieri</a>”, this is <strong>10.1021/ol2030907</strong>. Note that DOIs are uppercase-normalized in Wikidata.</p>
<h5 id="check-for-the-presence-of-your-compound-in-wikidata">Check for the presence of your compound in Wikidata <a name="check_compound"></a></h5>
<p>Using the compound’s InChIKey (i.e. <code>QOVGHDRCAGYGEB-FFZYJECLSA-N</code> for trigocherrin A), run a SPARQL query to check if your compound is present in Wikidata or not:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb13-1"><a href="#cb13-1"></a><span class="kw">SELECT</span> ?item ?itemLabel <span class="kw">WHERE</span> {</span>
<span id="cb13-2"><a href="#cb13-2"></a>  <span class="kw">VALUES</span> ?classes {</span>
<span id="cb13-3"><a href="#cb13-3"></a>    wd<span class="ch">:Q11173</span> # chemical compound</span>
<span id="cb13-4"><a href="#cb13-4"></a>    wd<span class="ch">:Q59199015</span> # <span class="kw">group</span> <span class="kw">of</span> stereoisomers</span>
<span id="cb13-5"><a href="#cb13-5"></a>    wd<span class="ch">:Q79529</span> # chemical substance</span>
<span id="cb13-6"><a href="#cb13-6"></a>    wd<span class="ch">:Q17339814</span> # <span class="kw">group</span> <span class="kw">of</span> chemical substances</span>
<span id="cb13-7"><a href="#cb13-7"></a>    wd<span class="ch">:Q47154513</span> # structural <span class="kw">class</span> <span class="kw">of</span> chemical compounds</span>
<span id="cb13-8"><a href="#cb13-8"></a>  }</span>
<span id="cb13-9"><a href="#cb13-9"></a>  ?item wdt<span class="ch">:P31</span> ?classes. # <span class="kw">instance</span> <span class="kw">of</span></span>
<span id="cb13-10"><a href="#cb13-10"></a>  ?item wdt<span class="ch">:P235</span> <span class="st">&#39;QOVGHDRCAGYGEB-FFZYJECLSA-N&#39;</span></span>
<span id="cb13-11"><a href="#cb13-11"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb13-12"><a href="#cb13-12"></a>}</span></code></pre></div>
<p><a href="https://w.wiki/32zU">Try this query</a>. You can adapt it by replacing the InChIKey with the one for your compound.</p>
<p>Alternatively you can use the following Scholia link (replace by your compounds InChIKey) <a href="https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N">https://scholia.toolforge.org/inchikey/QOVGHDRCAGYGEB-FFZYJECLSA-N</a></p>
<p>If your compound is already present on Wikidata, you can directly skip to the <a href="#add_bio">Add the biological source information</a> section below.</p>
<h5 id="add-your-data-manually-to-wikidata">Add your data manually to Wikidata <a name="add_your_data_manually"></a></h5>
<p>First, if you do not have a Wikidata account already, it is advisable that you create one via <a href="https://www.wikidata.org/wiki/Special:CreateAccount">https://www.wikidata.org/wiki/Special:CreateAccount</a>. While an account is not strictly required for manual edits, having one will be useful if you want to contribute more than once, and it helps in getting your contributions recognized. Note that Wikidata accounts are integrated with accounts across the Wikimedia ecosystem, so if you already have an account on, say, any Wikipedia or on Wikispecies, then you can use the same credentials on Wikidata.</p>
<p>If you are unfamiliar with how Wikidata works, you can start by reading the Wikidata introduction page <a href="https://www.wikidata.org/wiki/Wikidata:Introduction">https://www.wikidata.org/wiki/Wikidata:Introduction</a> and have a look at the Wikidata Tours page <a href="https://www.wikidata.org/wiki/Wikidata:Tours">https://www.wikidata.org/wiki/Wikidata:Tours</a>.</p>
<p>Now that you are all set up, you can go to Wikidata’s page for creating new items, <a href="https://www.wikidata.org/wiki/Special:NewItem">https://www.wikidata.org/wiki/Special:NewItem</a>:</p>
<figure>
<img src="images/tutorial/2021-02-25-14-18-03.png" alt="" /><figcaption>tutorial-image-03</figcaption>
</figure>
<p>An empty page with a new Wikidata identifier is created</p>
<figure>
<img src="images/tutorial/2021-02-25-14-18-50.png" alt="" /><figcaption>tutorial-image-04</figcaption>
</figure>
<h6 id="add-the-chemical-compound-information">Add the chemical compound information <a name="add_chemical"></a></h6>
<p>Create a new statement for <code>is an instance of</code></p>
<figure>
<img src="images/tutorial/2021-02-25-14-20-08.png" alt="" /><figcaption>tutorial-image-05</figcaption>
</figure>
<p>and select chemical compound (i.e. <a href="https://www.wikidata.org/wiki/Q11173">Q11173</a>):</p>
<figure>
<img src="images/tutorial/2021-02-25-14-20-58.png" alt="" /><figcaption>tutorial-image-06</figcaption>
</figure>
<p>Click <code>publish</code> to save your changes and make them public.</p>
<p>Since you created a new item about an instance of a chemical compound, the user interface will automatically propose to you a set of additional statements commonly found on items about chemical compounds.</p>
<figure>
<img src="images/tutorial/2021-02-25-14-22-44.png" alt="" /><figcaption>tutorial-image-07</figcaption>
</figure>
<p>You can then go on and fill these in.</p>
<p>Here, we start with the InChIKey.
Note the little flag which will automatically tell you if you have some problems with the recently created statements.</p>
<figure>
<img src="images/tutorial/2021-02-25-14-24-34.png" alt="" /><figcaption>tutorial-image-08</figcaption>
</figure>
<p>Here, Wikidata tells us that if we add an InChiKey, we will need to also add an InChI. Logical, but good to have a reminder !</p>
<p>Let’s go ahead and add the InChI string.</p>
<p>Likewise, the addition of an isomeric SMILES string will require us to add a Canonical SMILES.</p>
<p>Note that you might have to copy and paste the SMILES string from PubChem to a plain text editor and then back to Wikidata because of some formatting issues when copy pasting directly from PubChem.</p>
<pre><code>C[C@@H]1C2[C@]([C@H]3[C@H]4[C@]1(C5=CC(=C(Cl)Cl)[C@@H]([C@]5([C@@H]([C@@]6([C@H]4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O3)(O2)C8=CC=CC=C8)(C(=C)C)O

CC1C2C(C3C4C1(C5=CC(=C(Cl)Cl)C(C5(C(C6(C4O6)COC(=O)C)OC(=O)C)O)OC(=O)C7=CC=CC=C7)OC(O2)(O3)C8=CC=CC=C8)(C(=C)C)O</code></pre>
<h6 id="add-the-biological-source-information">Add the biological source information <a name="add_bio"></a></h6>
<p>Now let’s add the <code>found in taxon</code> property (<a href="https://www.wikidata.org/wiki/Property:P703">P703</a>).</p>
<p>Just click on <code>Add a new statement</code> and type in the first letters of the property you want to add:</p>
<figure>
<img src="images/tutorial/2021-02-25-14-31-56.png" alt="" /><figcaption>tutorial-image-09</figcaption>
</figure>
<p>Again, type in the first letters of the taxon, and if the organism is present, it will autocomplete. Here is how this looks like for <em>Trigonostemon cherrieri</em>:</p>
<figure>
<img src="images/tutorial/2021-02-25-14-33-14.png" alt="" /><figcaption>tutorial-image-10</figcaption>
</figure>
<p>Click <code>publish</code> to save your changes and make them public.</p>
<p>If your target taxon is not yet present on Wikidata and you are sure you have a valid taxon name that is spelled correctly, then you can go to <a href="https://www.wikidata.org/wiki/Special:NewItem">https://www.wikidata.org/wiki/Special:NewItem</a>, as described in the <a href="#add_your_data_manually">Add your data manually to Wikidata</a> section. For items about taxa, the <code>instance of</code> statement should have a value <code>taxon</code> (i.e. <a href="https://www.wikidata.org/wiki/Q16521">Q16521</a>). As for chemical compounds, the user interface will then suggest to you further statements to add. For taxa, these include taxon name, parent taxon and taxon rank.</p>
<h6 id="add-the-reference-documenting-the-structure-organism-pair">Add the reference documenting the structure-organism pair <a name="add_ref"></a></h6>
<p>Finally, since we report documented structure-organisms pairs, we need to add the reference for this newly created <code>compound found in taxon</code> relationship.
This happens on the item about the compound, just below the <code>found in taxon</code> statement. Click on the <code>0 references</code> link and then on <code>add reference</code>:</p>
<figure>
<img src="images/tutorial/2021-02-25-14-35-33.png" alt="" /><figcaption>tutorial-image-11</figcaption>
</figure>
<p>Here, we use the <code>stated in</code> property (<a href="https://www.wikidata.org/wiki/Property:P248">P248</a>):</p>
<figure>
<img src="images/tutorial/2021-02-25-14-36-33.png" alt="" /><figcaption>tutorial-image-12</figcaption>
</figure>
<p>Now, type in the first letters or word of the scientific publication documenting the natural product occurence, autocompletion happens again. Note that multiple publications might have the same title, and that there could be minor differences in punctuation or special characters between the information you and Wikidata have about the same reference. If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check, as outlined in the <a href="#check_ref">Check whether your target reference is already on Wikidata</a> section.</p>
<figure>
<img src="images/tutorial/2021-02-25-14-38-54.png" alt="" /><figcaption>tutorial-image-13</figcaption>
</figure>
<p>Click <code>publish</code> to save your changes and make them public.</p>
<figure>
<img src="images/tutorial/2021-02-25-14-39-48.png" alt="" /><figcaption>tutorial-image-14</figcaption>
</figure>
<h6 id="check-whether-your-target-reference-is-already-on-wikidata">Check whether your target reference is already on Wikidata <a name="check_ref"></a></h6>
<p>If you are not sure whether your target reference is already in Wikidata, you can use its DOI to check. For our DOI <code>10.1021/ol2030907</code>, the URL <a href="https://scholia.toolforge.org/doi/10.1021/ol2030907">https://scholia.toolforge.org/doi/<strong><em>10.1021/ol2030907</em></strong></a> will lead you to a <a href="https://www.wikidata.org/wiki/Wikidata:Scholia">Scholia</a> page about this publication: <a href="https://scholia.toolforge.org/work/Q83059010">https://scholia.toolforge.org/work/Q83059010</a>. Scholia visualizes information from Wikidata, so if it has an entry for your target reference, then so does Wikidata, and both of them will use the same identifier (in this case <a href="https://www.wikidata.org/wiki/Q83059010">Q83059010</a>). If you prefer to resolve your DOI to Wikidata directly, you can do so by using the uppercase-normalized DOI in the following URL pattern: <a href="https://hub.toolforge.org/P356:10.1021/OL2030907">https://hub.toolforge.org/P356:<strong>10.1021/OL2030907</strong></a>, which will lead you to the respective Wikidata page, in this case <a href="https://www.wikidata.org/wiki/Q83059010">Q83059010</a>.</p>
<p>If you think that no Wikidata entry exists for your target reference, you can use the DOI in the URL pattern <a href="https://sourcemd.toolforge.org/index_old.php?id=10.1021/ol2030907&amp;doit=Check+source">https://sourcemd.toolforge.org/index_old.php?id=<strong>10.1021/ol2030907</strong>&amp;doit=Check+source</a>, which will trigger a check with both Crossref and Wikidata, and if no Wikidata entry can be found, the metadata from Crossref will be fetched and presented to you for creating the respective Wikidata item semi-automatically. Using such semi-automated workflows does require and account that is a minimum number of days old and has made a minimum number of edits on Wikidata.</p>
<p>If you are interested the annotation of article with topics in Scholia here is a tutorial <a href="https://laurendupuis.github.io/Scholia_tutorial/">https://laurendupuis.github.io/Scholia_tutorial/</a></p>
<h5 id="voilà">Voilà !</h5>
<p>You have added your first documented structure-organism relationship to Wikidata and made a valuable contribution to the community.
You can add further statements, e.g. <code>molecular formula</code>, or <code>SPLASH code</code> for linking to spectral data.</p>
<p>The Wikidata entry <a href="https://www.wikidata.org/wiki/Q105674316">https://www.wikidata.org/wiki/Q105674316</a> has been started using these instructions.</p>
<p>You can run a SPARQL query and check that everything went smoothly by modifying the InChiKey line in the following <a href="https://w.wiki/32zb">SPARQL query</a>:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode sql"><code class="sourceCode sql"><span id="cb15-1"><a href="#cb15-1"></a><span class="kw">SELECT</span> ?item ?itemLabel ?taxonLabel ?artLabel <span class="kw">WHERE</span> {</span>
<span id="cb15-2"><a href="#cb15-2"></a>  <span class="kw">VALUES</span> ?classes {</span>
<span id="cb15-3"><a href="#cb15-3"></a>    wd<span class="ch">:Q11173</span> # chemical compound</span>
<span id="cb15-4"><a href="#cb15-4"></a>    wd<span class="ch">:Q59199015</span> # <span class="kw">group</span> <span class="kw">of</span> stereoisomers</span>
<span id="cb15-5"><a href="#cb15-5"></a>    wd<span class="ch">:Q79529</span> # chemical substance</span>
<span id="cb15-6"><a href="#cb15-6"></a>    wd<span class="ch">:Q17339814</span> # <span class="kw">group</span> <span class="kw">of</span> chemical substances</span>
<span id="cb15-7"><a href="#cb15-7"></a>    wd<span class="ch">:Q47154513</span> # structural <span class="kw">class</span> <span class="kw">of</span> chemical compounds</span>
<span id="cb15-8"><a href="#cb15-8"></a>  }</span>
<span id="cb15-9"><a href="#cb15-9"></a>  ?item wdt<span class="ch">:P31</span> ?classes. # <span class="kw">instance</span> <span class="kw">of</span></span>
<span id="cb15-10"><a href="#cb15-10"></a>  ?item wdt<span class="ch">:P235</span> <span class="st">&#39;QOVGHDRCAGYGEB-FFZYJECLSA-N&#39;</span> # InChiKey</span>
<span id="cb15-11"><a href="#cb15-11"></a>  {</span>
<span id="cb15-12"><a href="#cb15-12"></a>    ?item p<span class="ch">:P1582</span> ?stmt. # <span class="kw">natural</span> product <span class="kw">of</span> taxon</span>
<span id="cb15-13"><a href="#cb15-13"></a>    ?stmt ps<span class="ch">:P1582</span> ?taxon. # <span class="kw">natural</span> product <span class="kw">of</span> taxon</span>
<span id="cb15-14"><a href="#cb15-14"></a>    OPTIONAL {</span>
<span id="cb15-15"><a href="#cb15-15"></a>      ?stmt prov<span class="ch">:wasDerivedFrom</span> ?<span class="fu">ref</span>. </span>
<span id="cb15-16"><a href="#cb15-16"></a>      ?<span class="fu">ref</span> pr<span class="ch">:P248</span> ?art. # stated <span class="kw">in</span></span>
<span id="cb15-17"><a href="#cb15-17"></a>    }</span>
<span id="cb15-18"><a href="#cb15-18"></a>  }</span>
<span id="cb15-19"><a href="#cb15-19"></a>  <span class="kw">UNION</span></span>
<span id="cb15-20"><a href="#cb15-20"></a>  {</span>
<span id="cb15-21"><a href="#cb15-21"></a>    ?item p<span class="ch">:P703</span> ?stmt. # found <span class="kw">in</span> taxon</span>
<span id="cb15-22"><a href="#cb15-22"></a>    ?stmt ps<span class="ch">:P703</span> ?taxon. # found <span class="kw">in</span> taxon</span>
<span id="cb15-23"><a href="#cb15-23"></a>    OPTIONAL {</span>
<span id="cb15-24"><a href="#cb15-24"></a>      ?stmt prov<span class="ch">:wasDerivedFrom</span> ?<span class="fu">ref</span>.</span>
<span id="cb15-25"><a href="#cb15-25"></a>      ?<span class="fu">ref</span> pr<span class="ch">:P248</span> ?art. # stated <span class="kw">in</span></span>
<span id="cb15-26"><a href="#cb15-26"></a>    }</span>
<span id="cb15-27"><a href="#cb15-27"></a>  }</span>
<span id="cb15-28"><a href="#cb15-28"></a>  SERVICE wikibase<span class="ch">:label</span> { bd<span class="ch">:serviceParam</span> wikibase<span class="ch">:language</span> <span class="ot">&quot;[AUTO_LANGUAGE],en&quot;</span>. }</span>
<span id="cb15-29"><a href="#cb15-29"></a>}</span></code></pre></div>
<h3 class="page_break_before" id="si-5-complement-to-figure-7">SI 5 Complement to Figure 7</h3>
<div id="fig:magicTree-additional" class="fignos">
<figure>
<img src="images/magicTree_additional.svg" data-tag="SI-5" alt="" /><figcaption><span>Figure SI-5:</span> <strong>Complement to Figure 7</strong>: distribution of β-sitosterol and related chemical parents among families with at least 50 reported compounds present in LOTUS. Script used for the generation of each tree in the figure is the same (<a href="https://gitlab.com/lotus7/lotusProcessor/-/tree/preprint/src/4_visualizing/plot_magicTree.R">src/4_visualizing/plot_magicTree.R</a>) as for Figure <a href="#fig:magicTree">7</a> as both figures are related.</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="references-1">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-pIM4qxZA">
<p>Afendi FM, Okada T, Yamazaki M, Hirai-Morita A, Nakamura Y, Nakamura K, Ikeda S, Takahashi H, Altaf-Ul-Amin M, Darusman LK, Saito K, Kanaya S. 2012. KNApSAcK Family Databases: Integrated Metabolite–Plant Species Databases for Multifaceted Plant Research. <em>Plant and Cell Physiology</em> <strong>53</strong>:e1–e1. doi:<a href="https://doi.org/10.1093/pcp/pcr165">10.1093/pcp/pcr165</a></p>
</div>
<div id="ref-QBTqPbZC">
<p>Allard P-M, Bisson J, Azzollini A, Pauli GF, Cordell GA, Wolfender J-L. 2018. Pharmacognosy in the digital era: shifting to contextualized metabolomics. <em>Current Opinion in Biotechnology</em> <strong>54</strong>:57–64. doi:<a href="https://doi.org/10.1016/j.copbio.2018.02.010">10.1016/j.copbio.2018.02.010</a></p>
</div>
<div id="ref-hd3vl6GL">
<p>All natural. 2007.. <em>Nature Chemical Biology</em> <strong>3</strong>:351–351. doi:<a href="https://doi.org/10.1038/nchembio0707-351">10.1038/nchembio0707-351</a></p>
</div>
<div id="ref-iFXju5sG">
<p>Balietti S, Mäs M, Helbing D. 2015. On Disciplinary Fragmentation and Scientific Progress. <em>PLOS ONE</em> <strong>10</strong>:e0118747. doi:<a href="https://doi.org/10.1371/journal.pone.0118747">10.1371/journal.pone.0118747</a></p>
</div>
<div id="ref-WwEEYuQ9">
<p>Bisson J, McAlpine JB, Friesen JB, Chen S-N, Graham J, Pauli GF. 2015. Can Invalid Bioactives Undermine Natural Product-Based Drug Discovery? <em>Journal of Medicinal Chemistry</em> <strong>59</strong>:1671–1690. doi:<a href="https://doi.org/10.1021/acs.jmedchem.5b01009">10.1021/acs.jmedchem.5b01009</a></p>
</div>
<div id="ref-3cUZctro">
<p>Bisson J, Simmler C, Chen S-N, Friesen JB, Lankin DC, McAlpine JB, Pauli GF. 2016. Dissemination of original NMR data enhances reproducibility and integrity in chemical research. <em>Natural Product Reports</em> <strong>33</strong>:1028–1033. doi:<a href="https://doi.org/10.1039/c6np00022c">10.1039/c6np00022c</a></p>
</div>
<div id="ref-Go2AAIYt">
<p>Boonen J, Bronselaer A, Nielandt J, Veryser L, De Tré G, De Spiegeleer B. 2012. Alkamid database: Chemistry, occurrence and functionality of plant N-alkylamides. <em>Journal of Ethnopharmacology</em> <strong>142</strong>:563–590. doi:<a href="https://doi.org/10.1016/j.jep.2012.05.038">10.1016/j.jep.2012.05.038</a></p>
</div>
<div id="ref-13wsG3vmM">
<p>Brunson J. 2020. ggalluvial: Layered Grammar for Alluvial Plots. <em>Journal of Open Source Software</em> <strong>5</strong>:2017. doi:<a href="https://doi.org/10.21105/joss.02017">10.21105/joss.02017</a></p>
</div>
<div id="ref-7J4ZPUL7">
<p>Campitelli E. 2021. ggnewscale: Multiple fill and colour scales in “ggplot2” (manual).</p>
</div>
<div id="ref-zukFZ2aq">
<p>Cao Y, Charisi A, Cheng L-C, Jiang T, Girke T. 2008. ChemmineR: a compound mining framework for R. <em>Bioinformatics</em> <strong>24</strong>:1733–1734. doi:<a href="https://doi.org/10.1093/bioinformatics/btn307">10.1093/bioinformatics/btn307</a></p>
</div>
<div id="ref-2HOkPhmh">
<p>Capecchi A, Probst D, Reymond J-L. 2020. One molecular fingerprint to rule them all: drugs, biomolecules, and the metabolome. <em>Journal of Cheminformatics</em> <strong>12</strong>:43. doi:<a href="https://doi.org/10.1186/s13321-020-00445-4">10.1186/s13321-020-00445-4</a></p>
</div>
<div id="ref-UHC5x6VA">
<p>Chamberlain S, Zhu H, Jahn N, Boettiger C, Ram K. 2020. rcrossref: Client for Various “CrossRef” “APIs”.</p>
</div>
<div id="ref-1DPlqketM">
<p>Choi H, Cho SY, Pak HJ, Kim Y, Choi J-y, Lee YJ, Gong BH, Kang YS, Han T, Choi G, Cho Y, Lee S, Ryoo D, Park H. 2017. NPCARE: database of natural products and fractional extracts for cancer regulation. <em>Journal of Cheminformatics</em> <strong>9</strong>:2. doi:<a href="https://doi.org/10.1186/s13321-016-0188-5">10.1186/s13321-016-0188-5</a></p>
</div>
<div id="ref-MGL5MqgM">
<p>Cordell GA. 2017a. Cognate and cognitive ecopharmacognosy — in an anthropogenic era. <em>Phytochemistry Letters</em> <strong>20</strong>:540–549. doi:<a href="https://doi.org/10.1016/j.phytol.2016.10.009">10.1016/j.phytol.2016.10.009</a></p>
</div>
<div id="ref-NtzQIgT9">
<p>Cordell GA. 2017b. Sixty Challenges – A 2030 Perspective on Natural Products and Medicines Security. <em>Natural Product Communications</em> <strong>12</strong>:1934578X1701200. doi:<a href="https://doi.org/10.1177/1934578x1701200849">10.1177/1934578x1701200849</a></p>
</div>
<div id="ref-Ebuu5mfF">
<p>Cousijn H, Feeney P, Lowenberg D, Presani E, Simons N. 2019. Bringing Citations and Usage Metrics Together to Make Data Count. <em>Data Science Journal</em> <strong>18</strong>:9. doi:<a href="https://doi.org/10.5334/dsj-2019-009">10.5334/dsj-2019-009</a></p>
</div>
<div id="ref-IN34cxcI">
<p>Cousijn H, Kenall A, Ganley E, Harrison M, Kernohan D, Lemberger T, Murphy F, Polischuk P, Taylor S, Martone M, Clark T. 2018. A data citation roadmap for scientific publishers. <em>Scientific Data</em> <strong>5</strong>:180259. doi:<a href="https://doi.org/10.1038/sdata.2018.259">10.1038/sdata.2018.259</a></p>
</div>
<div id="ref-tPji342r">
<p>Davis GDJ, Vasanthi AHR. 2011. Seaweed metabolite database (SWMD): A database of natural compounds from marine algae. <em>Bioinformation</em> <strong>5</strong>:361–364. doi:<a href="https://doi.org/10.6026/97320630005361">10.6026/97320630005361</a></p>
</div>
<div id="ref-xVEzsOX1">
<p>de Candolle AP de. 1816. Essai sur les propriétés médicales des plantes, comparées avec leurs formes extérieures et leur classification naturelle. Paris: Chez Crochard, Libraire, doi:<a href="https://doi.org/10.5962/bhl.title.112422">10.5962/bhl.title.112422</a></p>
</div>
<div id="ref-ToSx7lEy">
<p>Defossez E, Pitteloud C, Descombes P, Glauser G, Allard P-M, Walker TWN, Fernandez-Conradi P, Wolfender J-L, Pellissier L, Rasmann S. 2021. Spatial and evolutionary predictability of phytochemical diversity. <em>Proceedings of the National Academy of Sciences</em> <strong>118</strong>:e2013344118. doi:<a href="https://doi.org/10.1073/pnas.2013344118">10.1073/pnas.2013344118</a></p>
</div>
<div id="ref-fAKz2qgk">
<p>Derese, Solomon, Ndakala, Albert, Rogo, Michael, Maynim, Cholastica, Oyim, James. 31st August to 3rd September 2015. Mitishamba database: a web based in silico database of natural products from Kenya plants.</p>
</div>
<div id="ref-1GFQixHzv">
<p>Djoumbou Feunang Y, Eisner R, Knox C, Chepelev L, Hastings J, Owen G, Fahy E, Steinbeck C, Subramanian S, Bolton E, Greiner R, Wishart DS. 2016. ClassyFire: automated chemical classification with a comprehensive, computable taxonomy. <em>Journal of Cheminformatics</em> <strong>8</strong>:61. doi:<a href="https://doi.org/10.1186/s13321-016-0174-y">10.1186/s13321-016-0174-y</a></p>
</div>
<div id="ref-hR0edX6b">
<p>Dowle M, Srinivasan A. 2020. data.table: Extension of `data.frame`.</p>
</div>
<div id="ref-gYzuBVpk">
<p>Dührkop K, Nothias LF, Fleischauer M, Ludwig M, Hoffmann MA, Rousu J, Dorrestein PC, Böcker S. 2020. Classes for the masses: Systematic classification of unknowns using fragmentation spectra. <em>Cold Spring Harbor Laboratory</em>. doi:<a href="https://doi.org/10.1101/2020.04.17.046672">10.1101/2020.04.17.046672</a></p>
</div>
<div id="ref-156ifmHvV">
<p>Flor M. 2020. chorddiag: Interactive Chord Diagrams.</p>
</div>
<div id="ref-12mR83H1T">
<p>Gagolewski M. 2020. R package stringi: Character string processing facilities.</p>
</div>
<div id="ref-aBMvYmj0">
<p>GBIF.org. 2020.. <em>GBIF Home Page</em>. <a href="https://www.gbif.org">https://www.gbif.org</a></p>
</div>
<div id="ref-T2D9o6RC">
<p>Gehlenborg N. 2019. UpSetR: A More Scalable Alternative to Venn and Euler Diagrams for Visualizing Intersecting Sets.</p>
</div>
<div id="ref-1APrBLQ3X">
<p>Giacomoni F, Bento Da Silva AL, Bronze M, Gladine C, Hollman P, Kopec R, Low Yanwen D, Micheau P, Nunes Dos Santos MC, Pavot B, Schmidt G, Morand C, Urpi Sarda M, Vazquez Manjarrez N, Verny M-A, Wiczkowski W, Knox C, Manach C. 2017. PhytoHub, an online platform to gather expert knowledge on polyphenols and other dietary phytochemicals.</p>
</div>
<div id="ref-15tHEF73t">
<p>Gottlieb OR. 1982. Micromolecular Evolution, Systematics and Ecology. Springer Science and Business Media LLC. doi:<a href="https://doi.org/10.1007/978-3-642-68641-2">10.1007/978-3-642-68641-2</a></p>
</div>
<div id="ref-KmGsdzOE">
<p>Graham JG, Farnsworth NR. 2010. The NAPRALERT Database as an Aid for Discovery of Novel Bioactive Compounds. Elsevier BV. pp. 81–94. doi:<a href="https://doi.org/10.1016/b978-008045382-8.00060-5">10.1016/b978-008045382-8.00060-5</a></p>
</div>
<div id="ref-W6BmnntI">
<p>Gu J, Gui Y, Chen L, Yuan G, Lu H-Z, Xu X. 2013. Use of Natural Products as Chemical Library for Drug Discovery and Network Pharmacology. <em>PLoS ONE</em> <strong>8</strong>:e62839. doi:<a href="https://doi.org/10.1371/journal.pone.0062839">10.1371/journal.pone.0062839</a></p>
</div>
<div id="ref-fP0lai2j">
<p>Günthardt BF, Hollender J, Hungerbühler K, Scheringer M, Bucheli TD. 2018. Comprehensive Toxic Plants–Phytotoxins Database and Its Application in Assessing Aquatic Micropollution Potential. <em>Journal of Agricultural and Food Chemistry</em> <strong>66</strong>:7577–7588. doi:<a href="https://doi.org/10.1021/acs.jafc.8b01639">10.1021/acs.jafc.8b01639</a></p>
</div>
<div id="ref-ZRFusbLt">
<p>Hatherley R, Brown DK, Musyoka TM, Penkler DL, Faya N, Lobb KA, Tastan Bishop Ö. 2015. SANCDB: a South African natural compound database. <em>Journal of Cheminformatics</em> <strong>7</strong>:29. doi:<a href="https://doi.org/10.1186/s13321-015-0080-8">10.1186/s13321-015-0080-8</a></p>
</div>
<div id="ref-1D8ShigK1">
<p>Haug K, Cochrane K, Nainala VC, Williams M, Chang J, Jayaseelan KV, O’Donovan C. 2019. MetaboLights: a resource evolving in response to the needs of its scientific community. <em>Nucleic Acids Research</em> gkz1019. doi:<a href="https://doi.org/10.1093/nar/gkz1019">10.1093/nar/gkz1019</a></p>
</div>
<div id="ref-18uusIb9e">
<p>Hegnauer R. 1986a. Phytochemistry and plant taxonomy — an essay on the chemotaxonomy of higher plants. <em>Phytochemistry</em> <strong>25</strong>:1519–1535. doi:<a href="https://doi.org/10.1016/s0031-9422(00)81204-2">10.1016/s0031-9422(00)81204-2</a></p>
</div>
<div id="ref-AgCpbRSb">
<p>Hegnauer R. 1986b. Chemotaxonomie der Pflanzen. Springer Science and Business Media LLC. doi:<a href="https://doi.org/10.1007/978-3-0348-9314-5">10.1007/978-3-0348-9314-5</a></p>
</div>
<div id="ref-AP6h5CYA">
<p>Heller S, McNaught A, Stein S, Tchekhovskoi D, Pletnev I. 2013. InChI - the worldwide chemical structure identifier standard. <em>Journal of Cheminformatics</em> <strong>5</strong>:7. doi:<a href="https://doi.org/10.1186/1758-2946-5-7">10.1186/1758-2946-5-7</a></p>
</div>
<div id="ref-18PujMmok">
<p>Helmy M, Crits-Christoph A, Bader GD. 2016. Ten Simple Rules for Developing Public Biological Databases. <em>PLOS Computational Biology</em> <strong>12</strong>:e1005128. doi:<a href="https://doi.org/10.1371/journal.pcbi.1005128">10.1371/journal.pcbi.1005128</a></p>
</div>
<div id="ref-otfsIdc0">
<p>Hester J, Wickham H. 2020. vroom: Read and write rectangular text data quickly (manual).</p>
</div>
<div id="ref-OceiRpiP">
<p>Hoffmann MA, Nothias L-F, Ludwig M, Fleischauer M, Gentry EC, Witting M, Dorrestein PC, Dührkop K, Böcker S. 2021. Assigning confidence to structural annotations from mass spectra with COSMIC. <em>Cold Spring Harbor Laboratory</em>. doi:<a href="https://doi.org/10.1101/2021.03.18.435634">10.1101/2021.03.18.435634</a></p>
</div>
<div id="ref-jTCd8yG0">
<p>Horai H, Arita M, Kanaya S, Nihei Y, Ikeda T, Suwa K, Ojima Y, Tanaka K, Tanaka S, Aoshima K, Oda Y, Kakazu Y, Kusano M, Tohge T, Matsuda F, Sawada Y, Hirai MY, Nakanishi H, Ikeda K, Akimoto N, Maoka T, Takahashi H, Ara T, Sakurai N, Suzuki H, Shibata D, Neumann S, Iida T, Tanaka K, Funatsu K, Matsuura F, Soga T, Taguchi R, Saito K, Nishioka T. 2010. MassBank: a public repository for sharing mass spectral data for life sciences. <em>Journal of Mass Spectrometry</em> <strong>45</strong>:703–714. doi:<a href="https://doi.org/10.1002/jms.1777">10.1002/jms.1777</a></p>
</div>
<div id="ref-yJZjE7TJ">
<p>Huang W, Brewer LK, Jones JW, Nguyen AT, Marcu A, Wishart DS, Oglesby-Sherrouse AG, Kane MA, Wilks A. 2018. PAMDB: a comprehensive Pseudomonas aeruginosa metabolome database. <em>Nucleic Acids Research</em> <strong>46</strong>:D575–D580. doi:<a href="https://doi.org/10.1093/nar/gkx1061">10.1093/nar/gkx1061</a></p>
</div>
<div id="ref-1026Gxdsi">
<p>Hunter JD. 2007. Matplotlib: A 2D Graphics Environment. <em>Computing in Science &amp; Engineering</em> <strong>9</strong>:90–95. doi:<a href="https://doi.org/10.1109/mcse.2007.55">10.1109/mcse.2007.55</a></p>
</div>
<div id="ref-mpr5sPI7">
<p>Ibezim A, Debnath B, Ntie-Kang F, Mbah CJ, Nwodo NJ. 2017. Binding of anti-Trypanosoma natural products from African flora against selected drug targets: a docking study. <em>Medicinal Chemistry Research</em> <strong>26</strong>:562–579. doi:<a href="https://doi.org/10.1007/s00044-016-1764-y">10.1007/s00044-016-1764-y</a></p>
</div>
<div id="ref-vgWLelZQ">
<p>Jarmusch AK, Wang M, Aceves CM, Advani RS, Aguirre S, Aksenov AA, Aleti G, Aron AT, Bauermeister A, Bolleddu S, Bouslimani A, Caraballo Rodriguez AM, Chaar R, Coras R, Elijah EO, Ernst M, Gauglitz JM, Gentry EC, Husband M, Jarmusch SA, Jones KL, Kamenik Z, Le Gouellec A, Lu A, McCall L-I, McPhail KL, Meehan MJ, Melnik AV, Menezes RC, Montoya Giraldo YA, Nguyen NH, Nothias LF, Nothias-Esposito M, Panitchpakdi M, Petras D, Quinn RA, Sikora N, van der Hooft JJJ, Vargas F, Vrbanac A, Weldon KC, Knight R, Bandeira N, Dorrestein PC. 2020. ReDU: a framework to find and reanalyze public mass spectrometry data. <em>Nature Methods</em> <strong>17</strong>:901–904. doi:<a href="https://doi.org/10.1038/s41592-020-0916-7">10.1038/s41592-020-0916-7</a></p>
</div>
<div id="ref-OhONtrcV">
<p>Jones MR, Pinto E, Torres MA, Dörr F, Mazur-Marzec H, Szubert K, Tartaglione L, Dell’Aversano C, Miles CO, Beach DG, McCarron P, Sivonen K, Fewer DP, Jokela J, Janssen EM-L. 2021. CyanoMetDB, a comprehensive public database of secondary metabolites from cyanobacteria. <em>Water Research</em> <strong>196</strong>:117017. doi:<a href="https://doi.org/10.1016/j.watres.2021.117017">10.1016/j.watres.2021.117017</a></p>
</div>
<div id="ref-dkabTMGA">
<p>Jr FEH, Dupont with contributions from C, others many. 2020. Hmisc: Harrell Miscellaneous.</p>
</div>
<div id="ref-F5HJbaQH">
<p>Kautsar SA, Blin K, Shaw S, Navarro-Muñoz JC, Terlouw BR, van der Hooft JJJ, van Santen JA, Tracanna V, Suarez Duran HG, Pascal Andreu V, Selem-Mojica N, Alanjary M, Robinson SL, Lund G, Epstein SC, Sisto AC, Charkoudian LK, Collemare J, Linington RG, Weber T, Medema MH. 2019. MIBiG 2.0: a repository for biosynthetic gene clusters of known function. <em>Nucleic Acids Research</em> gkz882. doi:<a href="https://doi.org/10.1093/nar/gkz882">10.1093/nar/gkz882</a></p>
</div>
<div id="ref-1Hc8Xn2Ff">
<p>Kessler A, Kalske A. 2018. Plant Secondary Metabolite Diversity and Species Interactions. <em>Annual Review of Ecology, Evolution, and Systematics</em> <strong>49</strong>:115–138. doi:<a href="https://doi.org/10.1146/annurev-ecolsys-110617-062406">10.1146/annurev-ecolsys-110617-062406</a></p>
</div>
<div id="ref-lUVqn8aT">
<p>kim H, Wang M, Leber C, Nothias L-F, Reher R, Kang KB, van der Hooft JJJ, Dorrestein P, Gerwick W, Cottrell G. 2020. NPClassifier: A Deep Neural Network-Based Structural Classification Tool for Natural Products. <em>American Chemical Society (ACS)</em>. doi:<a href="https://doi.org/10.26434/chemrxiv.12885494.v1">10.26434/chemrxiv.12885494.v1</a></p>
</div>
<div id="ref-1BsHixMtu">
<p>Kim S, Chen J, Cheng T, Gindulyte A, He J, He S, Li Q, Shoemaker BA, Thiessen PA, Yu B, Zaslavsky L, Zhang J, Bolton EE. 2019. PubChem 2019 update: improved access to chemical data. <em>Nucleic Acids Research</em> <strong>47</strong>:D1102–D1109. doi:<a href="https://doi.org/10.1093/nar/gky1033">10.1093/nar/gky1033</a></p>
</div>
<div id="ref-a48CrnhC">
<p>Kim S-K, Nam S, Jang H, Kim A, Lee J-J. 2015. TM-MC: a database of medicinal materials and chemical compounds in Northeast Asian traditional medicine. <em>BMC Complementary and Alternative Medicine</em> <strong>15</strong>:218. doi:<a href="https://doi.org/10.1186/s12906-015-0758-5">10.1186/s12906-015-0758-5</a></p>
</div>
<div id="ref-15rL9Psbo">
<p>Klementz D, Döring K, Lucas X, Telukunta KK, Erxleben A, Deubel D, Erber A, Santillana I, Thomas OS, Bechthold A, Günther S. 2016. StreptomeDB 2.0—an extended resource of natural products produced by streptomycetes. <em>Nucleic Acids Research</em> <strong>44</strong>:D509–D514. doi:<a href="https://doi.org/10.1093/nar/gkv1319">10.1093/nar/gkv1319</a></p>
</div>
<div id="ref-13RtkyRi5">
<p>Kratochvíl M, Vondrášek J, Galgonek J. 2019. Interoperable chemical structure search service. <em>Journal of Cheminformatics</em> <strong>11</strong>:45. doi:<a href="https://doi.org/10.1186/s13321-019-0367-2">10.1186/s13321-019-0367-2</a></p>
</div>
<div id="ref-mjVvWLK2">
<p>Kratochvíl M, Vondrášek J, Galgonek J. 2018. Sachem: a chemical cartridge for high-performance substructure search. <em>Journal of Cheminformatics</em> <strong>10</strong>:27. doi:<a href="https://doi.org/10.1186/s13321-018-0282-y">10.1186/s13321-018-0282-y</a></p>
</div>
<div id="ref-XiPNsL5P">
<p>Kuang K, Kong Q, Napolitano F. 2019. pbmcapply: Tracking the Progress of Mc*pply with Progress Bar.</p>
</div>
<div id="ref-TceFUyOp">
<p>Lang DT. 2020. XML: Tools for Parsing and Generating XML Within R and S-Plus.</p>
</div>
<div id="ref-WNY3oEAs">
<p>Lee CJ, Sugimoto CR, Zhang G, Cronin B. 2013. Bias in peer review. <em>Journal of the American Society for Information Science and Technology</em> <strong>64</strong>:2–17. doi:<a href="https://doi.org/10.1002/asi.22784">10.1002/asi.22784</a></p>
</div>
<div id="ref-MMp6cQ1x">
<p>Lin D, Crabtree J, Dillo I, Downs RR, Edmunds R, Giaretta D, De Giusti M, L’Hours H, Hugo W, Jenkyns R, Khodiyar V, Martone ME, Mokrane M, Navale V, Petters J, Sierman B, Sokolova DV, Stockhause M, Westbrook J. 2020. The TRUST Principles for digital repositories. <em>Scientific Data</em> <strong>7</strong>:144. doi:<a href="https://doi.org/10.1038/s41597-020-0486-7">10.1038/s41597-020-0486-7</a></p>
</div>
<div id="ref-LmFzkeBM">
<p>Loo M. 2014. The stringdist Package for Approximate String Matching. <em>The R Journal</em> <strong>6</strong>:111. doi:<a href="https://doi.org/10.32614/rj-2014-011">10.32614/rj-2014-011</a></p>
</div>
<div id="ref-pf43Zseq">
<p>Lowe DM, Corbett PT, Murray-Rust P, Glen RC. 2011. Chemical Name to Structure: OPSIN, an Open Source Solution. <em>Journal of Chemical Information and Modeling</em> <strong>51</strong>:739–753. doi:<a href="https://doi.org/10.1021/ci100384d">10.1021/ci100384d</a></p>
</div>
<div id="ref-dyiEPToF">
<p>Madariaga-Mazón A, Naveja JJ, Medina-Franco JL, Noriega-Colima KO, Martinez-Mayorga K. 2021. DiaNat-DB: a molecular database of antidiabetic compounds from medicinal plants. <em>RSC Advances</em> <strong>11</strong>:5172–5178. doi:<a href="https://doi.org/10.1039/d0ra10453a">10.1039/d0ra10453a</a></p>
</div>
<div id="ref-NfiSMvAq">
<p>Mahto A. 2019. splitstackshape: Stack and Reshape Datasets After Splitting Concatenated Values.</p>
</div>
<div id="ref-1CVDln27l">
<p>Main Page - WAKANKENSAKU. n.d. <a href="https://wakankensaku.inm.u-toyama.ac.jp/wiki/Main_Page">https://wakankensaku.inm.u-toyama.ac.jp/wiki/Main_Page</a></p>
</div>
<div id="ref-pzgOjFLZ">
<p>Martens M, Ammar A, Riutta A, Waagmeester A, Slenter D, Hanspers K, A. Miller R, Digles D, Lopes E, Ehrhart F, Dupuis LJ, Winckers LA, Coort S, Willighagen EL, Evelo CT, Pico AR, Kutmon M. 2021. WikiPathways: connecting communities. <em>Nucleic Acids Research</em> <strong>49</strong>:D613–D621. doi:<a href="https://doi.org/10.1093/nar/gkaa1024">10.1093/nar/gkaa1024</a></p>
</div>
<div id="ref-R6AOorG5">
<p>McAlpine JB, Chen S-N, Kutateladze A, MacMillan JB, Appendino G, Barison A, Beniddir MA, Biavatti MW, Bluml S, Boufridi A, Butler MS, Capon RJ, Choi YH, Coppage D, Crews P, Crimmins MT, Csete M, Dewapriya P, Egan JM, Garson MJ, Genta-Jouve G, Gerwick WH, Gross H, Harper MK, Hermanto P, Hook JM, Hunter L, Jeannerat D, Ji N-Y, Johnson TA, Kingston DGI, Koshino H, Lee H-W, Lewin G, Li J, Linington RG, Liu M, McPhail KL, Molinski TF, Moore BS, Nam J-W, Neupane RP, Niemitz M, Nuzillard J-M, Oberlies NH, Ocampos FMM, Pan G, Quinn RJ, Reddy DS, Renault J-H, Rivera-Chávez J, Robien W, Saunders CM, Schmidt TJ, Seger C, Shen B, Steinbeck C, Stuppner H, Sturm S, Taglialatela-Scafati O, Tantillo DJ, Verpoorte R, Wang B-G, Williams CM, Williams PG, Wist J, Yue J-M, Zhang C, Xu Z, Simmler C, Lankin DC, Bisson J, Pauli GF. 2019. The value of universally available raw NMR data for transparency, reproducibility, and integrity in natural product research. <em>Natural Product Reports</em> <strong>36</strong>:35–107. doi:<a href="https://doi.org/10.1039/c7np00064b">10.1039/c7np00064b</a></p>
</div>
<div id="ref-hPmhpanN">
<p>Michonneau F, Brown JW, Winter DJ. 2016. rotl: an R package to interact with the Open Tree of Life data. <em>Methods in Ecology and Evolution</em> <strong>7</strong>:1476–1481. doi:<a href="https://doi.org/10.1111/2041-210x.12593">10.1111/2041-210x.12593</a></p>
</div>
<div id="ref-VNMRDIw3">
<p>Mohamed A, Abuoda G, Ghanem A, Kaoudi Z, Aboulnaga A. 2020. RDFFrames: Knowledge Graph Access for Machine Learning Tools.</p>
</div>
<div id="ref-8sNOa0nm">
<p>Mongia M, Mohimani H. 2021. Repository scale classification and decomposition of tandem mass spectral data. <em>Scientific Reports</em> <strong>11</strong>:8314. doi:<a href="https://doi.org/10.1038/s41598-021-87796-6">10.1038/s41598-021-87796-6</a></p>
</div>
<div id="ref-LnilgHS9">
<p>Müller K, Wickham H, James DA, Falcon S. 2021. RSQLite: “SQLite” interface for r (manual).</p>
</div>
<div id="ref-jqMOfVhn">
<p>Nielsen FÅ, Mietchen D, Willighagen E. 2017. Scholia And Scientometrics With Wikidata. <em>Zenodo</em>. doi:<a href="https://doi.org/10.5281/zenodo.1036595">10.5281/zenodo.1036595</a></p>
</div>
<div id="ref-17CeNW62K">
<p>Nissen SB, Magidson T, Gross K, Bergstrom CT. 2016. Publication bias and the canonization of false facts. <em>eLife</em> <strong>5</strong>:e21451. doi:<a href="https://doi.org/10.7554/elife.21451">10.7554/elife.21451</a></p>
</div>
<div id="ref-IhdADmDO">
<p>Noteborn HPJM, Lommen A, van der Jagt RC, Weseman JM. 2000. Chemical fingerprinting for the evaluation of unintended secondary metabolic changes in transgenic food crops. <em>Journal of Biotechnology</em> <strong>77</strong>:103–114. doi:<a href="https://doi.org/10.1016/s0168-1656(99)00210-2">10.1016/s0168-1656(99)00210-2</a></p>
</div>
<div id="ref-6xTnsq5S">
<p>Ntie-Kang F, Telukunta KK, Döring K, Simoben CV, A. Moumbock AF, Malange YI, Njume LE, Yong JN, Sippl W, Günther S. 2017. NANPDB: A Resource for Natural Products from Northern African Sources. <em>Journal of Natural Products</em> <strong>80</strong>:2067–2076. doi:<a href="https://doi.org/10.1021/acs.jnatprod.7b00283">10.1021/acs.jnatprod.7b00283</a></p>
</div>
<div id="ref-Pf1Puxxg">
<p>Nupur LNU, Vats A, Dhanda SK, Raghava GPS, Pinnaka AK, Kumar A. 2016. ProCarDB: a database of bacterial carotenoids. <em>BMC Microbiology</em> <strong>16</strong>:96. doi:<a href="https://doi.org/10.1186/s12866-016-0715-6">10.1186/s12866-016-0715-6</a></p>
</div>
<div id="ref-Jdw1vygr">
<p>Ooms J. 2014. The jsonlite Package: A Practical and Consistent Mapping Between JSON Data and R Objects.</p>
</div>
<div id="ref-gdTufCbt">
<p>Pedersen TL. 2020. ggraph: An Implementation of Grammar of Graphics for Graphs and Networks.</p>
</div>
<div id="ref-xf44BZw9">
<p>Pierce HH, Dev A, Statham E, Bierer BE. 2019. Credit data generators for data reuse. <em>Nature</em> <strong>570</strong>:30–32. doi:<a href="https://doi.org/10.1038/d41586-019-01715-4">10.1038/d41586-019-01715-4</a></p>
</div>
<div id="ref-1EBJWvXBS">
<p>Pilon AC, Valli M, Dametto AC, Pinto MEF, Freire RT, Castro-Gamboa I, Andricopulo AD, Bolzani VS. 2017. NuBBEDB: an updated database to uncover chemical and biological information from Brazilian biodiversity. <em>Scientific Reports</em> <strong>7</strong>:7215. doi:<a href="https://doi.org/10.1038/s41598-017-07451-x">10.1038/s41598-017-07451-x</a></p>
</div>
<div id="ref-Ue7W1SPi">
<p>Pilón-Jiménez B, Saldívar-González F, Díaz-Eufracio B, Medina-Franco J. 2019. BIOFACQUIM: A Mexican Compound Database of Natural Products. <em>Biomolecules</em> <strong>9</strong>:31. doi:<a href="https://doi.org/10.3390/biom9010031">10.3390/biom9010031</a></p>
</div>
<div id="ref-1DcpY38I1">
<p>Probst D, Reymond J-L. 2020. Visualization of very large high-dimensional data sets as minimum spanning trees. <em>Journal of Cheminformatics</em> <strong>12</strong>:12. doi:<a href="https://doi.org/10.1186/s13321-020-0416-x">10.1186/s13321-020-0416-x</a></p>
</div>
<div id="ref-VMUBsFFG">
<p>Probst D, Reymond J-L. 2018a. FUn: a framework for interactive visualizations of large, high-dimensional datasets on the web. <em>Bioinformatics</em> <strong>34</strong>:1433–1435. doi:<a href="https://doi.org/10.1093/bioinformatics/btx760">10.1093/bioinformatics/btx760</a></p>
</div>
<div id="ref-eLVYiiAg">
<p>Probst D, Reymond J-L. 2018b. SmilesDrawer: Parsing and Drawing SMILES-Encoded Molecular Structures Using Client-Side JavaScript. <em>Journal of Chemical Information and Modeling</em> <strong>58</strong>:1–7. doi:<a href="https://doi.org/10.1021/acs.jcim.7b00425">10.1021/acs.jcim.7b00425</a></p>
</div>
<div id="ref-ZG9WyHw3">
<p>Rasberry L, Willighagen E, Nielsen F, Mietchen D. 2019. Robustifying Scholia: paving the way for knowledge discovery and research assessment through Wikidata. <em>Research Ideas and Outcomes</em> <strong>5</strong>:e35820. doi:<a href="https://doi.org/10.3897/rio.5.e35820">10.3897/rio.5.e35820</a></p>
</div>
<div id="ref-1BXLY6HF1">
<p>RDKit: Open-source cheminformatics. n.d.</p>
</div>
<div id="ref-zEc5CK8P">
<p>Reback J, McKinney W, Jbrockmendel, Bossche JVD, Augspurger T, Cloud P, Gfyoung, Sinhrks, Hawkins S, Roeschke M, Klein A, Terji Petersen, Tratner J, She C, Ayd W, Naveh S, Garcia M, Schendel J, Hayden A, Saxton D, Jancauskas V, McMaster A, Battiston P, Skipper Seabold, Chris-B1, H-Vetinari, Kaiqi Dong, Hoyer S, Overmeire W, Gorelli M. 2020. pandas-dev/pandas: Pandas 1.1.4. Zenodo. doi:<a href="https://doi.org/10.5281/zenodo.4161697">10.5281/zenodo.4161697</a></p>
</div>
<div id="ref-1F083MXGq">
<p>Rees J, Cranston K. 2017. Automated assembly of a reference taxonomy for phylogenetic data synthesis. <em>Biodiversity Data Journal</em> <strong>5</strong>:e12581. doi:<a href="https://doi.org/10.3897/bdj.5.e12581">10.3897/bdj.5.e12581</a></p>
</div>
<div id="ref-9GVjNRag">
<p>Rothwell JA, Perez-Jimenez J, Neveu V, Medina-Remon A, M’Hiri N, Garcia-Lobato P, Manach C, Knox C, Eisner R, Wishart DS, Scalbert A. 2013. Phenol-Explorer 3.0: a major update of the Phenol-Explorer database to incorporate data on the effects of food processing on polyphenol content. <em>Database</em> <strong>2013</strong>:bat070–bat070. doi:<a href="https://doi.org/10.1093/database/bat070">10.1093/database/bat070</a></p>
</div>
<div id="ref-7JFonzW0">
<p>R Special Interest Group on Databases (R-SIG-DB), Wickham H, Müller K. 2021. DBI: R database interface (manual).</p>
</div>
<div id="ref-1EaGvQl0B">
<p>Rutz A, Dounoue-Kubo M, Ollivier S, Bisson J, Bagheri M, Saesong T, Ebrahimi SN, Ingkaninan K, Wolfender J-L, Allard P-M. 2019. Taxonomically Informed Scoring Enhances Confidence in Natural Products Annotation. <em>Frontiers in Plant Science</em> <strong>10</strong>:1329. doi:<a href="https://doi.org/10.3389/fpls.2019.01329">10.3389/fpls.2019.01329</a></p>
</div>
<div id="ref-BSCQ82T">
<p>SAIKKONEN K. 2004. Evolution of endophyte?plant symbioses. <em>Trends in Plant Science</em> <strong>9</strong>:275–280. doi:<a href="https://doi.org/10.1016/j.tplants.2004.04.005">10.1016/j.tplants.2004.04.005</a></p>
</div>
<div id="ref-bOvsdNwM">
<p>Sander T, Freyss J, von Korff M, Rufener C. 2015. DataWarrior: An Open-Source Program For Chemistry Aware Data Visualization And Analysis. <em>Journal of Chemical Information and Modeling</em> <strong>55</strong>:460–473. doi:<a href="https://doi.org/10.1021/ci500588j">10.1021/ci500588j</a></p>
</div>
<div id="ref-12Ui1Rc6x">
<p>Sawada Y, Nakabayashi R, Yamada Y, Suzuki M, Sato M, Sakata A, Akiyama K, Sakurai T, Matsuda F, Aoki T, Hirai MY, Saito K. 2012. RIKEN tandem mass spectral database (ReSpect) for phytochemicals: A plant-specific MS/MS-based data resource and database. <em>Phytochemistry</em> <strong>82</strong>:38–45. doi:<a href="https://doi.org/10.1016/j.phytochem.2012.07.007">10.1016/j.phytochem.2012.07.007</a></p>
</div>
<div id="ref-1Ck6coooY">
<p>Sedio BE. 2017. Recent breakthroughs in metabolomics promise to reveal the cryptic chemical traits that mediate plant community composition, character evolution and lineage diversification. <em>New Phytologist</em> <strong>214</strong>:952–958. doi:<a href="https://doi.org/10.1111/nph.14438">10.1111/nph.14438</a></p>
</div>
<div id="ref-GFBBPIgV">
<p>Sharma A, Dutta P, Sharma M, Rajput NK, Dodiya B, Georrge JJ, Kholia T, Bhardwaj A, OSDD Consortium. 2014. BioPhytMol: a drug discovery community resource on anti-mycobacterial phytomolecules and plant extracts. <em>Journal of Cheminformatics</em> <strong>6</strong>:46. doi:<a href="https://doi.org/10.1186/s13321-014-0046-2">10.1186/s13321-014-0046-2</a></p>
</div>
<div id="ref-2fEiIH9c">
<p>Shinbo Y, Nakamura Y, Altaf-Ul-Amin M, Asahi H, Kurokawa K, Arita M, Saito K, Ohta D, Shibata D, Kanaya S. 2006. KNApSAcK: A Comprehensive Species-Metabolite Relationship Database. Springer Science and Business Media LLC. pp. 165–181. doi:<a href="https://doi.org/10.1007/3-540-29782-0_13">10.1007/3-540-29782-0_13</a></p>
</div>
<div id="ref-kVXkIA2g">
<p>Sievert C. 2020. Interactive Web-Based Data Visualization with R, plotly, and shiny. Chapman and Hall/CRC.</p>
</div>
<div id="ref-1CVZzR5SC">
<p>Slenter DN, Kutmon M, Hanspers K, Riutta A, Windsor J, Nunes N, Mélius J, Cirillo E, Coort SL, Digles D, Ehrhart F, Giesbertz P, Kalafati M, Martens M, Miller R, Nishida K, Rieswijk L, Waagmeester A, Eijssen LMT, Evelo CT, Pico AR, Willighagen EL. 2018. WikiPathways: a multifaceted pathway database bridging metabolomics to other omics research. <em>Nucleic Acids Research</em> <strong>46</strong>:D661–D667. doi:<a href="https://doi.org/10.1093/nar/gkx1064">10.1093/nar/gkx1064</a></p>
</div>
<div id="ref-16ijqhOBz">
<p>Sorokina M, Merseburger P, Rajan K, Yirik MA, Steinbeck C. 2021. COCONUT online: Collection of Open Natural Products database. <em>Journal of Cheminformatics</em> <strong>13</strong>:2. doi:<a href="https://doi.org/10.1186/s13321-020-00478-9">10.1186/s13321-020-00478-9</a></p>
</div>
<div id="ref-R06ZE1xj">
<p>Sorokina M, Steinbeck C. 2020a. Review on natural products databases: where to find data in 2020. <em>Journal of Cheminformatics</em> <strong>12</strong>:20. doi:<a href="https://doi.org/10.1186/s13321-020-00424-9">10.1186/s13321-020-00424-9</a></p>
</div>
<div id="ref-1EoPmmMQv">
<p>Sorokina M, Steinbeck C. 2020b. COCONUT: the COlleCtion of Open NatUral producTs. doi:<a href="https://doi.org/10.5281/zenodo.3778405">10.5281/zenodo.3778405</a></p>
</div>
<div id="ref-kM2kraS2">
<p>Szöcs E, Stirling T, Scott ER, Scharmüller A, Schäfer RB. 2020. <strong>webchem</strong> : An <em>R</em> Package to Retrieve Chemical Information from the Web. <em>Journal of Statistical Software</em> <strong>93</strong>. doi:<a href="https://doi.org/10.18637/jss.v093.i13">10.18637/jss.v093.i13</a></p>
</div>
<div id="ref-TgDlrv2t">
<p>Tsugawa H. 2018. Advances in computational metabolomics and databases deepen the understanding of metabolisms. <em>Current Opinion in Biotechnology</em> <strong>54</strong>:10–17. doi:<a href="https://doi.org/10.1016/j.copbio.2018.01.008">10.1016/j.copbio.2018.01.008</a></p>
</div>
<div id="ref-1HQ8C1kF4">
<p>U.S. Department of Agriculture, Agricultural Research Service. Dr. Duke’s Phytochemical and Ethnobotanical Databases. 1992–2016.. <em>Home Page</em>. <a href="https://phytochem.nal.usda.gov/">https://phytochem.nal.usda.gov/</a></p>
</div>
<div id="ref-1H1kBOCRr">
<p>van Santen JA, Jacob G, Singh AL, Aniebok V, Balunas MJ, Bunsko D, Neto FC, Castaño-Espriu L, Chang C, Clark TN, Cleary Little JL, Delgadillo DA, Dorrestein PC, Duncan KR, Egan JM, Galey MM, Haeckl FPJ, Hua A, Hughes AH, Iskakova D, Khadilkar A, Lee J-H, Lee S, LeGrow N, Liu DY, Macho JM, McCaughey CS, Medema MH, Neupane RP, O’Donnell TJ, Paula JS, Sanchez LM, Shaikh AF, Soldatou S, Terlouw BR, Tran TA, Valentine M, van der Hooft JJJ, Vo DA, Wang M, Wilson D, Zink KE, Linington RG. 2019. The Natural Products Atlas: An Open Access Knowledge Base for Microbial Natural Products Discovery. <em>ACS Central Science</em> <strong>5</strong>:1824–1833. doi:<a href="https://doi.org/10.1021/acscentsci.9b00806">10.1021/acscentsci.9b00806</a></p>
</div>
<div id="ref-8Miti2Gz">
<p>Virtanen P, Gommers R, Oliphant TE, Haberland M, Reddy T, Cournapeau D, Burovski E, Peterson P, Weckesser W, Bright J, van der Walt SJ, Brett M, Wilson J, Millman KJ, Mayorov N, Nelson ARJ, Jones E, Kern R, Larson E, Carey CJ, Polat İ, Feng Y, Moore EW, VanderPlas J, Laxalde D, Perktold J, Cimrman R, Henriksen I, Quintero EA, Harris CR, Archibald AM, Ribeiro AH, Pedregosa F, van Mulbregt P, SciPy 1.0 Contributors. 2020. SciPy 1.0: fundamental algorithms for scientific computing in Python. <em>Nature Methods</em> <strong>17</strong>:261–272. doi:<a href="https://doi.org/10.1038/s41592-019-0686-2">10.1038/s41592-019-0686-2</a></p>
</div>
<div id="ref-HUodcdz1">
<p>Waagmeester A, Stupp G, Burgstaller-Muehlbacher S, Good BM, Griffith M, Griffith OL, Hanspers K, Hermjakob H, Hudson TS, Hybiske K, Keating SM, Manske M, Mayers M, Mietchen D, Mitraka E, Pico AR, Putman T, Riutta A, Queralt-Rosinach N, Schriml LM, Shafee T, Slenter D, Stephan R, Thornton K, Tsueng G, Tu R, Ul-Hasan S, Willighagen E, Wu C, Su AI. 2020. Wikidata as a knowledge graph for the life sciences. <em>eLife</em> <strong>9</strong>:e52614. doi:<a href="https://doi.org/10.7554/elife.52614">10.7554/elife.52614</a></p>
</div>
<div id="ref-1EloBUtuI">
<p>Wang L-G, Lam TT-Y, Xu S, Dai Z, Zhou L, Feng T, Guo P, Dunn CW, Jones BR, Bradley T, Zhu H, Guan Y, Jiang Y, Yu G. 2020. Treeio: An R Package for Phylogenetic Tree Input and Output with Richly Annotated and Associated Data. <em>Molecular Biology and Evolution</em> <strong>37</strong>:599–603. doi:<a href="https://doi.org/10.1093/molbev/msz240">10.1093/molbev/msz240</a></p>
</div>
<div id="ref-7AFWc7sN">
<p>Warnes GR, Bolker B, Gorjanc G, Grothendieck G, Korosec A, Lumley T, MacQueen D, Magnusson A, Rogers J, others. 2017. gdata: Various r programming tools for data manipulation (manual).</p>
</div>
<div id="ref-t7Cw5tqt">
<p>Weininger D. 1988. SMILES, a chemical language and information system. 1. Introduction to methodology and encoding rules. <em>Journal of Chemical Information and Modeling</em> <strong>28</strong>:31–36. doi:<a href="https://doi.org/10.1021/ci00057a005">10.1021/ci00057a005</a></p>
</div>
<div id="ref-1ELrikmXF">
<p>Wickham H. 2020. rvest: Easily Harvest (Scrape) Web Pages.</p>
</div>
<div id="ref-EWll2pyF">
<p>Wickham H, Averick M, Bryan J, Chang W, McGowan L, François R, Grolemund G, Hayes A, Henry L, Hester J, Kuhn M, Pedersen T, Miller E, Bache S, Müller K, Ooms J, Robinson D, Seidel D, Spinu V, Takahashi K, Vaughan D, Wilke C, Woo K, Yutani H. 2019. Welcome to the Tidyverse. <em>Journal of Open Source Software</em> <strong>4</strong>:1686. doi:<a href="https://doi.org/10.21105/joss.01686">10.21105/joss.01686</a></p>
</div>
<div id="ref-xN7NElEK">
<p>Wickham H, Bryan J. 2019. readxl: Read Excel Files.</p>
</div>
<div id="ref-2keqmgUs">
<p>Wickham H, Hester J, Ooms J. 2020. xml2: Parse XML (manual).</p>
</div>
<div id="ref-ejxxPICa">
<p>Wilkins D. 2020. ggfittext: Fit Text Inside a Box in “ggplot2”.</p>
</div>
<div id="ref-6DjakjNS">
<p>Wilkinson MD, Dumontier M, Aalbersberg IJ, Appleton G, Axton M, Baak A, Blomberg N, Boiten J-W, da Silva Santos LB, Bourne PE, Bouwman J, Brookes AJ, Clark T, Crosas M, Dillo I, Dumon O, Edmunds S, Evelo CT, Finkers R, Gonzalez-Beltran A, Gray AJG, Groth P, Goble C, Grethe JS, Heringa J, ’t Hoen PAC, Hooft R, Kuhn T, Kok R, Kok J, Lusher SJ, Martone ME, Mons A, Packer AL, Persson B, Rocca-Serra P, Roos M, van Schaik R, Sansone S-A, Schultes E, Sengstag T, Slater T, Strawn G, Swertz MA, Thompson M, van der Lei J, van Mulligen E, Velterop J, Waagmeester A, Wittenburg P, Wolstencroft K, Zhao J, Mons B. 2016. The FAIR Guiding Principles for scientific data management and stewardship. <em>Scientific Data</em> <strong>3</strong>:160018. doi:<a href="https://doi.org/10.1038/sdata.2016.18">10.1038/sdata.2016.18</a></p>
</div>
<div id="ref-xpBWVpji">
<p>Willighagen EL, Mayfield JW, Alvarsson J, Berg A, Carlsson L, Jeliazkova N, Kuhn S, Pluskal T, Rojas-Chertó M, Spjuth O, Torrance G, Evelo CT, Guha R, Steinbeck C. 2017. The Chemistry Development Kit (CDK) v2.0: atom typing, depiction, molecular formulas, and substructure searching. <em>Journal of Cheminformatics</em> <strong>9</strong>:33. doi:<a href="https://doi.org/10.1186/s13321-017-0220-4">10.1186/s13321-017-0220-4</a></p>
</div>
<div id="ref-13ZrxSjw4">
<p>Winter D. 2017. rentrez: An R package for the NCBI eUtils API. <em>The R Journal</em> <strong>9</strong>:520. doi:<a href="https://doi.org/10.32614/rj-2017-058">10.32614/rj-2017-058</a></p>
</div>
<div id="ref-WRJsojww">
<p>Wohlgemuth G, Haldiya PK, Willighagen E, Kind T, Fiehn O. 2010. The Chemical Translation Service–a web-based tool to improve standardization of metabolomic reports. <em>Bioinformatics</em> <strong>26</strong>:2647–2648. doi:<a href="https://doi.org/10.1093/bioinformatics/btq476">10.1093/bioinformatics/btq476</a></p>
</div>
<div id="ref-17NxHx9HG">
<p>Xu S. 2021. ggstar: Star Layer for “ggplot2” (manual).</p>
</div>
<div id="ref-121McWIls">
<p>Xu S, Dai Z, Guo P, Fu X, Liu S, Zhou L, Tang W, Feng T, Chen M, Zhan L, Wu T, Hu E, Yu G. 2021. ggtreeExtra: Compact visualization of richly annotated phylogenetic data. <em>Research Square Platform LLC</em>. doi:<a href="https://doi.org/10.21203/rs.3.rs-155672/v2">10.21203/rs.3.rs-155672/v2</a></p>
</div>
<div id="ref-FzUbylmA">
<p>Yabuzaki J. 2017. Carotenoids Database: structures, chemical fingerprints and distribution among organisms. <em>Database</em> <strong>2017</strong>. doi:<a href="https://doi.org/10.1093/database/bax004">10.1093/database/bax004</a></p>
</div>
<div id="ref-YdA5Ivd8">
<p>Yu G, Smith DK, Zhu H, Guan Y, Lam TT. 2016.: an package for visualization and annotation of phylogenetic trees with their covariates and other associated data. <em>Methods in Ecology and Evolution</em> <strong>8</strong>:28–36. doi:<a href="https://doi.org/10.1111/2041-210x.12628">10.1111/2041-210x.12628</a></p>
</div>
<div id="ref-16mSGZn7C">
<p>Yue Y, Chu G-X, Liu X-S, Tang X, Wang W, Liu G-J, Yang T, Ling T-J, Wang X-G, Zhang Z-Z, Xia T, Wan X-C, Bao G-H. 2014. TMDB: A literature-curated database for small molecular compounds found from tea. <em>BMC Plant Biology</em> <strong>14</strong>:243. doi:<a href="https://doi.org/10.1186/s12870-014-0243-1">10.1186/s12870-014-0243-1</a></p>
</div>
<div id="ref-4RKmYOPE">
<p>Zeng X, Zhang P, He W, Qin C, Chen S, Tao L, Wang Y, Tan Y, Gao D, Wang B, Chen Z, Chen W, Jiang YY, Chen YZ. 2018. NPASS: natural product activity and species source database for natural product research, discovery and tool development. <em>Nucleic Acids Research</em> <strong>46</strong>:D1217–D1222. doi:<a href="https://doi.org/10.1093/nar/gkx1026">10.1093/nar/gkx1026</a></p>
</div>
<div id="ref-lB6krLxn">
<p>Zhang R, Lin J, Zou Y, Zhang X-J, Xiao W-L. 2018. Chemical Space and Biological Target Network of Anti-Inflammatory Natural Products. <em>Journal of Chemical Information and Modeling</em> <strong>59</strong>:66–73. doi:<a href="https://doi.org/10.1021/acs.jcim.8b00560">10.1021/acs.jcim.8b00560</a></p>
</div>
<div id="ref-mtVu8pJx">
<p>富木毅, 斎藤臣, 植木雅, 今野英, 浅岡丈, 鈴木龍, 浦本昌, 掛谷秀, 長田裕. 2006. 理化学研究所・天然化合物バンク(RIKEN NPDepo)の化合物データベース“理化学研究所・天然化合物エンサイクロペディア(RIKEN NPEdia)”. <em>ケモインフォマティクス討論会予稿集</em>. doi:<a href="https://doi.org/10.11545/ciqs.2006.0.jl6.0">10.11545/ciqs.2006.0.jl6.0</a></p>
</div>
</div>
<!-- default theme -->

<style>
  /* import google fonts */
  @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
  @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

  /* -------------------------------------------------- */
  /* global */
  /* -------------------------------------------------- */

  /* all elements */
  * {
    /* force sans-serif font unless specified otherwise */
    font-family: "Open Sans", "Helvetica", sans-serif;

    /* prevent text inflation on some mobile browsers */
    -webkit-text-size-adjust: none !important;
    -moz-text-size-adjust: none !important;
    -o-text-size-adjust: none !important;
    text-size-adjust: none !important;
  }

  @media only screen {
    /* "page" element */
    body {
      position: relative;
      box-sizing: border-box;
      font-size: 12pt;
      line-height: 1.5;
      max-width: 8.5in;
      margin: 20px auto;
      padding: 40px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* "page" element */
    body {
      padding: 20px;
      margin: 0;
      border-radius: 0;
      border: none;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
      background: none;
    }
  }

  /* -------------------------------------------------- */
  /* headings */
  /* -------------------------------------------------- */

  /* all headings */
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 20px 0;
    padding: 0;
    font-weight: bold;
  }

  /* biggest heading */
  h1 {
    margin: 40px 0;
    text-align: center;
  }

  /* second biggest heading */
  h2 {
    margin-top: 30px;
    padding-bottom: 5px;
    border-bottom: solid 1px #bdbdbd;
  }

  /* heading font sizes */
  h1 {
    font-size: 2em;
  }
  h2 {
    font-size: 1.5em;
  }
  h3 {
    font-size: 1.35em;
  }
  h4 {
    font-size: 1.25em;
  }
  h5 {
    font-size: 1.15em;
  }
  h6 {
    font-size: 1em;
  }

  /* -------------------------------------------------- */
  /* manuscript header */
  /* -------------------------------------------------- */

  /* manuscript title */
  header > h1 {
    margin: 0;
  }

  /* manuscript title caption text (ie "automatically generated on") */
  header + p {
    text-align: center;
    margin-top: 10px;
  }

  /* -------------------------------------------------- */
  /* text elements */
  /* -------------------------------------------------- */

  /* links */
  a {
    color: #2196f3;
    overflow-wrap: break-word;
  }

  /* superscripts and subscripts */
  sub,
  sup {
    /* prevent from affecting line height */
    line-height: 0;
  }

  /* unordered and ordered lists*/
  ul,
  ol {
    padding-left: 20px;
  }

  /* class for styling text semibold */
  .semibold {
    font-weight: 600;
  }

  /* class for styling elements horizontally left aligned */
  .left {
    display: block;
    text-align: left;
    margin-left: auto;
    margin-right: 0;
    justify-content: left;
  }

  /* class for styling elements horizontally centered */
  .center {
    display: block;
    text-align: center;
    margin-left: auto;
    margin-right: auto;
    justify-content: center;
  }

  /* class for styling elements horizontally right aligned */
  .right {
    display: block;
    text-align: right;
    margin-left: 0;
    margin-right: auto;
    justify-content: right;
  }

  /* -------------------------------------------------- */
  /* section elements */
  /* -------------------------------------------------- */

  /* horizontal divider line */
  hr {
    border: none;
    height: 1px;
    background: #bdbdbd;
  }

  /* paragraphs, horizontal dividers, figures, tables, code */
  p {
  text-align: justify;
  }

  #refs p {
  text-align: left;
  },
  hr,
  figure,
  table,
  pre {
    /* treat all as "paragraphs", with consistent vertical margins */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* figures */
  /* -------------------------------------------------- */

  /* figure */
  figure {
    max-width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure caption */
  figcaption {
    padding: 0;
    padding-top: 10px;
  }

  /* figure image element */
  figure > img,
  figure > svg {
    max-width: 100%;
    display: block;
    margin-left: auto;
    margin-right: auto;
  }

  /* figure auto-number */
  img + figcaption > span:first-of-type,
  svg + figcaption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* tables */
  /* -------------------------------------------------- */

  /* table */
  table {
    border-collapse: collapse;
    border-spacing: 0;
    width: 100%;
    margin-left: auto;
    margin-right: auto;
  }

  /* table cells */
  th,
  td {
    border: solid 1px #bdbdbd;
    padding: 10px;
    /* squash table if too wide for page by forcing line breaks */
    overflow-wrap: break-word;
    word-break: break-word;
  }

  /* header row and even rows */
  th,
  tr:nth-child(2n) {
    background-color: #fafafa;
  }

  /* odd rows */
  tr:nth-child(2n + 1) {
    background-color: #ffffff;
  }

  /* table caption */
  caption {
    text-align: left;
    padding: 0;
    padding-bottom: 10px;
  }

  /* table auto-number */
  table > caption > span:first-of-type,
  div.table_wrapper > table > caption > span:first-of-type {
    font-weight: bold;
    margin-right: 5px;
  }

  /* -------------------------------------------------- */
  /* code */
  /* -------------------------------------------------- */

  /* multi-line code block */
  pre {
    padding: 10px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
    break-inside: avoid;
    text-align: left;
  }

  /* inline code, ie code within normal text */
  :not(pre) > code {
    padding: 0 4px;
    background-color: #eeeeee;
    color: #000000;
    border-radius: 5px;
  }

  /* code text */
  /* apply all children, to reach syntax highlighting sub-elements */
  code,
  code * {
    /* force monospace font */
    font-family: "Source Code Pro", "Courier New", monospace;
  }

  /* -------------------------------------------------- */
  /* quotes */
  /* -------------------------------------------------- */

  /* quoted text */
  blockquote {
    margin: 0;
    padding: 0;
    border-left: 4px solid #bdbdbd;
    padding-left: 16px;
    break-inside: avoid;
  }

  /* -------------------------------------------------- */
  /* banners */
  /* -------------------------------------------------- */

  /* info banners */
  .banner {
    box-sizing: border-box;
    display: block;
    position: relative;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 20px;
    padding: 20px;
    text-align: center;
  }

  /* paragraph in banner */
  .banner > p {
    margin: 0;
  }

  /* -------------------------------------------------- */
  /* highlight colors */
  /* -------------------------------------------------- */

  .white {
    background: #ffffff;
  }
  .lightgrey {
    background: #eeeeee;
  }
  .grey {
    background: #757575;
  }
  .darkgrey {
    background: #424242;
  }
  .black {
    background: #000000;
  }
  .lightred {
    background: #ffcdd2;
  }
  .lightyellow {
    background: #ffecb3;
  }
  .lightgreen {
    background: #dcedc8;
  }
  .lightblue {
    background: #e3f2fd;
  }
  .lightpurple {
    background: #f3e5f5;
  }
  .red {
    background: #f44336;
  }
  .orange {
    background: #ff9800;
  }
  .yellow {
    background: #ffeb3b;
  }
  .green {
    background: #4caf50;
  }
  .blue {
    background: #2196f3;
  }
  .purple {
    background: #9c27b0;
  }
  .white,
  .lightgrey,
  .lightred,
  .lightyellow,
  .lightgreen,
  .lightblue,
  .lightpurple,
  .orange,
  .yellow,
  .white a,
  .lightgrey a,
  .lightred a,
  .lightyellow a,
  .lightgreen a,
  .lightblue a,
  .lightpurple a,
  .orange a,
  .yellow a {
    color: #000000;
  }
  .grey,
  .darkgrey,
  .black,
  .red,
  .green,
  .blue,
  .purple,
  .grey a,
  .darkgrey a,
  .black a,
  .red a,
  .green a,
  .blue a,
  .purple a {
    color: #ffffff;
  }

  /* -------------------------------------------------- */
  /* buttons */
  /* -------------------------------------------------- */

  /* class for styling links like buttons */
  .button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    margin: 5px;
    padding: 10px 20px;
    font-size: 0.75em;
    font-weight: 600;
    text-transform: uppercase;
    text-decoration: none;
    letter-spacing: 1px;
    background: none;
    color: #2196f3;
    border: solid 1px #bdbdbd;
    border-radius: 5px;
  }

  /* buttons when hovered */
  .button:hover:not([disabled]),
  .icon_button:hover:not([disabled]) {
    cursor: pointer;
    background: #f5f5f5;
  }

  /* buttons when disabled */
  .button[disabled],
  .icon_button[disabled] {
    opacity: 0.35;
    pointer-events: none;
  }

  /* class for styling buttons containg only single icon */
  .icon_button {
    display: inline-flex;
    justify-content: center;
    align-items: center;
    text-decoration: none;
    margin: 0;
    padding: 0;
    background: none;
    border-radius: 5px;
    border: none;
    width: 20px;
    height: 20px;
    min-width: 20px;
    min-height: 20px;
  }

  /* icon button inner svg image */
  .icon_button > svg {
    height: 16px;
  }

  /* -------------------------------------------------- */
  /* icons */
  /* -------------------------------------------------- */

  /* class for styling icons inline with text */
  .inline_icon {
    height: 1em;
    position: relative;
    top: 0.125em;
  }

  /* -------------------------------------------------- */
  /* print control */
  /* -------------------------------------------------- */

  @media print {
    @page {
      /* suggested printing margin */
      margin: 0.5in;
    }

    /* document and "page" elements */
    html,
    body {
      margin: 0;
      padding: 0;
      width: 100%;
      height: 100%;
    }

    /* "page" element */
    body {
      font-size: 11pt !important;
      line-height: 1.35;
    }

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      margin: 15px 0;
    }

    /* figures and tables */
    figure,
    table {
      font-size: 0.85em;
    }

    /* table cells */
    th,
    td {
      padding: 5px;
    }

    /* shrink font awesome icons */
    i.fas,
    i.fab,
    i.far,
    i.fal {
      transform: scale(0.85);
    }

    /* decrease banner margins */
    .banner {
      margin-top: 15px;
      margin-bottom: 15px;
      padding: 15px;
    }

    /* class for centering an element vertically on its own page */
    .page_center {
      margin: auto;
      width: 100%;
      height: 100%;
      display: flex;
      align-items: center;
      vertical-align: middle;
      break-before: page;
      break-after: page;
    }

    /* always insert a page break before the element */
    .page_break_before {
      break-before: page;
    }

    /* always insert a page break after the element */
    .page_break_after {
      break-after: page;
    }

    /* avoid page break before the element */
    .page_break_before_avoid {
      break-before: avoid;
    }

    /* avoid page break after the element */
    .page_break_after_avoid {
      break-after: avoid;
    }

    /* avoid page break inside the element */
    .page_break_inside_avoid {
      break-inside: avoid;
    }
  }

  /* -------------------------------------------------- */
  /* override pandoc css quirks */
  /* -------------------------------------------------- */

  .sourceCode {
    /* prevent unsightly overflow in wide code blocks */
    overflow: auto !important;
  }

  div.sourceCode {
    /* prevent background fill on top-most code block  container */
    background: none !important;
  }

  .sourceCode * {
    /* force consistent line spacing */
    line-height: 1.5 !important;
  }

  div.sourceCode {
    /* style code block margins same as <pre> element */
    margin-top: 20px;
    margin-bottom: 20px;
  }

  /* -------------------------------------------------- */
  /* tablenos */
  /* -------------------------------------------------- */

  /* tablenos wrapper */
  .tablenos {
    /* show scrollbar on tables if necessary to prevent overflow */
    width: 100%;
    margin: 20px 0;
  }

  .tablenos > table {
    /* move margins from table to table_wrapper to allow margin collapsing */
    margin: 0;
  }

  @media only screen {
    /* tablenos wrapper */
    .tablenos {
      /* show scrollbar on tables if necessary to prevent overflow */
      overflow-x: auto !important;
    }

    .tablenos th,
    .tablenos td {
      overflow-wrap: unset !important;
      word-break: unset !important;
    }

    /* table in wrapper */
    .tablenos table,
    .tablenos table * {
      /* don't break table words */
      overflow-wrap: normal !important;
    }
  }
</style>
<!-- 
    Plugin Core

    Functions needed for and shared across all first-party plugins.
-->

<script>
  // get element that is target of hash (from link element or url)
  function getHashTarget(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector(`[id="${id}"]`);
    if (!target) return;

    // if figure or table, modify target to get expected element
    if (id.indexOf("fig:") === 0) target = target.querySelector("figure");
    if (id.indexOf("tbl:") === 0) target = target.querySelector("table");

    return target;
  }

  // get position/dimensions of element or viewport
  function getRectInView(element) {
    let rect = {};
    rect.left = 0;
    rect.top = 0;
    rect.right = document.documentElement.clientWidth;
    rect.bottom = document.documentElement.clientHeight;
    let style = {};

    if (element instanceof HTMLElement) {
      rect = element.getBoundingClientRect();
      style = window.getComputedStyle(element);
    }

    const margin = {};
    margin.left = parseFloat(style.marginLeftWidth) || 0;
    margin.top = parseFloat(style.marginTopWidth) || 0;
    margin.right = parseFloat(style.marginRightWidth) || 0;
    margin.bottom = parseFloat(style.marginBottomWidth) || 0;

    const border = {};
    border.left = parseFloat(style.borderLeftWidth) || 0;
    border.top = parseFloat(style.borderTopWidth) || 0;
    border.right = parseFloat(style.borderRightWidth) || 0;
    border.bottom = parseFloat(style.borderBottomWidth) || 0;

    const newRect = {};
    newRect.left = rect.left + margin.left + border.left;
    newRect.top = rect.top + margin.top + border.top;
    newRect.right = rect.right + margin.right + border.right;
    newRect.bottom = rect.bottom + margin.bottom + border.bottom;
    newRect.width = newRect.right - newRect.left;
    newRect.height = newRect.bottom - newRect.top;

    return newRect;
  }

  // get position of element relative to page
  function getRectInPage(element) {
    const rect = getRectInView(element);
    const body = getRectInView(document.body);

    const newRect = {};
    newRect.left = rect.left - body.left;
    newRect.top = rect.top - body.top;
    newRect.right = rect.right - body.left;
    newRect.bottom = rect.bottom - body.top;
    newRect.width = rect.width;
    newRect.height = rect.height;

    return newRect;
  }

  // get closest element before specified element that matches query
  function firstBefore(element, query) {
    while (element && element !== document.body && !element.matches(query))
      element = element.previousElementSibling || element.parentNode;

    return element;
  }

  // check if element is part of collapsed heading
  function isCollapsed(element) {
    while (element && element !== document.body) {
      if (element.dataset.collapsed === "true") return true;
      element = element.parentNode;
    }
    return false;
  }

  // expand any collapsed parent containers of element if necessary
  function expandElement(element) {
    if (isCollapsed(element)) {
      // accordion plugin
      const heading = firstBefore(element, "h2");
      if (heading) heading.click();
      // details/summary HTML element
      const summary = firstBefore(element, "summary");
      if (summary) summary.click();
    }
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);

    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // get list of elements after a start element up to element matching query
  function nextUntil(element, query, exclude) {
    const elements = [];
    while (((element = element.nextElementSibling), element)) {
      if (element.matches(query)) break;
      if (!element.matches(exclude)) elements.push(element);
    }
    return elements;
  }
</script>
<!--
  Accordion Plugin

  Allows sections of content under h2 headings to be collapsible.
-->

<script type="module">
  // whether to always start expanded ('false'), always start collapsed
  // ('true'), or start collapsed when screen small ('auto')
  const startCollapsed = "auto";

  // start script
  function start() {
    // run through each <h2> heading
    const headings = document.querySelectorAll("h2");
    for (const heading of headings) {
      addArrow(heading);

      // start expanded/collapsed based on option
      if (
        startCollapsed === "true" ||
        (startCollapsed === "auto" && isSmallScreen())
      )
        collapseHeading(heading);
      else expandElement(heading);
    }

    // attach hash change listener to window
    window.addEventListener("hashchange", onHashChange);
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) goToElement(target);
  }

  // add arrow to heading
  function addArrow(heading) {
    // add arrow button
    const arrow = document.createElement("button");
    arrow.innerHTML = document.querySelector(".icon_angle_down").innerHTML;
    arrow.classList.add("icon_button", "accordion_arrow");
    heading.insertBefore(arrow, heading.firstChild);

    // attach click listener to heading and button
    heading.addEventListener("click", onHeadingClick);
    arrow.addEventListener("click", onArrowClick);
  }

  // determine if on mobile-like device with small screen
  function isSmallScreen() {
    return Math.min(window.innerWidth, window.innerHeight) < 480;
  }

  // when <h2> heading is clicked
  function onHeadingClick(event) {
    // only collapse if <h2> itself is target of click (eg, user did
    // not click on anchor within <h2>)
    if (event.target === this) toggleCollapse(this);
  }

  // when arrow button is clicked
  function onArrowClick() {
    toggleCollapse(this.parentNode);
  }

  // collapse section if expanded, expand if collapsed
  function toggleCollapse(heading) {
    if (heading.dataset.collapsed === "false") collapseHeading(heading);
    else expandElement(heading);
  }

  // elements to exclude from collapse, such as table of contents panel,
  // hypothesis panel, etc
  const exclude = "#toc_panel, div.annotator-frame, #lightbox_overlay";

  // collapse section
  function collapseHeading(heading) {
    heading.setAttribute("data-collapsed", "true");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "true");
  }

  // expand section
  function expandElement(heading) {
    heading.setAttribute("data-collapsed", "false");
    const children = getChildren(heading);
    for (const child of children) child.setAttribute("data-collapsed", "false");
  }

  // get list of elements between this <h2> and next <h2> or <h1>
  // ("children" of the <h2> section)
  function getChildren(heading) {
    return nextUntil(heading, "h2, h1", exclude);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
  <!-- modified from: https://fontawesome.com/icons/angle-down -->
  <svg width="16" height="16" viewBox="0 0 448 512">
    <path
      fill="currentColor"
      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* accordion arrow button */
    .accordion_arrow {
      margin-right: 10px;
    }

    /* arrow icon when <h2> data-collapsed attribute true */
    h2[data-collapsed="true"] > .accordion_arrow > svg {
      transform: rotate(-90deg);
    }

    /* all elements (except <h2>'s) when data-collapsed attribute true */
    *:not(h2)[data-collapsed="true"] {
      display: none;
    }

    /* accordion arrow button when hovered and <h2>'s when hovered */
    .accordion_arrow:hover,
    h2[data-collapsed="true"]:hover,
    h2[data-collapsed="false"]:hover {
      cursor: pointer;
    }
  }

  /* always hide accordion arrow button on print */
  @media only print {
    .accordion_arrow {
      display: none;
    }
  }
</style>
<!--
  Anchors Plugin

  Adds an anchor next to each of a certain type of element that provides a
  human-readable url to that specific item/position in the document (e.g.
  "manuscript.html#abstract"). It also makes it such that scrolling out of view
  of a target removes its identifier from the url.
-->

<script type="module">
  // which types of elements to add anchors next to, in "document.querySelector"
  // format
  const typesQuery = 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]';

  // start script
  function start() {
    // add anchor to each element of specified types
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) addAnchor(element);

    // attach scroll listener to window
    window.addEventListener("scroll", onScroll);
  }

  // when window is scrolled
  function onScroll() {
    // if url has hash and user has scrolled out of view of hash
    // target, remove hash from url
    const tolerance = 100;
    const target = getHashTarget();
    if (target) {
      if (
        target.getBoundingClientRect().top > window.innerHeight + tolerance ||
        target.getBoundingClientRect().bottom < 0 - tolerance
      )
        history.pushState(null, null, " ");
    }
  }

  // add anchor to element
  function addAnchor(element) {
    let addTo; // element to add anchor button to

    // if figure or table, modify withId and addTo to get expected
    // elements
    if (element.id.indexOf("fig:") === 0) {
      addTo = element.querySelector("figcaption");
    } else if (element.id.indexOf("tbl:") === 0) {
      addTo = element.querySelector("caption");
    } else if (element.id.indexOf("eq:") === 0) {
      addTo = element.querySelector(".eqnos-number");
    }

    addTo = addTo || element;
    const id = element.id || null;

    // do not add anchor if element doesn't have assigned id.
    // id is generated by pandoc and is assumed to be unique and
    // human-readable
    if (!id) return;

    // create anchor button
    const anchor = document.createElement("a");
    anchor.innerHTML = document.querySelector(".icon_link").innerHTML;
    anchor.title = "Link to this part of the document";
    anchor.classList.add("icon_button", "anchor");
    anchor.dataset.ignore = "true";
    anchor.href = "#" + id;
    addTo.appendChild(anchor);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- link icon -->

<template class="icon_link">
  <!-- modified from: https://fontawesome.com/icons/link -->
  <svg width="16" height="16" viewBox="0 0 512 512">
    <path
      fill="currentColor"
      d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* anchor button */
    .anchor {
      opacity: 0;
      margin-left: 5px;
    }

    /* anchor buttons within <h2>'s */
    h2 .anchor {
      margin-left: 10px;
    }

    /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
    *:hover > .anchor,
    .anchor:hover,
    .anchor:focus {
      opacity: 1;
    }

    /* anchor button when hovered */
    .anchor:hover {
      cursor: pointer;
    }
  }

  /* always show anchor button on devices with no mouse/hover ability */
  @media (hover: none) {
    .anchor {
      opacity: 1;
    }
  }

  /* always hide anchor button on print */
  @media only print {
    .anchor {
      display: none;
    }
  }
</style>
<!-- 
    Attributes Plugin

    Allows arbitrary HTML attributes to be attached to (almost) any element.
    Place an HTML comment inside or next to the desired element with the content:
    $attribute="value"
-->

<script type="module">
  // start script
  function start() {
    // get list of comments in document
    const comments = findComments();

    for (const comment of comments)
      if (comment.parentElement)
        addAttributes(comment.parentElement, comment.nodeValue.trim());
  }

  // add html attributes to specified element based on string of
  // html attributes and values
  function addAttributes(element, text) {
    // regex's for finding attribute/value pairs in the format of
    // attribute="value" or attribute='value
    const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
    const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

    // loop through attribute/value pairs
    let match;
    while ((match = text.match(regex2) || text.match(regex1))) {
      // get attribute and value from regex capture groups
      let attribute = match[1];
      let value = match[2];

      // remove from string
      text = text.substring(match.index + match[0].length);

      if (!attribute || !value) break;

      // set attribute of parent element
      try {
        element.setAttribute(attribute, value);
      } catch (error) {
        console.log(error);
      }

      // special case for colspan
      if (attribute === "colspan") removeTableCells(element, value);
    }
  }

  // get list of comment elements in document
  function findComments() {
    const comments = [];

    // iterate over comment nodes in document
    function acceptNode(node) {
      return NodeFilter.FILTER_ACCEPT;
    }
    const iterator = document.createNodeIterator(
      document.body,
      NodeFilter.SHOW_COMMENT,
      acceptNode
    );
    let node;
    while ((node = iterator.nextNode())) comments.push(node);

    return comments;
  }

  // remove certain number of cells after specified cell
  function removeTableCells(cell, number) {
    number = parseInt(number);
    if (!number) return;

    // remove elements
    for (; number > 1; number--) {
      if (cell.nextElementSibling) cell.nextElementSibling.remove();
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>
<!--
  Jump to First Plugin

  Adds a button next to each reference entry, figure, and table that jumps the
  page to the first occurrence of a link to that item in the manuscript.
-->

<script type="module">
  // whether to add buttons next to reference entries
  const references = "true";
  // whether to add buttons next to figures
  const figures = "true";
  // whether to add buttons next to tables
  const tables = "true";

  // start script
  function start() {
    if (references !== "false") makeButtons("ref-", "p", "reference");
    if (figures !== "false") makeButtons("fig:", "figcaption", "figure");
    if (tables !== "false") makeButtons("tbl:", "caption", "table");
  }

  // when jump button clicked
  function onButtonClick() {
    const first = getFirstOccurrence(this.dataset.id);
    if (!first) return;

    // update url hash so navigating "back" in history will return user to button
    window.location.hash = this.dataset.id;
    // scroll to link
    const timeout = function () {
      goToElement(first, window.innerHeight * 0.5);
    };
    window.setTimeout(timeout, 0);
  }

  // get first occurrence of link to item in document
  function getFirstOccurrence(id) {
    let query = "a";
    query += '[href="#' + id + '"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelector(query);
  }

  // add button next to each reference entry, figure, or table
  function makeButtons(query, containerQuery, subject) {
    const elements = document.querySelectorAll(`[id^="${query}"`);
    for (const element of elements) {
      const id = element.id;
      const buttonContainer = element.querySelector(containerQuery);
      const first = getFirstOccurrence(id);

      // if can't find link to reference or place to put button, ignore
      if (!first || !buttonContainer) continue;

      // make jump button
      let button = document.createElement("button");
      button.classList.add("icon_button", "jump_arrow");
      button.title = `Jump to the first occurrence of this ${subject} in the document`;
      const icon = document.querySelector(".icon_angle_double_up");
      button.innerHTML = icon.innerHTML;
      button.dataset.id = id;
      button.dataset.ignore = "true";
      button.addEventListener("click", onButtonClick);
      buttonContainer.prepend(button);
    }
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
  <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
  <svg width="16" height="16" viewBox="0 0 320 512">
    <path
      fill="currentColor"
      d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* jump button */
    .jump_arrow {
      position: relative;
      top: 0.125em;
      margin-right: 5px;
    }
  }

  /* always hide jump button on print */
  @media only print {
    .jump_arrow {
      display: none;
    }
  }
</style>
<!-- 
    Lightbox Plugin

    Makes it such that when a user clicks on an image, the image fills the
    screen and the user can pan/drag/zoom the image and navigate between other
    images in the document.
-->

<script type="module">
  // list of possible zoom/scale factors
  const zooms =
    "0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1, 1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8";
  // whether to fit image to view ('fit'), display at 100% and shrink if
  // necessary ('shrink'), or always display at 100% ('100')
  const defaultZoom = "fit";
  // whether to zoom in/out toward center of view ('true') or mouse ('false')
  const centerZoom = "false";

  // start script
  function start() {
    // run through each <img> element
    const imgs = document.querySelectorAll("figure > img");
    let count = 1;
    for (const img of imgs) {
      img.classList.add("lightbox_document_img");
      img.dataset.number = count;
      img.dataset.total = imgs.length;
      img.addEventListener("click", openLightbox);
      count++;
    }

    // attach mouse and key listeners to window
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("keyup", onKeyUp);
  }

  // when mouse is moved anywhere in window
  function onWindowMouseMove(event) {
    window.mouseX = event.clientX;
    window.mouseY = event.clientY;
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("lightbox_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("lightbox_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeLightbox();
        break;
    }
  }

  // open lightbox
  function openLightbox() {
    const lightbox = makeLightbox(this);
    if (!lightbox) return;

    blurBody(lightbox);
    document.body.appendChild(lightbox);
  }

  // make lightbox
  function makeLightbox(img) {
    // delete lightbox if it exists, start fresh
    closeLightbox();

    // create screen overlay containing lightbox
    const overlay = document.createElement("div");
    overlay.id = "lightbox_overlay";

    // create image info boxes
    const numberInfo = document.createElement("div");
    const zoomInfo = document.createElement("div");
    numberInfo.id = "lightbox_number_info";
    zoomInfo.id = "lightbox_zoom_info";

    // create container for image
    const imageContainer = document.createElement("div");
    imageContainer.id = "lightbox_image_container";
    const lightboxImg = makeLightboxImg(
      img,
      imageContainer,
      numberInfo,
      zoomInfo
    );
    imageContainer.appendChild(lightboxImg);

    // create bottom container for caption and navigation buttons
    const bottomContainer = document.createElement("div");
    bottomContainer.id = "lightbox_bottom_container";
    const caption = makeCaption(img);
    const prevButton = makePrevButton(img);
    const nextButton = makeNextButton(img);
    bottomContainer.appendChild(prevButton);
    bottomContainer.appendChild(caption);
    bottomContainer.appendChild(nextButton);

    // attach top middle and bottom to overlay
    overlay.appendChild(numberInfo);
    overlay.appendChild(zoomInfo);
    overlay.appendChild(imageContainer);
    overlay.appendChild(bottomContainer);

    return overlay;
  }

  // make <img> object that is intuitively draggable and zoomable
  function makeLightboxImg(sourceImg, container, numberInfoBox, zoomInfoBox) {
    // create copy of source <img>
    const img = sourceImg.cloneNode(true);
    img.classList.remove("lightbox_document_img");
    img.removeAttribute("id");
    img.removeAttribute("width");
    img.removeAttribute("height");
    img.style.position = "unset";
    img.style.margin = "0";
    img.style.padding = "0";
    img.style.width = "";
    img.style.height = "";
    img.style.minWidth = "";
    img.style.minHeight = "";
    img.style.maxWidth = "";
    img.style.maxHeight = "";
    img.id = "lightbox_img";

    // build sorted list of zoomSteps
    const zoomSteps = zooms.split(/[^0-9.]/).map((step) => parseFloat(step));
    zoomSteps.sort((a, b) => a - b);

    // <img> object property variables
    let zoom = 1;
    let translateX = 0;
    let translateY = 0;
    let clickMouseX = undefined;
    let clickMouseY = undefined;
    let clickTranslateX = undefined;
    let clickTranslateY = undefined;

    updateNumberInfo();

    // update image numbers displayed in info box
    function updateNumberInfo() {
      numberInfoBox.innerHTML =
        sourceImg.dataset.number + " of " + sourceImg.dataset.total;
    }

    // update zoom displayed in info box
    function updateZoomInfo() {
      let zoomInfo = zoom * 100;
      if (!Number.isInteger(zoomInfo)) zoomInfo = zoomInfo.toFixed(2);
      zoomInfoBox.innerHTML = zoomInfo + "%";
    }

    // move to closest zoom step above current zoom
    const zoomIn = function () {
      for (const zoomStep of zoomSteps) {
        if (zoomStep > zoom) {
          zoom = zoomStep;
          break;
        }
      }
      updateTransform();
    };

    // move to closest zoom step above current zoom
    const zoomOut = function () {
      zoomSteps.reverse();
      for (const zoomStep of zoomSteps) {
        if (zoomStep < zoom) {
          zoom = zoomStep;
          break;
        }
      }
      zoomSteps.reverse();

      updateTransform();
    };

    // update display of <img> based on scale/translate properties
    const updateTransform = function () {
      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      // get new width/height after scale
      const rect = img.getBoundingClientRect();
      // limit translate
      translateX = Math.max(translateX, -rect.width / 2);
      translateX = Math.min(translateX, rect.width / 2);
      translateY = Math.max(translateY, -rect.height / 2);
      translateY = Math.min(translateY, rect.height / 2);

      // set transform
      img.style.transform =
        "translate(" +
        (translateX || 0) +
        "px," +
        (translateY || 0) +
        "px) scale(" +
        (zoom || 1) +
        ")";

      updateZoomInfo();
    };

    // fit <img> to container
    const fit = function () {
      // no x/y offset, 100% zoom by default
      translateX = 0;
      translateY = 0;
      zoom = 1;

      // widths of <img> and container
      const imgWidth = img.naturalWidth;
      const imgHeight = img.naturalHeight;
      const containerWidth = parseFloat(
        window.getComputedStyle(container).width
      );
      const containerHeight = parseFloat(
        window.getComputedStyle(container).height
      );

      // how much zooming is needed to fit <img> to container
      const xRatio = imgWidth / containerWidth;
      const yRatio = imgHeight / containerHeight;
      const maxRatio = Math.max(xRatio, yRatio);
      const newZoom = 1 / maxRatio;

      // fit <img> to container according to option
      if (defaultZoom === "shrink") {
        if (maxRatio > 1) zoom = newZoom;
      } else if (defaultZoom === "fit") zoom = newZoom;

      updateTransform();
    };

    // when mouse wheel is rolled anywhere in container
    const onContainerWheel = function (event) {
      if (!event) return;

      // let ctrl + mouse wheel to zoom behave as normal
      if (event.ctrlKey) return;

      // prevent normal scroll behavior
      event.preventDefault();
      event.stopPropagation();

      // point around which to scale img
      const viewRect = container.getBoundingClientRect();
      const viewX = (viewRect.left + viewRect.right) / 2;
      const viewY = (viewRect.top + viewRect.bottom) / 2;
      const originX = centerZoom === "true" ? viewX : mouseX;
      const originY = centerZoom === "true" ? viewY : mouseY;

      // get point on image under origin
      const oldRect = img.getBoundingClientRect();
      const oldPercentX = (originX - oldRect.left) / oldRect.width;
      const oldPercentY = (originY - oldRect.top) / oldRect.height;

      // increment/decrement zoom
      if (event.deltaY < 0) zoomIn();
      if (event.deltaY > 0) zoomOut();

      // get offset between previous image point and origin
      const newRect = img.getBoundingClientRect();
      const offsetX = originX - (newRect.left + newRect.width * oldPercentX);
      const offsetY = originY - (newRect.top + newRect.height * oldPercentY);

      // translate image to keep image point under origin
      translateX += offsetX;
      translateY += offsetY;

      // perform translate
      updateTransform();
    };

    // when container is clicked
    function onContainerClick(event) {
      // if container itself is target of click, and not other
      // element above it
      if (event.target === this) closeLightbox();
    }

    // when mouse button is pressed on image
    const onImageMouseDown = function (event) {
      // store original mouse position relative to image
      clickMouseX = window.mouseX;
      clickMouseY = window.mouseY;
      clickTranslateX = translateX;
      clickTranslateY = translateY;
      event.stopPropagation();
      event.preventDefault();
    };

    // when mouse button is released anywhere in window
    const onWindowMouseUp = function (event) {
      // reset original mouse position
      clickMouseX = undefined;
      clickMouseY = undefined;
      clickTranslateX = undefined;
      clickTranslateY = undefined;

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mouseup", onWindowMouseUp);
    };

    // when mouse is moved anywhere in window
    const onWindowMouseMove = function (event) {
      if (
        clickMouseX === undefined ||
        clickMouseY === undefined ||
        clickTranslateX === undefined ||
        clickTranslateY === undefined
      )
        return;

      // offset image based on original and current mouse position
      translateX = clickTranslateX + window.mouseX - clickMouseX;
      translateY = clickTranslateY + window.mouseY - clickMouseY;
      updateTransform();
      event.preventDefault();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("mousemove", onWindowMouseMove);
    };

    // when window is resized
    const onWindowResize = function (event) {
      fit();

      // remove global listener if lightbox removed from document
      if (!document.body.contains(container))
        window.removeEventListener("resize", onWindowResize);
    };

    // attach the necessary event listeners
    img.addEventListener("dblclick", fit);
    img.addEventListener("mousedown", onImageMouseDown);
    container.addEventListener("wheel", onContainerWheel);
    container.addEventListener("mousedown", onContainerClick);
    container.addEventListener("touchstart", onContainerClick);
    window.addEventListener("mouseup", onWindowMouseUp);
    window.addEventListener("mousemove", onWindowMouseMove);
    window.addEventListener("resize", onWindowResize);

    // run fit() after lightbox atttached to document and <img> Loaded
    // so needed container and img dimensions available
    img.addEventListener("load", fit);

    return img;
  }

  // make caption
  function makeCaption(img) {
    const caption = document.createElement("div");
    caption.id = "lightbox_caption";
    const captionSource = img.nextElementSibling;
    if (captionSource.tagName.toLowerCase() === "figcaption") {
      const captionCopy = makeCopy(captionSource);
      caption.innerHTML = captionCopy.innerHTML;
    }

    caption.addEventListener("touchstart", function (event) {
      event.stopPropagation();
    });

    return caption;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // make button to jump to previous image in document
  function makePrevButton(img) {
    const prevButton = document.createElement("button");
    prevButton.id = "lightbox_prev_button";
    prevButton.title = "Jump to the previous image in the document [←]";
    prevButton.classList.add("icon_button", "lightbox_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;

    // attach click listeners to button
    prevButton.addEventListener("click", function () {
      getPrevImg(img).click();
    });

    return prevButton;
  }

  // make button to jump to next image in document
  function makeNextButton(img) {
    const nextButton = document.createElement("button");
    nextButton.id = "lightbox_next_button";
    nextButton.title = "Jump to the next image in the document [→]";
    nextButton.classList.add("icon_button", "lightbox_button");
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;

    // attach click listeners to button
    nextButton.addEventListener("click", function () {
      getNextImg(img).click();
    });

    return nextButton;
  }

  // get previous image in document
  function getPrevImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if < 1
    if (index - 1 >= 0) index--;
    else index = imgs.length - 1;
    return imgs[index];
  }

  // get next image in document
  function getNextImg(img) {
    const imgs = document.querySelectorAll(".lightbox_document_img");

    // find index of provided img
    let index;
    for (index = 0; index < imgs.length; index++) {
      if (imgs[index] === img) break;
    }

    // wrap index to other side if > total
    if (index + 1 <= imgs.length - 1) index++;
    else index = 0;
    return imgs[index];
  }

  // close lightbox
  function closeLightbox() {
    focusBody();

    const lightbox = document.getElementById("lightbox_overlay");
    if (lightbox) lightbox.remove();
  }

  // make all elements behind lightbox non-focusable
  function blurBody(overlay) {
    const all = document.querySelectorAll("*");
    for (const element of all) element.tabIndex = -1;
    document.body.classList.add("body_no_scroll");
  }

  // make all elements focusable again
  function focusBody() {
    const all = document.querySelectorAll("*");
    for (const element of all) element.removeAttribute("tabIndex");
    document.body.classList.remove("body_no_scroll");
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* regular <img> in document when hovered */
    img.lightbox_document_img:hover {
      cursor: pointer;
    }

    .body_no_scroll {
      overflow: hidden !important;
    }

    /* screen overlay */
    #lightbox_overlay {
      display: flex;
      flex-direction: column;
      position: fixed;
      left: 0;
      top: 0;
      right: 0;
      bottom: 0;
      background: rgba(0, 0, 0, 0.75);
      z-index: 3;
    }

    /* middle area containing lightbox image */
    #lightbox_image_container {
      flex-grow: 1;
      display: flex;
      justify-content: center;
      align-items: center;
      overflow: hidden;
      position: relative;
      padding: 20px;
    }

    /* bottom area containing caption */
    #lightbox_bottom_container {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 100px;
      min-height: 100px;
      max-height: 100px;
      background: rgba(0, 0, 0, 0.5);
    }

    /* image number info text box */
    #lightbox_number_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      left: 2px;
      top: 0;
      z-index: 4;
    }

    /* zoom info text box */
    #lightbox_zoom_info {
      position: absolute;
      color: #ffffff;
      font-weight: 600;
      right: 2px;
      top: 0;
      z-index: 4;
    }

    /* copy of image caption */
    #lightbox_caption {
      box-sizing: border-box;
      display: inline-block;
      width: 100%;
      max-height: 100%;
      padding: 10px 0;
      text-align: center;
      overflow-y: auto;
      color: #ffffff;
    }

    /* navigation previous/next button */
    .lightbox_button {
      width: 100px;
      height: 100%;
      min-width: 100px;
      min-height: 100%;
      color: #ffffff;
    }

    /* navigation previous/next button when hovered */
    .lightbox_button:hover {
      background: none !important;
    }

    /* navigation button icon */
    .lightbox_button > svg {
      height: 25px;
    }

    /* figure auto-number */
    #lightbox_caption > span:first-of-type {
      font-weight: bold;
      margin-right: 5px;
    }

    /* lightbox image when hovered */
    #lightbox_img:hover {
      cursor: grab;
    }

    /* lightbox image when grabbed */
    #lightbox_img:active {
      cursor: grabbing;
    }
  }

  /* when on screen < 480px wide */
  @media only screen and (max-width: 480px) {
    /* make navigation buttons skinnier on small screens to make more room for caption text */
    .lightbox_button {
      width: 50px;
      min-width: 50px;
    }
  }

  /* always hide lightbox on print */
  @media only print {
    #lightbox_overlay {
      display: none;
    }
  }
</style>
<!-- 
  Link Highlight Plugin

  Makes it such that when a user hovers or focuses a link, other links that have
  the same target will be highlighted. It also makes it such that when clicking
  a link, the target of the link (eg reference, figure, table) is briefly
  highlighted.
-->

<script type="module">
  // whether to also highlight links that go to external urls
  const externalLinks = "false";
  // whether user must click off to unhighlight instead of just
  // un-hovering
  const clickUnhighlight = "false";
  // whether to also highlight links that are unique
  const highlightUnique = "true";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach mouse and focus listeners to link
      link.addEventListener("mouseenter", onLinkFocus);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("mouseleave", onLinkUnhover);
    }

    // attach click and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("hashchange", onHashChange);

    // run hash change on window load in case user has navigated
    // directly to hash
    onHashChange();
  }

  // when link is focused (tabbed to) or hovered
  function onLinkFocus() {
    highlight(this);
  }

  // when link is unhovered
  function onLinkUnhover() {
    if (clickUnhighlight !== "true") unhighlightAll();
  }

  // when the mouse is clicked anywhere in window
  function onClick(event) {
    unhighlightAll();
  }

  // when hash (eg manuscript.html#introduction) changes
  function onHashChange() {
    const target = getHashTarget();
    if (target) glowElement(target);
  }

  // start glow sequence on an element
  function glowElement(element) {
    const startGlow = function () {
      onGlowEnd();
      element.dataset.glow = "true";
      element.addEventListener("animationend", onGlowEnd);
    };
    const onGlowEnd = function () {
      element.removeAttribute("data-glow");
      element.removeEventListener("animationend", onGlowEnd);
    };
    startGlow();
  }

  // highlight link and all others with same target
  function highlight(link) {
    // force unhighlight all to start fresh
    unhighlightAll();

    // get links with same target
    if (!link) return;
    const sameLinks = getSameLinks(link);

    // if link unique and option is off, exit and don't highlight
    if (sameLinks.length <= 1 && highlightUnique !== "true") return;

    // highlight all same links, and "select" (special highlight) this
    // one
    for (const sameLink of sameLinks) {
      if (sameLink === link) sameLink.setAttribute("data-selected", "true");
      else sameLink.setAttribute("data-highlighted", "true");
    }
  }

  // unhighlight all links
  function unhighlightAll() {
    const links = getLinks();
    for (const link of links) {
      link.setAttribute("data-selected", "false");
      link.setAttribute("data-highlighted", "false");
    }
  }

  // get links with same target
  function getSameLinks(link) {
    const results = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        results.push(otherLink);
    }
    return results;
  }

  // get all links of types we wish to handle
  function getLinks() {
    let query = "a";
    if (externalLinks !== "true") query += '[href^="#"]';
    // exclude buttons, anchor links, toc links, etc
    query += ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    return document.querySelectorAll(query);
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  @media only screen {
    /* anything with data-highlighted attribute true */
    [data-highlighted="true"] {
      background: #ffeb3b;
    }

    /* anything with data-selected attribute true */
    [data-selected="true"] {
      background: #ff8a65 !important;
    }

    /* animation definition for glow */
    @keyframes highlight_glow {
      0% {
        background: none;
      }
      10% {
        background: #bbdefb;
      }
      100% {
        background: none;
      }
    }

    /* anything with data-glow attribute true */
    [data-glow="true"] {
      animation: highlight_glow 2s;
    }
  }
</style>
<!--
  Table of Contents Plugin

  Provides a "table of contents" (toc) panel on the side of the document that
  allows the user to conveniently navigate between sections of the document.
-->

<script type="module">
  // which types of elements to add links for, in "document.querySelector" format
  const typesQuery = "h1, h2, h3";
  // whether toc starts open. use 'true' or 'false', or 'auto' to
  // use 'true' behavior when screen wide enough and 'false' when not
  const startOpen = "false";
  // whether toc closes when clicking on toc link. use 'true' or
  // 'false', or 'auto' to use 'false' behavior when screen wide
  // enough and 'true' when not
  const clickClose = "auto";
  // if list item is more than this many characters, text will be
  // truncated
  const charLimit = "50";
  // whether or not to show bullets next to each toc item
  const bullets = "false";

  // start script
  function start() {
    // make toc panel and populate with entries (links to document
    // sections)
    const panel = makePanel();
    if (!panel) return;
    makeEntries(panel);
    // attach panel to document after making entries, so 'toc' heading
    // in panel isn't included in toc
    document.body.insertBefore(panel, document.body.firstChild);

    // initial panel state
    if (startOpen === "true" || (startOpen === "auto" && !isSmallScreen()))
      openPanel();
    else closePanel();

    // attach click, scroll, and hash change listeners to window
    window.addEventListener("click", onClick);
    window.addEventListener("scroll", onScroll);
    window.addEventListener("hashchange", onScroll);
    window.addEventListener("keyup", onKeyUp);
    onScroll();

    // add class to push document body down out of way of toc button
    document.body.classList.add("toc_body_nudge");
  }

  // determine if screen wide enough to fit toc panel
  function isSmallScreen() {
    // in default theme:
    // 816px = 8.5in = width of "page" (<body>) element
    // 260px = min width of toc panel (*2 for both sides of <body>)
    return window.innerWidth < 816 + 260 * 2;
  }

  // when mouse is clicked anywhere in window
  function onClick() {
    if (isSmallScreen()) closePanel();
  }

  // when window is scrolled or hash changed
  function onScroll() {
    highlightViewed();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    // close on esc
    if (event.key === "Escape") closePanel();
  }

  // find entry of currently viewed document section in toc and highlight
  function highlightViewed() {
    const firstId = getFirstInView(typesQuery);

    // get toc entries (links), unhighlight all, then highlight viewed
    const list = document.getElementById("toc_list");
    if (!firstId || !list) return;
    const links = list.querySelectorAll("a");
    for (const link of links) link.dataset.viewing = "false";
    const link = list.querySelector('a[href="#' + firstId + '"]');
    if (!link) return;
    link.dataset.viewing = "true";
  }

  // get first or previous toc listed element in top half of view
  function getFirstInView(query) {
    // get all elements matching query and with id
    const elements = document.querySelectorAll(query);
    const elementsWithIds = [];
    for (const element of elements) {
      if (element.id) elementsWithIds.push(element);
    }

    // get first or previous element in top half of view
    for (let i = 0; i < elementsWithIds.length; i++) {
      const element = elementsWithIds[i];
      const prevElement = elementsWithIds[Math.max(0, i - 1)];
      if (element.getBoundingClientRect().top >= 0) {
        if (element.getBoundingClientRect().top < window.innerHeight / 2)
          return element.id;
        else return prevElement.id;
      }
    }
  }

  // make panel
  function makePanel() {
    // create panel
    const panel = document.createElement("div");
    panel.id = "toc_panel";
    if (bullets === "true") panel.dataset.bullets = "true";

    // create header
    const header = document.createElement("div");
    header.id = "toc_header";

    // create toc button
    const button = document.createElement("button");
    button.id = "toc_button";
    button.innerHTML = document.querySelector(".icon_th_list").innerHTML;
    button.title = "Table of Contents";
    button.classList.add("icon_button");

    // create header text
    const text = document.createElement("h4");
    text.innerHTML = "Table of Contents";

    // create container for toc list
    const list = document.createElement("div");
    list.id = "toc_list";

    // attach click listeners
    panel.addEventListener("click", onPanelClick);
    header.addEventListener("click", onHeaderClick);
    button.addEventListener("click", onButtonClick);

    // attach elements
    header.appendChild(button);
    header.appendChild(text);
    panel.appendChild(header);
    panel.appendChild(list);

    return panel;
  }

  // create toc entries (links) to each element of the specified types
  function makeEntries(panel) {
    const elements = document.querySelectorAll(typesQuery);
    for (const element of elements) {
      // do not add link if element doesn't have assigned id
      if (!element.id) continue;

      // create link/list item
      const link = document.createElement("a");
      link.classList.add("toc_link");
      switch (element.tagName.toLowerCase()) {
        case "h1":
          link.dataset.level = "1";
          break;
        case "h2":
          link.dataset.level = "2";
          break;
        case "h3":
          link.dataset.level = "3";
          break;
        case "h4":
          link.dataset.level = "4";
          break;
      }
      link.title = element.innerText;
      let text = element.innerText;
      if (text.length > charLimit) text = text.slice(0, charLimit) + "...";
      link.innerHTML = text;
      link.href = "#" + element.id;
      link.addEventListener("click", onLinkClick);

      // attach link
      panel.querySelector("#toc_list").appendChild(link);
    }
  }

  // when panel is clicked
  function onPanelClick(event) {
    // stop click from propagating to window/document and closing panel
    event.stopPropagation();
  }

  // when header itself is clicked
  function onHeaderClick(event) {
    togglePanel();
  }

  // when button is clicked
  function onButtonClick(event) {
    togglePanel();
    // stop header underneath button from also being clicked
    event.stopPropagation();
  }

  // when link is clicked
  function onLinkClick(event) {
    if (clickClose === "true" || (clickClose === "auto" && isSmallScreen()))
      closePanel();
    else openPanel();
  }

  // open panel if closed, close if opened
  function togglePanel() {
    const panel = document.getElementById("toc_panel");
    if (!panel) return;

    if (panel.dataset.open === "true") closePanel();
    else openPanel();
  }

  // open panel
  function openPanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "true";
  }

  // close panel
  function closePanel() {
    const panel = document.getElementById("toc_panel");
    if (panel) panel.dataset.open = "false";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- th list icon -->

<template class="icon_th_list">
  <!-- modified from: https://fontawesome.com/icons/th-list -->
  <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
    <path
      fill="currentColor"
      d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* toc panel */
    #toc_panel {
      box-sizing: border-box;
      position: fixed;
      top: 0;
      left: 0;
      background: #ffffff;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      z-index: 2;
    }

    /* toc panel when closed */
    #toc_panel[data-open="false"] {
      min-width: 60px;
      width: 60px;
      height: 60px;
      border-right: solid 1px #bdbdbd;
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc panel when open */
    #toc_panel[data-open="true"] {
      min-width: 260px;
      max-width: 480px;
      /* keep panel edge consistent distance away from "page" edge */
      width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
      bottom: 0;
      border-right: solid 1px #bdbdbd;
    }

    /* toc panel header */
    #toc_header {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      height: 60px;
      margin: 0;
      padding: 20px;
    }

    /* toc panel header when hovered */
    #toc_header:hover {
      cursor: pointer;
    }

    /* toc panel header when panel open */
    #toc_panel[data-open="true"] > #toc_header {
      border-bottom: solid 1px #bdbdbd;
    }

    /* toc open/close header button */
    #toc_button {
      margin-right: 20px;
    }

    /* hide toc list and header text when closed */
    #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
    #toc_panel[data-open="false"] > #toc_list {
      display: none;
    }

    /* toc list of entries */
    #toc_list {
      box-sizing: border-box;
      width: 100%;
      padding: 20px;
      position: absolute;
      top: calc(60px + 1px);
      bottom: 0;
      overflow: auto;
    }

    /* toc entry, link to section in document */
    .toc_link {
      display: block;
      padding: 5px;
      position: relative;
      font-weight: 600;
      text-decoration: none;
    }

    /* toc entry when hovered or when "viewed" */
    .toc_link:hover,
    .toc_link[data-viewing="true"] {
      background: #f5f5f5;
    }

    /* toc entry, level 1 indentation */
    .toc_link[data-level="1"] {
      margin-left: 0;
    }

    /* toc entry, level 2 indentation */
    .toc_link[data-level="2"] {
      margin-left: 20px;
    }

    /* toc entry, level 3 indentation */
    .toc_link[data-level="3"] {
      margin-left: 40px;
    }

    /* toc entry, level 4 indentation */
    .toc_link[data-level="4"] {
      margin-left: 60px;
    }

    /* toc entry bullets */
    #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
      position: absolute;
      left: -15px;
      top: -1px;
      font-size: 1.5em;
    }

    /* toc entry, level 2 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
      content: "\2022";
    }

    /* toc entry, level 3 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
      content: "\25AB";
    }

    /* toc entry, level 4 bullet */
    #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
      content: "-";
    }
  }

  /* when on screen < 8.5in wide */
  @media only screen and (max-width: 8.5in) {
    /* push <body> ("page") element down to make room for toc icon */
    .toc_body_nudge {
      padding-top: 60px;
    }

    /* toc icon when panel closed and not hovered */
    #toc_panel[data-open="false"]:not(:hover) {
      background: rgba(255, 255, 255, 0.75);
    }
  }

  /* always hide toc panel on print */
  @media only print {
    #toc_panel {
      display: none;
    }
  }
</style>
<!-- 
  Tooltips Plugin

  Makes it such that when the user hovers or focuses a link to a citation or
  figure, a tooltip appears with a preview of the reference content, along with
  arrows to navigate between instances of the same reference in the document.
-->

<script type="module">
  // whether user must click off to close tooltip instead of just un-hovering
  const clickClose = "false";
  // delay (in ms) between opening and closing tooltip
  const delay = "100";

  // start script
  function start() {
    const links = getLinks();
    for (const link of links) {
      // attach hover and focus listeners to link
      link.addEventListener("mouseover", onLinkHover);
      link.addEventListener("mouseleave", onLinkUnhover);
      link.addEventListener("focus", onLinkFocus);
      link.addEventListener("touchend", onLinkTouch);
    }

    // attach mouse, key, and resize listeners to window
    window.addEventListener("mousedown", onClick);
    window.addEventListener("touchstart", onClick);
    window.addEventListener("keyup", onKeyUp);
    window.addEventListener("resize", onResize);
  }

  // when link is hovered
  function onLinkHover() {
    // function to open tooltip
    const delayOpenTooltip = function () {
      openTooltip(this);
    }.bind(this);

    // run open function after delay
    this.openTooltipTimer = window.setTimeout(delayOpenTooltip, delay);
  }

  // when mouse leaves link
  function onLinkUnhover() {
    // cancel opening tooltip
    window.clearTimeout(this.openTooltipTimer);

    // don't close on unhover if option specifies
    if (clickClose === "true") return;

    // function to close tooltip
    const delayCloseTooltip = function () {
      // if tooltip open and if mouse isn't over tooltip, close
      const tooltip = document.getElementById("tooltip");
      if (tooltip && !tooltip.matches(":hover")) closeTooltip();
    };

    // run close function after delay
    this.closeTooltipTimer = window.setTimeout(delayCloseTooltip, delay);
  }

  // when link is focused (tabbed to)
  function onLinkFocus(event) {
    openTooltip(this);
  }

  // when link is touched on touch screen
  function onLinkTouch(event) {
    // attempt to force hover state on first tap always, and trigger
    // regular link click (and navigation) on second tap
    if (event.target === document.activeElement) event.target.click();
    else {
      document.activeElement.blur();
      event.target.focus();
    }
    if (event.cancelable) event.preventDefault();
    event.stopPropagation();
    return false;
  }

  // when mouse is clicked anywhere in window
  function onClick(event) {
    closeTooltip();
  }

  // when key pressed
  function onKeyUp(event) {
    if (!event || !event.key) return;

    switch (event.key) {
      // trigger click of prev button
      case "ArrowLeft":
        const prevButton = document.getElementById("tooltip_prev_button");
        if (prevButton) prevButton.click();
        break;
      // trigger click of next button
      case "ArrowRight":
        const nextButton = document.getElementById("tooltip_next_button");
        if (nextButton) nextButton.click();
        break;
      // close on esc
      case "Escape":
        closeTooltip();
        break;
    }
  }

  // when window is resized or zoomed
  function onResize() {
    closeTooltip();
  }

  // get all links of types we wish to handle
  function getLinks() {
    const queries = [];
    // exclude buttons, anchor links, toc links, etc
    const exclude =
      ":not(.button):not(.icon_button):not(.anchor):not(.toc_link)";
    queries.push('a[href^="#ref-"]' + exclude); // citation links
    queries.push('a[href^="#fig:"]' + exclude); // figure links
    const query = queries.join(", ");
    return document.querySelectorAll(query);
  }

  // get links with same target, get index of link in set, get total
  // same links
  function getSameLinks(link) {
    const sameLinks = [];
    const links = getLinks();
    for (const otherLink of links) {
      if (otherLink.getAttribute("href") === link.getAttribute("href"))
        sameLinks.push(otherLink);
    }

    return {
      elements: sameLinks,
      index: sameLinks.indexOf(link),
      total: sameLinks.length,
    };
  }

  // open tooltip
  function openTooltip(link) {
    // delete tooltip if it exists, start fresh
    closeTooltip();

    // make tooltip element
    const tooltip = makeTooltip(link);

    // if source couldn't be found and tooltip not made, exit
    if (!tooltip) return;

    // make navbar elements
    const navBar = makeNavBar(link);
    if (navBar) tooltip.firstElementChild.appendChild(navBar);

    // attach tooltip to page
    document.body.appendChild(tooltip);

    // position tooltip
    const position = function () {
      positionTooltip(link);
    };
    position();

    // if tooltip contains images, position again after they've loaded
    const imgs = tooltip.querySelectorAll("img");
    for (const img of imgs) img.addEventListener("load", position);
  }

  // close (delete) tooltip
  function closeTooltip() {
    const tooltip = document.getElementById("tooltip");
    if (tooltip) tooltip.remove();
  }

  // make tooltip
  function makeTooltip(link) {
    // get target element that link points to
    const source = getSource(link);

    // if source can't be found, exit
    if (!source) return;

    // create new tooltip
    const tooltip = document.createElement("div");
    tooltip.id = "tooltip";
    const tooltipContent = document.createElement("div");
    tooltipContent.id = "tooltip_content";
    tooltip.appendChild(tooltipContent);

    // make copy of source node and put in tooltip
    const sourceCopy = makeCopy(source);
    tooltipContent.appendChild(sourceCopy);

    // attach mouse event listeners
    tooltip.addEventListener("click", onTooltipClick);
    tooltip.addEventListener("mousedown", onTooltipClick);
    tooltip.addEventListener("touchstart", onTooltipClick);
    tooltip.addEventListener("mouseleave", onTooltipUnhover);

    // (for interaction with lightbox plugin)
    // transfer click on tooltip copied img to original img
    const sourceImg = source.querySelector("img");
    const sourceCopyImg = sourceCopy.querySelector("img");
    if (sourceImg && sourceCopyImg) {
      const clickImg = function () {
        sourceImg.click();
        closeTooltip();
      };
      sourceCopyImg.addEventListener("click", clickImg);
    }

    return tooltip;
  }

  // make carbon copy of html dom element
  function makeCopy(source) {
    const sourceCopy = source.cloneNode(true);

    // delete elements marked with ignore (eg anchor and jump buttons)
    const deleteFromCopy = sourceCopy.querySelectorAll('[data-ignore="true"]');
    for (const element of deleteFromCopy) element.remove();

    // delete certain element attributes
    const attributes = [
      "id",
      "data-collapsed",
      "data-selected",
      "data-highlighted",
      "data-glow",
    ];
    for (const attribute of attributes) {
      sourceCopy.removeAttribute(attribute);
      const elements = sourceCopy.querySelectorAll("[" + attribute + "]");
      for (const element of elements) element.removeAttribute(attribute);
    }

    return sourceCopy;
  }

  // when tooltip is clicked
  function onTooltipClick(event) {
    // when user clicks on tooltip, stop click from transferring
    // outside of tooltip (eg, click off to close tooltip, or eg click
    // off to unhighlight same refs)
    event.stopPropagation();
  }

  // when tooltip is unhovered
  function onTooltipUnhover(event) {
    if (clickClose === "true") return;

    // make sure new mouse/touch/focus no longer over tooltip or any
    // element within it
    const tooltip = document.getElementById("tooltip");
    if (!tooltip) return;
    if (this.contains(event.relatedTarget)) return;

    closeTooltip();
  }

  // make nav bar to go betwen prev/next instances of same reference
  function makeNavBar(link) {
    // find other links to the same source
    const sameLinks = getSameLinks(link);

    // don't show nav bar when singular reference
    if (sameLinks.total <= 1) return;

    // find prev/next links with same target
    const prevLink = getPrevLink(link, sameLinks);
    const nextLink = getNextLink(link, sameLinks);

    // create nav bar
    const navBar = document.createElement("div");
    navBar.id = "tooltip_nav_bar";
    const text = sameLinks.index + 1 + " of " + sameLinks.total;

    // create nav bar prev/next buttons
    const prevButton = document.createElement("button");
    const nextButton = document.createElement("button");
    prevButton.id = "tooltip_prev_button";
    nextButton.id = "tooltip_next_button";
    prevButton.title =
      "Jump to the previous occurence of this item in the document [←]";
    nextButton.title =
      "Jump to the next occurence of this item in the document [→]";
    prevButton.classList.add("icon_button");
    nextButton.classList.add("icon_button");
    prevButton.innerHTML = document.querySelector(".icon_caret_left").innerHTML;
    nextButton.innerHTML = document.querySelector(
      ".icon_caret_right"
    ).innerHTML;
    navBar.appendChild(prevButton);
    navBar.appendChild(document.createTextNode(text));
    navBar.appendChild(nextButton);

    // attach click listeners to buttons
    prevButton.addEventListener("click", function () {
      onPrevNextClick(link, prevLink);
    });
    nextButton.addEventListener("click", function () {
      onPrevNextClick(link, nextLink);
    });

    return navBar;
  }

  // get previous link with same target
  function getPrevLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if < 1
    let index;
    if (sameLinks.index - 1 >= 0) index = sameLinks.index - 1;
    else index = sameLinks.total - 1;
    return sameLinks.elements[index];
  }

  // get next link with same target
  function getNextLink(link, sameLinks) {
    if (!sameLinks) sameLinks = getSameLinks(link);
    // wrap index to other side if > total
    let index;
    if (sameLinks.index + 1 <= sameLinks.total - 1) index = sameLinks.index + 1;
    else index = 0;
    return sameLinks.elements[index];
  }

  // get element that is target of link or url hash
  function getSource(link) {
    const hash = link ? link.hash : window.location.hash;
    const id = hash.slice(1);
    let target = document.querySelector('[id="' + id + '"]');
    if (!target) return;

    // if ref or figure, modify target to get expected element
    if (id.indexOf("ref-") === 0) target = target.querySelector("p");
    else if (id.indexOf("fig:") === 0) target = target.querySelector("figure");

    return target;
  }

  // when prev/next arrow button is clicked
  function onPrevNextClick(link, prevNextLink) {
    if (link && prevNextLink)
      goToElement(prevNextLink, window.innerHeight * 0.5);
  }

  // scroll to and focus element
  function goToElement(element, offset) {
    // expand accordion section if collapsed
    expandElement(element);
    const y =
      getRectInView(element).top -
      getRectInView(document.documentElement).top -
      (offset || 0);
    // trigger any function listening for "onscroll" event
    window.dispatchEvent(new Event("scroll"));
    window.scrollTo(0, y);
    document.activeElement.blur();
    element.focus();
  }

  // determine position to place tooltip based on link position in
  // viewport and tooltip size
  function positionTooltip(link, left, top) {
    const tooltipElement = document.getElementById("tooltip");
    if (!tooltipElement) return;

    // get convenient vars for position/dimensions of
    // link/tooltip/page/view
    link = getRectInPage(link);
    const tooltip = getRectInPage(tooltipElement);
    const view = getRectInPage();

    // horizontal positioning
    if (left)
      // use explicit value
      left = left;
    else if (link.left + tooltip.width < view.right)
      // fit tooltip to right of link
      left = link.left;
    else if (link.right - tooltip.width > view.left)
      // fit tooltip to left of link
      left = link.right - tooltip.width;
    // center tooltip in view
    else left = (view.right - view.left) / 2 - tooltip.width / 2;

    // vertical positioning
    if (top)
      // use explicit value
      top = top;
    else if (link.top - tooltip.height > view.top)
      // fit tooltip above link
      top = link.top - tooltip.height;
    else if (link.bottom + tooltip.height < view.bottom)
      // fit tooltip below link
      top = link.bottom;
    else {
      // center tooltip in view
      top = view.top + view.height / 2 - tooltip.height / 2;
      // nudge off of link to left/right if possible
      if (link.right + tooltip.width < view.right) left = link.right;
      else if (link.left - tooltip.width > view.left)
        left = link.left - tooltip.width;
    }

    tooltipElement.style.left = left + "px";
    tooltipElement.style.top = top + "px";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
  <!-- modified from: https://fontawesome.com/icons/caret-left -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
    ></path>
  </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
  <!-- modified from: https://fontawesome.com/icons/caret-right -->
  <svg width="16" height="16" viewBox="0 0 192 512">
    <path
      fill="currentColor"
      d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
    ></path>
  </svg>
</template>

<style>
  @media only screen {
    /* tooltip container */
    #tooltip {
      position: absolute;
      width: 50%;
      min-width: 240px;
      max-width: 75%;
      z-index: 1;
    }

    /* tooltip content */
    #tooltip_content {
      margin-bottom: 5px;
      padding: 20px;
      border-radius: 5px;
      border: solid 1px #bdbdbd;
      box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
      background: #ffffff;
      overflow-wrap: break-word;
    }

    /* tooltip copy of paragraphs and figures */
    #tooltip_content > p,
    #tooltip_content > figure {
      margin: 0;
      max-height: 320px;
      overflow-y: auto;
    }

    /* tooltip copy of <img> */
    #tooltip_content > figure > img,
    #tooltip_content > figure > svg {
      max-height: 260px;
    }

    /* navigation bar */
    #tooltip_nav_bar {
      margin-top: 10px;
      text-align: center;
    }

    /* navigation bar previous/next buton */
    #tooltip_nav_bar > .icon_button {
      position: relative;
      top: 3px;
    }

    /* navigation bar previous button */
    #tooltip_nav_bar > .icon_button:first-of-type {
      margin-right: 5px;
    }

    /* navigation bar next button */
    #tooltip_nav_bar > .icon_button:last-of-type {
      margin-left: 5px;
    }
  }

  /* always hide tooltip on print */
  @media only print {
    #tooltip {
      display: none;
    }
  }
</style>
<!--
  Analytics Plugin (third-party) 
  
  Copy and paste code from Google Analytics or similar service here.
-->
<!-- 
  Annotations Plugin

  Allows public annotation of the  manuscript. See https://web.hypothes.is/.
-->

<script>
  // configuration
  window.hypothesisConfig = function () {
    return {
      branding: {
        accentColor: "#2196f3",
        appBackgroundColor: "#f8f8f8",
        ctaBackgroundColor: "#f8f8f8",
        ctaTextColor: "#000000",
        selectionFontFamily: "Open Sans, Helvetica, sans serif",
        annotationFontFamily: "Open Sans, Helvetica, sans serif",
      },
    };
  };

  // hypothesis client script
  const embed = "https://hypothes.is/embed.js";
  // hypothesis annotation count query url
  const query = "https://api.hypothes.is/api/search?limit=0&url=";

  // start script
  function start() {
    const button = makeButton();
    document.body.insertBefore(button, document.body.firstChild);
    insertCount(button);
  }

  // make button
  function makeButton() {
    // create button
    const button = document.createElement("button");
    button.id = "hypothesis_button";
    button.innerHTML = document.querySelector(".icon_hypothesis").innerHTML;
    button.title = "Hypothesis annotations";
    button.classList.add("icon_button");

    function onClick(event) {
      onButtonClick(event, button);
    }

    // attach click listeners
    button.addEventListener("click", onClick);

    return button;
  }

  // insert annotations count
  async function insertCount(button) {
    // get annotation count from Hypothesis based on url
    let count = "-";
    try {
      const canonical = document.querySelector('link[rel="canonical"]');
      const location = window.location;
      const url = encodeURIComponent((canonical || location).href);
      const response = await fetch(query + url);
      const json = await response.json();
      count = json.total || "-";
    } catch (error) {
      console.log(error);
    }

    // put count into button
    const counter = document.createElement("span");
    counter.id = "hypothesis_count";
    counter.innerHTML = count;
    button.title = "View " + count + " Hypothesis annotations";
    button.append(counter);
  }

  // when button is clicked
  function onButtonClick(event, button) {
    const script = document.createElement("script");
    script.src = embed;
    document.body.append(script);
    button.remove();
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
  <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
  <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
    <path
      fill="currentColor"
      d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
      tabindex="-1"
    ></path>
  </svg>
</template>

<style>
  /* hypothesis activation button */
  #hypothesis_button {
    box-sizing: border-box;
    position: fixed;
    top: 0;
    right: 0;
    width: 60px;
    height: 60px;
    background: #ffffff;
    border-radius: 0;
    border-left: solid 1px #bdbdbd;
    border-bottom: solid 1px #bdbdbd;
    box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
    z-index: 2;
  }

  /* hypothesis button svg */
  #hypothesis_button > svg {
    position: relative;
    top: -4px;
  }

  /* hypothesis annotation count */
  #hypothesis_count {
    position: absolute;
    left: 0;
    right: 0;
    bottom: 5px;
  }

  /* side panel */
  .annotator-frame {
    width: 280px !important;
  }

  /* match highlight color to rest of theme */
  .annotator-highlights-always-on .annotator-hl {
    background-color: #ffeb3b !important;
  }

  /* match focused color to rest of theme */
  .annotator-hl.annotator-hl-focused {
    background-color: #ff8a65 !important;
  }

  /* match bucket bar color to rest of theme */
  .annotator-bucket-bar {
    background: #f5f5f5 !important;
  }

  /* always hide button, toolbar, and tooltip on print */
  @media only print {
    #hypothesis_button {
      display: none;
    }

    .annotator-frame {
      display: none !important;
    }

    hypothesis-adder {
      display: none !important;
    }
  }
</style>
<!-- 
  Mathjax Plugin (third-party) 

  Allows the proper rendering of math/equations written in LaTeX.
  See https://www.mathjax.org/.
-->

<script type="text/x-mathjax-config">
  // configuration
  MathJax.Hub.Config({
    "CommonHTML": { linebreaks: { automatic: true } },
    "HTML-CSS": { linebreaks: { automatic: true } },
    "SVG": { linebreaks: { automatic: true } },
    "fast-preview": { disabled: true }
  });
</script>

<script
  src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
  integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
  crossorigin="anonymous"
></script>

<style>
  /* mathjax containers */
  .math.display > span:not(.MathJax_Preview) {
    /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
    display: flex !important;
    overflow-x: auto !important;
    overflow-y: hidden !important;
    justify-content: center;
    align-items: center;
    margin: 0 !important;
  }

  /* right click menu */
  .MathJax_Menu {
    border-radius: 5px !important;
    border: solid 1px #bdbdbd !important;
    box-shadow: none !important;
  }

  /* equation auto-number */
  span[id^="eq:"] > span.math.display + span {
    font-weight: 600;
  }

  /* equation */
  span[id^="eq:"] > span.math.display > span {
    /* nudge to make room for equation auto-number and anchor */
    margin-right: 60px !important;
  }
</style>
<!-- 
  Scite Plugin (third party)

  Adds a Scite badge next to each citation in the references section.
  See https://scite.ai/.
-->

<script
  async
  type="application/javascript"
  src="https://cdn.scite.ai/badge/scite-badge-latest.min.js"
></script>

<script>
  // start script
  function start() {
    // if printing, exit and don't run badges
    if (window.matchMedia("print").matches) return;

    // get citation elements
    const query = ".references div[id*='ref-']";
    const elements = document.querySelectorAll(query);
    for (const element of elements) addBadge(element);
  }

  // add badge in citation element
  function addBadge(element) {
    // get citation text
    const text = element.innerText;
    // find doi string, eg "DOI: xxxxx/xxxxx"
    const match = text.match(/DOI:\s+(\S*)/) || {};
    const doi = match[1];
    if (!doi) return;
    // create cite badge and put after citation
    const badge = document.createElement("div");
    badge.innerHTML = "Loading Scite badge...";
    badge.classList.add("scite-badge");
    badge.setAttribute("data-doi", doi);
    badge.setAttribute("data-layout", "horizontal");
    badge.setAttribute("data-show-zero", "true");
    element.append(badge);
    // modify preceding citation <p> element
    const p = element.querySelector("p");
    if (p) p.style.marginBottom = "0";
  }

  // start script when document is finished loading
  window.addEventListener("load", start);
</script>

<style>
  .scite-badge {
    text-indent: 0;
    margin-top: 10px;
  }

  @media print {
    .scite-badge {
      display: none;
    }
  }
</style>
</body>
</html>
